From 624e87df12f01e5cfd05af66b3c4d5badeba1d6d Mon Sep 17 00:00:00 2001
From: Orion Poplawski <orion@nwra.com>
Date: Thu, 19 Oct 2023 21:54:03 -0600
Subject: [PATCH] Make plugins/mathematica/bin/realpath.py python 3

---
 plugins/mathematica/bin/realpath.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/mathematica/bin/realpath.py b/plugins/mathematica/bin/realpath.py
index 3581a7dc38..142ad6a68d 100755
--- a/plugins/mathematica/bin/realpath.py
+++ b/plugins/mathematica/bin/realpath.py
@@ -2,4 +2,4 @@
 from os.path import realpath
 from sys import argv,exit
 if len(argv)<2: exit(1)
-print realpath(argv[1])
+print(realpath(argv[1]))
