diff -up ghdl/Makefile.in.gcc15~ ghdl/Makefile.in
--- ghdl/Makefile.in.gcc15~	2025-06-18 08:42:30.000000000 +0200
+++ ghdl/Makefile.in	2025-08-10 22:52:59.301330658 +0200
@@ -301,9 +301,9 @@ copy-sources.gcc: version.ads scripts/gc
 	 6 | 6.*)  	    gcc_ortho_lang=ortho-lang-6.c ;; \
 	 7.*)      	    gcc_ortho_lang=ortho-lang-7.c ;; \
 	 8.*)      	    gcc_ortho_lang=ortho-lang-8.c ;; \
-	 9.* | 10.* | 11.* | 12.* | 13.* | 14.*) gcc_ortho_lang=ortho-lang-9.c ;; \
+	 9.* | 10.* | 11.* | 12.* | 13.* | 14.* | 15.*) gcc_ortho_lang=ortho-lang-9.c ;; \
 	 *) echo "Mismatch gcc version from $(gcc_src_dir)"; \
-	    echo "Need gcc version 4.9.x, 5.x to 14.x"; \
+	    echo "Need gcc version 4.9.x, 5.x to 15.x"; \
 	    exit 1 ;; \
 	esac; \
 	$(CP) -p $(srcdir)/src/ortho/gcc/$$gcc_ortho_lang \
@@ -318,7 +318,7 @@ copy-sources.gcc: version.ads scripts/gc
 #	For gcc 12.x, use .cc extension
 	base_ver=`cat $(gcc_src_dir)/gcc/BASE-VER`; \
 	case $$base_ver in \
-	12.* | 13.* | 14.*) for f in $(gcc_vhdl_dir)/*.c; do mv $$f $${f}c; done; \
+	12.* | 13.* | 14.* | 15.*) for f in $(gcc_vhdl_dir)/*.c; do mv $$f $${f}c; done; \
 	  sed -e 's/ortho-lang.c/ortho-lang.cc/' \
 	    < $(srcdir)/scripts/gcc/config-lang.in \
 	    > $(gcc_vhdl_dir)/config-lang.in ;; \
