--- tango-icon-theme-extras-0.1.0/configure~	2006-01-10 22:25:55.000000000 +0000
+++ tango-icon-theme-extras-0.1.0/configure	2013-08-24 13:51:15.000000000 +0100
@@ -2000,7 +2000,7 @@
    fi
 fi
 
-# Check for rsvg or ksvgtopng for creating large scale bitmaps from svg
+# Check for rsvg-convert or ksvgtopng for creating large scale bitmaps from svg
 # Check whether --enable-png-creation or --disable-png-creation was given.
 if test "${enable_png_creation+set}" = set; then
   enableval="$enable_png_creation"
@@ -2015,7 +2015,7 @@
 if test "x$enable_large_bitmaps" = "xyes"; then
    if test -n "$PKG_CONFIG" && \
     { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"librsvg-2.0 >= 2.12.3\"") >&5
-  ($PKG_CONFIG --exists --print-errors "librsvg-2.0 >= 2.12.3") 2>&5
+  ($PKG_CONFIG --exists --print-errors "librsvg-2.0 >= 2.35.2") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
@@ -2024,7 +2024,7 @@
   enable_large_bitmaps=no
 fi
    if test "x$enable_large_bitmaps" = "xyes"; then
-      svgconvert_prog="rsvg"
+      svgconvert_prog="rsvg-convert"
    else
       svgconvert_prog="ksvgtopng"
    fi
