From 823939555acf4460e962985276ef292d02cc3d46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert-Andr=C3=A9=20Mauchin?= <zebob.m@gmail.com>
Date: Sun, 15 Oct 2023 13:05:24 +0200
Subject: [PATCH] Use QtIoCompressor from Qt5

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 600dc6efe..5af6b2663 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -390,8 +390,8 @@ endif(USE_SYSTEM_QTSINGLEAPPLICATION)
 
 # QtIoCompressor isn't patched, so we can use a system version if it's
 # available
-find_path(QTIOCOMPRESSOR_INCLUDE_DIRS qtiocompressor.h PATH_SUFFIXES QtSolutions)
-find_library(QTIOCOMPRESSOR_LIBRARIES QtSolutions_IOCompressor-2.3)
+find_path(QTIOCOMPRESSOR_INCLUDE_DIRS qtiocompressor.h PATH_SUFFIXES qt5/QtSolutions)
+find_library(QTIOCOMPRESSOR_LIBRARIES Qt5Solutions_IOCompressor-2.3)
 if(NOT QTIOCOMPRESSOR_INCLUDE_DIRS OR NOT QTIOCOMPRESSOR_LIBRARIES)
   add_subdirectory(3rdparty/qtiocompressor)
   set(QTIOCOMPRESSOR_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtiocompressor)
-- 
2.41.0

