--- logjam-4.4.1/data/logjam.desktop.in.BAD	2005-04-14 09:04:52.000000000 -0500
+++ logjam-4.4.1/data/logjam.desktop.in	2005-04-14 09:05:55.000000000 -0500
@@ -1,9 +1,9 @@
 [Desktop Entry]
 _Name=LogJam
-_Comment=Manage journal/weblog
-Exec=logjam
-Icon=logjam.png
+_Comment=LiveJournal Client
+Exec=/usr/bin/logjam
+Icon=/usr/share/pixmaps/logjam_goat.png
 Terminal=false
 Type=Application
-Categories=Application;Network;
+Categories=Application;Network;X-Fedora;
 StartupNotify=true
