diff -up logjam-4.6.2/src/journalstore-sqlite.c.fix-verify-path-call logjam-4.6.2/src/journalstore-sqlite.c
--- logjam-4.6.2/src/journalstore-sqlite.c.fix-verify-path-call	2024-02-10 08:51:25.930516069 -0500
+++ logjam-4.6.2/src/journalstore-sqlite.c	2024-02-10 08:51:30.060579151 -0500
@@ -137,7 +137,7 @@ JournalStore* journal_store_open(JamAcco
 		goto out;
 	}
 
-	if (!verify_path(path, FALSE, &err))
+	if (!verify_path(path, FALSE, err))
 		goto out;
 
 	ret = sqlite3_open(path, &db);
