diff -up xgalaga-2.0.34/libsprite/Wlib.h~ xgalaga-2.0.34/libsprite/Wlib.h
--- xgalaga-2.0.34/libsprite/Wlib.h~	2009-11-05 11:58:13.000000000 +0100
+++ xgalaga-2.0.34/libsprite/Wlib.h	2009-11-05 12:00:02.000000000 +0100
@@ -94,7 +94,6 @@ extern unsigned long w_socket;
 
 /* For XF86VIDMODE Support */
 #ifdef XF86VIDMODE
-#include <X11/extensions/xf86dga.h>
 #include <X11/extensions/xf86vmode.h>   
 
 struct xf86vidmode_data {
diff -up xgalaga-2.0.34/Wlib.h~ xgalaga-2.0.34/Wlib.h
--- xgalaga-2.0.34/Wlib.h~	2009-11-05 12:02:14.000000000 +0100
+++ xgalaga-2.0.34/Wlib.h	2009-11-05 12:03:01.000000000 +0100
@@ -94,7 +94,6 @@ extern unsigned long w_socket;
 
 /* For XF86VIDMODE Support */
 #ifdef XF86VIDMODE
-#include <X11/extensions/xf86dga.h>
 #include <X11/extensions/xf86vmode.h>   
 
 struct xf86vidmode_data {
