--- src/pic16/glue.cold	2024-06-07 17:31:22.047242046 +1000
+++ src/pic16/glue.c	2024-06-07 17:19:39.997621665 +1000
@@ -1802,7 +1802,7 @@
       unsigned long ramsize = pic16 ? pic16->RAMsize : 0x200;
       symbol *sym;
 
-      reg = newReg (REG_SFR, PO_SFR_REGISTER, ramsize-1, "_sram_end", 0, 0, NULL);
+      reg = (memmap *)newReg (REG_SFR, PO_SFR_REGISTER, ramsize-1, "_sram_end", 0, 0, NULL);
       addSet (&pic16_fix_udata, reg);
 
       sym = newSymbol ("sram_end", 0);
