From a34009195f2ea05ed331abcbcdfc38505a426c3f Mon Sep 17 00:00:00 2001
From: Sergio Pascual <sergiopr@fis.ucm.es>
Date: Thu, 4 Jan 2024 20:10:07 +0100
Subject: [PATCH] Delete usage of function removed from numpy>=1.25

---
 astroML/tests/test_resample.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/astroML/tests/test_resample.py b/astroML/tests/test_resample.py
index e099fdd5..dd539978 100644
--- a/astroML/tests/test_resample.py
+++ b/astroML/tests/test_resample.py
@@ -1,5 +1,5 @@
 import numpy as np
-from numpy.testing import assert_allclose, run_module_suite
+from numpy.testing import assert_allclose
 
 from astroML.resample import bootstrap, jackknife
 from astroML.stats import mean_sigma
@@ -87,7 +87,3 @@ def test_jackknife_pass_indices():
                      pass_indices=True)
 
     assert_allclose(res1, res2)
-
-
-if __name__ == '__main__':
-    run_module_suite()
