--- a/bindings/Makefile.am
+++ b/bindings/Makefile.am
@@ -35,8 +35,9 @@
                 && $(MAKE) install ) \
            || true
 	$(AM_V_GEN)test -f ${builddir}/ruby/Makefile \
-           && ( cd ${builddir}/ruby \
-                && $(MAKE) EPREFIX=$(DESTDIR)$(exec_prefix) $(RUBY_MAKE_OPTIONS) install ) \
+           && ( $(MKDIR_P) $(DESTDIR)$(libdir)/ruby/vendor_ruby && cd ${builddir}/ruby \
+                && $(MAKE) EPREFIX=$(DESTDIR)$(exec_prefix) $(RUBY_MAKE_OPTIONS) \
+                hdrdir="$(includedir)" rubyhdrdir="$(includedir)" includedir="$(includedir)" install ) \
            || true
 	$(AM_V_GEN)test -f ${builddir}/python/dist/*.whl \
            && ( $(PYTHON) -m pip install --upgrade --no-index --no-deps \
@@ -59,7 +60,8 @@
              EPREFIX=$(exec_prefix) \
              ABS_TOP_SRCDIR=${abs_top_srcdir} \
              ABS_TOP_BUILDDIR=${abs_top_builddir} \
-             $(RUBY_MAKE_OPTIONS) RUBYARCHDIR= )
+             $(RUBY_MAKE_OPTIONS) hdrdir="$(includedir)" \
+             rubyhdrdir="$(includedir)" includedir="$(includedir)" RUBYARCHDIR= )
 
 # Build the python module as a wheel in ${builddir}/python/dist/; it is
 # installed from there (via pip) by install-data-local. The wheel.stamp
