Index: rpcsvc-proto-1.3.1/rpcgen/rpc_main.c
===================================================================
--- rpcsvc-proto-1.3.1.orig/rpcgen/rpc_main.c
+++ rpcsvc-proto-1.3.1/rpcgen/rpc_main.c
@@ -1346,15 +1346,6 @@ parseargs (int argc, const char *argv[],
   cmd->Scflag = flag['C'];
   cmd->makefileflag = flag['M'];
 
-#ifndef _RPC_THREAD_SAFE_
-  if (mtflag || newstyle)
-    {
-      /* glibc doesn't support these flags.  */
-      f_print (stderr,
-	       _("This implementation doesn't support newstyle or MT-safe code!\n"));
-      return (0);
-    }
-#endif
   if (tirpcflag)
     {
       pmflag = inetdflag ? 0 : 1;    /* pmflag or inetdflag is always TRUE */
