From 1314784b240db6dbe053922e69698c4f79827980 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Thu, 25 Apr 2019 17:45:30 +0200
Subject: [PATCH 4/5] build: align a couple of configure options with their
 documentation

Signed-off-by: Lubomir Rintel <lkundrak@v3.sk>
---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index d564899..e69e447 100644
--- a/configure.ac
+++ b/configure.ac
@@ -78,7 +78,7 @@ AC_ARG_WITH(xorg-module-dir,
 # Vivante GPU acceleration using libGAL
 AC_ARG_ENABLE(vivante,
 	      AC_HELP_STRING([--enable-vivante],
-			     [Enable Vivante libGAL acceleration support [[default=enable]]]),
+			     [Enable Vivante libGAL acceleration support [[default=no]]]),
 	      [ACCEL_GALCORE="$enableval"],
 	      [ACCEL_GALCORE=no])
 
@@ -233,12 +233,12 @@ AC_ARG_ENABLE(dri2, AC_HELP_STRING([--disable-dri2],
 AC_ARG_ENABLE(dri3, AC_HELP_STRING([--disable-dri3],
 		[Disable DRI3 support [[default=auto]]]),
 		[DRI3="$enableval"],
-		[DRI3=no])
+		[DRI3=auto])
 
 AC_ARG_ENABLE(present, AC_HELP_STRING([--disable-present],
 		[Disable PRESENT support [[default=auto]]]),
 		[PRESENT="$enableval"],
-		[PRESENT=no])
+		[PRESENT=auto])
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.9.99.1 xproto fontsproto pixman-1])
-- 
2.21.0

