diff -Naur bacula-15.0.2.old/autoconf/bacula-macros/db.m4 bacula-15.0.2/autoconf/bacula-macros/db.m4
--- bacula-15.0.2.old/autoconf/bacula-macros/db.m4	2024-06-02 10:15:17.084968302 +0200
+++ bacula-15.0.2/autoconf/bacula-macros/db.m4	2024-06-02 10:15:17.418968518 +0200
@@ -102,7 +102,7 @@
      fi
      DBI_INCLUDE=-I$DBI_INCDIR
      if test x$use_libtool != xno; then
-        DBI_LIBS="-R $DBI_LIBDIR -L$DBI_LIBDIR -ldbi"
+        DBI_LIBS="-L$DBI_LIBDIR -ldbi"
      else
         DBI_LIBS="-L$DBI_LIBDIR -ldbi"
      fi
@@ -494,7 +494,7 @@
         elif test -f $MYSQL_LIBDIR/libmysqlclient_r.a \
                -o -f $MYSQL_LIBDIR/libmysqlclient_r.so; then
            if test x$use_libtool != xno; then
-              MYSQL_LIBS="-R $MYSQL_LIBDIR -L$MYSQL_LIBDIR -lmysqlclient_r -lz"
+              MYSQL_LIBS="-L$MYSQL_LIBDIR -lmysqlclient_r -lz"
            else
               MYSQL_LIBS="-L$MYSQL_LIBDIR -lmysqlclient_r -lz"
            fi
@@ -625,7 +625,7 @@
         fi
      MYSQL_INCLUDE=-I$MYSQL_INCDIR
      if test x$use_libtool != xno; then
-        MYSQL_LIBS="-R $MYSQL_LIBDIR -L$MYSQL_LIBDIR -lmysqld -lz -lm -lcrypt"
+        MYSQL_LIBS="-L$MYSQL_LIBDIR -lmysqld -lz -lm -lcrypt"
      else
         MYSQL_LIBS="-L$MYSQL_LIBDIR -lmysqld -lz -lm -lcrypt"
      fi
@@ -715,7 +715,7 @@
      fi
      INGRES_INCLUDE=-I$INGRES_INCDIR
      if test x$use_libtool != xno; then
-        INGRES_LIBS="-R $INGRES_LIBDIR -L$INGRES_LIBDIR -lq.1 -lcompat.1 -lframe.1"
+        INGRES_LIBS="-L$INGRES_LIBDIR -lq.1 -lcompat.1 -lframe.1"
      else
         INGRES_LIBS="-L$INGRES_LIBDIR -lq.1 -lcompat.1 -lframe.1"
      fi
@@ -803,7 +803,7 @@
      fi
      SQLITE_INCLUDE=-I$SQLITE_INCDIR
      if test x$use_libtool != xno; then
-        SQLITE_LIBS="-R $SQLITE_LIBDIR -L$SQLITE_LIBDIR -lsqlite3"
+        SQLITE_LIBS="-L$SQLITE_LIBDIR -lsqlite3"
      else
         SQLITE_LIBS="-L$SQLITE_LIBDIR -lsqlite3"
      fi
@@ -936,7 +936,7 @@
 
      POSTGRESQL_INCLUDE=-I$POSTGRESQL_INCDIR
      if test x$use_libtool != xno; then
-        POSTGRESQL_LIBS="-R $POSTGRESQL_LIBDIR -L$POSTGRESQL_LIBDIR -lpq"
+        POSTGRESQL_LIBS="-L$POSTGRESQL_LIBDIR -lpq"
      else
         POSTGRESQL_LIBS="-L$POSTGRESQL_LIBDIR -lpq"
      fi
diff -Naur bacula-15.0.2.old/autoconf/configure.in bacula-15.0.2/autoconf/configure.in
--- bacula-15.0.2.old/autoconf/configure.in	2024-06-02 10:15:17.086968303 +0200
+++ bacula-15.0.2/autoconf/configure.in	2024-06-02 10:15:17.418968518 +0200
@@ -16,7 +16,7 @@
 cd ${BUILD_DIR}
 AC_SUBST(BUILD_DIR)
 AC_SUBST(TOP_DIR)
-AC_CONFIG_AUX_DIR(${BUILD_DIR}/autoconf)
+AC_CONFIG_AUX_DIR(autoconf)
 AC_CONFIG_HEADERS(src/config.h:autoconf/config.h.in)
 
 dnl minimal Win32 stuff for "make clean"
@@ -238,7 +238,6 @@
    QMAKE_LIBTOOL="${BUILD_DIR}/libtool"
    FD_PLUGIN_DIR="src/plugins/fd"
    DIR_PLUGIN_DIR="src/plugins/dir"
-   DLIB="-rpath \$(libdir)"
    have_plugins=yes
 else
    DEFAULT_OBJECT_TYPE=".o"
@@ -3748,11 +3747,6 @@
 AC_DEFINE(FDLIBS)
 
 CFLAGS=${CFLAGS--O}
-
-if test x$have_gcc = xyes ; then
-   CPPFLAGS="$CPPFLAGS -x c++ -fno-strict-aliasing -fno-exceptions -fno-rtti"
-   CFLAGS="$CFLAGS -x c++ -fno-strict-aliasing -fno-exceptions -fno-rtti"
-fi
 LDFLAGS=${LDFLAGS--O}
 CPPFLAGS="$CPPFLAGS"
 CFLAGS="$CFLAGS"
diff -Naur bacula-15.0.2.old/autoconf/Make.common.in bacula-15.0.2/autoconf/Make.common.in
--- bacula-15.0.2.old/autoconf/Make.common.in	2024-06-02 10:15:17.084968302 +0200
+++ bacula-15.0.2/autoconf/Make.common.in	2024-06-02 10:15:17.419968519 +0200
@@ -39,7 +39,7 @@
 mandir = @mandir@
 manext = 8
 
-NO_ECHO = @
+NO_ECHO =
 
 # Tools & program stuff
 CC = @CC@
@@ -77,12 +77,12 @@
 DEFAULT_ARCHIVE_TYPE = @DEFAULT_ARCHIVE_TYPE@
 DEFAULT_SHARED_OBJECT_TYPE = @DEFAULT_SHARED_OBJECT_TYPE@
 LIBTOOL = @BUILD_DIR@/libtool
-LIBTOOL_COMPILE = @LIBTOOL@ --silent --tag=CXX --mode=compile
-LIBTOOL_LINK = @LIBTOOL@ --silent --tag=CXX --mode=link
-LIBTOOL_INSTALL = @LIBTOOL@ --silent --tag=CXX --mode=install
-LIBTOOL_INSTALL_FINISH = @LIBTOOL@ --silent --tag=CXX --finish --mode=install
-LIBTOOL_UNINSTALL = @LIBTOOL@ --silent --tag=CXX --mode=uninstall
-LIBTOOL_CLEAN = @LIBTOOL@ --silent --tag=CXX --mode=clean
+LIBTOOL_COMPILE = @LIBTOOL@ --tag=CXX --mode=compile
+LIBTOOL_LINK = @LIBTOOL@ --tag=CXX --mode=link
+LIBTOOL_INSTALL = @LIBTOOL@ --tag=CXX --mode=install
+LIBTOOL_INSTALL_FINISH = @LIBTOOL@ --tag=CXX --finish --mode=install
+LIBTOOL_UNINSTALL = @LIBTOOL@ --tag=CXX --mode=uninstall
+LIBTOOL_CLEAN = @LIBTOOL@ --tag=CXX --mode=clean
 
 # Flags & libs
 CFLAGS = @CFLAGS@ @OPENSSL_INC@
diff -Naur bacula-15.0.2.old/src/qt-console/bat.pro.in bacula-15.0.2/src/qt-console/bat.pro.in
--- bacula-15.0.2.old/src/qt-console/bat.pro.in	2024-06-02 10:15:16.980968234 +0200
+++ bacula-15.0.2/src/qt-console/bat.pro.in	2024-06-02 10:15:17.419968519 +0200
@@ -23,8 +23,8 @@
 DEPENDPATH  += .
 INCLUDEPATH += .. . ./console ./restore ./select
 LIBS        += -L../lib -lbaccfg -lbac -L../findlib -lbacfind @OPENSSL_LIBS@
-LIBTOOL_LINK = @QMAKE_LIBTOOL@ --silent --tag=CXX --mode=link
-LIBTOOL_INSTALL = @QMAKE_LIBTOOL@ --silent --mode=install
+LIBTOOL_LINK = @QMAKE_LIBTOOL@ --tag=CXX --mode=link
+LIBTOOL_INSTALL = @QMAKE_LIBTOOL@ --mode=install
 QMAKE_LINK   = $${LIBTOOL_LINK} $(CXX)
 QMAKE_INSTALL_PROGRAM = $${LIBTOOL_INSTALL} install -m @SBINPERM@ -p
 QMAKE_CLEAN += .libs/* bat
@@ -195,3 +195,8 @@
 QMAKE_EXTRA_TARGETS += depend
 
 TRANSLATIONS += ts/bat_fr.ts ts/bat_de.ts
+
+QMAKE_CPPFLAGS += @CPPFLAGS@
+QMAKE_CFLAGS += @CFLAGS@
+QMAKE_CXXFLAGS += @CXXFLAGS@
+QMAKE_LFLAGS += @LDFLAGS@
diff -Naur bacula-15.0.2.old/src/qt-console/tray-monitor/tray-monitor.pro.in bacula-15.0.2/src/qt-console/tray-monitor/tray-monitor.pro.in
--- bacula-15.0.2.old/src/qt-console/tray-monitor/tray-monitor.pro.in	2024-06-02 10:15:16.983968236 +0200
+++ bacula-15.0.2/src/qt-console/tray-monitor/tray-monitor.pro.in	2024-06-02 10:15:17.419968519 +0200
@@ -36,8 +36,8 @@
 DEPENDPATH  += .
 INCLUDEPATH += ../.. .
 QMAKE_CC = $(CXX) 
-LIBTOOL_LINK = @QMAKE_LIBTOOL@ --silent --tag=CXX --mode=link
-LIBTOOL_INSTALL = @QMAKE_LIBTOOL@ --silent --mode=install
+LIBTOOL_LINK = @QMAKE_LIBTOOL@ --tag=CXX --mode=link
+LIBTOOL_INSTALL = @QMAKE_LIBTOOL@ --mode=install
 QMAKE_LINK   = $${LIBTOOL_LINK} $(CXX)
 QMAKE_INSTALL_PROGRAM = $${LIBTOOL_INSTALL} install -m @SBINPERM@ -p
 QMAKE_CLEAN += obj/* .libs/* bacula-tray-monitor release/bacula-tray-monitor
@@ -69,3 +69,8 @@
 INCLUDEPATH += $$JOURNAL_DIR
 
 TRANSLATIONS += ts/tm_fr.ts ts/tm_de.ts ts/tm_ja.ts
+
+QMAKE_CPPFLAGS += @CPPFLAGS@
+QMAKE_CFLAGS += @CFLAGS@
+QMAKE_CXXFLAGS += @CXXFLAGS@
+QMAKE_LFLAGS += @LDFLAGS@
