--- HPCombi-1.1.1/CMakeLists.txt.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/CMakeLists.txt	2025-11-24 19:30:56.514208745 -0700
@@ -119,7 +119,6 @@ add_subdirectory(doc)
 
 include_directories(
   ${CMAKE_SOURCE_DIR}/include
-  ${CMAKE_SOURCE_DIR}/third_party
   ${PROJECT_BINARY_DIR})
 
 # add_compile_options(-Wall -Wextra -Wpedantic)
@@ -159,9 +158,6 @@ install (
   DESTINATION include
   FILES_MATCHING PATTERN "*.hpp")
 install (
-  DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/simde-0.8.2
-  DESTINATION include/${HPCOMBI_INSTALL_DIR})
-install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/hpcombi.pc
   DESTINATION lib/pkgconfig)
 
--- HPCombi-1.1.1/examples/pattern.cpp.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/examples/pattern.cpp	2025-11-24 20:59:33.923874179 -0700
@@ -30,7 +30,7 @@
 #else
 #include <unordered_map>
 #endif
-#include "simde-0.8.2/x86/sse4.1.h"  // for simde_mm_max_epu8, simde...
+#include <simde/x86/sse4.1.h>  // for simde_mm_max_epu8, simde...
 
 #include "hpcombi/perm16.hpp"
 
--- HPCombi-1.1.1/examples/Renner.cpp.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/examples/Renner.cpp	2025-11-24 20:59:17.123980041 -0700
@@ -32,7 +32,7 @@
 #endif
 
 #include "hpcombi/perm16.hpp"
-#include "simde-0.8.2/x86/sse4.1.h"  // for simde_mm_max_epu8, simde...
+#include <simde/x86/sse4.1.h>  // for simde_mm_max_epu8, simde...
 
 template <typename T>
 std::ostream &operator<<(std::ostream &out, const std::vector<T> &v) {
--- HPCombi-1.1.1/include/hpcombi/bmat16.hpp.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/include/hpcombi/bmat16.hpp	2025-11-24 20:55:04.663038374 -0700
@@ -34,7 +34,7 @@
 #include "bmat8.hpp"  // for BMat8
 #include "debug.hpp"  // for HPCOMBI_ASSERT
 
-#include "simde-0.8.2/x86/avx2.h"
+#include <simde/x86/avx2.h>
 
 namespace HPCombi {
 using xpu16 = uint16_t __attribute__((vector_size(32)));
--- HPCombi-1.1.1/include/hpcombi/epu8.hpp.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/include/hpcombi/epu8.hpp	2025-11-24 20:55:36.118813661 -0700
@@ -45,8 +45,8 @@ eg simde_mm_testz_si128(a,a) → is_all_
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wbitwise-instead-of-logical"
 #endif
-#include "simde-0.8.2/x86/sse4.1.h"  // for simde_mm_max_epu8, simde...
-#include "simde-0.8.2/x86/sse4.2.h"  // for ???
+#include <simde/x86/sse4.1.h>  // for simde_mm_max_epu8, simde...
+#include <simde/x86/sse4.2.h>  // for ???
 #if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
--- HPCombi-1.1.1/include/hpcombi/perm16.hpp.orig	2025-11-22 07:43:56.000000000 -0700
+++ HPCombi-1.1.1/include/hpcombi/perm16.hpp	2025-11-24 20:56:01.478258883 -0700
@@ -48,8 +48,8 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wbitwise-instead-of-logical"
 #endif
-#include "simde-0.8.2/x86/sse4.1.h"
-#include "simde-0.8.2/x86/sse4.2.h"
+#include <simde/x86/sse4.1.h>
+#include <simde/x86/sse4.2.h>
 #if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
