diff --git a/lsfrom.cabal b/lsfrom.cabal
index 3430b61..b9d4d08 100644
--- a/lsfrom.cabal
+++ b/lsfrom.cabal
@@ -21,6 +21,10 @@ extra-doc-files:     ChangeLog.md
 extra-source-files:  test/files/A
                      test/files/B
                      test/files/C
+                     test/cases/A
+                     test/cases/B
+                     test/cases/a
+                     test/cases/b
 
 source-repository head
   type:                git
@@ -64,6 +68,7 @@ test-suite test
 
   ghc-options:   -Wall
   build-depends: base >= 4 && < 5
+               , extra
                , filepath
                , process
                , simple-cmd
diff --git a/test/tests.hs b/test/tests.hs
index e418899..09ae47f 100644
--- a/test/tests.hs
+++ b/test/tests.hs
@@ -1,30 +1,33 @@
 {-# LANGUAGE CPP #-}
 
-import Control.Monad (unless)
-import SimpleCmd (cmdN, error')
+import Control.Monad.Extra (unless, whenJust)
+import SimpleCmd (cmdLines, cmdN, error', (+-+))
 import System.IO (hGetContents)
 import System.Process (CreateProcess(..), StdStream(CreatePipe),
                        proc, withCreateProcess)
 
 type Test = (Maybe String, FilePath, [String], [String])
 
-lsfrom :: Test -> IO ()
-lsfrom (mlocale, dir, args, expect) = do
-  let menv = mlocale >>= \l -> Just [("LC_ALL",l)]
-  withCreateProcess
-    (proc "lsfrom" args)
-    { cwd = Just dir, env = menv, std_out = CreatePipe } $
-    \ _si mso _se _p ->
-      case mso of
-        Nothing -> error "no stdout handle found"
-        Just so -> do
-          out <- hGetContents so
-          unless (lines out == expect) $ do
-            print mlocale
-            cmdN "lsfrom" args
-            putStrLn $ "returned> " ++ show out
-            putStrLn $ "expected> " ++ show expect
-            error' "failed"
+lsfrom :: [String] -> Test -> IO ()
+lsfrom locales (mlocale, dir, args, expect) = do
+  if maybe True (`elem` locales) mlocale
+    then do
+    let menv = mlocale >>= \l -> Just [("LC_ALL",l)]
+    withCreateProcess
+      (proc "lsfrom" args)
+      { cwd = Just dir, env = menv, std_out = CreatePipe } $
+      \ _si mso _se _p ->
+        case mso of
+          Nothing -> error "no stdout handle found"
+          Just so -> do
+            out <- lines <$> hGetContents so
+            unless (out == expect) $ do
+              print mlocale
+              cmdN "lsfrom" args
+              putStrLn $ "returned>" +-+ show out
+              putStrLn $ "expected>" +-+ show expect
+              error' "failed"
+    else whenJust mlocale $ putStrLn . ("locale skipped:" +-+)
 
 tests :: [Test]
 tests =
@@ -41,14 +44,16 @@ tests =
   , (Nothing, "test/files", ["-b", "B", "-a", "B"], [])
   , (Nothing, "test/files", ["-s", "-u", "B", "-f", "A"], ["A", "B"])
   , (Just "C", "test/cases", [], ["A", "B", "a", "b"])
+  -- encodings must be lowercase ".utf8" to be detected
   , (Just "ja_JP.utf8", "test/cases", [], ["A", "B", "a", "b"])
-  , (Just "en_US.UTF-8", "test/cases", [], ["a", "A", "b", "B"])
+  , (Just "en_US.utf8", "test/cases", [], ["a", "A", "b", "B"])
   ]
 
 main :: IO ()
 main = do
-  mapM_ lsfrom tests
-  putStrLn $ show (length tests) ++ " tests run"
+  locales <- cmdLines "locale" ["-a"]
+  mapM_ (lsfrom locales) tests
+  putStrLn $ show (length tests) +-+ "tests run"
 
 #if !MIN_VERSION_simple_cmd(0,1,4)
 error' :: String -> a
