--- CToolbox/Modules/hess_ari.cpp.orig	2014-01-30 11:13:37.000000000 -0700
+++ CToolbox/Modules/hess_ari.cpp	2020-12-12 11:02:15.419543000 -0700
@@ -140,7 +140,7 @@ LowTriMatrix& LowTriMatrix::operator= (
   if (this == &a) return *this;
   delete [] rows;
   dim = a.dim;
-  if (dim == 0) { rows = NULL; return *this; }
+  if (dim <= 0) { rows = NULL; return *this; }
   rows = new ivector[dim];
   for (int i = 0; i < dim; i++) rows[i] = a.rows[i];
   return *this;
@@ -256,7 +256,7 @@ HTvector& HTvector::operator= ( const HT
   if (this == &v) return *this;
   delete [] ht;
   nmax = v.nmax;
-  if (nmax == 0) { ht = NULL; return *this; }
+  if (nmax <= 0) { ht = NULL; return *this; }
   ht = new HessType[nmax];
   for (int i = 0; i < nmax; i++) ht[i] = v.ht[i];
   return *this;
