From 37cb83d9c85f23893e2bc6b022a67db019d3ef3d Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Fri, 26 Dec 2025 21:02:44 +0100
Subject: [PATCH 3/4] port to nix 0.27

---
 src/connection/ll_conn.rs | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/src/connection/ll_conn.rs b/src/connection/ll_conn.rs
index 5ef91a3..90b51e9 100644
--- a/src/connection/ll_conn.rs
+++ b/src/connection/ll_conn.rs
@@ -10,15 +10,14 @@ use std::io::IoSlice;
 use std::io::IoSliceMut;
 use std::time;
 
+use std::os::unix::io::AsRawFd;
 use std::os::unix::io::RawFd;
-use std::os::unix::io::{AsRawFd, FromRawFd};
 use std::os::unix::net::UnixStream;
 
 use nix::cmsg_space;
-use nix::sys::socket::SockaddrStorage;
 use nix::sys::socket::{
     self, connect, recvmsg, sendmsg, socket, ControlMessage, ControlMessageOwned, MsgFlags,
-    UnixAddr,
+    SockaddrStorage, UnixAddr,
 };
 
 /// A lowlevel abstraction over the raw unix socket
@@ -45,14 +44,13 @@ pub struct DuplexConn {
 impl RecvConn {
     pub fn can_read_from_source(&self) -> nix::Result<bool> {
         let mut fdset = nix::sys::select::FdSet::new();
-        let fd = self.stream.as_raw_fd();
-        fdset.insert(fd);
+        fdset.insert(&self.stream);
 
         use nix::sys::time::TimeValLike;
         let mut zero_timeout = nix::sys::time::TimeVal::microseconds(0);
 
         nix::sys::select::select(None, Some(&mut fdset), None, None, Some(&mut zero_timeout))?;
-        Ok(fdset.contains(fd))
+        Ok(fdset.contains(&self.stream))
     }
 
     /// Reads from the source once but takes care that the internal buffer only reaches at maximum max_buffer_size
@@ -453,8 +451,8 @@ impl DuplexConn {
             None,
         )?;
 
-        connect(sock, &addr)?;
-        let mut stream = unsafe { UnixStream::from_raw_fd(sock) };
+        connect(sock.as_raw_fd(), &addr)?;
+        let mut stream = UnixStream::from(sock);
         match auth::do_auth(&mut stream)? {
             auth::AuthResult::Ok => {}
             auth::AuthResult::Rejected => return Err(Error::AuthFailed),
-- 
2.52.0

