From 24dcb6a3b37bc82826c7846ea69e1c42f5351f2d Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Mon, 27 Apr 2020 12:57:13 +0200
Subject: [PATCH 3/3] Disable reporting

Forwarded: not-needed
---
 .../plugin/generator/GeneratorUtils.java      | 61 +++----------------
 1 file changed, 7 insertions(+), 54 deletions(-)

diff --git a/maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java b/maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java
index 55ea5f9..82923b6 100644
--- a/maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java
+++ b/maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java
@@ -30,7 +30,6 @@ import org.apache.maven.artifact.DependencyResolutionRequiredException;
 import org.apache.maven.plugin.descriptor.MojoDescriptor;
 import org.apache.maven.plugin.descriptor.Parameter;
 import org.apache.maven.project.MavenProject;
-import org.apache.maven.reporting.MavenReport;
 import org.codehaus.plexus.util.DirectoryScanner;
 import org.codehaus.plexus.util.FileUtils;
 import org.codehaus.plexus.util.StringUtils;
@@ -141,7 +140,7 @@ public final class PluginUtils {
         try {
             Class<?> clazz = Class.forName(mojoClassName, false, classLoader);
 
-            return MavenReport.class.isAssignableFrom(clazz);
+            return false; // reporting support disabled
         } catch (ClassNotFoundException e) {
             return false;
         }
-- 
2.0
