diff --git a/pyproject.toml b/pyproject.toml
index 872c5bd..7fb3d71 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [build-system]
 requires = [
-    "setuptools==69.5.1",
-    "setuptools_scm==8.0.4",
+    "setuptools>=69.2.0",
+    "setuptools_scm>=8.0.4",
 ]
 build-backend = "setuptools.build_meta"
 
@@ -50,12 +50,12 @@ dependencies = [
 # Extras used by developers *of* gbulb are pinned to specific versions to
 # ensure environment consistency.
 dev = [
-    "coverage[toml] == 7.5.0",
+    "coverage[toml] >= 7.3.2",
     # Pre-commit 3.6.0 deprecated support for Python 3.8
     "pre-commit == 3.5.0 ; python_version < '3.9'",
-    "pre-commit == 3.7.0 ; python_version >= '3.9'",
-    "pytest == 8.1.1",
-    "tox == 4.15.0",
+    "pre-commit >= 3.7.0 ; python_version >= '3.9'",
+    "pytest >= 8.1.1",
+    "tox >= 4.15.0",
 ]
 
 [project.urls]
