From 1f6c4251d23ba6117943440bc1e3f43fe691da3a Mon Sep 17 00:00:00 2001
Subject: httpdomain: bump domain data version

---
 src/sphinxcontrib/httpdomain/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/sphinxcontrib/httpdomain/__init__.py b/src/sphinxcontrib/httpdomain/__init__.py
--- a/src/sphinxcontrib/httpdomain/__init__.py
+++ b/src/sphinxcontrib/httpdomain/__init__.py
@@ -640,1 +640,3 @@ class HTTPDomain(Domain):
-    initial_data = {
+    data_version = '0.1'
+
+    initial_data = {
