--- cmigemo-default-src/src/rxgen.c.keepregex	2011-03-09 16:07:58.000000000 +0900
+++ cmigemo-default-src/src/rxgen.c	2011-03-09 12:02:31.000000000 +0900
@@ -176,11 +176,11 @@
     }
 }
 
-    static rnode*
-search_rnode(rnode* node, unsigned int code)
+    static rnode**
+search_rnode(rnode** node, unsigned int code)
 {
-    while (node && node->code != code)
-	node = node->next;
+    while (*node && (*node)->code != code)
+	node = &(*node)->next;
     return node;
 }
 
@@ -211,13 +211,14 @@
 	    }
 	    break;
 	}
-	pnode = search_rnode(*ppnode, code);
+	ppnode = search_rnode(ppnode, code);
+	pnode = *ppnode;
 	if (pnode == NULL)
 	{
 	    /* code‚ðŽ‚Âƒm[ƒh‚ª–³‚¢ê‡Aì¬’Ç‰Á‚·‚é */
 	    pnode = rnode_new();
 	    pnode->code = code;
-	    pnode->next = *ppnode;
+	    /*pnode->next = *ppnode;*/
 	    *ppnode = pnode;
 	}
 	else if (pnode->child == NULL)
