diff --git a/core/rpicam_app.cpp b/core/rpicam_app.cpp
index 0f0b5ff..aa416db 100644
--- a/core/rpicam_app.cpp
+++ b/core/rpicam_app.cpp
@@ -638,6 +638,7 @@ void RPiCamApp::StartCamera()
 	// This makes all the Request objects that we shall need.
 	makeRequests();
 
+#ifndef DISABLE_RPI_FEATURES
 	// Build a list of initial controls that we must set in the camera before starting it.
 	// We don't overwrite anything the application may have set before calling us.
 	if (!controls_.get(controls::ScalerCrop) && !controls_.get(controls::rpi::ScalerCrops))
@@ -673,6 +674,7 @@ void RPiCamApp::StartCamera()
 		else
 			controls_.set(controls::rpi::ScalerCrops, libcamera::Span<const Rectangle>(crops.data(), crops.size()));
 	}
+#endif
 
 	if (!controls_.get(controls::AfWindows) && !controls_.get(controls::AfMetering) &&
 		options_->Get().afWindow_width != 0 && options_->Get().afWindow_height != 0)
