diff --git a/src/slices.rs b/src/slices.rs
index c01c52a0..141f798a 100644
--- a/src/slices.rs
+++ b/src/slices.rs
@@ -1,6 +1,6 @@
 // Copyright (c) Facebook, Inc. and its affiliates.
 use anyhow::Result;
-use enum_iterator::IntoEnumIterator;
+use enum_iterator::Sequence;
 use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 use std::ops::{Index, IndexMut};
@@ -26,7 +26,7 @@ const SLICE_DOC: &str = "\
 //
 ";
 
-#[derive(Debug, Copy, Clone, IntoEnumIterator, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, Sequence, PartialEq, Eq)]
 pub enum Slice {
     Init = 0,
     Host = 1,
@@ -181,7 +181,7 @@ pub struct SliceKnobs {
 impl Default for SliceKnobs {
     fn default() -> Self {
         let mut slices = BTreeMap::new();
-        for slc in Slice::into_enum_iter() {
+        for slc in enum_iterator::all::<Slice>() {
             slices.insert(slc.name().into(), SliceConfig::default(slc));
         }
         Self {
diff --git a/src/sysreqs.rs b/src/sysreqs.rs
index c2c8128f..8d20f8e8 100644
--- a/src/sysreqs.rs
+++ b/src/sysreqs.rs
@@ -1,5 +1,5 @@
 // Copyright (c) Facebook, Inc. and its affiliates.
-use enum_iterator::IntoEnumIterator;
+use enum_iterator::Sequence;
 use log::warn;
 use serde::{Deserialize, Serialize};
 use std::collections::{BTreeMap, BTreeSet};
@@ -21,21 +21,11 @@ const SYSREQ_DOC: &str = "\
 ";
 
 lazy_static::lazy_static! {
-    pub static ref ALL_SYSREQS_SET: BTreeSet<SysReq> = SysReq::into_enum_iter().collect();
+    pub static ref ALL_SYSREQS_SET: BTreeSet<SysReq> = enum_iterator::all::<SysReq>().collect();
 }
 
 #[derive(
-    Debug,
-    Clone,
-    Copy,
-    PartialEq,
-    Eq,
-    PartialOrd,
-    Ord,
-    Hash,
-    IntoEnumIterator,
-    Serialize,
-    Deserialize,
+    Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Sequence, Serialize, Deserialize,
 )]
 pub enum SysReq {
     Controllers,
