--- read-process-memory-0.1.6/Cargo.toml	1970-01-01T00:00:01+00:00
+++ read-process-memory-0.1.6/Cargo.toml	2023-03-15T14:22:23.090868+00:00
@@ -27,17 +27,3 @@
 [dependencies.log]
 version = "0.4"
 
-[target."cfg(target_os=\"macos\")".dependencies.mach]
-version = "0.3.2"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "std",
-    "basetsd",
-    "minwindef",
-    "handleapi",
-    "memoryapi",
-    "processthreadsapi",
-    "winnt",
-]
