From 7e16011acc25f255ebad51097fd2204c01a78e5f Mon Sep 17 00:00:00 2001
From: Carl George <carlwgeorge@gmail.com>
Date: Fri, 28 Mar 2025 18:09:17 -0500
Subject: [PATCH] Makefile: install manpage without execute permission

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3511802..848b43d 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ $(TARGET): $(OBJS)
 install:
 	install -d $(DESTDIR)$(BINDIR)
 	install -m 0755 $(TARGET) $(DESTDIR)$(BINDIR)
-	install -m 0744 $(MANUAL) $(DESTDIR)$(MANDIR)/man1/
+	install -m 0644 $(MANUAL) $(DESTDIR)$(MANDIR)/man1/
 
 clean:
 	rm -f $(OBJS) $(TARGET)
-- 
2.49.0

