diff -up fprettify-0.3.7/fprettify/__init__.py.pyenv fprettify-0.3.7/fprettify/__init__.py
--- fprettify-0.3.7/fprettify/__init__.py.pyenv	2020-11-20 15:15:51.000000000 +0000
+++ fprettify-0.3.7/fprettify/__init__.py	2022-05-25 14:21:50.493442338 +0000
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 ###############################################################################
 #    This file is part of fprettify.
diff -up fprettify-0.3.7/fprettify.py.pyenv fprettify-0.3.7/fprettify.py
--- fprettify-0.3.7/fprettify.py.pyenv	2020-11-20 15:15:51.000000000 +0000
+++ fprettify-0.3.7/fprettify.py	2022-05-25 14:21:38.992645986 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 ###############################################################################
 #    This file is part of fprettify.
diff -up fprettify-0.3.7/fprettify/tests/__init__.py.pyenv fprettify-0.3.7/fprettify/tests/__init__.py
--- fprettify-0.3.7/fprettify/tests/__init__.py.pyenv	2020-11-20 15:15:51.000000000 +0000
+++ fprettify-0.3.7/fprettify/tests/__init__.py	2022-05-25 14:22:25.477822858 +0000
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 ###############################################################################
 #    This file is part of fprettify.
diff -up fprettify-0.3.7/run_tests.py.pyenv fprettify-0.3.7/run_tests.py
--- fprettify-0.3.7/run_tests.py.pyenv	2020-11-20 15:15:51.000000000 +0000
+++ fprettify-0.3.7/run_tests.py	2022-05-25 14:21:11.973124423 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 ###############################################################################
 #    This file is part of fprettify.
diff -up fprettify-0.3.7/setup.py.pyenv fprettify-0.3.7/setup.py
--- fprettify-0.3.7/setup.py.pyenv	2020-11-20 15:15:51.000000000 +0000
+++ fprettify-0.3.7/setup.py	2022-05-25 14:21:22.958929897 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 from setuptools import setup
 from codecs import open
 from os import path
