diff --git a/src/python/grpcio_tests/tests/health_check/_health_servicer_test.py b/src/python/grpcio_tests/tests/health_check/_health_servicer_test.py
index 867fd28..81195e3 100644
--- a/src/python/grpcio_tests/tests/health_check/_health_servicer_test.py
+++ b/src/python/grpcio_tests/tests/health_check/_health_servicer_test.py
@@ -265,7 +265,7 @@ class BaseWatchTests:
 class HealthServicerTest(BaseWatchTests.WatchTests):
     def setUp(self):
         self._thread_pool = thread_pool.RecordingThreadPool(max_workers=None)
-        super(HealthServicerTest, self).start_server(
+        super().start_server(
             non_blocking=True, thread_pool=self._thread_pool
         )
 
@@ -304,7 +304,7 @@ class HealthServicerTest(BaseWatchTests.WatchTests):
 
 class HealthServicerBackwardsCompatibleWatchTest(BaseWatchTests.WatchTests):
     def setUp(self):
-        super(HealthServicerBackwardsCompatibleWatchTest, self).start_server(
+        super().start_server(
             non_blocking=False, thread_pool=None
         )
 
diff --git a/src/python/grpcio_tests/tests/protoc_plugin/_python_plugin_test.py b/src/python/grpcio_tests/tests/protoc_plugin/_python_plugin_test.py
index 9640305..77005cd 100644
--- a/src/python/grpcio_tests/tests/protoc_plugin/_python_plugin_test.py
+++ b/src/python/grpcio_tests/tests/protoc_plugin/_python_plugin_test.py
@@ -585,7 +585,7 @@ class SimpleStubsPluginTest(unittest.TestCase):
             )
 
     def setUp(self):
-        super(SimpleStubsPluginTest, self).setUp()
+        super().setUp()
         self._server = test_common.test_server()
         service_pb2_grpc.add_TestServiceServicer_to_server(
             self.Servicer(), self._server
@@ -596,7 +596,7 @@ class SimpleStubsPluginTest(unittest.TestCase):
 
     def tearDown(self):
         self._server.stop(None)
-        super(SimpleStubsPluginTest, self).tearDown()
+        super().tearDown()
 
     def testUnaryCall(self):
         request = request_pb2.SimpleRequest(response_size=13)
