diff --git a/CMakeLists.txt b/CMakeLists.txt
index 046e2e6..c826ec2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22127,6 +22127,18 @@ target_include_directories(http2_status_test
     ${_gRPC_ZLIB_INCLUDE_DIR}
     third_party/googletest/googletest/include
     third_party/googletest/googletest
+
+if(gRPC_BUILD_TESTS)
+  get_property(grpc_all_targets DIRECTORY PROPERTY BUILDSYSTEM_TARGETS)
+  foreach(grpc_target IN LISTS grpc_all_targets)
+    if(grpc_target MATCHES "test|benchmark|bm_")
+      get_target_property(grpc_target_type ${grpc_target} TYPE)
+      if(grpc_target_type MATCHES "EXECUTABLE|LIBRARY")
+        target_compile_options(${grpc_target} PRIVATE -O0)
+      endif()
+    endif()
+  endforeach()
+endif()
     third_party/googletest/googlemock/include
     third_party/googletest/googlemock
     ${_gRPC_PROTO_GENS_DIR}
