diff --git a/src/python/grpcio_tests/tests/unit/_dns_resolver_test.py b/src/python/grpcio_tests/tests/unit/_dns_resolver_test.py
index 2f8ff58..91ccd3c 100644
--- a/src/python/grpcio_tests/tests/unit/_dns_resolver_test.py
+++ b/src/python/grpcio_tests/tests/unit/_dns_resolver_test.py
@@ -14,6 +14,7 @@
 """Tests for an actual dns resolution."""
 
 import logging
+import os
 import unittest
 
 import grpc
@@ -46,6 +47,8 @@ class DNSResolverTest(unittest.TestCase):
     def tearDown(self):
         self._server.stop(None)
 
+    @unittest.skipIf('FEDORA_NO_NETWORK_TESTS' in os.environ,
+                     'Network tests disabled')
     def test_connect_loopback(self):
         # NOTE(https://github.com/grpc/grpc/issues/18422)
         # In short, Gevent + C-Ares = Segfault. The C-Ares driver is not
