diff --git a/test/core/util/http_client/httpcli_test_util.cc b/test/core/util/http_client/httpcli_test_util.cc
index 217914f..5923e50 100644
--- a/test/core/util/http_client/httpcli_test_util.cc
+++ b/test/core/util/http_client/httpcli_test_util.cc
@@ -53,7 +53,7 @@ HttpRequestTestServer StartHttpRequestTestServer(int argc, char** argv,
       root = absl::StrCat(dirname, "/../../../..");
     } else {
       // Cmake paths will be "cmake/build".
-      root = absl::StrCat(dirname, "/../..");
+      root = absl::StrCat(dirname, "/..");
     }
   } else {
     root = ".";
