--- sequoia-chameleon-gnupg-0.13.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sequoia-chameleon-gnupg-0.13.1/Cargo.toml	2026-03-17T09:52:46.588208+00:00
@@ -50,18 +50,9 @@
 repository = "https://gitlab.com/sequoia-pgp/sequoia-chameleon-gnupg"
 
 [features]
-crypto-botan = ["sequoia-openpgp/crypto-botan"]
-crypto-botan2 = ["sequoia-openpgp/crypto-botan2"]
-crypto-cng = ["sequoia-openpgp/crypto-cng"]
 crypto-nettle = ["sequoia-openpgp/crypto-nettle"]
 crypto-openssl = ["sequoia-openpgp/crypto-openssl"]
-default = ["crypto-nettle"]
-the-recorder-steals-secrets-and-i-wont-package-it-in-slash-bin = []
-
-[[bin]]
-name = "gpg-recorder"
-path = "src/gpg-recorder.rs"
-required-features = ["the-recorder-steals-secrets-and-i-wont-package-it-in-slash-bin"]
+default = ["crypto-openssl"]
 
 [[bin]]
 name = "gpg-sq"
@@ -208,8 +199,7 @@
 features = ["backtrace"]
 
 [dev-dependencies.bzip2]
-version = "0.4"
-default-features = false
+version = ">=0.4,<0.7"
 
 [dev-dependencies.diff]
 version = "0.1"
@@ -228,7 +218,7 @@
 version = "0.9"
 
 [dev-dependencies.pty-process]
-version = "0.4"
+version = "0.5"
 default-features = false
 
 [dev-dependencies.regex]
