diff -rupN --no-dereference openambit-0.5/tools/openambit2gpx.py openambit-0.5-new/tools/openambit2gpx.py
--- openambit-0.5/tools/openambit2gpx.py	2019-11-12 22:27:23.000000000 +0100
+++ openambit-0.5-new/tools/openambit2gpx.py	2025-07-17 22:52:42.725731362 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 """ converts the *.log files produced by openambit in ~/.openambit/ to standard gpx format.
 usage: ./openambit2gpx.py inputfile outputFile
diff -rupN --no-dereference openambit-0.5/wireshark_dissector/tools/make-dissector-reg.py openambit-0.5-new/wireshark_dissector/tools/make-dissector-reg.py
--- openambit-0.5/wireshark_dissector/tools/make-dissector-reg.py	2019-11-12 22:27:23.000000000 +0100
+++ openambit-0.5-new/wireshark_dissector/tools/make-dissector-reg.py	2025-07-17 22:52:42.726023878 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
@@ -10,7 +10,7 @@
 # less processes would have to be started.
 #
 # $Id: make-dissector-reg.py 30447 2009-10-09 20:47:18Z krj $
-from __future__ import print_function
+
 
 import os
 import sys
@@ -119,7 +119,7 @@ if cache_filename:
 		cache_file = open(cache_filename, 'rb')
 		cache = pickle.load(cache_file)
 		cache_file.close()
-		if not cache.has_key(VERSION_KEY) or cache[VERSION_KEY] != CUR_VERSION:
+		if VERSION_KEY not in cache or cache[VERSION_KEY] != CUR_VERSION:
 			cache = {VERSION_KEY: CUR_VERSION}
 	except:
 		cache = {VERSION_KEY: CUR_VERSION}
@@ -128,7 +128,7 @@ if cache_filename:
 for filename in filenames:
 	file = open(filename)
 	cur_mtime = os.fstat(file.fileno())[ST_MTIME]
-	if cache and cache.has_key(filename):
+	if cache and filename in cache:
 		cdict = cache[filename]
 		if cur_mtime == cdict['mtime']:
 #			print("Pulling %s from cache" % (filename))
