--- io-lifetimes-3.0.1/Cargo.toml
+++ io-lifetimes-3.0.1/Cargo.toml
@@ -46,11 +46,7 @@
 features = ["close"]
 
 [features]
-close = [
-    "libc",
-    "hermit-abi",
-    "windows-sys",
-]
+close = ["hermit-abi"]
 default = []
 
 [lib]
@@ -91,25 +87,10 @@
 ]
 optional = true
 
-[target."cfg(not(windows))".dependencies.libc]
-version = "0.2.96"
-optional = true
-
 [target.'cfg(target_os = "hermit")'.dependencies.hermit-abi]
 version = ">=0.3, <=0.5"
 optional = true
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.52, <=0.60"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_Networking_WinSock",
-    "Win32_Security",
-    "Win32_System_IO",
-]
-optional = true
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
