diff -ruN inkscape-silhouette-1.29/silhouette/beutil.py inkscape-silhouette-1.29-mod/silhouette/beutil.py
--- inkscape-silhouette-1.29/silhouette/beutil.py	2024-09-12 23:18:56.000000000 +0300
+++ inkscape-silhouette-1.29-mod/silhouette/beutil.py	2025-07-27 15:49:18.072020385 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Conversion utilities for printer's binary encoding commands
 # (c) 2018 D. Bajar
diff -ruN inkscape-silhouette-1.29/silhouette/convert2dashes.py inkscape-silhouette-1.29-mod/silhouette/convert2dashes.py
--- inkscape-silhouette-1.29/silhouette/convert2dashes.py	2024-09-12 23:18:56.000000000 +0300
+++ inkscape-silhouette-1.29-mod/silhouette/convert2dashes.py	2025-07-27 15:49:06.080207580 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # coding=utf-8
 #
 # Extracted from inkscape extension; original comments below:
