--- tree-house-bindings-0.2.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ tree-house-bindings-0.2.1/Cargo.toml	2025-07-25T20:32:14.782427+00:00
@@ -23,7 +23,7 @@
 autobenches = false
 description = "Homey Rust bindings for the tree-sitter C library"
 readme = "README.md"
-license = "MPL-2.0"
+license = "MPL-2.0 AND MIT AND Unicode-DFS-2016"
 repository = "https://github.com/helix-editor/tree-house"
 
 [features]
