diff --git a/command.c b/command.c
index 99db955..ac9ba50 100644
--- a/command.c
+++ b/command.c
@@ -5,6 +5,7 @@
 #include <unistd.h>
 #include <time.h>
 #include <fcntl.h>
+#include <signal.h>
 #include <sys/utsname.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -29,8 +30,6 @@ struct cmd *Nodecmds = NULL;
 
 int meminfo(const char *s);
 
-int kill();
-
 void init_nodecmds(void)
 {
   add_internal_cmd(&Nodecmds, "?",        1, do_help);
