diff --git a/crates/hyfetch/Cargo.toml b/crates/hyfetch/Cargo.toml
index 6781017..90323ec 100644
--- a/crates/hyfetch/Cargo.toml
+++ b/crates/hyfetch/Cargo.toml
@@ -50,12 +50,6 @@ serde_json = { workspace = true, features = ["std"] }
 anyhow = { workspace = true, features = ["std"] }
 heck = "0.5.0"
 
-[target.'cfg(windows)'.dependencies]
-enable-ansi-support = { workspace = true, features = [] }
-normpath = { workspace = true, features = [] }
-same-file = { workspace = true, features = [] }
-crossterm = { workspace = true, features = ["windows"] }
-
 [features]
 default = ["autocomplete", "color", "macchina"]
 autocomplete = ["bpaf/autocomplete"]
