From 77c73c29a9ef92b6eee1260c1c12ccac3db7f661 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sun, 30 Jun 2024 04:11:08 -0400
Subject: [PATCH 3/3] Remove pytest-cov from test requirements

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 pyproject.toml | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 310a4a0..7aecfe1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -43,7 +43,6 @@ changelog = "https://github.com/pycountry/pycountry/blob/main/HISTORY.txt"
 mypy = ">1.7.1"
 pre-commit = ">3.4.0"
 pytest = ">7.1.3"
-pytest-cov = ">4.0.0"
 tox = ">4.11.4"
 zest-releaser = "^9.1.3"
 # For zest-releaser
@@ -51,9 +50,8 @@ docutils = "<0.21"
 
 [tool.pytest.ini_options]
 pythonpath = "src"
-addopts = "--doctest-glob='*.rst' --cov --cov-report=html --cov-report=term-missing"
+addopts = "--doctest-glob='*.rst'"
 doctest_optionflags = "NORMALIZE_WHITESPACE IGNORE_EXCEPTION_DETAIL"
-required_plugins = "pytest-cov>=4.0.0"
 
 [tool.ruff]
 # See https://docs.astral.sh/ruff/settings/ for documentation and other options
-- 
2.53.0

