--- frotz-d9676545/Makefile.orig	2025-01-27 19:04:47.000000000 -0500
+++ frotz-d9676545/Makefile	2025-01-29 15:32:54.278650829 -0500
@@ -375,7 +375,7 @@
 SDL_SOUND_PKGS = SDL2_mixer
 SDL_LDFLAGS += -lm
 
-SUBDIRS = $(COMMON_DIR) $(CURSES_DIR) $(X11_DIR) $(SDL_DIR) $(DUMB_DIR) $(BLORB_DIR) $(DOS_DIR) $(FONTS_DIR)
+SUBDIRS = $(COMMON_DIR) $(CURSES_DIR) $(X11_DIR) $(SDL_DIR) $(DUMB_DIR) $(BLORB_DIR) $(DOS_DIR)
 SUB_CLEAN = $(SUBDIRS:%=%-clean)
 
 FROTZ_BIN = frotz$(EXTENSION)
@@ -739,9 +739,6 @@
 	rm -f $(DESTDIR)$(BINDIR)/dfrotz
 	rm -f $(DESTDIR)$(MANDIR)/man6/dfrotz.6
 
-fonts_pcf:
-	$(MAKE) -C $(FONTS_DIR) pcf
-
 install_x11: install_xfrotz
 install-x11: install_xfrotz
 install-xfrotz: install_xfrotz
@@ -750,7 +747,6 @@
 	mkdir -p "$(DESTDIR)$(MANDIR)/man6" && test -w $(DESTDIR)$(MANDIR)/man6
 	install "$(XFROTZ_BIN)" "$(DESTDIR)$(PREFIX)/bin/"
 	install -m 644 doc/xfrotz.6 "$(DESTDIR)$(MANDIR)/man6/"
-	$(MAKE) -C $(FONTS_DIR) install_pcf
 
 uninstall_x11: uninstall_xfrotz
 uninstall-x11: uninstall_xfrotz
@@ -758,7 +754,6 @@
 uninstall_xfrotz:
 	rm -f "$(DESTDIR)$(PREFIX)/bin/xfrotz"
 	rm -f "$(DESTDIR)$(MANDIR)/man6/xfrotz.6"
-	$(MAKE) -C $(FONTS_DIR) uninstall_pcf
 
 install-sdl: install_sfrotz
 install_sdl: install_sfrotz
