--- libreoffice-26.2.0.1/include/vcl/filter/PDFiumLibrary.hxx~	2026-01-14 20:09:52.814590776 +0000
+++ libreoffice-26.2.0.1/include/vcl/filter/PDFiumLibrary.hxx	2026-01-14 20:10:07.285630397 +0000
@@ -11,6 +11,7 @@
 #pragma once
 
 #include <memory>
+#include <stdint.h>
 
 #include <com/sun/star/util/DateTime.hpp>
 
--- libreoffice-26.2.0.1/oox/inc/drawingml/textbodycontext.hxx~	2026-01-14 20:45:34.712221028 +0000
+++ libreoffice-26.2.0.1/oox/inc/drawingml/textbodycontext.hxx	2026-01-14 20:45:38.434319255 +0000
@@ -23,6 +23,7 @@
 #include <drawingml/textbody.hxx>
 #include <drawingml/textrun.hxx>
 #include <oox/core/contexthandler2.hxx>
+#include <stdint.h>
 
 namespace oox::drawingml {
 
--- libreoffice-26.2.0.1/oox/inc/drawingml/textparagraphpropertiescontext.hxx~	2026-01-14 21:17:49.730802524 +0000
+++ libreoffice-26.2.0.1/oox/inc/drawingml/textparagraphpropertiescontext.hxx	2026-01-14 21:17:51.924074095 +0000
@@ -21,6 +21,7 @@
 #define INCLUDED_OOX_DRAWINGML_TEXTPARAGRAPHPROPERTIESCONTEXT_HXX
 
 #include <vector>
+#include <stdint.h>
 
 #include <com/sun/star/style/TabStop.hpp>
 #include <drawingml/textparagraphproperties.hxx>
