diff -up quake3-1.36/code/qcommon/files.c~ quake3-1.36/code/qcommon/files.c
--- quake3-1.36/code/qcommon/files.c~	2011-07-28 10:55:29.000000000 +0200
+++ quake3-1.36/code/qcommon/files.c	2011-07-28 14:10:08.294992989 +0200
@@ -3268,11 +3268,10 @@ static void FS_CheckPak0( void )
 		
 		curpack = path->pack;
 
-		if(!Q_stricmpn( curpack->pakGamename, "demoq3", MAX_OSPATH )
-		   && !Q_stricmpn( pakBasename, "pak0", MAX_OSPATH ))
+		if(!Q_stricmpn( pakBasename, "pak0", MAX_OSPATH ) &&
+		   curpack->checksum == DEMO_PAK0_CHECKSUM)
 		{
-			if(curpack->checksum == DEMO_PAK0_CHECKSUM)
-				founddemo = qtrue;
+			founddemo = qtrue;
 		}
 
 		else if(!Q_stricmpn( curpack->pakGamename, BASEGAME, MAX_OSPATH )
