diff -up LuxCore-luxcorerender_v2.6/src/slg/CMakeLists.txt.system2 LuxCore-luxcorerender_v2.6/src/slg/CMakeLists.txt
--- LuxCore-luxcorerender_v2.6/src/slg/CMakeLists.txt.system2	2022-06-13 17:38:17.613625980 +0200
+++ LuxCore-luxcorerender_v2.6/src/slg/CMakeLists.txt	2022-06-13 17:40:16.007801268 +0200
@@ -115,14 +115,14 @@ include_directories(${LuxRays_SOURCE_DIR
 #	${LuxRays_SOURCE_DIR}/deps/openvdb-7.0.0/openvdb/util/Util.cc
 #)
 
-SOURCE_GROUP("Source Files\\OpenVDB Library" FILES ${OPENVDB_SRCS})
+#SOURCE_GROUP("Source Files\\OpenVDB Library" FILES ${OPENVDB_SRCS})
 
-include_directories(${LuxRays_SOURCE_DIR}/deps/openvdb-7.0.0)
+#include_directories(${LuxRays_SOURCE_DIR}/deps/openvdb-7.0.0)
 
 # Required by OpenVDB to read ABI 3
 #add_definitions("-D OPENVDB_USE_DEPRECATED_ABI -DOPENVDB_3_ABI_COMPATIBLE -DOPENVDB_STATICLIB -DOPENVDB_OPENEXR_STATICLIB")
-add_definitions("-DOPENVDB_USE_BLOSC -DOPENVDB_STATICLIB -DOPENVDB_OPENEXR_STATICLIB")
-add_library(openvdb STATIC ${OPENVDB_SRCS})
+#add_definitions("-DOPENVDB_USE_BLOSC -DOPENVDB_STATICLIB -DOPENVDB_OPENEXR_STATICLIB")
+#add_library(openvdb STATIC ${OPENVDB_SRCS})
 
 ###########################################################################
 #
@@ -179,9 +179,9 @@ add_library(openvdb STATIC ${OPENVDB_SRC
 #include_directories(${LuxRays_SOURCE_DIR}/deps/opensubdiv-3.4.0)
 
 # Required by VisualStudio to have M_PI defined
-add_definitions("-D_USE_MATH_DEFINES")
+#add_definitions("-D_USE_MATH_DEFINES")
 
-add_library(opensubdiv SHARED IMPORTED)
+#add_library(opensubdiv SHARED IMPORTED)
 
 ###########################################################################
 #
@@ -424,7 +424,7 @@ if(WIN32)
 	add_definitions("-DUSE_SSE")
 endif(WIN32)
 
-add_library(opencolorio SHARED IMPORTED)
+#add_library(opencolorio SHARED IMPORTED)
 
 if(WIN32)
 	# OCIO 2.0.0 is the only module not supporting -D_UNICODE
