From fb7400610cd32b12840951bb4685b1bab1c53a6f Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Fri, 30 Apr 2021 12:42:41 -0400
Subject: [PATCH 3/3] Fix a leaked cairo context in render_to_surface() in
 waveform.c

---
 src/waveform.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/waveform.c b/src/waveform.c
index d8d8d80..02444a2 100644
--- a/src/waveform.c
+++ b/src/waveform.c
@@ -912,6 +912,7 @@ render_to_surface (RENDER * render, SNDFILE *infile, SF_INFO *info, cairo_surfac
 			render_timeaxis (surface, render, info, LEFT_BORDER, width, TOP_BORDER, height) ;
 		} ;
 
+	cairo_destroy (cr) ;
 	return ;
 } /* render_to_surface */
 
-- 
2.30.2

