diff -ruN a/GNUmakefile b/GNUmakefile
--- a/GNUmakefile	2020-07-09 06:10:19.000000000 -0700
+++ b/GNUmakefile	2026-08-12 07:25:57.972537261 -0700
@@ -12,7 +12,7 @@
 #
 # G N U  M A K E  F U N C T I O N S
 #
-KNOWN_APIS = 5.1 5.2 5.3 5.4
+KNOWN_APIS = 5.1 5.2 5.3 5.4 5.5
 
 # template for invoking luapath script
 LUAPATH := $(d)/mk/luapath
@@ -42,6 +42,8 @@
 lua53path ?= $(datadir)/lua/5.3
 lua54cpath ?= $(libdir)/lua/5.4
 lua54path ?= $(datadir)/lua/5.4
+lua55cpath ?= $(libdir)/lua/5.5
+lua55path ?= $(datadir)/lua/5.5
 
 
 AR ?= ar
@@ -99,7 +101,7 @@
 
 # set LUA_APIS if empty or "?"
 ifeq ($(or $(strip $(LUA_APIS)),?),?)
-override LUA_APIS := $(call HAVE_API_FN,5.1) $(call HAVE_API_FN,5.2) $(call HAVE_API_FN,5.3) $(call HAVE_API_FN,5.4)
+override LUA_APIS := $(call HAVE_API_FN,5.1) $(call HAVE_API_FN,5.2) $(call HAVE_API_FN,5.3) $(call HAVE_API_FN,5.4) $(call HAVE_API_FN,5.5)
 endif
 
 define LUAXY_template
@@ -135,6 +137,7 @@
 $(eval $(call LUAXY_template,5.2))
 $(eval $(call LUAXY_template,5.3))
 $(eval $(call LUAXY_template,5.4))
+$(eval $(call LUAXY_template,5.5))
 
 #
 # A U T O D E T E C T  C O M P I L A T I O N  F L A G S
diff -ruN a/src/GNUmakefile b/src/GNUmakefile
--- a/src/GNUmakefile	2020-07-09 06:10:19.000000000 -0700
+++ b/src/GNUmakefile	2026-08-12 07:25:57.972635141 -0700
@@ -41,6 +41,7 @@
 
 ifneq ($(1), 5.3)
 ifneq ($(1), 5.4)
+ifneq ($(1), 5.5)
 $$(d)/$(1)/compat53.o: $$(d)/../vendor/compat53/c-api/compat-5.3.c $$(d)/../vendor/compat53/c-api/compat-5.3.h $$(d)/config.h
 	$$(MKDIR) -p $$(@D)
 	$$(CC) $$(CFLAGS_$(d)) $$(ALL_LUA$(subst .,,$(1))_CPPFLAGS) $$(CPPFLAGS_$(d)) -c -o $$@ $$<
@@ -50,6 +51,7 @@
 $$(d)/$(1)/openssl.so: $$(d)/$(1)/compat53.o
 endif
 endif
+endif
 
 .SECONDARY: liblua$(1)-openssl openssl$(1) openssl
 
@@ -61,14 +63,16 @@
 $(eval $(call BUILD_$(d),5.2))
 $(eval $(call BUILD_$(d),5.3))
 $(eval $(call BUILD_$(d),5.4))
+$(eval $(call BUILD_$(d),5.5))
 
 ifneq "$(filter $(abspath $(d)/..)/%, $(abspath $(firstword $(MAKEFILE_LIST))))" ""
-.SECONDARY: all all5.1 all5.2 all5.3 all5.4
+.SECONDARY: all all5.1 all5.2 all5.3 all5.4 all5.5
 
 all5.1: liblua5.1-openssl
 all5.2: liblua5.2-openssl
 all5.3: liblua5.3-openssl
 all5.4: liblua5.4-openssl
+all5.5: liblua5.5-openssl
 all: $(foreach API,$(strip $(LUA_APIS)),all$(API))
 
 endif
@@ -157,23 +161,26 @@
 $(eval $(call INSTALL_$(d),5.2,$$(lua52cpath),$$(lua52path)))
 $(eval $(call INSTALL_$(d),5.3,$$(lua53cpath),$$(lua53path)))
 $(eval $(call INSTALL_$(d),5.4,$$(lua54cpath),$$(lua54path)))
+$(eval $(call INSTALL_$(d),5.5,$$(lua55cpath),$$(lua55path)))
 
 ifneq "$(filter $(abspath $(d)/..)/%, $(abspath $(firstword $(MAKEFILE_LIST))))" ""
 
-.SECONDARY: install5.1 install5.2 install5.3 install5.4 install
+.SECONDARY: install5.1 install5.2 install5.3 install5.4 install5.5 install
 
 install5.1: liblua5.1-openssl-install
 install5.2: liblua5.2-openssl-install
 install5.3: liblua5.3-openssl-install
 install5.4: liblua5.4-openssl-install
+install5.5: liblua5.5-openssl-install
 install: $(foreach API,$(strip $(LUA_APIS)),install$(API))
 
-.PHONY: uninstall5.1 uninstall5.2 uninstall5.3 uninstall5.4 uninstall
+.PHONY: uninstall5.1 uninstall5.2 uninstall5.3 uninstall5.4 uninstall5.5 uninstall
 
 uninstall5.1: liblua5.1-openssl-uninstall
 uninstall5.2: liblua5.2-openssl-uninstall
 uninstall5.3: liblua5.3-openssl-uninstall
 uninstall5.4: liblua5.4-openssl-uninstall
+uninstall5.5: liblua5.5-openssl-uninstall
 uninstall: $(foreach API,$(strip $(LUA_APIS)),uninstall$(API))
 
 endif
