--- tokio-1.53.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ tokio-1.53.1/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -845,6 +845,11 @@
 [dependencies.pin-project-lite]
 version = "0.2.11"
 
+[dependencies.io-uring]
+version = "0.7.11"
+optional = true
+default-features = false
+
 [dependencies.tokio-macros]
 version = "~2.7.0"
 optional = true
@@ -968,9 +973,6 @@
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-optional = true
 
 [target."cfg(windows)".dev-dependencies.windows-sys]
 version = "0.61"
