--- a/t/13_TCP_ipvANY.t
+++ b/t/13_TCP_ipvANY.t
@@ -8,6 +8,7 @@
 use lib $Bin;
 use NetServerTest qw(prepare_test ok use_ok note skip_without_ipv6);
 skip_without_ipv6;
+require Net::Server::Proto;
 exit 0+!print "1..0 # SKIP Platform does not permit disabling V6ONLY\n" if !Net::Server::Proto->CAN_DISABLE_V6ONLY;
 my $IPv4 = "127.0.0.1"; # Should connect to IPv4
 my $IPv6 = "::1"; # Should also connect to IPv6
