diff -ruN disktype-9-orig/detect.c disktype-9-20140129/detect.c
--- disktype-9-orig/detect.c	2006-06-04 13:05:06.000000000 -0700
+++ disktype-9-20140129/detect.c	2014-01-15 16:08:12.000000000 -0800
@@ -50,6 +50,7 @@
 void detect_ntfs(SECTION *section, int level);
 void detect_hpfs(SECTION *section, int level);
 void detect_dos_loader(SECTION *section, int level);
+void detect_exfat(SECTION *section, int level);
 
 /* in cdrom.c */
 void detect_iso(SECTION *section, int level);
@@ -59,7 +60,10 @@
 void detect_udf(SECTION *section, int level);
 
 /* in linux.c */
-void detect_ext23(SECTION *section, int level);
+void detect_luks(SECTION * section, int level);
+void detect_f2fs(SECTION * section, int level);
+void detect_btrfs(SECTION * section, int level);
+void detect_ext234(SECTION *section, int level);
 void detect_reiser(SECTION *section, int level);
 void detect_reiser4(SECTION *section, int level);
 void detect_linux_raid(SECTION *section, int level);
@@ -131,12 +135,16 @@
   detect_amiga_fs,
   detect_apple_volume,
   detect_fat,
+  detect_exfat, 
   detect_ntfs,
   detect_hpfs,
   detect_udf,
   detect_cdrom_misc,
   detect_iso,
-  detect_ext23,
+  detect_luks,
+  detect_f2fs,
+  detect_btrfs,
+  detect_ext234,
   detect_reiser,
   detect_reiser4,
   detect_linux_raid,
diff -ruN disktype-9-orig/dos.c disktype-9-20140129/dos.c
--- disktype-9-orig/dos.c	2006-06-04 13:05:06.000000000 -0700
+++ disktype-9-20140129/dos.c	2014-01-15 16:11:57.000000000 -0800
@@ -606,4 +606,59 @@
     print_line(level, "Windows / MS-DOS boot loader");
 }
 
+/*
+ * exfat file system
+ */
+
+void detect_exfat(SECTION *section, int level)
+{
+  u4 sectsize, clustersize, allocatedpercent, versionminor, versionmajor;
+  u8 sectcount;
+  unsigned char *buf;
+  char s[256];
+
+  // retrieve the header of 1024 bytes
+  if (get_buffer(section, 0, 1024, (void **)&buf) < 1024)
+    return;
+
+  // see if the EXFAT string exists at position 3, if it doesn't then return
+  get_string(buf + 3, 7, s);
+  if (!s[0])
+    return;
+  if (memcmp(s, "EXFAT  ", 7) != 0)
+    return;
+
+  print_line(level, "exFAT file system");
+
+  /* sector size: must be a power of two */
+  sectsize = 1 << buf[0x6c];
+
+  //  printf("%u", sectsize);
+  if (sectsize < 512 || (sectsize & (sectsize - 1)))
+    return;
+  /* sectors per cluster: must be a power of two */
+  clustersize = (1 << buf[0x6d]) * sectsize;
+  if (clustersize == 0 || (clustersize & (clustersize - 1)))
+    return;
+
+  /* get size in sectors */
+  sectcount = get_le_quad(buf + 0x48);
+
+  /* get FS version minor and major */
+  versionminor = buf[0x68];
+  versionmajor = buf[0x69];
+  print_line(level +1, "Version %u.%u", versionmajor, versionminor);
+
+  ///* get allocation percentage */
+  //allocatedpercent = buf[0x70];
+  //print_line(level + 1, "Allocation Percentage: %s", allocatedpercent);
+
+  /* print volume size info */
+  format_blocky_size(s, sectcount, 512, "sectors", NULL);
+  print_line(level + 1, "Volume size %s", s);
+
+  return;
+}
+
+
 /* EOF */
diff -ruN disktype-9-orig/ewf.c disktype-9-20140129/ewf.c
--- disktype-9-orig/ewf.c	1969-12-31 16:00:00.000000000 -0800
+++ disktype-9-20140129/ewf.c	2014-01-31 04:30:50.191172685 -0800
@@ -0,0 +1,141 @@
+/*
+ * ewf.c
+ * Layered data source for EWF images via libewf.
+ * For use with disktype, Copyright (c) 2003 Christoph Pfisterer
+ *
+ * Copyright (c) 2007 David Loveall
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE. 
+ */
+
+#include "global.h"
+
+#ifdef USE_LIBEWF
+#include <libewf.h>
+
+/*
+ * types
+ */
+
+typedef struct ewf_source {
+  SOURCE c;
+  libewf_handle_t *e;
+  libewf_error_t **error;
+} EWF_SOURCE;
+
+/*
+ * helper functions
+ */
+
+static SOURCE *init_ewf_source(char * const filenames[], uint16_t file_amount);
+static int read_block_ewf(SOURCE *s, u8 pos, void *buf);
+static void close_ewf(SOURCE *s);
+
+/*
+ * ewf analyzer
+ */
+
+void analyze_ewf(char * const filenames[], uint16_t file_amount)
+{
+  SOURCE *s;
+
+  /* create and analyze wrapped source */
+  s = init_ewf_source(filenames, file_amount);
+  analyze_source(s, 1);
+  close_source(s);
+}
+
+/*
+ * initialize the ewf source
+ */
+
+static SOURCE *init_ewf_source(char * const filenames[], uint16_t file_amount)
+{
+  EWF_SOURCE *src;
+
+  src = (EWF_SOURCE *)malloc(sizeof(EWF_SOURCE));
+  if (src == NULL)
+    bailout("Out of memory");
+  memset(src, 0, sizeof(EWF_SOURCE));
+
+  src->error = NULL;
+
+  if( libewf_handle_initialize(&src->e, src->error) != 1) {
+    bailout("unable to initialize handle");
+  }
+
+
+  if (libewf_handle_open(src->e, filenames, file_amount, LIBEWF_OPEN_READ, src->error) != 1)
+    bailout("Can't open EWF file");
+
+  src->c.size_known = 1;
+
+#if defined( LIBEWF_STRING_DIGEST_HASH_LENGTH_MD5 )
+  libewf_handle_get_media_size(src->e, src->c.size, src->error);
+  libewf_handle_get_bytes_per_sector(src->e, src->c.blocksize, src->error);
+#else
+  if( libewf_handle_get_media_size( src->e, (uint64_t *) &( src->c.size ), src->error ) != 1 )
+    bailout("Unable to get media size of EWF file");
+  if( libewf_handle_get_bytes_per_sector( src->e, (uint32_t *) &(src->c.blocksize), src->error) != 1)
+    bailout("Unable to get sector size of EWF file");
+#endif
+  src->c.read_block = read_block_ewf;
+  src->c.close = close_ewf;
+
+  return (SOURCE *)src;
+}
+
+/*
+ * raw read
+ */
+
+static int read_block_ewf(SOURCE *s, u8 pos, void *buf)
+{
+  EWF_SOURCE *es = (EWF_SOURCE *)s;
+  
+  /* read from lower layer */
+  if(libewf_handle_read_random(es->e, buf, es->c.blocksize, pos, NULL) == -1)  {
+    return 0;
+  }
+
+  return 1;
+}
+
+/*
+ * cleanup
+ */
+
+static void close_ewf(SOURCE *s)
+{
+  EWF_SOURCE *es = (EWF_SOURCE *)s;
+
+/* free(es); */
+
+  if( libewf_handle_close( es->e, NULL ) != 0 )
+    bailout("Unable to close handle");
+
+  if( libewf_handle_free( &es->e, NULL) != 1 )
+    bailout("Unable to free handle");
+
+}
+
+#endif
+/* EOF */
diff -ruN disktype-9-orig/HISTORY disktype-9-20140129/HISTORY
--- disktype-9-orig/HISTORY	2006-06-05 00:57:51.000000000 -0700
+++ disktype-9-20140129/HISTORY	2014-01-29 11:10:40.000000000 -0800
@@ -103,5 +103,10 @@
 Improved other structures: Amiga "Rigid Disk" partitioning, LILO,
   ISO9660 El Torito.
 
+Version 9-14 contains the following improvements:
+Added file systems: btrfs, ext4, f2fs, exfat
+Added LUKS encryption container detection/identification
 
+Version 9-15 contains the following improvements:
+Updated libewf function calls in ewf.c
 EOF
diff -ruN disktype-9-orig/linux.c disktype-9-20140129/linux.c
--- disktype-9-orig/linux.c	2006-06-04 13:05:06.000000000 -0700
+++ disktype-9-20140129/linux.c	2014-01-15 15:57:25.000000000 -0800
@@ -28,10 +28,111 @@
 #include "global.h"
 
 /*
- * ext2/ext3 file system
+ * luks encryption detection
  */
 
-void detect_ext23(SECTION *section, int level)
+void detect_luks(SECTION *section, int level)
+{
+  unsigned char *buf;
+  char s[256];
+
+  if (get_buffer(section, 0x0, 1024, (void **)&buf) < 1024)
+    return ;
+
+  if (memcmp(buf, "LUKS\xBA\xBE", 6) == 0) {
+    print_line(level, "LUKS Encryption Detected");
+
+    get_string(buf + 0xa8, 36, s);
+    print_line(level + 1, "luksUUID: %s", s);
+
+    get_string(buf + 0x28, 24, s);
+    print_line(level + 1, "type: %s", s);
+  }
+
+}
+
+/*
+ * f2fs file system
+ */
+
+void detect_f2fs(SECTION *section, int level)
+{
+  unsigned char *buf;
+  char s[256];
+  u4 log_sectorsize, log_sectors_per_block, log_blocksize, log_blocks_per_seg, segs_per_sec, secs_per_zone, checksum_offset;;
+  u8 block_count;
+  u4 sector_size;
+
+// retrieve the header of 65536 bytes
+  if (get_buffer(section, 0x1000, 1024, (void **)&buf) < 1024)
+    return;
+
+  /* signature */
+  if (get_le_long(buf) != 0xF2F52010)
+    return;
+
+  print_line(level, "f2fs file system");
+
+  // label information
+  format_utf16_le(buf + 0x8c, 512, s);
+  if (s[0])
+    print_line(level + 1, "Volume name \"%s\"", s);
+}
+
+/*
+ * btrfs file system
+ */
+
+void detect_btrfs(SECTION *section, int level)
+{
+  unsigned char *buf;
+  char s[256];
+  u4 sectorsize, nodesize, leafsize, stripesize;
+  u8 blockcount, totalbytes, bytesused;
+
+  // retrieve the header of 65536 bytes
+  if (get_buffer(section, 64 * 1024, 1024, (void **)&buf) < 1024)
+    return;
+
+  // get the 8 character string that is 64 bytes into the header
+  get_string(buf + 64, 8, s);
+  if (!s[0])
+    return;
+  if (memcmp(s, "_BHRfS_M", 8) != 0)
+    return;
+
+  print_line(level, "Btrfs file system");
+
+  // label information
+  get_string(buf + 299, 256, s);
+  if (s[0])
+    print_line(level + 1, "Volume name \"%s\"", s);
+
+  // guid information
+  format_guid(buf + 32, s);
+  print_line(level + 1, "fsid %s", s);
+
+  /* collect size information */
+  totalbytes = get_le_quad(buf + 0x70);
+  bytesused = get_le_quad(buf + 0x78);
+  sectorsize = get_le_long(buf + 0x90);
+  nodesize = get_le_long(buf + 0x94);
+  leafsize = get_le_long(buf + 0x98);
+  stripesize = get_le_long(buf + 0x9c);
+
+  /* print the size information */
+  print_line(level +1, "Bytes in use: %u; node size: %u; leaf size: %u; stripe size: %u", bytesused, nodesize, leafsize, stripesize);
+
+  format_blocky_size(s, totalbytes/sectorsize, sectorsize, "blocks", NULL);
+  print_line(level + 1, "Volume size: %s", s);
+}
+
+/*
+ * ext2/ext3/ext4 file system
+ */
+
+
+void detect_ext234(SECTION *section, int level)
 {
   unsigned char *buf;
   char s[256];
@@ -44,7 +145,9 @@
   if (get_le_short(buf + 56) == 0xEF53) {
     if (get_le_long(buf + 96) & 0x0008)       /* JOURNAL_DEV flag */
       print_line(level, "Ext3 external journal");
-    else if (get_le_long(buf + 92) & 0x0004)  /* HAS_JOURNAL flag */
+    else if (get_le_long(buf + 96) > 0x003f || get_le_long(buf + 100) > 7 )
+      print_line(level, "Ext4 file system");
+    else if (get_le_long(buf + 92) & 0x0004 && ((get_le_long(buf + 100) < 8)))  /* HAS_JOURNAL flag */
       print_line(level, "Ext3 file system");
     else
       print_line(level, "Ext2 file system");
diff -ruN disktype-9-orig/main.c disktype-9-20140129/main.c
--- disktype-9-orig/main.c	2003-05-24 10:35:44.000000000 -0700
+++ disktype-9-20140129/main.c	2014-01-29 08:16:28.000000000 -0800
@@ -27,6 +27,10 @@
 
 #include "global.h"
 
+#ifdef USE_LIBEWF
+#include <libewf.h>
+#endif
+
 #ifdef USE_MACOS_TYPE
 #include <CoreServices/CoreServices.h>
 #endif
@@ -38,6 +42,10 @@
 static void analyze_file(const char *filename);
 static void print_kind(int filekind, u8 size, int size_known);
 
+#ifdef USE_LIBEWF
+void analyze_ewf(char * const filenames[], uint16_t file_amount);
+#endif
+
 #ifdef USE_MACOS_TYPE
 static void show_macos_type(const char *filename);
 #endif
@@ -59,7 +67,19 @@
   /* loop over filenames */
   print_line(0, "");
   for (i = 1; i < argc; i++) {
+#ifndef USE_LIBEWF
     analyze_file(argv[i]);
+#endif
+#ifdef USE_LIBEWF
+    if (libewf_check_file_signature(argv[i],NULL) == 1) {
+      print_line(0, "EWF image");
+      analyze_ewf(&argv[i], argc-i);
+      return 0;
+    }
+    else {
+      analyze_file(argv[i]);
+    }
+#endif
     print_line(0, "");
   }
 
diff -ruN disktype-9-orig/Makefile disktype-9-20140129/Makefile
--- disktype-9-orig/Makefile	2006-01-12 09:55:15.000000000 -0800
+++ disktype-9-20140129/Makefile	2014-01-29 08:21:17.000000000 -0800
@@ -9,12 +9,12 @@
          buffer.o file.o cdaccess.o cdimage.o vpc.o compressed.o \
          detect.o apple.o amiga.o atari.o dos.o cdrom.o \
          linux.o unix.o beos.o archives.o \
-         udf.o blank.o cloop.o
+         udf.o blank.o cloop.o ewf.o
 
 TARGET = disktype
 
 CPPFLAGS = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS   = -Wall
+CFLAGS   = $(RPM_OPT_FLAGS)
 LDFLAGS  =
 LIBS     =
 
@@ -43,6 +43,12 @@
   endif
 endif
 
+
+ifneq ($(LIBEWF),)
+  CPPFLAGS += -DUSE_LIBEWF
+  LIBS     += -lewf
+endif
+
 # real making
 
 all: $(TARGET)
diff -ruN disktype-9-orig/Seedfile disktype-9-20140129/Seedfile
--- disktype-9-orig/Seedfile	2006-01-12 09:55:15.000000000 -0800
+++ disktype-9-20140129/Seedfile	2014-01-29 08:21:35.000000000 -0800
@@ -7,7 +7,7 @@
          buffer.c file.c cdaccess.c cdimage.c vpc.c compressed.c
          detect.c apple.c amiga.c atari.c dos.c cdrom.c
          linux.c unix.c beos.c archives.c
-         udf.c blank.c cloop.c;
+         udf.c blank.c cloop.c ewf.c;
 
   cflags "-D_LARGEFILE_SOURCE" "-D_FILE_OFFSET_BITS=64";
 }
