--- ghc-9.8.0.20230727/libraries/Cabal/Cabal/src/Distribution/Simple/Install.hs~	2023-07-27 21:34:28.000000000 +0100
+++ ghc-9.8.0.20230727/libraries/Cabal/Cabal/src/Distribution/Simple/Install.hs	2023-08-04 14:47:19.782822654 +0100
@@ -269,9 +269,8 @@
   when (not inPath) $
     warn
       verbosity
-      ( "The directory "
+      ( "Executable installed in "
           ++ binPref
-          ++ " is not in the system search path."
       )
   case compilerFlavor (compiler lbi) of
     GHC -> GHC.installExe verbosity lbi binPref buildPref progFix pkg_descr exe
