--- extrema-4.4.5/src/Graphics/GRA_thiessenTriangulation.cpp.orig	2011-01-31 20:54:31.951467491 +0100
+++ extrema-4.4.5/src/Graphics/GRA_thiessenTriangulation.cpp	2011-01-31 20:55:54.879463653 +0100
@@ -15,6 +15,7 @@
 along with this program; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
+#include <cstddef>
 #include <cmath>
 
 #include "GRA_thiessenTriangulation.h"
