--- lib/callable/src/perl/Main.cc.orig	2022-01-14 03:46:51.000000000 -0700
+++ lib/callable/src/perl/Main.cc	2022-03-05 21:24:12.840080722 -0700
@@ -110,6 +110,8 @@ void deduce_install_dirs(std::string lib
 {
    const char stem_name[]="libpolymake";
    constexpr size_t stem_size=sizeof(stem_name)-1;
+   if (lib_path.rfind("/lib", 0) == 0)
+      lib_path = "/usr" + lib_path;
    auto basename_start=lib_path.rfind(stem_name);
    if (basename_start == std::string::npos)
       throw std::runtime_error("polymake::Main - weird callable library path " + lib_path);
