diff -up normalize-0.7.7/configure.ac.orig normalize-0.7.7/configure.ac
--- normalize-0.7.7/configure.ac.orig	2005-09-14 09:23:16.000000000 +0200
+++ normalize-0.7.7/configure.ac	2022-07-21 18:25:44.190984478 +0200
@@ -221,9 +221,6 @@ AC_ARG_ENABLE(lookup-table,
 
 dnl *** Stuff for xmms plugin ***
 have_xmms=true
-AM_PATH_GLIB(1.2.2, , [ have_xmms=false ])
-AM_PATH_GTK(1.2.2, , [ have_xmms=false ], gthread)
-AM_PATH_XMMS(1.0.0, , [ have_xmms=false ])
 AM_DISABLE_STATIC
 AM_PROG_LIBTOOL
 AC_ARG_ENABLE(xmms,
