--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2065,7 +2065,7 @@
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/jconfig.h
   ${CMAKE_CURRENT_SOURCE_DIR}/src/jerror.h
   ${CMAKE_CURRENT_SOURCE_DIR}/src/jmorecfg.h
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/jpeglib.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/jpeglib.h ${CMAKE_CURRENT_SOURCE_DIR}/src/jpegint.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT include)
 
 include(cmakescripts/BuildPackages.cmake)
