From 6c1eb15bc53597a4fc290ce661c89ae91a0f2c09 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:47 +0200
Subject: [PATCH 11/41] From Debian: 107_fix_xsynchronize.diff

---
 xlib/init.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/xlib/init.c b/xlib/init.c
index 17722a8..cff33bf 100644
--- a/xlib/init.c
+++ b/xlib/init.c
@@ -386,6 +386,9 @@ initialize (char **argv, int argc)
       printf ("could not open display\n");
       exit (0);
     }
+
+  if (Sync)
+     XSynchronize (dp, 1);
   XSetCloseDownMode (dp, DestroyAll);
   screen = DefaultScreen (dp);
   fadeenable = 1;
@@ -578,8 +581,6 @@ initialize (char **argv, int argc)
     }
 #endif
 
-  if (Sync)
-    XSynchronize (dp, 1);
 #if defined(NAS_SOUND)||defined(RSOUND)
   printf ("Initializing sound server...\n");
   if (sndinit)
@@ -666,7 +667,7 @@ main (int argc, char **argv)
 	  "\n\n\n\n"
 	  "                     Copyright(c) Jan Hubicka 1995, 1996\n\n\n");
   useprivate = 0;
-  while ((c = mygetopt (argc, argv, "KWD:P:L:C:SxslEMmpdhfb")) != -1)
+  while ((c = mygetopt (argc, argv, "KWD:P:L:C:SxyslEMmpdhfb")) != -1)
     {
       switch (c)
 	{
-- 
2.5.0

