From 4925aff0536657d18178b0b2cfc9de183795d2dd Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:48 +0200
Subject: [PATCH 16/41] DEFAULTINITPORT is not defined if building without
 NETSUPPORT

---
 xlib/init.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/xlib/init.c b/xlib/init.c
index e35a2e4..3950319 100644
--- a/xlib/init.c
+++ b/xlib/init.c
@@ -787,7 +787,6 @@ main (int argc, char **argv)
 #endif
 	  break;
 	default:
-#ifdef NETSUPPORT
 #define USAGE_NETSUPPORT " -S run koules as network server\n \
 -C<host> run koules as network client\n \
 -P<port> select port. Default is:%i\n \
@@ -800,9 +799,6 @@ main (int argc, char **argv)
     3: easy\n \
     4: very easy\n \
 -K run server in deathmatch mode\n "
-#else
-#define USAGE_NETSUPPORT ""
-#endif
 #ifdef SOUND
 #define USAGE_SOUND " -d Disable sound support\n"
 #else
@@ -825,7 +821,10 @@ main (int argc, char **argv)
 		  USAGE_SOUND
 		  " -x Disable X11 pointer\n"
 		  USAGE_MITSHM
-		  USAGE_NETSUPPORT, DEFAULTINITPORT
+#ifdef NETSUPPORT
+		  USAGE_NETSUPPORT,
+		  DEFAULTINITPORT
+#endif
 	    );
 	  exit (2);
 	}
-- 
2.5.0

