From a6fa9324f8baed19d8fab2b02944d33cfb691428 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:50 +0200
Subject: [PATCH 19/41] Fix banner placement with OS/2

It would overlay otherwise.
---
 menu.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/menu.c b/menu.c
index e871385..659c168 100644
--- a/menu.c
+++ b/menu.c
@@ -645,15 +645,15 @@ draw_menu (CONST int draw)
       DrawBlackMaskedText (MAPWIDTH / 2 - 10 * 4 + 1, 11, "K O U L E S");
       DrawBlackMaskedText (MAPWIDTH / 2 - 2 * 4 + 1, 21, "BY");
       DrawBlackMaskedText (MAPWIDTH / 2 - 8 * 4 + 1, 31, "JAHUSOFT");
-      DrawBlackMaskedText (MAPWIDTH / 2 - 30 * 4 + 1, 46, "PRESS P FOR PAUSE / H FOR HELP");
+      DrawBlackMaskedText (MAPWIDTH / 2 - 30 * 4 + 1, 56, "PRESS P FOR PAUSE / H FOR HELP");
 #ifdef __OS2__
-      DrawBlackMaskedText (MAPWIDTH / 2 - 28 * 4, 40, "OS/2 Warp VERSION BY T.A.K.K");
+      DrawBlackMaskedText (MAPWIDTH / 2 - 28 * 4, 41, "OS/2 Warp VERSION BY T.A.K.K");
 #endif
       DrawWhiteMaskedText (MAPWIDTH / 2 - 8 * 4, 0, "THE GAME");
       DrawWhiteMaskedText (MAPWIDTH / 2 - 10 * 4, 10, "K O U L E S");
       DrawWhiteMaskedText (MAPWIDTH / 2 - 2 * 4, 20, "BY");
       DrawWhiteMaskedText (MAPWIDTH / 2 - 8 * 4, 30, "JAHUSOFT");
-      DrawWhiteMaskedText (MAPWIDTH / 2 - 30 * 4, 45, "PRESS P FOR PAUSE / H FOR HELP");
+      DrawWhiteMaskedText (MAPWIDTH / 2 - 30 * 4, 55, "PRESS P FOR PAUSE / H FOR HELP");
 #ifdef __OS2__
       DrawWhiteMaskedText (MAPWIDTH / 2 - 28 * 4, 40, "OS/2 Warp VERSION BY T.A.K.K");
 #endif
-- 
2.5.0

