From 024fc2a18e54bd9ae3be29ade94c785ed4fe05cb Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 9 Sep 2013 19:17:42 +0200
Subject: [PATCH 20/41] Fix rocketcolor signedness

---
 koules.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/koules.h b/koules.h
index c252896..d188141 100644
--- a/koules.h
+++ b/koules.h
@@ -284,7 +284,7 @@ extern BitmapType bball_bitmap, apple_bitmap, inspector_bitmap, mouse_bitmap,
                 lunatic_bitmap, lball_bitmap[NLETTERS], circle_bitmap,
                 hole_bitmap, ehole_bitmap, ball_bitmap, eye_bitmap[MAXROCKETS],
                 rocket_bitmap[MAXROCKETS];
-extern char     rocketcolor[5];
+extern unsigned char rocketcolor[5];
 
 
 #endif
-- 
2.5.0

