From 70636465d0808b6fce0be4ad624fc485e3a0c430 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:45 +0200
Subject: [PATCH 21/41] Fix background color calculation

---
 framebuffer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/framebuffer.c b/framebuffer.c
index 6a3b44f..341cff6 100644
--- a/framebuffer.c
+++ b/framebuffer.c
@@ -391,7 +391,7 @@ createbackground ()
 #ifndef NODIRECT
 	    i += *(pixel - 1) - back (0);
 #else
-	    i += SGetPixel (x - 1, y);
+	    i += SGetPixel (x - 1, y) - back (0);
 #endif
 	    n++;
 	  }
@@ -400,7 +400,7 @@ createbackground ()
 #ifndef NODIRECT
 	    i += *(pixel - MAPWIDTH) - back (0);
 #else
-	    i += SGetPixel (x, y - 1);
+	    i += SGetPixel (x, y - 1) - back (0);
 #endif
 	    n++;
 	  }
-- 
2.5.0

