From 10404c8c9b703f2c4aa3c8a3bc8b92aeac3e0d8d Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:49 +0200
Subject: [PATCH 22/41] Fix string quoting

---
 xlib/inlstring.h | 43 ++++++++++++++++++++-----------------------
 1 file changed, 20 insertions(+), 23 deletions(-)

diff --git a/xlib/inlstring.h b/xlib/inlstring.h
index 7218cf7..46a68f4 100644
--- a/xlib/inlstring.h
+++ b/xlib/inlstring.h
@@ -238,13 +238,12 @@ extern          __memcpyasm_regargs_aligned ();		/* nu_bytes >= 32 */
 static INLINE void *
 __memcpy_aligndest (void *dest, const void *src, int n)
 {
-  __asm__         __volatile__ ("
-				cmpl $3, %%ecx
-				ja 1f
-				call * __memcpy_jumptable (, %%ecx, 4)
-				jmp 2f
-				1:call __memcpyasm_regargs
-				"
+  __asm__         __volatile__ (
+				"cmpl $3, %%ecx\n\t"
+				"ja 1f\n\t"
+				"call * __memcpy_jumptable (, %%ecx, 4)\n\t"
+				"jmp 2f\n\t"
+				"1:call __memcpyasm_regargs\n\t"
 				:
 				:"b"            (dest), "d" (src), "c" (n)
 				:"ax", "0", "1", "2");
@@ -255,14 +254,13 @@ __memcpy_aligndest (void *dest, const void *src, int n)
 static INLINE void *
 __memcpy_destaligned (void *dest, const void *src, int n)
 {
-  __asm__         __volatile__ ("
-				cmpl $32, %%ecx
-				ja 1f
-				call * __memcpy_jumptable (, %%ecx, 4)
-				jmp 2f
-				1:call __memcpyasm_regargs_aligned
-				2:
-				"
+  __asm__         __volatile__ (
+				"cmpl $32, %%ecx\n\t"
+				"ja 1f\n\t"
+				"call * __memcpy_jumptable (, %%ecx, 4)\n\t"
+				"jmp 2f\n\t"
+				"1:call __memcpyasm_regargs_aligned\n\t"
+				"2:\n\t"
 				:
 				:"b"            (dest), "d" (src), "c" (n)
 				:"ax", "0", "1", "2");
@@ -273,14 +271,13 @@ __memcpy_destaligned (void *dest, const void *src, int n)
 static INLINE void *
 __memcpy_balanced (void *dest, const void *src, int n)
 {
-  __asm__         __volatile__ ("
-				cmpl $19, %%ecx
-				ja 1f
-				call * __memcpy_jumptable (, %%ecx, 4)
-				jmp 2f
-				1:call __memcpyasm_regargs
-				2:
-				"
+  __asm__         __volatile__ (
+				"cmpl $19, %%ecx\n\t"
+				"ja 1f\n\t"
+				"call * __memcpy_jumptable (, %%ecx, 4)\n\t"
+				"jmp 2f\n\t"
+				"1:call __memcpyasm_regargs\n\t"
+				"2:\n\t"
 				:
 	     :"b"            ((long) dest), "d" ((long) src), "c" ((long) n)
 				:"ax", "bx", "cx", "dx");
-- 
2.5.0

