From 41ff7475d138c8f84428ed972d84f2d932849dd4 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sun, 8 Nov 2015 18:14:10 +0100
Subject: [PATCH 34/41] Make compiler bounds-checking happy

---
 server.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/server.c b/server.c
index 3e8be27..93b6e04 100644
--- a/server.c
+++ b/server.c
@@ -510,6 +510,7 @@ ssendbuffer (int client)
 	{
 	  perror ("Socket error");
 	  Quit ("Server error:can not send message to client ");
+	  return;
 	}
     }
   packets++;
@@ -550,6 +551,12 @@ ssendreliable (int client, unsigned char *message, int size)
   if (conn[client].rpos + size + 6 > BUFFERSIZE)
     {
       Quit ("Connection to client broken too long! Reliable buffer owerfllow\n");
+      return;
+    }
+  if (message[0] >= MAXPACKET)
+    {
+      Quit ("Bad packet\n");
+      return;
     }
 
   PUTLONG ((conn[client].rbuffer + conn[client].rpos), conn[client].rcount);
@@ -842,6 +849,7 @@ client_message (struct conn *c)
       perror ("Socket error");
       GetSocketError (c->socket);
       Quit ("Connection to client broken-exiting game\n");
+      return;
     }
   isize += bytes;
   GETLONG (ibuffer, scount);
-- 
2.5.0

