--- manedit-1.1.1/manedit/ca_string.c.compile	2008-06-17 15:00:00.000000000 +0900
+++ manedit-1.1.1/manedit/ca_string.c	2008-06-30 15:31:48.000000000 +0900
@@ -1,4 +1,5 @@
 #include <glib.h>
+#include <string.h>
 #include "ca_string.h"
 
 
--- manedit-1.1.1/manedit/editor_find_replace.c.compile	2008-06-17 15:00:00.000000000 +0900
+++ manedit-1.1.1/manedit/editor_find_replace.c	2008-06-30 15:34:25.000000000 +0900
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
+
 #include <string.h>
 #include <gtk/gtk.h>
 
--- manedit-1.1.1/manedit/manpage_view.c.compile	2008-06-17 15:00:00.000000000 +0900
+++ manedit-1.1.1/manedit/manpage_view.c	2008-06-30 15:30:10.000000000 +0900
@@ -1,4 +1,5 @@
 #include <gtk/gtk.h>
+#include <string.h>
 
 #include "guiutils.h"
 
