diff -rU4 lazpaint-7.3--orig/lazpaint/ugraph.pas lazpaint-7.3--patched/lazpaint/ugraph.pas
--- lazpaint-7.3--orig/lazpaint/ugraph.pas	2025-05-24 22:04:28.000000000 +0200
+++ lazpaint-7.3--patched/lazpaint/ugraph.pas	2025-08-15 12:46:51.591556532 +0200
@@ -1200,15 +1200,17 @@
 function NiceFrameBounds(ACanvasScale: integer; APoints: array of TPointF): TRect;
 var
   rF: TRectF;
   w: Extended;
-  pt: TPointF;
+  pt, topleft, bottomright: TPointF;
 begin
   w := FramePenWidth*ACanvasScale/10 / 2 + 1;
   rF := EmptyRectF;
-  for pt in APoints do
-    rF := rF.Union(RectF(pt + PointF(0.5, 0.5) - PointF(w, w),
-      pt + PointF(0.5, 0.5) + PointF(w, w)));
+  for pt in APoints do begin
+    topleft := pt + PointF(0.5, 0.5) - PointF(w, w);
+    bottomright := pt + PointF(0.5, 0.5) + PointF(w, w);
+    rF := rF.Union(RectF(topleft.x, topleft.y, bottomright.x, bottomright.y));
+  end;
   result := rect(floor(rF.Left), floor(rF.Top),
     ceil(rF.Right), ceil(rF.Bottom));
 end;
 
