diff --color -urN collada-dom-2.5.0/CMakeLists.txt collada-dom-2.5.0-cmake-version/CMakeLists.txt
--- collada-dom-2.5.0/CMakeLists.txt	2017-10-21 07:05:24.000000000 +0200
+++ collada-dom-2.5.0-cmake-version/CMakeLists.txt	2025-10-19 10:26:49.633304059 +0200
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.6.0)
+cmake_minimum_required (VERSION 3.5...4.0)
 project (collada-dom)
 set( CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS TRUE )
 
