diff --color -urN collada-dom-2.5.0/CMakeLists.txt collada-dom-2.5.0-cmake-drop-vars/CMakeLists.txt
--- collada-dom-2.5.0/CMakeLists.txt	2017-10-21 07:05:24.000000000 +0200
+++ collada-dom-2.5.0-cmake-drop-vars/CMakeLists.txt	2025-10-19 10:42:55.608316411 +0200
@@ -17,6 +17,8 @@
 # http://www.cmake.org/cmake/help/cmake-2.6.html#policy:CMP0003
 cmake_policy(SET CMP0003 NEW)
 
+include(GNUInstallDirs)
+
 # Use, i.e. don't skip the full RPATH for the build tree
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 
@@ -24,6 +26,9 @@
 # (but later on when installing)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 
+if(CMAKE_INSTALL_LIBDIR MATCHES "lib64$")
+  set(LIB_SUFFIX "64")
+endif()
 # The RPATH to be used when installing
 set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 
