From 6c1ed7a711b182b67b6220feee795606f14a576d Mon Sep 17 00:00:00 2001
From: serge-sans-paille <serge.guelton@telecom-bretagne.eu>
Date: Thu, 27 Feb 2025 18:48:12 +0100
Subject: [PATCH] Remove tests that depend on nanoarrow

---
 test/CMakeLists.txt | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index abba8ea..3343153 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 if(NOT FETCH_DEPENDENCIES_WITH_CMAKE STREQUAL "ON")
     find_package(doctest ${FIND_PACKAGE_OPTIONS})
     # find_package(Catch2 3 ${FIND_PACKAGE_OPTIONS})
-    find_package(nanoarrow ${FIND_PACKAGE_OPTIONS})
+    #find_package(nanoarrow ${FIND_PACKAGE_OPTIONS})
 endif()
 
 Set(FETCHCONTENT_QUIET FALSE)
@@ -67,19 +67,19 @@ if(FETCH_DEPENDENCIES_WITH_CMAKE STREQUAL "ON" OR FETCH_DEPENDENCIES_WITH_CMAKE
     #     FetchContent_MakeAvailable(Catch2)
     # endif()
 
-    if(NOT nanoarrow_FOUND)
-        message(STATUS "📦 Fetching nanoarrow 0.6")
-        FetchContent_Declare(
-            nanoarrow
-            GIT_SHALLOW TRUE
-            GIT_REPOSITORY https://github.com/apache/arrow-nanoarrow.git
-            GIT_TAG apache-arrow-nanoarrow-0.6.0
-            GIT_PROGRESS TRUE
-            SYSTEM
-            EXCLUDE_FROM_ALL)
-        FetchContent_MakeAvailable(nanoarrow)
-        target_compile_options(nanoarrow PRIVATE -w)
-    endif()
+    # if(NOT nanoarrow_FOUND)
+    #     message(STATUS "📦 Fetching nanoarrow 0.6")
+    #     FetchContent_Declare(
+    #         nanoarrow
+    #         GIT_SHALLOW TRUE
+    #         GIT_REPOSITORY https://github.com/apache/arrow-nanoarrow.git
+    #         GIT_TAG apache-arrow-nanoarrow-0.6.0
+    #         GIT_PROGRESS TRUE
+    #         SYSTEM
+    #         EXCLUDE_FROM_ALL)
+    #     FetchContent_MakeAvailable(nanoarrow)
+    #     target_compile_options(nanoarrow PRIVATE -w)
+    # endif()
 endif()
 
 if(NOT CMAKE_BUILD_TYPE)
@@ -102,7 +102,7 @@ else()
         better_junit_reporter.hpp
         external_array_data_creation.cpp
         external_array_data_creation.hpp
-        nanoarrow_utils.hpp
+        #nanoarrow_utils.hpp
         junit.cpp
         junit.hpp
         junit_xml_writer.hpp
@@ -147,12 +147,12 @@ else()
         test_nested_comperators.cpp
         test_null_array.cpp
         test_nullable.cpp
-        test_primitive_array.cpp
+        #test_primitive_array.cpp
         test_ranges.cpp
         test_record_batch.cpp
         test_repeat_container.cpp
         test_run_end_encoded_array.cpp
-        test_string_array.cpp
+        #test_string_array.cpp
         test_struct_array.cpp
         test_time_array.cpp
         test_timestamp_array.cpp
@@ -171,7 +171,7 @@ target_link_libraries(${test_target}
     PRIVATE
         sparrow
         doctest::doctest
-        nanoarrow::nanoarrow
+        #nanoarrow::nanoarrow
         $<$<BOOL:USE_SANITIZER>:${SANITIZER_LINK_LIBRARIES}>)
 
 # set(test_target_catch2 "test_sparrow_lib_catch2")
@@ -326,4 +326,4 @@ add_custom_target(run_tests_with_junit_report
 )
 
 set_target_properties(run_tests_with_junit_report PROPERTIES FOLDER "Tests utilities")
-    
\ No newline at end of file
+    
-- 
2.48.1

