--- zbus-4.4.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ zbus-4.4.0/Cargo.toml	2024-12-31T18:42:10.173137+00:00
@@ -212,12 +212,6 @@
     "dep:async-io",
 ]
 
-[target."cfg(any(target_os = \"macos\", windows))".dependencies.async-recursion]
-version = "1.1.1"
-
-[target."cfg(target_os = \"macos\")".dependencies.async-process]
-version = "2.2.2"
-
 [target."cfg(unix)".dependencies.nix]
 version = "0.29"
 features = [
@@ -227,18 +221,3 @@
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.uds_windows]
-version = "1.1.0"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52"
-features = [
-    "Win32_Foundation",
-    "Win32_Security_Authorization",
-    "Win32_System_Memory",
-    "Win32_Networking",
-    "Win32_Networking_WinSock",
-    "Win32_NetworkManagement",
-    "Win32_NetworkManagement_IpHelper",
-    "Win32_System_Threading",
-]
