From e49974075fb2bbc8b9fbaef1d1ee4358e004a0cc Mon Sep 17 00:00:00 2001
From: Zack Scholl <zack.scholl@gmail.com>
Date: Mon, 7 Feb 2022 14:40:29 -0800
Subject: [PATCH] add curve point tests

---
 pake.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pake.go b/pake.go
index 6d2a799..f43fb1e 100644
--- a/pake.go
+++ b/pake.go
@@ -102,6 +102,14 @@ func initCurve(curve string) (ellipticCurve EllipticCurve, Ux *big.Int, Uy *big.
 		err = errors.New("no such curve")
 		return
 	}
+	if err == nil {
+		if !ellipticCurve.IsOnCurve(Ux, Uy) {
+			err = fmt.Errorf("Ux/Uy not on curve")
+		}
+		if !ellipticCurve.IsOnCurve(Vx, Vy) {
+			err = fmt.Errorf("Vx/Vy not on curve")
+		}
+	}
 	return
 }
 
