diff -Naur cdsclient-3.78.orig/abibcode cdsclient-3.78.new/abibcode
--- cdsclient-3.78.orig/abibcode	2015-01-25 14:50:09.196503545 +0100
+++ cdsclient-3.78.new/abibcode	2015-01-25 15:19:56.046725512 +0100
@@ -15,6 +15,8 @@
 #----------------
 set D_BIN = `dirname $0`
 
+set AWKSCRIPT = '/usr/share/cdsclient/abibcode.awk'
+
 ## Examine the installation
 set AWK = `which gawk`
 if ($#AWK != 1) set AWK = `which nawk`
@@ -41,9 +43,9 @@
 #\bibitem	bibcode_19	free_bibcode
 #rm -rf $tt.*
 set file = $1
-trim $file | sed 's/, */ /g' | $AWK -f $0.awk > $tt.3
+#trim $file | sed 's/, */ /g' | $AWK -f $AWKSCRIPT > $tt.3
 #cat $tt.3
-#tr -d , < $1 | $AWK -f $0.awk > $tt.3
+tr -d , < $file | $AWK -f $AWKSCRIPT > $tt.3
 
 # Interrogate SIMBAD from 3rd column of $tt.3, and generates
 #             2-column table with input SIMBAD_output
@@ -101,7 +103,7 @@
 #cat $tt.fin | acut -i"#" -c1-
 
 # Final: merge the original text with the appopriate macro
-trim $file | $AWK -v f3=$tt.fin '\
+tr -d , < $file | $AWK -v f3=$tt.fin '\
   /^ *\\bibitem[[{]/{ if (ref!="") print "\t" ref; print; \
 	getline result < f3 ; nr = split(result, c, "\t")\
 	if (c[3] != "") ref = "\\simOK{" c[3] "}";  \
