--- git-delta-0.19.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ git-delta-0.19.1/Cargo.toml	2026-03-22T22:31:23.105139+00:00
@@ -85,7 +85,7 @@
 version = "4.4.4"
 
 [dependencies.console]
-version = "0.15.0"
+version = ">=0.15.0,<0.17.0"
 
 [dependencies.ctrlc]
 version = "3.2.5"
@@ -138,7 +138,7 @@
 default-features = false
 
 [dependencies.terminal-colorsaurus]
-version = "0.4.8"
+version = "1.0.3"
 
 [dependencies.unicode-segmentation]
 version = "1.10.1"
@@ -160,7 +160,7 @@
 version = "1.4"
 
 [dev-dependencies.rstest]
-version = "0.21.0"
+version = "0.26.0"
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
