--- ./docs/doxygen/Makefile.in.old	2025-07-10 10:59:00.684720980 +0200
+++ ./docs/doxygen/Makefile.in	2025-07-10 10:59:14.030677121 +0200
@@ -487,9 +487,7 @@
 
 @WITH_LIBZMF_DOCS_TRUE@install-data-am:
 @WITH_LIBZMF_DOCS_TRUE@	mkdir -p $(DESTDIR)$(docdir)/html
-@WITH_LIBZMF_DOCS_TRUE@	$(INSTALL_DATA) html/*.html $(DESTDIR)$(docdir)/html/
-@WITH_LIBZMF_DOCS_TRUE@	$(INSTALL_DATA) html/*.png $(DESTDIR)$(docdir)/html/
-@WITH_LIBZMF_DOCS_TRUE@	$(INSTALL_DATA) html/*.css $(DESTDIR)$(docdir)/html/
+@WITH_LIBZMF_DOCS_TRUE@	$(INSTALL_DATA) html/* $(DESTDIR)$(docdir)/html/
 
 @WITH_LIBZMF_DOCS_TRUE@uninstall-am:
 @WITH_LIBZMF_DOCS_TRUE@	-rm -rf $(DESTDIR)$(docdir)/html
