--- maxima-5.45.1/src/maxima.system.orig	2021-05-15 17:20:53.000000000 -0600
+++ maxima-5.45.1/src/maxima.system	2022-03-07 11:23:14.847637296 -0700
@@ -48,6 +48,12 @@
 (export '*maxima-build-time*)
 
 #+ecl
+(require :cmp)
+
+#+ecl
+(setf c::*compile-in-constants* t)
+
+#+ecl
 (defun do-compile-ecl (file &rest args)
   ;; Not sure about this.  Compile twice.  The first produces a .o
   ;; file.  The second produces the .fas file to keep defsystem happy
