--- frama-c-32.1-Germanium/src/plugins/aorai/dune.orig	2026-03-15 18:00:00.000000000 -0600
+++ frama-c-32.1-Germanium/src/plugins/aorai/dune	2026-03-20 11:47:32.527775996 -0600
@@ -8,6 +8,9 @@
 ;;                                                                            ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(dirs ../eva)
+(include_subdirs unqualified)
+
 (rule
  (alias frama-c-configure)
  (deps (universe))
