--- frama-c-32.0-Germanium/src/dune.orig	2025-12-01 17:00:00.000000000 -0700
+++ frama-c-32.0-Germanium/src/dune	2025-12-04 14:15:00.042836553 -0700
@@ -46,7 +46,7 @@
   (public_name frama-c.kernel)
   (foreign_stubs (language c) (names c_bindings floating_point))
   (flags :standard -w -9)
-  (libraries frama-c.init fpath str unix zarith ocamlgraph dynlink bytes yaml.unix yojson menhirLib dune-site dune-site.plugins zip)
+  (libraries frama-c.init fpath str unix zarith ocamlgraph dynlink yaml.unix yojson menhirLib dune-site dune-site.plugins zip)
   (instrumentation (backend landmarks))
   (instrumentation (backend bisect_ppx))
   (inline_tests (libraries frama-c.init.cmdline))
