From 15cf268b2dfb6bf7ef0564197f01aee0773ee97a Mon Sep 17 00:00:00 2001
From: Matthias Wolf <matthias.wolf@epfl.ch>
Date: Fri, 27 Sep 2024 12:19:43 +0200
Subject: [PATCH 3/5] Substitute ghc in tests.

---
 tests/test_config.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/tests/test_config.cpp b/tests/test_config.cpp
index d4485b4..ef8e62a 100644
--- a/tests/test_config.cpp
+++ b/tests/test_config.cpp
@@ -2,11 +2,10 @@
 
 #include <nlohmann/json.hpp>
 
-#include "../extlib/filesystem.hpp"
-
 #include <bbp/sonata/config.h>
 
 #include <cstdio>
+#include <filesystem>
 #include <fstream>
 #include <iostream>
 #include <string>
@@ -316,7 +315,7 @@ TEST_CASE("SimulationConfig") {
         CHECK(config.getRun().minisSeed == 333);
         CHECK(config.getRun().synapseSeed == 444);
 
-        namespace fs = ghc::filesystem;
+        namespace fs = std::filesystem;
         const auto basePath = fs::absolute(
             fs::path("./data/config/simulation_config.json").parent_path());
 
@@ -651,7 +650,7 @@ TEST_CASE("SimulationConfig") {
             "tstop": 1000
           }
         })";
-        namespace fs = ghc::filesystem;
+        namespace fs = std::filesystem;
         const auto basePath = fs::absolute(fs::path("./").parent_path());
         const auto config = SimulationConfig(contents, basePath);
         const auto network = fs::absolute(basePath / "circuit" / fs::path("circuit_config.json"));
-- 
2.48.1

