--- interprocess-2.2.3/Cargo.toml	1970-01-01T00:00:01+00:00
+++ interprocess-2.2.3/Cargo.toml	2025-04-29T14:32:17.337758+00:00
@@ -150,39 +150,6 @@
 version = "0.2.137"
 features = ["extra_traits"]
 
-[target."cfg(windows)".dependencies.recvmsg]
-version = "1.0.0"
-
-[target."cfg(windows)".dependencies.tokio]
-version = "1.36.0"
-features = [
-    "sync",
-    "rt-multi-thread",
-    "fs",
-    "net",
-    "time",
-    "io-util",
-]
-optional = true
-
-[target."cfg(windows)".dependencies.widestring]
-version = "1.0.2"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_Security_Authorization",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_Pipes",
-    "Win32_System_Threading",
-    "Win32_System_Memory",
-    "Win32_System_SystemServices",
-    "Win32_System_LibraryLoader",
-]
-
 [lints.clippy]
 as_conversions = "deny"
 dbg_macro = "warn"
@@ -198,3 +165,4 @@
 [lints.rust.rust_2018_idioms]
 level = "deny"
 priority = -1
+
