--- dune-3.24.0/src/dune_lang/dune_project.ml.orig	2026-06-21 14:57:05.000000000 -0600
+++ dune-3.24.0/src/dune_lang/dune_project.ml	2026-06-24 14:53:47.543763375 -0600
@@ -484,7 +484,7 @@ let interpret_lang_and_extensions ~(lang
 let filename = Filename.dune_project
 let opam_file_location_default ~lang:_ = `Relative_to_project
 let wrapped_executables_default ~(lang : Lang.Instance.t) = lang.version >= (2, 0)
-let map_workspace_root_default ~(lang : Lang.Instance.t) = lang.version >= (3, 0)
+let map_workspace_root_default ~(lang : Lang.Instance.t) = false
 
 let executables_implicit_empty_intf_default ~(lang : Lang.Instance.t) =
   lang.version >= (3, 0)
