--- dune-3.24.0/boot/duneboot.ml.orig	2026-06-24 14:51:18.137646266 -0600
+++ dune-3.24.0/boot/duneboot.ml	2026-06-24 14:52:17.800527553 -0600
@@ -2087,7 +2087,7 @@ let build
             ; allow_unstable_sources
             ; external_includes
             ; [ m ]
-            ; [ "-cclib"; "-llmdb" ]
+            ; [ "-cclib"; "-llmdb"; "-cclib"; "-lev" ]
             ]
           |> Process.run ~cwd:build_dir Config.compiler)
       in
@@ -2164,7 +2164,7 @@ let build
     ; link_flags
     ; (if static then ccopt "-static" else [])
     ; allow_unstable_sources
-    ; [ "-cclib"; "-llmdb" ]
+    ; [ "-cclib"; "-llmdb"; "-cclib"; "-lev" ]
     ]
   |> Process.run ~cwd:build_dir Config.compiler
 ;;
--- dune-3.24.0/src/lev/src/dune.orig	2026-06-21 14:57:05.000000000 -0600
+++ dune-3.24.0/src/lev/src/dune	2026-06-24 14:53:16.502775794 -0600
@@ -1,42 +1,8 @@
-(copy_files# ../vendor/*.c)
-
-(copy_files# ../vendor/{ev,ev_vars,ev_wrap}.h)
-
-(rule
- (enabled_if
-  (= %{ocaml-config:ccomp_type} "msvc"))
- (target c_flags.sexp)
- (action
-  (write-file %{target} "(\"/w\")")))
-
-(rule
- (enabled_if
-  (<> %{ocaml-config:ccomp_type} "msvc"))
- (target c_flags.sexp)
- (action
-  (write-file %{target} "(-w)")))
-
-(foreign_library
- (archive_name ev)
- (extra_deps
-  ev_kqueue.c
-  ev_epoll.c
-  ev_iouring.c
-  ev_poll.c
-  ev_port.c
-  ev_select.c
-  ev_win32.c)
- (language c)
- (flags
-  :standard
-  (:include c_flags.sexp))
- (names ev))
-
 (library
  (name lev)
  (synopsis "libev bindings")
  (libraries stdune unix)
- (foreign_archives ev)
+ (c_library_flags -lev)
  (foreign_stubs
   (language c)
   (names lev_stubs)))
