diff -up st2205tool/setpic/main.c~ st2205tool/setpic/main.c
--- st2205tool/setpic/main.c~	2008-01-22 00:53:24.000000000 +0100
+++ st2205tool/setpic/main.c	2011-02-20 10:05:54.659573470 +0100
@@ -53,8 +53,8 @@ int sendpic(st2205_handle *h, char* file
 	return 0;
     }
 
-    for (y=0; y<h->width; y++) {
-	for (x=0; x<h->height; x++) {
+    for (y=0; y<h->height; y++) {
+	for (x=0; x<h->width; x++) {
 	    c = gdImageGetPixel(im, x, y);
 	    if (gdImageTrueColor(im) ) {
 		r=gdTrueColorGetRed(c);
