diff -rupN --no-dereference gtk+-2.24.33/gdk/win32/gdkgc-win32.c gtk+-2.24.33-new/gdk/win32/gdkgc-win32.c
--- gtk+-2.24.33/gdk/win32/gdkgc-win32.c	2020-12-18 06:29:39.000000000 +0100
+++ gtk+-2.24.33-new/gdk/win32/gdkgc-win32.c	2024-07-24 15:08:25.373063701 +0200
@@ -744,7 +744,7 @@ static GdkDrawableImplWin32 *
 get_impl_drawable (GdkDrawable *drawable)
 {
   if (GDK_IS_OFFSCREEN_WINDOW (drawable))
-    return _gdk_offscreen_window_get_real_drawable (GDK_OFFSCREEN_WINDOW (drawable));
+    return (GdkDrawableImplWin32 *) _gdk_offscreen_window_get_real_drawable (GDK_OFFSCREEN_WINDOW (drawable));
   if (GDK_IS_DRAWABLE_IMPL_WIN32 (drawable))
     return GDK_DRAWABLE_IMPL_WIN32(drawable);
   else if (GDK_IS_WINDOW (drawable))
diff -rupN --no-dereference gtk+-2.24.33/gtk/gtkprintoperation-win32.c gtk+-2.24.33-new/gtk/gtkprintoperation-win32.c
--- gtk+-2.24.33/gtk/gtkprintoperation-win32.c	2020-12-18 06:29:40.000000000 +0100
+++ gtk+-2.24.33-new/gtk/gtkprintoperation-win32.c	2024-07-24 15:08:25.374063811 +0200
@@ -938,7 +938,7 @@ devmode_from_settings (GtkPrintSettings
   char *extras;
   GtkPaperSize *paper_size;
   const char *extras_base64;
-  int extras_len;
+  gsize extras_len;
   const char *val;
 
   extras = NULL;
@@ -1345,7 +1345,7 @@ plug_grab_notify (GtkWidget        *widg
 }
 
 
-static BOOL CALLBACK
+static INT_PTR CALLBACK
 pageDlgProc (HWND wnd, UINT message, WPARAM wparam, LPARAM lparam)
 {
   GtkPrintOperation *op;
@@ -1374,7 +1374,7 @@ pageDlgProc (HWND wnd, UINT message, WPA
 
       /* When we lose the grab we need to disable the print dialog */
       g_signal_connect (plug, "grab-notify", G_CALLBACK (plug_grab_notify), op);
-      return FALSE;
+      return 0;
     }
   else if (message == WM_DESTROY)
     {
@@ -1391,11 +1391,11 @@ pageDlgProc (HWND wnd, UINT message, WPA
       op = GTK_PRINT_OPERATION (GetWindowLongPtrW (wnd, GWLP_USERDATA));
       op_win32 = op->priv->platform_data;
 
-      return _gtk_win32_embed_widget_dialog_procedure (GTK_WIN32_EMBED_WIDGET (op_win32->embed_widget),
+      return (INT_PTR) _gtk_win32_embed_widget_dialog_procedure (GTK_WIN32_EMBED_WIDGET (op_win32->embed_widget),
 						       wnd, message, wparam, lparam);
     }
   
-  return FALSE;
+  return 0;
 }
 
 static HPROPSHEETPAGE
diff -rupN --no-dereference gtk+-2.24.33/gtk/gtksocket-win32.c gtk+-2.24.33-new/gtk/gtksocket-win32.c
--- gtk+-2.24.33/gtk/gtksocket-win32.c	2020-12-18 06:29:40.000000000 +0100
+++ gtk+-2.24.33-new/gtk/gtksocket-win32.c	2024-07-24 15:08:25.375063921 +0200
@@ -46,7 +46,7 @@ _gtk_socket_windowing_get_id (GtkSocket
   g_return_val_if_fail (GTK_IS_SOCKET (socket), 0);
   g_return_val_if_fail (GTK_WIDGET_ANCHORED (socket), 0);
 
-  if (!gtk_widget_get_realized (socket))
+  if (!gtk_widget_get_realized (GTK_WIDGET (socket)))
     gtk_widget_realize (GTK_WIDGET (socket));
 
   return (GdkNativeWindow) GDK_WINDOW_HWND (GTK_WIDGET (socket)->window);
diff -rupN --no-dereference gtk+-2.24.33/gtk/updateiconcache.c gtk+-2.24.33-new/gtk/updateiconcache.c
--- gtk+-2.24.33/gtk/updateiconcache.c	2020-12-18 06:50:58.000000000 +0100
+++ gtk+-2.24.33-new/gtk/updateiconcache.c	2024-07-24 15:08:25.375063921 +0200
@@ -74,7 +74,7 @@ static gchar *var_name = "-";
 
 #include <ftw.h>
 
-static struct stat cache_stat;
+static GStatBuf cache_stat;
 static gboolean cache_up_to_date;
 
 static int check_dir_mtime (const char        *dir, 
