diff -up ./lib/CGI/Application/Structured/Tools/Starter.pm.orig ./lib/CGI/Application/Structured/Tools/Starter.pm
--- ./lib/CGI/Application/Structured/Tools/Starter.pm.orig	2015-08-20 12:42:45.029820837 +0200
+++ ./lib/CGI/Application/Structured/Tools/Starter.pm	2015-08-20 12:44:12.274294867 +0200
@@ -160,7 +160,9 @@ sub create_distro {
 	my @files2 = 
 	my $file = File::Spec->catfile( $self->{basedir}, 'MANIFEST' );
 	open my $fh, '>', $file or die "Can't open file $file: $!\n";
-	map{print $fh "$_\n"} grep { $_ ne 't/boilerplate.t' } @files;    
+        foreach my $file ( grep !m{^\Qt/boilerplate.t\E$}, sort @files ) {
+            print {$fh} "$file\n" or croak "$OS_ERROR\n";
+        }
 	close $fh or die "Can't close file $file: $!\n";	
 			    }
 	);
@@ -184,13 +186,7 @@ sub create_t {
     #
     # 
     #
-    my %t_files = $self->t_guts(@modules);
-
-    my @files = map { 
-	$self->_create_t( $_, 
-			  $t_files{$_} 
-	    ) 
-    } keys %t_files;
+    my @files = $self->SUPER::create_t(@modules);
 
     # This next part is for the static files dir t/www
     my @dirparts = ( $self->{basedir}, 't', 'www' );
