diff -rupN --no-dereference urllib3-2.6.3/pyproject.toml urllib3-2.6.3-new/pyproject.toml
--- urllib3-2.6.3/pyproject.toml	2026-01-07 17:18:09.000000000 +0100
+++ urllib3-2.6.3-new/pyproject.toml	2026-01-09 12:39:15.529898983 +0100
@@ -1,8 +1,8 @@
 # This file is protected via CODEOWNERS
 
 [build-system]
-requires = ["hatchling>=1.27.0,<2", "hatch-vcs>=0.4.0,<0.6.0", "setuptools-scm>=8,<10"]
-build-backend = "hatchling.build"
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
 
 [project]
 name = "urllib3"
@@ -113,25 +113,6 @@ docs = [
 "Code" = "https://github.com/urllib3/urllib3"
 "Issue tracker" = "https://github.com/urllib3/urllib3/issues"
 
-[tool.hatch.version]
-source = "vcs"
-[tool.hatch.version.raw-options]
-local_scheme = "no-local-version"
-[tool.hatch.build.hooks.vcs]
-version-file = "src/urllib3/_version.py"
-
-[tool.hatch.build.targets.sdist]
-include = [
-  "/docs",
-  "/dummyserver",
-  "/src",
-  "/test",
-  "/uv.lock",
-  "/CHANGES.rst",
-  "/README.md",
-  "/LICENSE.txt",
-]
-
 [tool.pytest.ini_options]
 xfail_strict = true
 python_classes = ["Test", "*TestCase"]
