--- rustsec-0.31.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ rustsec-0.31.0/Cargo.toml	2025-12-12T11:46:16.647007+00:00
@@ -66,13 +66,6 @@
 ]
 
 [features]
-binary-scanning = [
-    "dep:auditable-info",
-    "dep:auditable-serde",
-    "dep:binfarce",
-    "dep:quitters",
-    "dep:once_cell",
-]
 default = ["gix-reqwest"]
 dependency-tree = ["cargo-lock/dependency-tree"]
 git = [
@@ -115,19 +108,6 @@
 name = "query"
 path = "tests/query.rs"
 
-[dependencies.auditable-info]
-version = "0.10"
-features = ["wasm"]
-optional = true
-
-[dependencies.auditable-serde]
-version = "0.9"
-optional = true
-
-[dependencies.binfarce]
-version = "0.2"
-optional = true
-
 [dependencies.cargo-lock]
 version = "11"
 
@@ -139,7 +119,7 @@
 version = "3"
 
 [dependencies.gix]
-version = "0.74"
+version = ">=0.73, <0.76"
 features = [
     "worktree-mutation",
     "revision",
@@ -152,17 +132,9 @@
 version = "0.5"
 optional = true
 
-[dependencies.once_cell]
-version = "1.15.0"
-optional = true
-
 [dependencies.platforms]
 version = "3"
 features = ["serde"]
-
-[dependencies.quitters]
-version = "0.1.0"
-optional = true
 
 [dependencies.semver]
 version = "1.0.23"
@@ -173,7 +145,7 @@
 features = ["serde_derive"]
 
 [dependencies.tame-index]
-version = "0.24"
+version = ">=0.24, <0.26"
 features = [
     "sparse",
     "native-certs",
