--- python-launcher-1.0.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ python-launcher-1.0.1/Cargo.toml	2024-07-07T15:43:51.290162+00:00
@@ -45,10 +45,6 @@
 name = "py"
 path = "src/main.rs"
 
-[[bench]]
-name = "list"
-harness = false
-
 [dependencies.comfy-table]
 version = "7.1.1"
 
@@ -71,9 +67,6 @@
 [dev-dependencies.assert_cmd]
 version = "2.0.14"
 
-[dev-dependencies.criterion]
-version = "0.5.1"
-
 [dev-dependencies.predicates]
 version = "3.1.0"
 
