diff -rupN --no-dereference qtwebkit-5.212.0-alpha4/Source/WebCore/xml/XSLTProcessor.h qtwebkit-5.212.0-alpha4-new/Source/WebCore/xml/XSLTProcessor.h
--- qtwebkit-5.212.0-alpha4/Source/WebCore/xml/XSLTProcessor.h	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4-new/Source/WebCore/xml/XSLTProcessor.h	2025-07-18 10:44:50.644794558 +0200
@@ -64,7 +64,7 @@ public:
 
     void reset();
 
-    static void parseErrorFunc(void* userData, xmlError*);
+    static void parseErrorFunc(void* userData, const xmlError*);
     static void genericErrorFunc(void* userData, const char* msg, ...);
     
     // Only for libXSLT callbacks
diff -rupN --no-dereference qtwebkit-5.212.0-alpha4/Source/WebCore/xml/XSLTProcessorLibxslt.cpp qtwebkit-5.212.0-alpha4-new/Source/WebCore/xml/XSLTProcessorLibxslt.cpp
--- qtwebkit-5.212.0-alpha4/Source/WebCore/xml/XSLTProcessorLibxslt.cpp	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4-new/Source/WebCore/xml/XSLTProcessorLibxslt.cpp	2025-07-18 10:44:50.645048762 +0200
@@ -78,7 +78,7 @@ void XSLTProcessor::genericErrorFunc(voi
     // It would be nice to do something with this error message.
 }
 
-void XSLTProcessor::parseErrorFunc(void* userData, xmlError* error)
+void XSLTProcessor::parseErrorFunc(void* userData, const xmlError* error)
 {
     PageConsoleClient* console = static_cast<PageConsoleClient*>(userData);
     if (!console)
