diff -up libclaw-1.7.4/doc/Doxyfile.noarch libclaw-1.7.4/doc/Doxyfile
--- libclaw-1.7.4/doc/Doxyfile.noarch	2013-07-03 13:34:38.000000000 -0600
+++ libclaw-1.7.4/doc/Doxyfile	2016-12-29 16:44:08.659098775 -0700
@@ -989,7 +989,7 @@ HTML_COLORSTYLE_GAMMA  = 80
 # page will contain the date and time when the page was generated. Setting
 # this to NO can help when comparing the output of multiple runs.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO
 
 # If the HTML_DYNAMIC_SECTIONS tag is set to YES then the generated HTML
 # documentation will contain sections that can be hidden and shown after the
