diff -up mathgl-8.0.1/CMakeLists.txt.orig mathgl-8.0.1/CMakeLists.txt
--- mathgl-8.0.1/CMakeLists.txt.orig   2022-01-11 13:48:53.000000000 -0700
+++ mathgl-8.0.1/CMakeLists.txt        2023-08-03 15:47:52.703488202 -0600
@@ -469,7 +469,7 @@ endif(enable-openmp)
 if(enable-gsl)
 	set(MGL_HAVE_GSL 1)
 	find_library(GSL_LIB gsl)
-	find_library(GSL_CBLAS_LIB gslcblas)
+	find_library(GSL_CBLAS_LIB flexiblas)
 	find_path(GSL_INCLUDE_DIR gsl/gsl_fft_complex.h)
 	if(NOT GSL_LIB OR NOT GSL_CBLAS_LIB OR NOT GSL_INCLUDE_DIR)
 		message(SEND_ERROR "${GSL_LIB}")
