diff -up alienarena-7.71.7/source/unix/minizip/zip.c.fixme alienarena-7.71.7/source/unix/minizip/zip.c
--- alienarena-7.71.7/source/unix/minizip/zip.c.fixme	2025-08-06 16:58:12.846803789 -0400
+++ alienarena-7.71.7/source/unix/minizip/zip.c	2025-08-06 16:58:30.977614943 -0400
@@ -157,7 +157,7 @@ typedef struct
     ZPOS64_T totalUncompressedData;
 #ifndef NOCRYPT
     unsigned long keys[3];     /* keys defining the pseudo-random sequence */
-    const z_crc_t* pcrc_32_tab;
+    const long unsigned int* pcrc_32_tab;
     int crypt_header_size;
 #endif
 } curfile64_info;
