diff -up libltc-1.3.2/Doxyfile.in.multilib libltc-1.3.2/Doxyfile.in
--- libltc-1.3.2/Doxyfile.in.multilib	2023-05-08 17:21:52.964908557 +0200
+++ libltc-1.3.2/Doxyfile.in	2023-05-08 17:23:20.051423901 +0200
@@ -1189,7 +1189,7 @@ HTML_COLORSTYLE_GAMMA  = 80
 # The default value is: NO.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO
 
 # If the HTML_DYNAMIC_SECTIONS tag is set to YES then the generated HTML
 # documentation will contain sections that can be hidden and shown after the
