diff -Naur zidrav4unix-1.2.0.orig/core.h zidrav4unix-1.2.0/core.h
--- zidrav4unix-1.2.0.orig/core.h	2016-02-15 05:21:09.143118636 +0100
+++ zidrav4unix-1.2.0/core.h	2016-02-15 05:22:02.503690038 +0100
@@ -48,7 +48,7 @@
 #define QSV		KSigver( ZQ, QVER )
 
 
-const long crc_table[256] = {		// it's just easier this way, alright? It's one K of data, I'm allowed a global if I want one
+const unsigned long crc_table[256] = {		// it's just easier this way, alright? It's one K of data, I'm allowed a global if I want one
   0x00000000L, 0x77073096L, 0xee0e612cL, 0x990951baL, 0x076dc419L,
   0x706af48fL, 0xe963a535L, 0x9e6495a3L, 0x0edb8832L, 0x79dcb8a4L,
   0xe0d5e91eL, 0x97d2d988L, 0x09b64c2bL, 0x7eb17cbdL, 0xe7b82d07L,
