diff -rupN --no-dereference pyproj-3.7.2/setup.py pyproj-3.7.2-new/setup.py
--- pyproj-3.7.2/setup.py	2025-08-14 13:05:05.000000000 +0200
+++ pyproj-3.7.2-new/setup.py	2025-09-14 17:13:43.251321613 +0200
@@ -194,9 +194,7 @@ def get_extension_modules():
     ext_options = {
         "include_dirs": include_dirs,
         "library_dirs": library_dirs,
-        "runtime_library_dirs": (
-            library_dirs if os.name != "nt" and sys.platform != "cygwin" else None
-        ),
+        "runtime_library_dirs": None,
         "libraries": get_libraries(library_dirs),
     }
     # setup cythonized modules
