--- warp-0.4.3/Cargo.toml
+++ warp-0.4.3/Cargo.toml
@@ -46,7 +46,6 @@
     "compression-brotli",
     "compression-gzip",
 ]
-compression-brotli = ["async-compression/brotli"]
 compression-gzip = [
     "async-compression/deflate",
     "async-compression/gzip",
@@ -162,7 +161,7 @@
 ]
 
 [dependencies.tokio-tungstenite]
-version = "0.29"
+version = "0.23"
 optional = true
 
 [dependencies.tokio-util]
