--- ghc-9.10.1.20250416/compiler/cbits/genSym.c~	2025-04-17 11:00:13.000000000 +0800
+++ ghc-9.10.1.20250416/compiler/cbits/genSym.c	2025-04-21 19:50:22.620607456 +0800
@@ -13,9 +13,6 @@
 #if MIN_VERSION_GLASGOW_HASKELL(9,9,0,0)
 // Unique64 patch was present in 9.10 and later
 #define HAVE_UNIQUE64 1
-#elif !MIN_VERSION_GLASGOW_HASKELL(9,9,0,0) && MIN_VERSION_GLASGOW_HASKELL(9,8,4,0)
-// Unique64 patch was backported to 9.8.4
-#define HAVE_UNIQUE64 1
 #elif !MIN_VERSION_GLASGOW_HASKELL(9,7,0,0) && MIN_VERSION_GLASGOW_HASKELL(9,6,7,0)
 // Unique64 patch was backported to 9.6.7
 #define HAVE_UNIQUE64 1
