--- src/Makefile~	2010-05-06 09:27:32.000000000 -0500
+++ src/Makefile	2010-05-06 09:28:47.867409952 -0500
@@ -37,7 +37,7 @@
 
 $(MAINFILE): $(OBJ1) $(OBJ2) $(OBJ3)
 	@echo Linking $(MAINFILE)
-	@$(CC) -o $(MAINFILE) $(OBJ1) $(OBJ2) $(OBJ3) $(LIB)
+	$(CC) -o $(MAINFILE) $(OBJ1) $(OBJ2) $(OBJ3) $(LIB)
 #	@strip $(MAINFILE)
 	@echo done
 
@@ -55,7 +55,7 @@
 
 .c.o:
 	@echo Compiling $*.o 
-	@$(CC) $(CFLAGS) -c $*.c
+	$(CC) $(CFLAGS) -c $*.c
 
 
 clean:
