diff -up swatchdog-3.2.4/swatchdog.mail-at-fix swatchdog-3.2.4/swatchdog
--- swatchdog-3.2.4/swatchdog.mail-at-fix	2022-09-23 14:56:05.641957705 -0400
+++ swatchdog-3.2.4/swatchdog	2022-09-23 14:56:11.873989065 -0400
@@ -574,14 +574,17 @@ sub action_def_to_subroutine_call {
 		 "write" => { 'sub_name' => "&Swatchdog::Actions::write_message", 'def_arg' => 'USERS' },
 		};
 
+  my $opt;
   my %options;
   my $have_opts = 0;
 
   foreach my $v (split(/,/, $optstr)) {
     if ($v =~ /(\w+)\s*=\s*"?(\S+[^"]*)/) {
-      $options{uc $1} = $2;
+      $opt = uc($1);
+      $options{$opt} = $2;
+      $options{$opt} =~ s/@/\\@/g if ($key eq 'mail');
     } else {
-      my $opt = $v;
+      $opt = $v;
       $opt =~ s/@/\\@/g;
       $opt =~ s/^\s+//o;
       $opt =~ s/^\s+$//o;
