diff --git a/build/cmake/CondorPackageConfig.cmake b/build/cmake/CondorPackageConfig.cmake
index ed894501fc..4a296051d0 100644
--- a/build/cmake/CondorPackageConfig.cmake
+++ b/build/cmake/CondorPackageConfig.cmake
@@ -490,7 +490,11 @@ elseif( "${OS_NAME}" STREQUAL "LINUX" AND CONDOR_PACKAGE_BUILD )
 		set( C_BIN			usr/bin )
 		set( C_LIBEXEC		usr/libexec/condor )
 		set( C_LIBEXEC_BLAHP		usr/libexec/blahp )
-		set( C_SBIN			usr/sbin )
+		if ( ${LINUX_NAME} MATCHES "Fedora"  AND  ${LINUX_VER} GREATER 41 )
+			set( C_SBIN			usr/bin )
+		else()
+			set( C_SBIN			usr/sbin )
+		endif()
 		set( C_INCLUDE		usr/include/condor )
 		set( C_INCLUDE_PUBLIC		usr/include )
 		set( C_MAN			usr/share/man/man1 )
