diff --git a/src/X11/xfbbabtd.c b/src/X11/xfbbabtd.c
index d08a019..3289a7b 100644
--- a/src/X11/xfbbabtd.c
+++ b/src/X11/xfbbabtd.c
@@ -20,8 +20,8 @@
     Thanks for the help.
 ***********************************************************************/
 
-#include <serv.h>
 #include <xfbb.h>
+#include <serv.h>
 
 static Widget about_dialog = NULL;
 static Widget copy_dialog = NULL;
diff --git a/src/X11/xfbbcnsl.c b/src/X11/xfbbcnsl.c
index d665b9c..170c631 100644
--- a/src/X11/xfbbcnsl.c
+++ b/src/X11/xfbbcnsl.c
@@ -20,7 +20,6 @@
     Thanks for the help.
 ***********************************************************************/
 
-#include <serv.h>
 #include <xfbb.h>
 
 #include <stdio.h>
@@ -41,6 +40,7 @@
 #include <Xm/Protocols.h>
 
 #include <stdint.h>
+#include <serv.h>
 
 typedef struct {
   char text[82];
