From e5c842003a94f46506c8ec1b44afcc2d0c5a79db Mon Sep 17 00:00:00 2001
From: Tom Hughes <tom@compton.nu>
Date: Sun, 2 Mar 2025 12:24:19 +0000
Subject: [PATCH 3/3] Allow some minor differences in images

---
 test/visual/renderer.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/test/visual/renderer.hpp b/test/visual/renderer.hpp
index e5e6278de..9650d5ff9 100644
--- a/test/visual/renderer.hpp
+++ b/test/visual/renderer.hpp
@@ -80,7 +80,7 @@ struct raster_renderer_base
         mapnik::image_any ref_image_any = reader->read(0, 0, reader->width(), reader->height());
         ImageType const& reference_image = mapnik::util::get<ImageType>(ref_image_any);
 
-        return mapnik::compare(actual, reference_image, 0, true);
+        return mapnik::compare(actual, reference_image, 300, true);
     }
 
     void save(image_type const& image, mapnik::fs::path const& path) const
@@ -103,7 +103,8 @@ struct vector_renderer_base
             throw std::runtime_error("Could not open: " + reference.string());
         }
         std::string expected(std::istreambuf_iterator<char>(stream.rdbuf()), std::istreambuf_iterator<char>());
-        return std::max(actual.size(), expected.size()) - std::min(actual.size(), expected.size());
+        unsigned diff = std::max(actual.size(), expected.size()) - std::min(actual.size(), expected.size());
+        return diff > 1 ? diff : 0;
     }
 
     void save(image_type const& image, mapnik::fs::path const& path) const
-- 
2.51.1

