Index: src/misc.cxx
===================================================================
RCS file: /cvsroot/atlas/Atlas/src/misc.cxx,v
retrieving revision 1.24
diff -u -p -r1.24 misc.cxx
--- src/misc.cxx	17 Aug 2014 20:20:19 -0000	1.24
+++ src/misc.cxx	19 Feb 2016 18:55:26 -0000
@@ -400,7 +400,7 @@ double magneticVariation(double lat, dou
 
     lon *= SGD_DEGREES_TO_RADIANS;
     lat *= SGD_DEGREES_TO_RADIANS;
-    t1.update(lon, lat, 0, 0);
+    t1.update(SGGeod::fromDeg(lon, lat), 0, 0);
 
     return sgGetMagVar(lon, lat, elev, t1.getJD()) * SGD_RADIANS_TO_DEGREES;
 }
