--- a/unittests/machinefiletests.py
+++ b/unittests/machinefiletests.py
@@ -1033,6 +1033,8 @@
             check_target(build_tuple)
 
         with self.subTest('properties and compiler'):
+            if build_tuple == host_tuple:
+                raise SkipTest('Do not run aarch64 cross compilation test on aarch64 itself.')
             self.new_builddir()
             config = self.helper_create_cross_file({
                 'binaries': {'rust': [rustc, f'--target={build_tuple}']},
