diff --git a/Cargo.toml b/Cargo.toml
index fe7ef641..f6f6fa9d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,15 +7,12 @@ edition = "2021"
 crate-type = ["cdylib", "lib"]
 
 [features]
-default = ["pg14"]
-#pg11 = ["pgrx/pg11", "pgrx-tests/pg11" ]
-#pg12 = ["pgrx/pg12", "pgrx-tests/pg12" ]
-#pg13 = ["pgrx/pg13", "pgrx-tests/pg13" ]
-pg14 = ["pgrx/pg14", "pgrx-tests/pg14" ]
-pg15 = ["pgrx/pg15", "pgrx-tests/pg15" ]
-pg16 = ["pgrx/pg16", "pgrx-tests/pg16" ]
-pg17 = ["pgrx/pg17", "pgrx-tests/pg17" ]
-pg18 = ["pgrx/pg18", "pgrx-tests/pg18" ]
+default = ["pg16"]
+pg14 = ["pgrx/pg14"]
+pg15 = ["pgrx/pg15"]
+pg16 = ["pgrx/pg16"]
+pg17 = ["pgrx/pg17"]
+pg18 = ["pgrx/pg18"]
 pg_test = []
 
 [dependencies]
