--- libgda-5.2.10/providers/web/gda-web-provider.c~	2020-11-08 14:23:20.000000000 -0600
+++ libgda-5.2.10/providers/web/gda-web-provider.c	2021-12-22 09:00:22.980170873 -0600
@@ -385,8 +385,8 @@
 	cdata->mutex = gda_mutex_new ();
 	cdata->server_id = NULL;
 	cdata->forced_closing = FALSE;
-	cdata->worker_session = soup_session_sync_new ();
-	cdata->front_session = soup_session_sync_new_with_options ("max-conns-per-host", 1, NULL);
+        cdata->worker_session = soup_session_new_with_options ("ssl-use-system-ca-file", TRUE, NULL);
+        cdata->front_session = soup_session_new_with_options ("max-conns-per-host", 1, "ssl-use-system-ca-file", TRUE, NULL);
 	if (use_ssl) {
 		server_url = g_string_new ("https://");
 		g_print ("USING SSL\n");
