--- clamav-0.104.3/clamconf/clamconf.c
+++ clamav-0.104.3/clamconf/clamconf.c
@@ -63,9 +63,9 @@
     const char *name;
     int tool;
 } cfgfile[] = {
-    {"clamd.conf", OPT_CLAMD},
+    {"clamd.d/scan.conf", OPT_CLAMD},
     {"freshclam.conf", OPT_FRESHCLAM},
-    {"clamav-milter.conf", OPT_MILTER},
+    {"mail/clamav-milter.conf", OPT_MILTER},
     {NULL, 0}};
 
 static void printopts(struct optstruct *opts, int nondef)
--- clamav-0.104.3/docs/man/clamav-milter.8.in
+++ clamav-0.104.3/docs/man/clamav-milter.8.in
@@ -27,7 +27,7 @@
 Read configuration from FILE.
 .SH "FILES"
 .LP
-@CONFDIR@/clamav-milter.conf
+@CONFDIR@/mail/clamav-milter.conf
 .SH "AUTHOR"
 .LP
 aCaB <acab@clamav.net>
--- clamav-0.104.3/docs/man/clamav-milter.conf.5.in
+++ clamav-0.104.3/docs/man/clamav-milter.conf.5.in
@@ -239,7 +239,7 @@
 All options expressing a size are limited to max 4GB. Values in excess will be reset to the maximum.
 .SH "FILES"
 .LP
-@CONFDIR@/clamav-milter.conf
+@CONFDIR@/mail/clamav-milter.conf
 .SH "AUTHOR"
 .LP
 aCaB <acab@clamav.net>
--- clamav-0.104.3/docs/man/clamd.8.in
+++ clamav-0.104.3/docs/man/clamd.8.in
@@ -7,7 +7,7 @@
 clamd [options]
 .SH "DESCRIPTION"
 .LP
-The daemon listens for incoming connections on Unix and/or TCP socket and scans files or directories on demand. It reads the configuration from @CONFDIR@/clamd.conf
+The daemon listens for incoming connections on Unix and/or TCP socket and scans files or directories on demand. It reads the configuration from @CONFDIR@/clamd.d/scan.conf
 .SH "COMMANDS"
 .LP
 It's recommended to prefix clamd commands with the letter \fBz\fR (eg. zSCAN) to indicate that the command will be delimited by a NULL character and that clamd should continue reading command data until a NULL character is read. The null delimiter assures that the complete command and its entire argument will be processed as a single command. Alternatively commands may be prefixed with the letter \fBn\fR (e.g. nSCAN) to use a newline character as the delimiter. Clamd replies will honour the requested terminator in turn.
@@ -147,7 +147,7 @@
 Perform a clean exit.
 .SH "FILES"
 .LP
-@CONFDIR@/clamd.conf
+@CONFDIR@/clamd.d/scan.conf
 .SH "CREDITS"
 Please check the full documentation for credits.
 .SH "AUTHOR"
--- clamav-0.104.3/docs/man/clamd.conf.5.in
+++ clamav-0.104.3/docs/man/clamd.conf.5.in
@@ -875,7 +875,7 @@
 All options expressing a size are limited to max 4GB. Values in excess will be reset to the maximum.
 .SH "FILES"
 .LP
-@CONFDIR@/clamd.conf
+@CONFDIR@/clamd.d/scan.conf
 .SH "AUTHORS"
 .LP
 Tomasz Kojm <tkojm@clamav.net>, Kevin Lin <klin@sourcefire.com>
--- clamav-0.104.3/common/optparser.h
+++ clamav-0.104.3/common/optparser.h
@@ -63,9 +63,9 @@
     #define OPT_DATADIR           DATADIR
     #define OPT_CONFDIR           CONFDIR
     #define OPT_CERTSDIR          CERTSDIR
-    #define OPT_CONFDIR_CLAMD     CONFDIR PATHSEP "clamd.conf"
+    #define OPT_CONFDIR_CLAMD     CONFDIR PATHSEP "clamd.d/scan.conf"
     #define OPT_CONFDIR_FRESHCLAM CONFDIR PATHSEP "freshclam.conf"
-    #define OPT_CONFDIR_MILTER    CONFDIR PATHSEP "clamav-milter.conf"
+    #define OPT_CONFDIR_MILTER    CONFDIR PATHSEP "mail/clamav-milter.conf"
 
 #endif
 // clang-format on
