From 833412c316ba370718cdb46368199c9bae6fa5af Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Thu, 17 Jul 2025 10:44:23 +0200
Subject: [PATCH] use zend_ce_exception instead of zend_exception_get_default()
 for 8.5

---
 geos.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/geos.c b/geos.c
index ad2dce2..8f4bdd3 100644
--- a/geos.c
+++ b/geos.c
@@ -150,7 +150,11 @@ static void errorHandler(const char *fmt, ...)
     va_end(args);
 
     /* TODO: use a GEOSException ? */
+#if PHP_VERSION_ID < 80000
     zend_throw_exception_ex(zend_exception_get_default(TSRMLS_C),
+#else
+    zend_throw_exception_ex(zend_ce_exception,
+#endif
         1 TSRMLS_CC, "%s", message);
 
 }
-- 
2.50.1

