From 5bfa43a8504015ad410bd54eec0363e848c7cda2 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Fri, 9 Sep 2022 10:44:25 +0200
Subject: [PATCH 6/6] fix __toString with 8.2

---
 arginfo.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arginfo.h b/arginfo.h
index 3bee988..17cb3b3 100644
--- a/arginfo.h
+++ b/arginfo.h
@@ -36,7 +36,12 @@ ZEND_END_ARG_INFO()
 
 #define arginfo_Geometry_construct       arginfo_void
 
+#if PHP_VERSION_ID >= 80200
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_Geometry_toString, 0, 0, IS_STRING, 0)
+ZEND_END_ARG_INFO();
+#else
 #define arginfo_Geometry_toString        arginfo_void
+#endif
 
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Geometry_project, 0, 0, 1)
     ZEND_ARG_INFO(0, geom)
-- 
2.37.2

