diff -rupN --no-dereference freetype-2.14.1/modules.cfg freetype-2.14.1-new/modules.cfg
--- freetype-2.14.1/modules.cfg	2025-09-08 04:48:18.000000000 +0200
+++ freetype-2.14.1-new/modules.cfg	2025-10-19 17:37:12.149221621 +0200
@@ -117,7 +117,7 @@ AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs `ftgxval.c' below.
 #
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid
 
 # Support for streams compressed with gzip (files with suffix .gz).
 #
@@ -136,7 +136,7 @@ AUX_MODULES += bzip2
 
 # OpenType table validation.  Needs `ftotval.c' below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
