diff --git a/pyproject.toml b/pyproject.toml
index 33aafde..6949945 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,8 +25,8 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = ">=3.9,<4.0"
-tomli = {version = "^2.1.0", optional = true}
-tomli-w = {version = "^1.1.0", optional = true}
+tomli = {version = "^2", optional = true}
+tomli-w = {version = "<2", optional = true}
 pyyaml = {version = "^6.0.1", optional = true}
 
 [tool.poetry.extras]
@@ -46,8 +46,8 @@ enum-tools = {extras = ["sphinx"], version = "^0.10.0"}
 docutils = "^0.21.1"  # NOTE: https://github.com/python-poetry/poetry/issues/9293
 
 [tool.poetry.group.dev.dependencies]
-tomli = "^2.1.0"
-tomli-w = "^1.1.0"
+tomli = "^2"
+tomli-w = "<2"
 pyyaml = "^6.0.1"
 pytest = "^7.4.4"
 hypothesis = "^6.92.6"
