diff -up ./t/01-wikiconverter.t.orig ./t/01-wikiconverter.t
--- ./t/01-wikiconverter.t.orig	2019-08-11 11:55:01.490595423 +0200
+++ ./t/01-wikiconverter.t	2019-08-11 11:48:14.352041012 +0200
@@ -157,7 +157,7 @@ is( $wc4->html2wiki( html => '&lt;' ), '
 
 SKIP: {
   skip "LWP::UserAgent required for testing how content is fetched from URIs" => 4 unless $have_lwp;
-  skip "Couldn't fetch test website http://www.perl.org. Perhaps you don't have internet access?" => 4 unless LWP::UserAgent->new->get('http://www.perl.org')->is_success;
+  skip "Couldn't fetch test website http://diberri.dyndns.org. Perhaps you don't have internet access?" => 4 unless LWP::UserAgent->new->get('http://diberri.dyndns.org')->is_success;
 
   is( $wc4->html2wiki( uri => 'http://diberri.dyndns.org/wikipedia/html2wiki-old/test.html', strip_tags => ['head'] ), '**test**', 'fetch uri, no ua' );
   is( $wc4->user_agent->agent, $wc4->__default_ua_string, 'using default ua' );
