diff --git a/install.sh b/install.sh
index ae841ae3d34d..3e793df3f494 100755
--- a/install.sh
+++ b/install.sh
@@ -388,7 +388,7 @@ else
 		# Not implemented on OpenWRT
 		/bin/true
 	    elif [ "$HOST" != debian ]; then
-		make_parent_directory ${DESTDIR}/${ETC}/NetworkManager/dispatcher.d 0755
+		make_parent_directory ${DESTDIR}${PREFIX}/lib/NetworkManager/dispatcher.d 0755
 	    fi
 	fi
     fi
@@ -415,12 +415,12 @@ if [ $HOST != openwrt ]; then
     install_file ifupdown ${DESTDIR}${LIBEXECDIR}/$PRODUCT/ifupdown 0544
 fi
 
-if [ -d ${DESTDIR}/etc/NetworkManager ]; then
+if [ -d ${DESTDIR}/${PREFIX}/lib/NetworkManager ]; then
     if [ "$HOST" = debian ]; then
 	rm -f ${DESTDIR}${ETC}/NetworkManager/dispatcher.d/01-shorewall
     else
-	[ $configure -eq 1 ] || make_parent_directory ${DESTDIR}${CONFDIR}/NetworkManager/dispatcher.d 0755
-	install_file ifupdown ${DESTDIR}${ETC}/NetworkManager/dispatcher.d/01-shorewall 0544
+	[ $configure -eq 1 ] || make_parent_directory ${DESTDIR}${PREFIX}/lib/NetworkManager/dispatcher.d 0755
+	install_file ifupdown ${DESTDIR}${PREFIX}/lib/NetworkManager/dispatcher.d/01-shorewall 0544
     fi
 fi
 
