diff -up nogravity-2.00/src/lt_func.h~ nogravity-2.00/src/lt_func.h
--- nogravity-2.00/src/lt_func.h~	2006-01-25 11:04:18.000000000 +0100
+++ nogravity-2.00/src/lt_func.h	2026-01-24 13:19:46.017445485 +0100
@@ -125,7 +125,7 @@ void NG_StretchFont(GXSPRITEGROUP *pSpri
 void NG_GamePlay(void);    
 void NG_DrawVideoSubtitles(FLI_STRUCT *);
 void NG_ReadLanguagePack(void);
-void NG_SetLanguage();
+void NG_SetLanguage(int l);
 int NG_ExecMainMenu(char **menu, int def, u_int32_t defColor, u_int8_t spacing);
 
 // AI
