From 254572fc2f38b8d847175c472486eb583744f906 Mon Sep 17 00:00:00 2001
From: Keith Bowes <keithbowes@users.noreply.github.com>
Date: Tue, 21 Jan 2025 17:41:21 -0500
Subject: [PATCH] Support for newer glslang versions without SPIRV and HLSL
 libraries

---
 qb/config.libs.sh | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index d5e30762957a..419f8bb964a0 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -666,7 +666,6 @@ if [ "$HAVE_GLSLANG" != no ]; then
       glslang/Public/ShaderLang.h \
       glslang/SPIRV/GlslangToSpv.h
 
-   check_lib cxx GLSLANG -lglslang '' '-lSPIRV'
    check_lib cxx GLSLANG_OSDEPENDENT -lOSDependent
    check_lib cxx GLSLANG_OGLCOMPILER -lOGLCompiler
    check_lib cxx GLSLANG_MACHINEINDEPENDENT -lMachineIndependent
@@ -675,10 +674,9 @@ if [ "$HAVE_GLSLANG" != no ]; then
    check_lib cxx GLSLANG_SPIRV -lSPIRV
    check_lib cxx GLSLANG_SPIRV_TOOLS_OPT -lSPIRV-Tools-opt
    check_lib cxx GLSLANG_SPIRV_TOOLS -lSPIRV-Tools
+   check_lib cxx GLSLANG -lglslang '' "$GLSLANG_SPIRV_LIBS"
 
    if [ "$HAVE_GLSLANG" = no ] ||
-      [ "$HAVE_GLSLANG_HLSL" = no ] ||
-      [ "$HAVE_GLSLANG_SPIRV" = no ] ||
       [ "$HAVE_GLSLANG_SPIRV_TOOLS_OPT" = no ] ||
       [ "$HAVE_GLSLANG_SPIRV_TOOLS" = no ]; then
       if [ "$HAVE_BUILTINGLSLANG" != yes ]; then
