Fix double assignment (make it a single assignment)

diff -ru cstream-2.7.6/cstream.c cstream-2.7.6-ndim/cstream.c
--- cstream-2.7.6/cstream.c	2010-09-09 22:28:20.748308391 +0200
+++ cstream-2.7.6-ndim/cstream.c	2010-09-09 22:30:15.809288769 +0200
@@ -1195,7 +1195,7 @@
 	  exit(1);
 	}
 
-	state->b = state->b = frames * 2 * 2; /* 16 bit Stereo */
+	state->b = frames * 2 * 2; /* 16 bit Stereo */
 	for (i = 0; i < frames; i++) {
 	  val = sin((double)i * 2.0 * pi / (double)frames) * 32767.0;
 	  buf[i * 4] = buf[i * 4 + 2] = val % 256;
