--- a/DES.xs
+++ b/DES.xs
@@ -36,7 +36,7 @@
 		if (key_len != sizeof(des_user_key))
 			croak("Invalid key");
 
-		perl_des_expand_key((i8 *)key, ks);
+		perl_des_expand_key((unsigned char *)key, ks);
 
 		ST(0) = sv_2mortal(newSVpv((char *)ks, sizeof(ks)));
 	}
@@ -66,7 +66,8 @@
 
 		(SvUPGRADE(output, SVt_PV));
 
-		perl_des_crypt(input, SvGROW(output, output_len), (i32 *)ks, enc_flag);
+		perl_des_crypt((unsigned char *)input, (unsigned char *)SvGROW(output, output_len),
+			       (unsigned long *)ks, enc_flag);
 
 		SvCUR_set(output, output_len);
 		*SvEND(output) = '\0';
