diff -up gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/gitg/gitg-plugins-engine.vala.orig gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/gitg/gitg-plugins-engine.vala
--- gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/gitg/gitg-plugins-engine.vala.orig	2025-05-12 23:54:48.000000000 +0200
+++ gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/gitg/gitg-plugins-engine.vala	2025-09-10 12:24:52.921168645 +0200
@@ -28,7 +28,7 @@ public class PluginsEngine : Peas.Engine
 	{
 		enable_loader("python");
 
-		var repo = Introspection.Repository.get_default();
+		var repo = Introspection.Repository.dup_default();
 
 		try
 		{
diff -up gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/vapi/gobject-introspection-1.0.vapi.orig gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/vapi/gobject-introspection-1.0.vapi
--- gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/vapi/gobject-introspection-1.0.vapi.orig	2025-05-12 23:54:48.000000000 +0200
+++ gitg-f7501bc827f1d4476336b0db0791335cf8a613c4/vapi/gobject-introspection-1.0.vapi	2025-09-10 12:26:07.403400436 +0200
@@ -1,4 +1,4 @@
-[CCode (cprefix = "GI", lower_case_cprefix = "g_i", cheader_filename = "girepository.h")]
+[CCode (cprefix = "GI", lower_case_cprefix = "gi_", cheader_filename = "girepository/girepository.h")]
 namespace Introspection
 {
 	[CCode (cprefix = "G_IREPOSITORY_ERROR_")]
@@ -16,7 +16,7 @@ namespace Introspection
 
 	[CCode (ref_function = "", unref_function = "")]
 	public class Repository {
-		public static unowned Repository get_default();
+		public static unowned Repository dup_default();
 		public static void prepend_search_path(string directory);
 		public static unowned GLib.SList<string> get_search_path();
 
