From: Than Ngo <than@redhat.com>
Date: Mon, 24 Feb 2020 11:17:04 +0100
Subject: [PATCH] Verbose build


diff --git a/make.sh b/make.sh
index 217967d59..4978d1cde 100755
--- a/make.sh
+++ b/make.sh
@@ -24,7 +24,7 @@ export LANG LC_ALL
 # thing" when run on the target machine, with the minor caveat that
 # any --xc-host parameter should be suitable for the host machine
 # instead of the target.
-sh make-config.sh "$@" --check-host-lisp || exit $?
+sh -x make-config.sh "$@" --check-host-lisp || exit $?
 
 . output/prefix.def
 . output/build-config
@@ -81,11 +81,11 @@ maybetime() {
         $@
     fi
 }
-maybetime sh make-host-1.sh
-maybetime sh make-target-1.sh
-maybetime sh make-host-2.sh
-maybetime sh make-target-2.sh
-maybetime sh make-target-contrib.sh
+maybetime sh -x make-host-1.sh
+maybetime sh -x make-target-1.sh
+maybetime sh -x make-host-2.sh
+maybetime sh -x make-target-2.sh
+maybetime sh -x make-target-contrib.sh
 
 # Confirm that default evaluation strategy is :INTERPRET if sb-fasteval was built
 src/runtime/sbcl --core output/sbcl.core --lose-on-corruption --noinform \
