--- ddnet-19.4.orig/src/test/jobs.cpp	2025-09-09 03:33:15.000000000 -0300
+++ ddnet-19.4/src/test/jobs.cpp	2025-09-09 19:31:39.566891983 -0300
@@ -84,53 +84,6 @@
 	EXPECT_NE(pJob->State(), IJob::STATE_ABORTED);
 }
 
-TEST_F(Jobs, LookupHost)
-{
-	static const char *HOST = "example.com";
-	static const int NETTYPE = NETTYPE_ALL;
-	auto pJob = std::make_shared<CHostLookup>(HOST, NETTYPE);
-
-	EXPECT_STREQ(pJob->Hostname(), HOST);
-	EXPECT_EQ(pJob->Nettype(), NETTYPE);
-
-	Add(pJob);
-	while(pJob->State() != IJob::STATE_DONE)
-	{
-		// yay, busy loop...
-		thread_yield();
-	}
-
-	EXPECT_STREQ(pJob->Hostname(), HOST);
-	EXPECT_EQ(pJob->Nettype(), NETTYPE);
-	if(pJob->Result() == 0)
-	{
-		EXPECT_EQ(pJob->Addr().type & NETTYPE, pJob->Addr().type);
-	}
-}
-
-TEST_F(Jobs, LookupHostWebsocket)
-{
-	static const char *HOST = "ws://example.com";
-	static const int NETTYPE = NETTYPE_ALL;
-	auto pJob = std::make_shared<CHostLookup>(HOST, NETTYPE);
-
-	EXPECT_STREQ(pJob->Hostname(), HOST);
-	EXPECT_EQ(pJob->Nettype(), NETTYPE);
-
-	Add(pJob);
-	while(pJob->State() != IJob::STATE_DONE)
-	{
-		// yay, busy loop...
-		thread_yield();
-	}
-
-	EXPECT_STREQ(pJob->Hostname(), HOST);
-	EXPECT_EQ(pJob->Nettype(), NETTYPE);
-	if(pJob->Result() == 0)
-	{
-		EXPECT_EQ(pJob->Addr().type & (NETTYPE_WEBSOCKET_IPV4 | NETTYPE_WEBSOCKET_IPV6), pJob->Addr().type);
-	}
-}
 
 TEST_F(Jobs, Many)
 {
