From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 1 Jun 2016 15:47:41 +0300
Subject: [PATCH] No unicode symbols in file extensions allowed

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/priv/mime.types b/priv/mime.types
index 1657386..a2728d0 100644
--- a/priv/mime.types
+++ b/priv/mime.types
@@ -571,7 +571,7 @@ application/vnd.fut-misnet
 application/vnd.fuzzysheet			fzs
 application/vnd.genomatix.tuxedo		txd
 application/vnd.geo+json			geojson
-application/vnd.geocube+xml			g3 g³
+application/vnd.geocube+xml			g3
 application/vnd.geogebra.file			ggb
 application/vnd.geogebra.tool			ggt
 application/vnd.geometry-explorer		gex gre
