diff --git a/src/RulesDigikamHost.cmake b/src/RulesDigikamHost.cmake
index f147e84..fb153ca 100644
--- a/src/RulesDigikamHost.cmake
+++ b/src/RulesDigikamHost.cmake
@@ -95,9 +95,11 @@ endif()
 # execute the command "qmake -query QT_INSTALL_PLUGINS" to get the path of plugins dir.
 
 execute_process(COMMAND ${QT_QMAKE_EXECUTABLE} -query QT_INSTALL_PLUGINS
-                OUTPUT_VARIABLE QT_PLUGINS_DIR
+                OUTPUT_VARIABLE QT_INSTALL_PLUGINS
                 OUTPUT_STRIP_TRAILING_WHITESPACE)
 
+set(QT_PLUGINS_DIR ${QT_INSTALL_PLUGINS} CACHE PATH "Qt plugin directory")
+
 if(NOT QT_PLUGINS_DIR)
 
     message(FATAL_ERROR "Qt plugin directory cannot be detected.")
