diff --git a/taglib-config.cmake b/taglib-config.cmake
--- a/taglib-config.cmake
+++ b/taglib-config.cmake
@@ -23,11 +23,6 @@
 else
   includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 fi
-if test -z "$exec_prefix"; then
-  libdir=@CMAKE_INSTALL_FULL_LIBDIR@
-else
-  libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
-fi
 
 flags=""
 
@@ -39,7 +34,7 @@
 do
   case $1 in
     --libs)
-      flags="$flags -L$libdir -ltag@TAGLIB_INSTALL_SUFFIX@ @ZLIB_LIBRARIES_FLAGS@"
+      flags="$flags -ltag@TAGLIB_INSTALL_SUFFIX@"
       ;;
     --cflags)
       flags="$flags -I$includedir -I$includedir/taglib@TAGLIB_INSTALL_SUFFIX@"
