--- cvc5-cvc5-1.1.2/src/theory/ff/uni_roots.cpp.orig	2024-03-01 10:21:00.000000000 -0700
+++ cvc5-cvc5-1.1.2/src/theory/ff/uni_roots.cpp	2024-03-29 10:15:03.668891240 -0600
@@ -29,7 +29,6 @@
 #include <CoCoA/RingZZ.H>
 #include <CoCoA/SmallFpImpl.H>
 #include <CoCoA/SparsePolyOps-RingElem.H>
-#include <CoCoA/SparsePolyOps-vector.H>
 #include <CoCoA/factor.H>
 #include <CoCoA/factorization.H>
 #include <CoCoA/random.H>
