From 9eb32427fcc727d0e4c55d16cfe1b43942ef66b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 27 Apr 2023 15:30:36 +0200
Subject: [PATCH] Do not use /bin/env in shebangs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/use.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/use.t b/t/use.t
index 7823f36..63ac6b1 100644
--- a/t/use.t
+++ b/t/use.t
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 
 use Test::More tests => 1;
 
-- 
2.40.1

