diff --git a/startx.in b/startx.in
index 777a6b2..e1b7c18 100644
--- a/startx.in
+++ b/startx.in
@@ -248,8 +248,10 @@ for i in $serverargs; do
         have_vtarg="yes"
     fi
 done
-if [ "$have_vtarg" = "no" ]; then
+if [ "$have_vtarg" = "no" -a x"$vtarg" != x ]; then
     serverargs="$serverargs $vtarg"
+    # Fedora specific mod to make X run as non root
+    export XORG_RUN_AS_USER_OK=1
 fi
 
 # if no display, use default
