From 2197eb772c8f3bf893ffc46841315ca7c0f1a30c Mon Sep 17 00:00:00 2001
From: Marcus Meissner <marcus@jet.franken.de>
Date: Mon, 29 Dec 2025 14:40:54 +0100
Subject: [PATCH] merge music-players.h from libmtp

---
 camlibs/ptp2/music-players.h | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/camlibs/ptp2/music-players.h
+++ b/camlibs/ptp2/music-players.h
@@ -3305,6 +3305,9 @@
   /* https://github.com/libmtp/libmtp/issues/334 */
   { "Huawei", 0x12d1, "Nova14", 0x1101,
       DEVICE_FLAGS_ANDROID_BUGS },
+  /* https://github.com/libmtp/libmtp/issues/334 */
+  { "Huawei", 0x12d1, "Nova14", 0x1101,
+      DEVICE_FLAGS_ANDROID_BUGS },
   /* https://sourceforge.net/p/libmtp/support-requests/276/ */
   { "Huawei", 0x12d1, "Y600", 0x2008,
       DEVICE_FLAGS_ANDROID_BUGS },
@@ -3619,9 +3622,11 @@
   { "Foxconn (for Vizio)", 0x0489, "VTAB1008", 0xe040,
       DEVICE_FLAGS_ANDROID_BUGS },
 /* ... seems to cause problems, see https://github.com/libmtp/libmtp/issues/346 
+/* ... seems to cause problems, see https://github.com/libmtp/libmtp/issues/346 
   { "Foxconn (for Lenovo)", 0x0489, "IdeaTab A2109/A2110/Medion LIFETAB S9714", 0xe111,
       DEVICE_FLAGS_ANDROID_BUGS },
 */
+*/
 
 
   /*
@@ -4175,6 +4180,8 @@
   /* https://github.com/libmtp/libmtp/issues/207 */
   { "Garmin", 0x091e, "Tactix 7 AMOLED", 0x51be, DEVICE_FLAGS_ANDROID_BUGS },
   /* https://github.com/libmtp/libmtp/issues/305 */
+  { "Garmin", 0x091e, "Edge 850", 0x521a, DEVICE_FLAGS_ANDROID_BUGS },
+  { "Garmin", 0x091e, "Venu 4 - 41mm", 0x5224, DEVICE_FLAGS_ANDROID_BUGS },
   { "Garmin", 0x091e, "Forerunner 970", 0x51d5, DEVICE_FLAGS_ANDROID_BUGS },
   /* https://github.com/libmtp/libmtp/issues/302 */
   { "Garmin", 0x091e, "Forerunner 570", 0x51de, DEVICE_FLAGS_ANDROID_BUGS },
