diff -r -U4 icon-9.5.24b--orig/src/icont/Makefile icon-9.5.24b--patched/src/icont/Makefile
--- icon-9.5.24b--orig/src/icont/Makefile	2024-11-19 22:01:39.000000000 +0100
+++ icon-9.5.24b--patched/src/icont/Makefile	2025-05-23 18:28:34.769905430 +0200
@@ -21,9 +21,8 @@
 icont:		$(OBJS) $(COBJS)
 		$(CC) $(CFLAGS) $(LDFLAGS) -o icont $(OBJS) $(COBJS)
 		rm -f ../../bin/icont$(EXE)
 		cp icont ../../bin
-		strip ../../bin/icont$(EXE)
 		(cd ../../bin; rm -f icon icon.exe; ln -s icont icon)
 
 $(OBJS):	$(HFILES) tproto.h
 
diff -r -U4 icon-9.5.24b--orig/src/runtime/Makefile icon-9.5.24b--patched/src/runtime/Makefile
--- icon-9.5.24b--orig/src/runtime/Makefile	2024-11-19 22:01:39.000000000 +0100
+++ icon-9.5.24b--patched/src/runtime/Makefile	2025-05-23 18:28:49.325970391 +0200
@@ -32,9 +32,8 @@
 	cd ../common; $(MAKE)
 	$(CC) $(RLINK) -o iconx $(XOBJS) $(COBJS) $(XL) $(RLIBS) $(TLIBS)
 	rm -f ../../bin/iconx$(EXE)
 	cp iconx ../../bin
-	strip $(SFLAGS) ../../bin/iconx$(EXE)
 
 $(COBJS):
 	cd ../common; $(MAKE)
 
