diff -uNdr reaver-1.6.6_orig/src/iface.c reaver-1.6.6_new/src/iface.c
--- reaver-1.6.6_orig/src/iface.c	2020-03-16 19:42:56.000000000 +0300
+++ reaver-1.6.6_new/src/iface.c	2020-03-17 11:36:12.239264539 +0300
@@ -32,7 +32,7 @@
  */
 
 #include "iface.h"
-#include "lwe/iwlib.h"
+#include "iwlib.h"
 #include "globule.h"
 #include <net/if.h>
 #include <netinet/in.h>
diff -uNdr reaver-1.6.6_orig/src/Makefile reaver-1.6.6_new/src/Makefile
--- reaver-1.6.6_orig/src/Makefile	2020-03-16 19:42:56.000000000 +0300
+++ reaver-1.6.6_new/src/Makefile	2020-03-17 11:37:55.636768800 +0300
@@ -34,8 +34,6 @@
 wps/wps.o wps/wps_common.o wps/wps_dev_attr.o wps/wps_enrollee.o \
 wps/wps_registrar.o wps/wps_ufd.o
 
-LWE_OBJS=lwe/iwlib.o
-
 TLS_OBJS= \
  tls/asn1.o \
  tls/bignum.o \
@@ -90,7 +88,7 @@
 
 
 LIB_OBJS=libwps/libwps.o $(WPS_OBJS) $(UTILS_OBJS) \
-	$(TLS_OBJS) $(CRYPTO_OBJS) $(LWE_OBJS)
+	$(TLS_OBJS) $(CRYPTO_OBJS)
 
 
 MAIN_OBJS=globule.o init.o sigint.o iface.o sigalrm.o \
@@ -99,14 +97,6 @@
 
 PROG_OBJS=$(MAIN_OBJS) exchange.o argsparser.o wpscrack.o wpsmon.o cracker.o main.o
 
-# Version of the Wireless Tools
-WT_VERSION := $(shell sed -ne "/WT_VERSION/{s:\([^0-9]*\)::;p;q;}" < lwe/iwlib.h )
-# Version of Wireless Extensions.
-WE_VERSION := $(shell sed -ne "/WE_VERSION/{s:\([^0-9]*\)::;p;q;}" < lwe/iwlib.h )
-# Always use local header for wireless extensions
-WEXT_HEADER = lwe/wireless.$(WE_VERSION).h
-
-GENH = lwe/wireless.h version.h
 
 all: wash reaver
 
@@ -120,7 +110,7 @@
 	ln -sf ./reaver wash
 
 reaver: $(PROG_OBJS) $(LIB_OBJS)
-	$(CC) $(CFLAGS) $(INC) $(PROG_OBJS) $(LIB_OBJS) $(LDFLAGS) -lpthread -o reaver
+	$(CC) $(CFLAGS) $(INC) $(PROG_OBJS) $(LIB_OBJS) $(LDFLAGS) -lpthread -liw -o reaver
 
 extest.o: exchange.c
 	$(CC) $(CFLAGS) -g3 -O0 -DEX_TEST -c exchange.c -o extest.o
@@ -131,10 +121,6 @@
 version.h: $(wildcard $(srcdir)/VERSION $(srcdir)/../.git)
 	printf '#define R_VERSION "%s"\n' "$$(cd $(srcdir); sh version.sh)"  > $@
 
-lwe/wireless.h: $(WEXT_HEADER)
-	cp $(WEXT_HEADER) lwe/wireless.h
-
-$(PROG_OBJS) $(LWE_OBJS): lwe/wireless.h
 
 install: wash reaver
 	$(INSTALL) -D -m 755 wash $(DESTDIR)$(exec_prefix)/bin/wash
