diff -up pnm2ppa-1.13/Makefile.am.ldflags pnm2ppa-1.13/Makefile.am
--- pnm2ppa-1.13/Makefile.am.ldflags	2023-10-20 13:52:38.470722125 +0200
+++ pnm2ppa-1.13/Makefile.am	2023-10-20 13:53:24.625056218 +0200
@@ -125,7 +125,7 @@ CLEANFILES = hash_ink.c make_hash_ink
 CC_FOR_BUILD = @CC_FOR_BUILD@
 CFLAGS_FOR_BUILD = $(RPM_OPT_FLAGS) -O2
 CPPFLAGS_FOR_BUILD = -I$(top_srcdir)
-LDFLAGS_FOR_BUILD =
+LDFLAGS_FOR_BUILD = $(LDFLAGS)
 LDLIBS_FOR_BUILD =
 
 make_hash_ink: make_hash_ink.c
