From f41c7acdfb1a1e21011fc34ff3ceeb9582e734ef Mon Sep 17 00:00:00 2001
From: "Ankur Sinha (Ankur Sinha Gmail)" <sanjay.ankur@gmail.com>
Date: Sun, 5 Apr 2020 11:10:50 +0100
Subject: [PATCH 4/5] Remove pysteps flags

---
 pysteps/CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/pysteps/CMakeLists.txt b/pysteps/CMakeLists.txt
index 140b5a3..de83408 100644
--- a/pysteps/CMakeLists.txt
+++ b/pysteps/CMakeLists.txt
@@ -35,11 +35,7 @@ add_custom_command(OUTPUT ${cython_output}
 
 
 # .cpp -> .so (it will be a single .so without dxependencies on libsteps)
-set(CYSTEPS_COMPILE_FLAGS
-    "-Wno-strict-aliasing -Wno-tautological-compare -Wno-useless-cast \
-     -Wno-format -Wno-double-promotion -Wno-unsequenced \
-     -Wno-unused-parameter -Wno-old-style-cast -Wno-unused-variable \
-     -Wno-shadow -Wno-parentheses -Wno-cast-align")
+set(CYSTEPS_COMPILE_FLAGS "")
 
 python_add_module(cysteps ${cython_output})
 
-- 
2.26.0

