--- a/macros.d/macros.cargo	2025-12-09 18:56:21.000000000 +0200
+++ b/macros.d/macros.cargo	2026-07-30 10:04:07.730465480 +0300
@@ -81,7 +81,8 @@
 #   -v <directory> - use vendored sources from <directory> instead of
 #                    %%cargo_registry
 #   -N             - Don't set up any registry. Only set up the build configuration.
-%cargo_prep(v:N)\
+%cargo_prep(V:v:N)\
+%{-V:%{?fedora:%{error:The -V flag for cargo_prep is not supported in Fedora}}}\
 %{-v:%{-N:%{error:-v and -N are mutually exclusive!}}}\
 (\
 set -euo pipefail\
@@ -140,6 +141,7 @@
 \
 EOF}}\
 %{__rm} -f Cargo.toml.orig\
+%{__cargo_to_rpm} --path Cargo.toml strip-foreign-targets\
 )
 
 # cargo_generate_buildrequires: prints BuildRequires that are needed for
