diff -rupN --no-dereference libxlsxwriter-RELEASE_1.1.7/cmake/FindMINIZIP.cmake libxlsxwriter-RELEASE_1.1.7-new/cmake/FindMINIZIP.cmake
--- libxlsxwriter-RELEASE_1.1.7/cmake/FindMINIZIP.cmake	2024-04-06 01:08:20.000000000 +0200
+++ libxlsxwriter-RELEASE_1.1.7-new/cmake/FindMINIZIP.cmake	2024-04-08 09:43:01.009358524 +0200
@@ -90,10 +90,10 @@ FindStaticLibs(MINIZIP)
 # -------
 
 if(MINIZIP_FOUND)
-    file(STRINGS "${MINIZIP_INCLUDE_DIRS}/zlib.h" MINIZIP_VERSION_CONTENTS REGEX "version [0-9]+\\.[0-9]+(\\.[0-9]+)?")
-    string(REGEX REPLACE ".*version ([0-9]+)\\.[0-9]+" "\\1" MINIZIP_VERSION_MAJOR "${MINIZIP_VERSION_CONTENTS}")
-    string(REGEX REPLACE ".*version [0-9]+\\.([0-9]+)" "\\1" MINIZIP_VERSION_MINOR "${MINIZIP_VERSION_CONTENTS}")
-    set(MINIZIP_VERSION_PATCH 0)
+    file(STRINGS "${MINIZIP_INCLUDE_DIRS}/minizip/mz.h" MINIZIP_VERSION_CONTENTS REGEX "MZ_VERSION.*\(\"[0-9]+\\.[0-9]+\\.[0-9]+\"\)")
+    string(REGEX REPLACE ".*([0-9]+)\\.[0-9]+\\.[0-9]+.*" "\\1" MINIZIP_VERSION_MAJOR "${MINIZIP_VERSION_CONTENTS}")
+    string(REGEX REPLACE ".*[0-9]+\\.([0-9]+)\\.[0-9]+.*" "\\1" MINIZIP_VERSION_MINOR "${MINIZIP_VERSION_CONTENTS}")
+    string(REGEX REPLACE ".*[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" MINIZIP_VERSION_PATCH "${MINIZIP_VERSION_CONTENTS}")
 
     set(MINIZIP_VERSION_STRING "${MINIZIP_VERSION_MAJOR}.${MINIZIP_VERSION_MINOR}.${MINIZIP_VERSION_PATCH}")
     set(MINIZIP_VERSION ${MINIZIP_VERSION_STRING})
