--- a/Makefile	2022-07-25 06:40:09.051418416 +0300
+++ b/Makefile	2022-07-25 06:54:30.208675877 +0300
@@ -75,8 +75,8 @@
 	make -C po install DESTDIR=$(DESTDIR)
 
 	# PolicyKit
-	mkdir -p $(DESTDIR)$(datadir)/polkit-1/actions
-	$(INSTALL_DATA) org.bleachbit.policy $(DESTDIR)$(datadir)/polkit-1/actions/
+	#mkdir -p $(DESTDIR)$(datadir)/polkit-1/actions
+	#$(INSTALL_DATA) org.bleachbit.policy $(DESTDIR)$(datadir)/polkit-1/actions/
 
 lint:
 	[ -x "$$(command -v pychecker)" ] ||  echo "WARNING: pychecker not found"
