diff -up netcdf-cxx4-4.3.0/cxx4/test_type.cpp.bigendian netcdf-cxx4-4.3.0/cxx4/test_type.cpp
--- netcdf-cxx4-4.3.0/cxx4/test_type.cpp.bigendian	2016-05-13 17:14:42.000000000 -0600
+++ netcdf-cxx4-4.3.0/cxx4/test_type.cpp	2017-02-14 14:26:12.246923900 -0700
@@ -267,9 +267,9 @@ try
     cout <<"    -----------   passed\n";
 
     cout <<left<<setw(57)<<"Testing NcEnumType::addMember()";
-    enumType1.addMember("Monday",1);
-    enumType1.addMember("Tuesday",7);
-    enumType1.addMember("Wednesday",-20);
+    enumType1.addMember("Monday",(short)1);
+    enumType1.addMember("Tuesday",(short)7);
+    enumType1.addMember("Wednesday",(short)-20);
     cout <<"    -----------   passed\n";
 
     cout <<left<<setw(57)<<"Testing NcEnumType::getBaseType() == and !=";
