From 1c53815f4ca68a9994ecfd1c295907a361f5e0d4 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 5 Oct 2021 16:39:13 +0100
Subject: [PATCH 1/1] Use /usr/lib64 for Fedora

---
 tests/merlin-wrapper | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/merlin-wrapper b/tests/merlin-wrapper
index 5b8d8184..2f7020d8 100755
--- a/tests/merlin-wrapper
+++ b/tests/merlin-wrapper
@@ -13,6 +13,6 @@ ocamlmerlin "$@" \
     | jq 'del(.heap_mbytes)' \
     | jq 'del(.cache)' \
     | jq 'del(.query_num)' \
-    | sed -e 's:"[^"]*lib/ocaml:"lib/ocaml:g' \
+    | sed -re 's:"[^"]*lib(64)?/ocaml:"lib/ocaml:g' \
     | sed -e 's:\\n:\
 :g'
-- 
2.32.0

