From c44222aadcfd365e7f3f9dcd4137f4c3a716f495 Mon Sep 17 00:00:00 2001
From: Montel Laurent <montel@kde.org>
Date: Fri, 25 Apr 2014 08:24:50 +0200
Subject: [PATCH 4/9] Fix option description

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c26bb1..68ad2fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@ find_package(Dmtx REQUIRED)
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-option(QT5_BUILD "Build Akonadi using the Qt5 framework" FALSE)
+option(QT5_BUILD "Build libprison using the Qt5 framework" FALSE)
 
 
 include_directories(${CMAKE_SOURCE_DIR}/lib ${QT_INCLUDES} ${CMAKE_CURRENT_BINARY_DIR} )
-- 
2.5.0

