diff -up abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/abookrc.5.p abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/abookrc.5
--- abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/abookrc.5.p	2014-01-16 12:51:40.000000000 +0100
+++ abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/abookrc.5	2014-02-27 13:44:46.473852144 +0100
@@ -80,7 +80,7 @@ only preserve the standard fields (see a
 discards any unknown field.
 .RE
 .IP
-Default is \fIstandard\fP.
+Default is \fIall\fP.
 
 .TP
 \fBindex_format\fP=format_string
diff -up abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/options.c.p abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/options.c
--- abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/options.c.p	2014-02-27 13:44:46.474852147 +0100
+++ abook-git-5840fceb1f91b066a4c4361b0599c417aa111386/options.c	2014-02-27 13:45:10.105923099 +0100
@@ -64,7 +64,7 @@ static struct option abook_vars[] = {
 	{ "use_ascii_only", OT_BOOL, BOOL_USE_ASCII_ONLY, FALSE },
 
 	{ "add_email_prevent_duplicates", OT_BOOL, BOOL_ADD_EMAIL_PREVENT_DUPLICATES, FALSE },
-	{ "preserve_fields", OT_STR, STR_PRESERVE_FIELDS, UL "standard" },
+	{ "preserve_fields", OT_STR, STR_PRESERVE_FIELDS, UL "all" },
 	{ "sort_field", OT_STR, STR_SORT_FIELD, UL "nick" },
 	{ "show_cursor", OT_BOOL, BOOL_SHOW_CURSOR, FALSE },
 	{ "use_mouse", OT_BOOL, BOOL_USE_MOUSE, FALSE },
