From 16df1e79cf2f2c2e429529c6ace3a9379922b90d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 12 Mar 2020 15:34:26 +0100
Subject: [PATCH] Disable autoinstalling the dependencies
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Makefile.PL | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index 66617a4..d41ac82 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -10,8 +10,5 @@ requires        'Test::MockObject'  => '1.08';
 license         'perl';
 version_from    'lib/Test/Mock/LWP.pm';
 
-# Auto-install all dependencies from CPAN
-auto_install;
-
 # Generate the Makefile
 WriteAll;
-- 
2.21.1

