From 02f24798d0f51ffae3e3d75522a4d5924f0b2309 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@fedoraproject.org>
Date: Wed, 15 Feb 2017 10:17:20 +0100
Subject: [PATCH 9/9] gcc7.0 compatibility

---
 src/uipp/dxuilib/EditorWindow.C | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/uipp/dxuilib/EditorWindow.C b/src/uipp/dxuilib/EditorWindow.C
index c406cc1..35c4682 100644
--- a/src/uipp/dxuilib/EditorWindow.C
+++ b/src/uipp/dxuilib/EditorWindow.C
@@ -5997,7 +5997,7 @@ char msg[128];
     //
     const char *tmpdir = theDXApplication->getTmpDirectory();
     int tmpdirlen = STRLEN(tmpdir);
-    if (!tmpdirlen) return FALSE;
+    if (!tmpdirlen) return 0;
     if (tmpdir[tmpdirlen-1] == '/') {
 	sprintf(netfilename, "%sdx%d.net", tmpdir, getpid());
 	sprintf(cfgfilename, "%sdx%d.cfg", tmpdir, getpid());
-- 
2.9.3

