--- p0rn-comfort-0.0.4/p0rn-proxy.static	2004-11-20 22:08:11.000000000 +0100
+++ p0rn-comfort-0.0.4/p0rn-proxy	2006-05-24 04:43:41.000000000 +0200
@@ -177,7 +177,7 @@
 #
 
 my $CVSVERSION = do { my @r = (q$Revision: 1.26 $ =~ /\d+/g); sprintf "%d."."%02d" x $#r, @r };
-my $URLBASE = $Static::PROXY_CTRL_BASE;
+my $URLBASE = $P0rn::Static::PROXY_CTRL_BASE;
 my ($ua, $thumbs, $pics, $downs);
 
 my @downs = ( 'queued', 'running', 'finished' );
@@ -207,35 +207,35 @@
 	my $uri = $2;
 	my $urishort = shorten_uri($uri);
 	
-	if ($cmd eq ${Static::PROXY_MARK_THUMB} ) {
+	if ($cmd eq ${P0rn::Static::PROXY_MARK_THUMB} ) {
 	    $thumbs->{$urishort} = 0;
 	    $content .= "<a href=\"$uri\">$urishort</a> has been marked as a thumbnail page.";
 	    $response->header(refresh => '5; url=javascript:window.close()');
 	    
-	} elsif ($cmd eq ${Static::PROXY_SINGLE_THUMB} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_SINGLE_THUMB} ) {
 	    $thumbs->{$urishort} = 1;
 	    $content .= "<a href=\"$uri\">$urishort</a> (thumbnail) can now be accessed once.";
 	    $response->header(refresh => "1; $uri");
 	    
-	} elsif ($cmd eq ${Static::PROXY_UNMARK_THUMB} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_UNMARK_THUMB} ) {
 	    delete $thumbs->{$urishort};
 	    $content .= "<a href=\"$uri\">$urishort</a> has been removed from the thumbnail list.";
 	    
-	} elsif ($cmd eq ${Static::PROXY_MARK_PICTURE} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_MARK_PICTURE} ) {
 	    $pics->{$urishort} = 0;
 	    $content .= "<a href=\"$uri\">$urishort</a> has been marked as a picture page.";
 	    $response->header(refresh => '5; url=javascript:window.close()');
 	    
-	} elsif ($cmd eq ${Static::PROXY_SINGLE_PICTURE} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_SINGLE_PICTURE} ) {
 	    $pics->{$urishort} = 1;
 	    $content .= "<a href=\"$uri\">$urishort</a> (picture) can now be accessed once.";
 	    $response->header(refresh => "1; $uri");
 	    
-	} elsif ($cmd eq ${Static::PROXY_UNMARK_PICTURE} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_UNMARK_PICTURE} ) {
 	    delete $pics->{$urishort};
 	    $content .= "<a href=\"$uri\">$urishort</a> has been removed from the picture list.";
 	    
-	} elsif ($cmd eq ${Static::PROXY_MARK_DOWNLOAD} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_MARK_DOWNLOAD} ) {
 	    $pics->{$urishort} = 0;
 	    $content .= "<a href=\"$uri\">$urishort</a> has been marked as a picture page.<br>";
 	    if (exists $downs->{$urishort}) {
@@ -246,59 +246,59 @@
 	    }
 	    $response->header(refresh => '5; url=javascript:window.close()');
 
-	} elsif ($cmd eq ${Static::PROXY_UNMARK_DOWNLOAD} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_UNMARK_DOWNLOAD} ) {
 	    delete $downs->{$urishort};
 	    $content .= "<a href=\"$uri\">$urishort</a> has been removed from the download list";
 
-	} elsif ($cmd eq ${Static::PROXY_MARK_DOWNLOAD} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_MARK_DOWNLOAD} ) {
 
 	    $content .= '<h1>p0rn-proxy main menu</h1><ul><ul><ul>' .
-		"<li><a href=\"$URLBASE/${Static::PROXY_LIST_THUMB}/_nouri_\">list thumbnail pages</a></li>" .
-		"<li><a href=\"$URLBASE/${Static::PROXY_LIST_PICTURE}/_nouri_\">list picture pages</a></li>" .
-		"<li><a href=\"$URLBASE/${Static::PROXY_LIST_DOWNLOAD}/_nouri_\">list download pages</a></li>" .
-		"</ul></ul><hr>p0rn-proxy ${Static::VERSION}/$CVSVERSION</ul>";
+		"<li><a href=\"$URLBASE/${P0rn::Static::PROXY_LIST_THUMB}/_nouri_\">list thumbnail pages</a></li>" .
+		"<li><a href=\"$URLBASE/${P0rn::Static::PROXY_LIST_PICTURE}/_nouri_\">list picture pages</a></li>" .
+		"<li><a href=\"$URLBASE/${P0rn::Static::PROXY_LIST_DOWNLOAD}/_nouri_\">list download pages</a></li>" .
+		"</ul></ul><hr>p0rn-proxy ${P0rn::Static::VERSION}/$CVSVERSION</ul>";
 	    
-	} elsif ($cmd eq ${Static::PROXY_LIST_THUMB} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_LIST_THUMB} ) {
 
 	    $content .= '<h1>thumbnail pages</h1>';
 
 	    foreach my $u (sort keys %{$thumbs}) {
-		$content .= "[<a href=\"$URLBASE/${Static::PROXY_UNMARK_THUMB}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
+		$content .= "[<a href=\"$URLBASE/${P0rn::Static::PROXY_UNMARK_THUMB}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
 	    }
 
-	} elsif ($cmd eq ${Static::PROXY_LIST_PICTURE} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_LIST_PICTURE} ) {
 
 	    $content .= '<h1>picture pages</h1>';
 
 	    foreach my $u (sort keys %{$pics}) {
-		$content .= "[<a href=\"$URLBASE/${Static::PROXY_UNMARK_PICTURE}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
+		$content .= "[<a href=\"$URLBASE/${P0rn::Static::PROXY_UNMARK_PICTURE}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
 	    }
 
-	} elsif ($cmd eq ${Static::PROXY_LIST_DOWNLOAD} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_LIST_DOWNLOAD} ) {
 
 	    $content .= '<h1>download pages</h1>';
 
 	    $content .= '<h2>queued</h2>';
 	    foreach my $u (sort keys %{$downs}) {
 		next unless $downs->{$u} == 0;
-		$content .= "[<a href=\"$URLBASE/${Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
+		$content .= "[<a href=\"$URLBASE/${P0rn::Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
 	    }
 
 	    $content .= '<h2>running</h2>';
 	    foreach my $u (sort keys %{$downs}) {
 		next unless $downs->{$u} == 1;
-		$content .= "[<a href=\"$URLBASE/${Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
+		$content .= "[<a href=\"$URLBASE/${P0rn::Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
 	    }
 
 	    $content .= '<h2>finished</h2>';
 	    foreach my $u (sort keys %{$downs}) {
 		next unless $downs->{$u} == 2;
-		$content .= "[<a href=\"$URLBASE/${Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
+		$content .= "[<a href=\"$URLBASE/${P0rn::Static::PROXY_UNMARK_DOWNLOAD}/$u\">del</a>] <a href=\"$u\">$u</a><br>";
 	    }
 
-	} elsif ($cmd eq ${Static::PROXY_GET_VERSION} ) {
+	} elsif ($cmd eq ${P0rn::Static::PROXY_GET_VERSION} ) {
 
-	    $content .= "\nVERSION: ${Static::VERSION}/$CVSVERSION\n";
+	    $content .= "\nVERSION: ${P0rn::Static::VERSION}/$CVSVERSION\n";
 	    
 	} else {
 	    $content .= "unknown command <tt>$cmd</tt>: <tt>".$request->uri().'</tt>';
@@ -342,8 +342,8 @@
 	    my $content = << "EOF";
 <html><head><title>thumbnail page</title></head><body>
 <a href="$uri">$urishort</a> is considered a thumbnail page and thus not displayed.<br>[
-<a href="$URLBASE/${Static::PROXY_SINGLE_THUMB}/$uri">access once</a>
-| <a href="$URLBASE/${Static::PROXY_UNMARK_THUMB}/$uri">access always</a>
+<a href="$URLBASE/${P0rn::Static::PROXY_SINGLE_THUMB}/$uri">access once</a>
+| <a href="$URLBASE/${P0rn::Static::PROXY_UNMARK_THUMB}/$uri">access always</a>
 ]</body></html>
 EOF
 ;
@@ -360,8 +360,8 @@
 		my $content = << "EOF";
 <html><head><title>already seen</title></head><body>
 <a href="$uri">$urishort</a> is considered an already seen picture page and thus not displayed.<br>[
-<a href="$URLBASE/${Static::PROXY_SINGLE_PICTURE}/$uri">access once</a>
-| <a href="$URLBASE/${Static::PROXY_UNMARK_PICTURE}/$uri">access always</a>
+<a href="$URLBASE/${P0rn::Static::PROXY_SINGLE_PICTURE}/$uri">access once</a>
+| <a href="$URLBASE/${P0rn::Static::PROXY_UNMARK_PICTURE}/$uri">access always</a>
 ]</body></html>
 EOF
 ;
@@ -389,10 +389,10 @@
 		    my $content = $response->content();
 		    
 		    my $linkline = '<p align="center">' .
-			'[ th: <a target="_top" href="'."$URLBASE/${Static::PROXY_MARK_THUMB}/$uri".'">mark as thumbnail page</a> ' .
-			'| ad: <a target="_top" href="'."$URLBASE/${Static::PROXY_ADMIN}/_nouri_".'">proxy admin</a> ' .
-			'| pi: <a target="_top" href="'."$URLBASE/${Static::PROXY_MARK_PICTURE}/$uri".'">mark as picture page</a> ' .
-			'| do: <a target="_top" href="'."$URLBASE/${Static::PROXY_MARK_DOWNLOAD}/$uri".'">mark as picture page and download</a> ' .
+			'[ th: <a target="_top" href="'."$URLBASE/${P0rn::Static::PROXY_MARK_THUMB}/$uri".'">mark as thumbnail page</a> ' .
+			'| ad: <a target="_top" href="'."$URLBASE/${P0rn::Static::PROXY_ADMIN}/_nouri_".'">proxy admin</a> ' .
+			'| pi: <a target="_top" href="'."$URLBASE/${P0rn::Static::PROXY_MARK_PICTURE}/$uri".'">mark as picture page</a> ' .
+			'| do: <a target="_top" href="'."$URLBASE/${P0rn::Static::PROXY_MARK_DOWNLOAD}/$uri".'">mark as picture page and download</a> ' .
 			']<br><br><p>';
 		    
 		    $content =~ s/<body([^>]*)>/<body$1>$linkline/i;
@@ -438,7 +438,7 @@
 ######[ Main program ]
 #
    
-print "this is p0rn-proxy ${Static::VERSION}/${CVSVERSION}\n";
+print "this is p0rn-proxy ${P0rn::Static::VERSION}/${CVSVERSION}\n";
 
 # define configuration options
 my $config = AppConfig->new( { CASE => 1 } );
@@ -486,7 +486,7 @@
 
 # create UserAgent
 $ua = LWP::UserAgent->new;
-$ua->agent("p0rn-proxy ${Static::VERSION}/${CVSVERSION}");
+$ua->agent("p0rn-proxy ${P0rn::Static::VERSION}/${CVSVERSION}");
 
 if ( defined $config->proxy()
      and $config->proxy() ne ""
@@ -496,11 +496,11 @@
 }
 
 # open dbs
-$thumbs = opendb('thumbz');
+$thumbs = P0rn::DB::opendb('thumbz');
 print "- thumbpages db loaded\n";
-$pics = opendb('picz');
+$pics = P0rn::DB::opendb('picz');
 print "- picpages db loaded\n";
-$downs = opendb('downz');
+$downs = P0rn::DB::opendb('downz');
 print "- download db loaded\n";
 
 print "- proxy started on ".$config->addr().":".$config->port()."\n";
