diff -urN clamav-unofficial-sigs-7.2.5/clamav-unofficial-sigs.sh clamav-unofficial-sigs-7.2.5.fixed/clamav-unofficial-sigs.sh
--- clamav-unofficial-sigs-7.2.5/clamav-unofficial-sigs.sh	2021-03-20 03:49:21.000000000 +0100
+++ clamav-unofficial-sigs-7.2.5.fixed/clamav-unofficial-sigs.sh	2023-04-17 10:20:54.020368888 +0200
@@ -1832,7 +1832,7 @@
     fi
 
     # Check there is an = for every set of "" optional whitespace \s* between = and "
-    config_check_vars="$(echo "$clean_config" | $grep_bin -c '=[[:space:]]*\"' )"
+    config_check_vars="$(echo "$clean_config" | $grep_bin -c '=[[:space:]]*"' )"
 
     if [ $(( ${#config_check} / 2 )) -ne "$config_check_vars" ] ; then
       xshok_pretty_echo_and_log "ERROR: Your configuration has errors, every = requires a pair of \"\""
