--- faust-2.81.8/build/CMakeLists.txt~	2025-09-29 08:03:58.000000000 -0500
+++ faust-2.81.8/build/CMakeLists.txt	2025-10-02 11:08:05.291803190 -0500
@@ -41,8 +41,8 @@
 option ( NONDETERMINISM_LINT "Add compiler warning for nondeterminism" off)
 option ( INCLUDE_WASMTIME    "Compile with wasmtime support" off)
 option ( SELF_CONTAINED_LIBRARY    "Don't search system architecture files." off)
-option ( LINK_LLVM_STATIC          "Link LLVM statically" on )
-option ( INCLUDE_LLVM_STATIC_IN_ARCHIVE "Combine static LLVM .a files and libfaust.a in a single .a file" on )
+option ( LINK_LLVM_STATIC          "Link LLVM statically" off )
+option ( INCLUDE_LLVM_STATIC_IN_ARCHIVE "Combine static LLVM .a files and libfaust.a in a single .a file" off )
 
 if (INCLUDE_WASMTIME)
     set(WASMTIME_LIB /usr/local/lib/libwasmtime.a)
