--- tree-sitter-0.24.7/Cargo.toml	1970-01-01T00:00:01+00:00
+++ tree-sitter-0.24.7/Cargo.toml	2026-01-23T19:39:35.973336+00:00
@@ -43,7 +43,7 @@
     "parsing",
     "text-editors",
 ]
-license = "MIT"
+license = "MIT AND Unicode-DFS-2016 AND BSD-2-Clause AND BSD-3-Clause AND LicenseRef-Fedora-Public-Domain"
 repository = "https://github.com/tree-sitter/tree-sitter"
 
 [lib]
@@ -65,16 +65,8 @@
 [dependencies.tree-sitter-language]
 version = "0.1"
 
-[dependencies.wasmtime-c-api]
-version = "25.0.2"
-features = ["cranelift"]
-optional = true
-default-features = false
-package = "wasmtime-c-api-impl"
-
 [build-dependencies.bindgen]
-version = "0.70.1"
-optional = true
+version = "0.72.1"
 
 [build-dependencies.cc]
 version = "1.1.22"
@@ -86,7 +78,7 @@
     "regex/perf",
     "regex-syntax/unicode",
 ]
-wasm = ["wasmtime-c-api"]
+wasm = []
 
 [lints.clippy]
 branches_sharing_code = "allow"
