From 5bc91bdebc42bb3a441c87b706288cea05fcd345 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Wed, 28 Aug 2024 10:42:48 -0400
Subject: [PATCH 1/6] Downstream-only: use the system double-conversion library

---
 pxr/base/tf/CMakeLists.txt | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/pxr/base/tf/CMakeLists.txt b/pxr/base/tf/CMakeLists.txt
index 6fc9c680e..551b5a3eb 100644
--- a/pxr/base/tf/CMakeLists.txt
+++ b/pxr/base/tf/CMakeLists.txt
@@ -105,11 +105,14 @@ function(add_py_dll_link_test)
 
 endfunction()
 
+find_library(double-conversion_LIBRARY NAMES double-conversion REQUIRED)
+
 pxr_library(tf
     LIBRARIES
         arch
         ${WINLIBS}
         TBB::tbb
+	${double-conversion_LIBRARY}
 
     PUBLIC_CLASSES
         anyUniquePtr
@@ -272,14 +275,6 @@ pxr_library(tf
 
     CPPFILES
         initConfig.cpp
-        pxrDoubleConversion/bignum-dtoa.cc
-        pxrDoubleConversion/bignum.cc
-        pxrDoubleConversion/cached-powers.cc
-        pxrDoubleConversion/double-to-string.cc
-        pxrDoubleConversion/fast-dtoa.cc
-        pxrDoubleConversion/fixed-dtoa.cc
-        pxrDoubleConversion/string-to-double.cc
-        pxrDoubleConversion/strtod.cc
         pxrLZ4/lz4.cpp
 
     PYMODULE_CPPFILES
-- 
2.53.0

