From 0caa009abce9d66d441cd548f45ca8fef8d4846d Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Thu, 29 Aug 2024 13:30:18 -0400
Subject: [PATCH 2/6] Downstream-only: use the system lz4 library

---
 pxr/base/tf/CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pxr/base/tf/CMakeLists.txt b/pxr/base/tf/CMakeLists.txt
index 551b5a3eb..772524a19 100644
--- a/pxr/base/tf/CMakeLists.txt
+++ b/pxr/base/tf/CMakeLists.txt
@@ -106,6 +106,8 @@ function(add_py_dll_link_test)
 endfunction()
 
 find_library(double-conversion_LIBRARY NAMES double-conversion REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LZ4 REQUIRED liblz4)
 
 pxr_library(tf
     LIBRARIES
@@ -113,6 +115,10 @@ pxr_library(tf
         ${WINLIBS}
         TBB::tbb
 	${double-conversion_LIBRARY}
+	${LZ4_LIBRARIES}
+
+    INCLUDE_DIRS
+        ${LZ4_INCLUDE_DIRS}
 
     PUBLIC_CLASSES
         anyUniquePtr
@@ -275,7 +281,6 @@ pxr_library(tf
 
     CPPFILES
         initConfig.cpp
-        pxrLZ4/lz4.cpp
 
     PYMODULE_CPPFILES
         module.cpp
-- 
2.53.0

