--- a/helix-stdx/Cargo.toml
+++ b/helix-stdx/Cargo.toml
@@ -22,9 +22,6 @@ once_cell = "1.21"
 regex-automata = "0.4.9"
 unicode-segmentation.workspace = true

-[target.'cfg(windows)'.dependencies]
-windows-sys = { version = "0.60", features = ["Win32_Foundation", "Win32_Security", "Win32_Security_Authorization", "Win32_Storage_FileSystem", "Win32_System_Threading"] }
-
 [target.'cfg(unix)'.dependencies]
 rustix = { version = "1.0", features = ["fs"] }

--- a/helix-view/Cargo.toml
+++ b/helix-view/Cargo.toml
@@ -52,9 +52,6 @@ log = "~0.4"
 parking_lot.workspace = true
 thiserror.workspace = true

-[target.'cfg(windows)'.dependencies]
-clipboard-win = { version = "5.4", features = ["std"] }
-
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
 rustix = { version = "1.0", features = ["fs"] }
