diff --git a/package/yapsy/PluginManager.py b/package/yapsy/PluginManager.py
index b3e2118..ddc672c 100644
--- a/package/yapsy/PluginManager.py
+++ b/package/yapsy/PluginManager.py
@@ -144,6 +144,7 @@ from yapsy import PLUGIN_NAME_FORBIDEN_STRING
 # imported for backward compatibility (this PluginInfo was imported
 # here before 1.10)
 from yapsy.PluginInfo import PluginInfo
+import inspect
 
 
 class PluginManager(object):
@@ -529,7 +530,7 @@ class PluginManager(object):
 				plugin_info_reference = None
 				for category_name in self.categories_interfaces:
 					try:
-						is_correct_subclass = issubclass(element, self.categories_interfaces[category_name])
+						is_correct_subclass = inspect.isclass(element) and issubclass(element, self.categories_interfaces[category_name])
 					except Exception:
 						exc_info = sys.exc_info()
 						log.debug("correct subclass tests failed for: %s in %s" % (element_name, candidate_filepath), exc_info=exc_info)
@@ -573,19 +574,20 @@ class PluginManager(object):
 
 		.. note:: Isolated and provided to be reused, but not to be reimplemented !
 		"""
-		# use imp to correctly load the plugin as a module
 		candidate_module = None
-		filepath_base = candidate_filepath.split('/')[-1]
+		submodule_search_locations = None
 		if os.path.isdir(candidate_filepath):
 			location = candidate_filepath + '/__init__.py'
+			submodule_search_locations = []
 		else:
 			location = candidate_filepath + '.py'
-
-		if (spec := importlib.util.spec_from_file_location(filepath_base, location)):
+		spec = importlib.util.spec_from_file_location(plugin_module_name, location, submodule_search_locations=submodule_search_locations)
+		if spec != None:
 			candidate_module = importlib.util.module_from_spec(spec)
 			sys.modules[plugin_module_name] = candidate_module
 			spec.loader.exec_module(candidate_module)
 		return candidate_module
+
 	
 	def instanciateElementWithImportInfo(self, element, element_name,
 										 plugin_module_name, candidate_filepath):
