Index: soqt-SoQt-1.6.0/CMakeLists.txt
===================================================================
--- soqt-SoQt-1.6.0.orig/CMakeLists.txt
+++ soqt-SoQt-1.6.0/CMakeLists.txt
@@ -3,8 +3,8 @@ cmake_minimum_required(VERSION 3.0)
 set(gui qt)
 set(Gui Qt)
 set(GUI QT)
-set(WIDGET QWidget*)
-set(EVENT QEvent*)
+set(WIDGET "QWidget *")
+set(EVENT "QEvent *")
 set(COMPONENTHEADER Q_OBJECT)
 
 function(dump_variable)
Index: soqt-SoQt-1.6.0/src/CMakeLists.txt
===================================================================
--- soqt-SoQt-1.6.0.orig/src/CMakeLists.txt
+++ soqt-SoQt-1.6.0/src/CMakeLists.txt
@@ -88,6 +88,7 @@ endif()
 # Create the library target
 if(SO${GUI}_BUILD_SHARED_LIBS)
   add_library(${PROJECT_NAME} SHARED ${HDRS} ${COMMON_HDRS} ${SRCS} ${COMMON_SRCS} ${COMMON_RESOURCES} ${MOC_SRCS})
+  set(SOQT_EXTRA_LIBS "-l${PROJECT_NAME}")
 else()
   add_library(${PROJECT_NAME} STATIC ${HDRS} ${COMMON_HDRS} ${SRCS} ${COMMON_SRCS} ${COMMON_RESOURCES} ${MOC_SRCS})
 endif()
