From 420d0a508202a31cef8ae052f6b144ebea0d7804 Mon Sep 17 00:00:00 2001
From: clover2 Fedora maintainer <clover2-owner@fedoraproject.org>
Date: Thu, 8 May 2025 16:29:54 +0900
Subject: [PATCH] block TCGETA usage on ppc64le

---
 src/class_system.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/class_system.c b/src/class_system.c
index 3fd5c33..9eb6bea 100644
--- a/src/class_system.c
+++ b/src/class_system.c
@@ -5203,6 +5203,7 @@ BOOL System_initialize_system_calls_system(CLVALUE** stack_ptr, CLVALUE* lvar, s
 #ifdef TCSETSF
     system->mClassFields[LAST_INITIALIZE_FIELD_NUM_ON_COMMAND_SYSTEM+125].mValue.mIntValue = TCSETSF;
 #endif
+#ifndef __powerpc64__
 #ifdef TCGETA
     system->mClassFields[LAST_INITIALIZE_FIELD_NUM_ON_COMMAND_SYSTEM+126].mValue.mIntValue = TCGETA;
 #endif
@@ -5215,6 +5216,7 @@ BOOL System_initialize_system_calls_system(CLVALUE** stack_ptr, CLVALUE* lvar, s
 #ifdef TCSETAF
     system->mClassFields[LAST_INITIALIZE_FIELD_NUM_ON_COMMAND_SYSTEM+129].mValue.mIntValue = TCSETAF;
 #endif
+#endif
 #ifdef TCSBRK
     system->mClassFields[LAST_INITIALIZE_FIELD_NUM_ON_COMMAND_SYSTEM+130].mValue.mIntValue = TCSBRK;
 #endif
-- 
2.49.0

