--- proptest-derive-0.8.0/Cargo.toml
+++ proptest-derive-0.8.0/Cargo.toml
@@ -51,10 +51,6 @@
 path = "tests/assoc.rs"
 
 [[test]]
-name = "compiletest"
-path = "tests/compiletest.rs"
-
-[[test]]
 name = "enum"
 path = "tests/enum.rs"
 
@@ -118,11 +114,6 @@
 name = "weight"
 path = "tests/weight.rs"
 
-[[bench]]
-name = "large_enum"
-path = "benches/large_enum.rs"
-harness = false
-
 [dependencies.proc-macro2]
 version = "1.0"
 
@@ -137,15 +128,5 @@
     "full",
 ]
 
-[dev-dependencies.compiletest_rs]
-version = "0.11"
-features = [
-    "tmp",
-    "stable",
-]
-
-[dev-dependencies.criterion]
-version = "0.8"
-
 [dev-dependencies.proptest]
 version = "1.10.0"
