From e5def9657bb4b661b191c8844c62ddcbf073ff12 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Sat, 4 Mar 2023 12:08:02 +0100
Subject: [PATCH 2/7] Update mapbox-gl-native

---
 src/3rdparty/mapbox-gl-native | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/3rdparty/mapbox-gl-native b/src/3rdparty/mapbox-gl-native
index d3101bbc2..d8054f1ef 160000
--- a/src/3rdparty/mapbox-gl-native
+++ b/src/3rdparty/mapbox-gl-native
@@ -1 +1 @@
-Subproject commit d3101bbc22edd41c9036ea487d4a71eabd97823d
+Subproject commit d8054f1efa0b97b291a6f8c7fff9090b0fd4328e
-- 
2.51.1

