From 5a84c3dc8ea04a524cb40efb21f07bf557d7f693 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 3 Sep 2019 12:29:04 +0200
Subject: [PATCH] Install an uncompressed manual page
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Makefile | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index c3fe89c..db3984b 100644
--- a/Makefile
+++ b/Makefile
@@ -11,28 +11,30 @@ DOCDIR = $(PREFIX)/share/doc/root-tail
 
 SOURCES = root-tail.c
 
-all: root-tail man
+all: root-tail root-tail.1
 
 root-tail: $(SOURCES) config.h
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(SOURCES) -lX11 -lXfixes
 
-man: root-tail.1.gz
+man: root-tail.1
 
-root-tail.1.gz: root-tail.man
+root-tail.1: root-tail.man
 	cp root-tail.man root-tail.1
+
+root-tail.1.gz: root-tail.1
 	gzip -f9 root-tail.1
 
 clean:
-	rm -f root-tail root-tail.o root-tail.1.gz
+	rm -f root-tail root-tail.o root-tail.1 root-tail.1.gz
 
 install: all
 	install -D -o root -g root root-tail $(BINDIR)
-	install -D -m 0644 -o root -g root root-tail.1.gz $(MANDIR)
+	install -D -m 0644 -o root -g root root-tail.1 $(MANDIR)
 	install -D -m 0644 -o root -g root README $(DOCDIR)
 	install -m 0644 -o root -g root Changes $(DOCDIR)
 
 uninstall:
-	rm -f $(BINDIR)/root-tail $(MANDIR)/root-tail.1.gz
+	rm -f $(BINDIR)/root-tail $(MANDIR)/root-tail.1
 	rm -f $(DOCDIR)/README $(DOCDIR)/Changes
 	rmdir --ignore-fail-on-non-empty $(DOCDIR)
 
-- 
2.21.0

