diff -up ./Bugzilla/Constants.pm.orig ./Bugzilla/Constants.pm
--- ./Bugzilla/Constants.pm.orig	2018-02-18 12:22:45.541837039 +0100
+++ ./Bugzilla/Constants.pm	2018-02-18 12:22:53.538784733 +0100
@@ -213,7 +213,7 @@ use constant LOCAL_FILE  => 'bugzilla-up
 # When true CSS and JavaScript assets will be concatanted and minified at
 # run-time, to reduce the number of requests required to render a page.
 # Setting this to a false value can help debugging.
-use constant CONCATENATE_ASSETS => 1;
+use constant CONCATENATE_ASSETS => 0;
 
 # These are unique values that are unlikely to match a string or a number,
 # to be used in criteria for match() functions and other things. They start
