diff --git a/pyobd b/pyobd
index 9852fac..e015105 100755
--- a/pyobd
+++ b/pyobd
@@ -594,7 +594,7 @@ the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  0211
         available = []
         for i in range(256):
           try: #scan standart ttyS*
-            s = serial.Serial(i)
+            s = serial.Serial("/dev/ttyS"+str(i))
             available.append(s.portstr)
             s.close()   # explicit close 'cause of delayed GC in java
           except serial.SerialException:
