--- a/llvm-11.1.0.src/include/llvm/Support/Base64.h	2021-02-03 14:51:10.000000000 -0700
+++ b/llvm-11.1.0.src/include/llvm/Support/Base64.h	2023-01-30 10:58:39.424270016 -0700
@@ -13,6 +13,7 @@
 #ifndef LLVM_SUPPORT_BASE64_H
 #define LLVM_SUPPORT_BASE64_H
 
+#include <cstdint>
 #include <string>
 
 namespace llvm {
--- a/llvm-11.1.0.src/include/llvm/Support/Signals.h	2021-02-03 14:51:10.000000000 -0700
+++ b/llvm-11.1.0.src/include/llvm/Support/Signals.h	2023-01-30 09:51:35.823901524 -0700
@@ -14,6 +14,7 @@
 #ifndef LLVM_SUPPORT_SIGNALS_H
 #define LLVM_SUPPORT_SIGNALS_H
 
+#include <cstdint>
 #include <string>
 
 namespace llvm {
