--- MinicomputerV1.41.old/SConstruct	2010-01-30 12:18:53.000000000 -0500
+++ MinicomputerV1.41/SConstruct	2010-02-13 16:32:00.000000000 -0500
@@ -45,6 +45,9 @@ if not conf.CheckLibWithHeader('asound',
 if not conf.CheckLibWithHeader('pthread', 'pthread.h','c'):
 	print 'Did not find pthread library, exiting!'
 	Exit(1)
+if not conf.CheckLibWithHeader('m', 'math.h','c'):
+	print 'Did not find math library, exiting!'
+	Exit(1)
 env = conf.Finish()
 
 print"-                    checking dependencies for the editor:"
