diff -rU3 MinicomputerV1.41-orig/SConstruct MinicomputerV1.41/SConstruct
--- MinicomputerV1.41-orig/SConstruct	2019-11-01 17:38:52.141938193 +0100
+++ MinicomputerV1.41/SConstruct	2019-11-01 17:41:06.219307341 +0100
@@ -1,6 +1,6 @@
-print" "
-print"Minicomputer-------------- "
-print"-                     1/2:configuring"
+print(" ")
+print("Minicomputer-------------- ")
+print("-                     1/2:configuring")
 
 if ARGUMENTS.get('64bit', 0):
 	env = Environment(CCFLAGS = '-m64')
@@ -34,47 +34,47 @@
 conf = Configure(env)
 
 if not conf.CheckLibWithHeader('jack', 'jack/jack.h','c'):
-	print 'Did not find jack, exiting!'
+	print('Did not find jack, exiting!')
 	Exit(1)
 if not conf.CheckLibWithHeader('lo', 'lo/lo.h','c'):
-	print 'Did not find liblo for OSC, exiting!'
+	print('Did not find liblo for OSC, exiting!')
 	Exit(1)
 if not conf.CheckLibWithHeader('asound', 'alsa/asoundlib.h','c'):
-	print 'Did not find alsa, exiting!'
+	print('Did not find alsa, exiting!')
 	Exit(1)
 if not conf.CheckLibWithHeader('pthread', 'pthread.h','c'):
-	print 'Did not find pthread library, exiting!'
+	print('Did not find pthread library, exiting!')
 	Exit(1)
 if not conf.CheckLibWithHeader('m', 'math.h','c'):
-	print 'Did not find math library, exiting!'
+	print('Did not find math library, exiting!')
 	Exit(1)
 env = conf.Finish()
 
-print"-                    checking dependencies for the editor:"
+print("-                    checking dependencies for the editor:")
 
 guiconf = Configure(guienv)
 if not guiconf.CheckLibWithHeader('lo', 'lo/lo.h','c'):
-	print 'Did not find liblo for OSC, exiting!'
+	print('Did not find liblo for OSC, exiting!')
 	Exit(1)
 if not guiconf.CheckLibWithHeader('fltk', 'FL/Fl.H','c++'):
-	print 'Did not find FLTK for the gui, exiting!'
+	print('Did not find FLTK for the gui, exiting!')
 	Exit(1)
 if not guiconf.CheckLibWithHeader('asound', 'alsa/asoundlib.h','c'):
-	print 'Did not find alsa, exiting!'
+	print('Did not find alsa, exiting!')
 	Exit(1)
 if not guiconf.CheckLibWithHeader('pthread', 'pthread.h','c'):
-	print 'Did not find pthread library, exiting!'
+	print('Did not find pthread library, exiting!')
 	Exit(1)
 guienv = guiconf.Finish()
 guienv.Append(CPPFLAGS = ['-O3','-Wall','-fmessage-length=0'])
 
-print"-                     2/2:compiling"
-print"-                     building the engine:"
+print("-                     2/2:compiling")
+print("-                     building the engine:")
 
 env.Program('minicomputerCPU','cpu/main.c');
 
-print""
-print"-                     building the editor:"
+print("")
+print("-                     building the editor:")
 
 guienv.Program('minicomputer',['editor/main.cpp','editor/Memory.cpp','editor/syntheditor.cxx']);
 
