diff -ur xterm-388.orig/aclocal.m4 xterm-388/aclocal.m4
--- xterm-388.orig/aclocal.m4	2023-10-02 23:23:32.000000000 +0200
+++ xterm-388/aclocal.m4	2023-11-28 15:32:31.790171161 +0100
@@ -4759,7 +4759,7 @@
 	int x = (XkbBI_Info |XkbBI_MinorError |XkbBI_MajorError |XkbBI_TerminalBell |XkbBI_MarginBell);
 	Atom y = 0;
 	(void)x;
-	XkbBell((Display *)0, (Widget)0, 0, y);
+	XkbBell((Display *)0, (Window)0, 0, y);
 ],[cf_cv_xkb_bell_ext=yes],[cf_cv_xkb_bell_ext=no])
 ])
 test "$cf_cv_xkb_bell_ext" = yes && AC_DEFINE(HAVE_XKB_BELL_EXT,1,[Define 1 if we have XKB Bell extension])
diff -ur xterm-388.orig/configure xterm-388/configure
--- xterm-388.orig/configure	2023-10-08 23:26:36.000000000 +0200
+++ xterm-388/configure	2023-11-28 15:32:53.598984387 +0100
@@ -17023,7 +17023,7 @@
 	int x = (XkbBI_Info |XkbBI_MinorError |XkbBI_MajorError |XkbBI_TerminalBell |XkbBI_MarginBell);
 	Atom y = 0;
 	(void)x;
-	XkbBell((Display *)0, (Widget)0, 0, y);
+	XkbBell((Display *)0, (Window)0, 0, y);
 
   ;
   return 0;
