--- a/src/syncevo/CurlTransportAgent.cpp
+++ b/src/syncevo/CurlTransportAgent.cpp
@@ -47,10 +47,10 @@ CurlTransportAgent::CurlTransportAgent() :
      * its read callback and reply is stored in write callback
      */
     CURLcode code;
-    auto readDataCallback = [] (void *buffer, size_t size, size_t nmemb, void *stream) noexcept {
+    auto readDataCallback = +[] (void *buffer, size_t size, size_t nmemb, void *stream) noexcept {
         return static_cast<CurlTransportAgent *>(stream)->readData(buffer, size * nmemb);
     };
-    auto writeDataCallback = [] (void *buffer, size_t size, size_t nmemb, void *stream) noexcept {
+    auto writeDataCallback = +[] (void *buffer, size_t size, size_t nmemb, void *stream) noexcept {
         return static_cast<CurlTransportAgent *>(stream)->writeData(buffer, size * nmemb);
     };
     if ((code = curl_easy_setopt(m_easyHandle, CURLOPT_NOPROGRESS, false)) ||
