--- CMakeLists.txt.qtnetwork	2008-02-17 23:03:38.000000000 +0100
+++ CMakeLists.txt	2009-04-28 16:33:20.393489520 +0200
@@ -8,7 +8,7 @@
 
 kde4_add_plugin( kio_ftps ${mySources} )
 
-target_link_libraries(kio_ftps ${KDE4_KDECORE_LIBS} kio )
+target_link_libraries(kio_ftps ${KDE4_KDECORE_LIBS} kio QtNetwork)
 
 install(TARGETS kio_ftps  DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES ftps.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
