From a9206d4ddff7257c0d16a014bdff3e8a6359479a Mon Sep 17 00:00:00 2001
From: Philipp Kaeser <kaeser@gubbe.ch>
Date: Sat, 24 Jan 2026 15:14:22 +0100
Subject: [PATCH] build: Forces /etc only for /usr, and irrespective of
 CMAKE_INSTALL_SYSCONFDIR.

---
 CMakeLists.txt | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cd3bbe4..47b6a34a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,8 +13,8 @@
 # limitations under the License.
 #
 # Default arguments:
-# cmake -DCMAKE_INSTALL_PREFIX:PATH=${HOME}/.local -DCMAKE_INSTALL_SYSCONFDIR=/etc -Dconfig_DOXYGEN_CRITICAL=ON -Dconfig_WERROR=ON -DIWYU_MODE=FAIL -B build
-# CC=clang cmake -DCMAKE_INSTALL_PREFIX:PATH="${HOME}/.local" -DCMAKE_INSTALL_SYSCONFDIR=/etc -Dconfig_DOXYGEN_CRITICAL=ON -Dconfig_WERROR=ON -DIWYU_MODE=FAIL -B build-clang
+# cmake -DCMAKE_INSTALL_PREFIX:PATH=${HOME}/.local -Dconfig_DOXYGEN_CRITICAL=ON -Dconfig_WERROR=ON -DIWYU_MODE=FAIL -B build
+# CC=clang cmake -DCMAKE_INSTALL_PREFIX:PATH="${HOME}/.local" -Dconfig_DOXYGEN_CRITICAL=ON -Dconfig_WERROR=ON -DIWYU_MODE=FAIL -B build-clang
 #
 # Useor "/etc".
 
@@ -70,10 +70,9 @@ ENDIF()
 
 # Defaults to /usr/local/lib for installation.
 INCLUDE(GNUInstallDirs)
-# Override: If using a non-standard prefix, install the config files beneath
-# the prefix. Otherwise, fall back to use ${CMAKE_INSTALL_SYSCONFDIR} (/etc/).
-IF(NOT CMAKE_INSTALL_PREFIX STREQUAL "/usr" AND NOT CMAKE_INSTALL_PREFIX STREQUAL "/usr/local")
-  SET(WLMAKER_INSTALL_CONFIG_DIR "${CMAKE_INSTALL_PREFIX}/etc/xdg/wlmaker")
+IF(CMAKE_INSTALL_PREFIX STREQUAL "/usr")
+  # For targets /usr, forces configurations into XDG config default path.
+  SET(WLMAKER_INSTALL_CONFIG_DIR "/etc/xdg/wlmaker")
 ELSE()
   SET(WLMAKER_INSTALL_CONFIG_DIR "${CMAKE_INSTALL_SYSCONFDIR}/xdg/wlmaker")
 ENDIF()
