diff -rupN --no-dereference qtactiveqt-everywhere-src-5.15.18/src/tools/idc/idc.pro qtactiveqt-everywhere-src-5.15.18-new/src/tools/idc/idc.pro
--- qtactiveqt-everywhere-src-5.15.18/src/tools/idc/idc.pro	2025-03-30 13:49:30.000000000 +0200
+++ qtactiveqt-everywhere-src-5.15.18-new/src/tools/idc/idc.pro	2025-11-08 23:29:22.507571992 +0100
@@ -1,4 +1,3 @@
-option(host_build)
 CONFIG += force_bootstrap
 SOURCES = main.cpp
 
