diff --git a/src/Common.pri b/src/Common.pri
index eb717f4..6c6be38 100644
--- a/src/Common.pri
+++ b/src/Common.pri
@@ -38,10 +38,8 @@ target.path = $$PWD/../dist
 INSTALLS += target
 
 unix:linux* {
-    QMAKE_CXXFLAGS += -Wall -Woverloaded-virtual -Wno-deprecated -Wuninitialized -O
+    QMAKE_CXXFLAGS += -Wall -Woverloaded-virtual -Wno-deprecated -Wuninitialized
     QMAKE_CFLAGS += -Wstrict-prototypes -Wmissing-prototypes
-
-    QMAKE_LFLAGS += -Wl,--rpath=\'\$\$ORIGIN\'
 }
 
 win32-g++ {
