diff -uNr golly-4.3-src.old/gui-wx/makefile-gtk golly-4.3-src/gui-wx/makefile-gtk
--- golly-4.3-src.old/gui-wx/makefile-gtk	2024-06-18 23:16:15.533911573 +0200
+++ golly-4.3-src/gui-wx/makefile-gtk	2024-06-18 23:23:17.923706409 +0200
@@ -61,13 +61,13 @@
 endif
 
 # For Python script support
-PYTHON_INCLUDE = -I`$(PYTHON) -c "import distutils.sysconfig as s; print(s.get_python_inc())"`
+PYTHON_INCLUDE = -I`$(PYTHON) -c "import sysconfig; print(sysconfig.get_paths()['include'])"`
 # we don't want to link against a specific library: PYTHON_LINK = -lpython2.3
 # following generates "-Xlinker -export-dynamic" but doesn't seem to be needed
 # PYTHON_LINK = `python -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LINKFORSHARED')"`
 PYTHON_LINK =
 # the following determines the filename for the python dynamic library (e.g. libpython3.8.so)
-PYTHON_SHLIB = `$(PYTHON) -c "import distutils.sysconfig as s; print(s.get_config_var('LDLIBRARY'))"`
+PYTHON_SHLIB = `$(PYTHON) -c "import sysconfig; print(sysconfig.get_config_var('LDLIBRARY'))"`
 PYTHON_PREFS = -DPYTHON_SHLIB="$(PYTHON_SHLIB)"
 
 LUA_INCLUDE = `pkg-config --cflags lua`
