--- cargo-util-terminal-0.1.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ cargo-util-terminal-0.1.2/Cargo.toml	2026-10-03T01:29:11.207416+00:00
@@ -70,18 +70,6 @@
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.184"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_Console",
-    "Win32_System_JobObjects",
-    "Win32_System_Threading",
-]
-
 [lints.clippy]
 dbg_macro = "warn"
 disallowed_methods = "warn"
@@ -103,3 +91,4 @@
 
 [lints.rustdoc]
 private_intra_doc_links = "allow"
+
