diff --git a/Makefile b/Makefile
index 20ebc7a..5d0ebdf 100644
--- a/Makefile
+++ b/Makefile
@@ -128,10 +128,6 @@ FILES	=	\
 		Makefile \
 		foo2zjs.c \
 		foo2zjs.1in \
-		jbig.c \
-		jbig.h \
-		jbig_ar.c \
-		jbig_ar.h \
 		zjsdecode.c \
 		zjsdecode.1in \
 		zjs.h \
@@ -296,7 +292,7 @@ MANPAGES+=	foo2ddst-wrapper.1 foo2ddst.1 ddstdecode.1
 MANPAGES+=	gipddecode.1
 MANPAGES+=	foo2zjs-pstops.1 arm2hpdl.1 usb_printerid.1
 MANPAGES+=	printer-profile.1
-LIBJBG	=	jbig.o jbig_ar.o
+LIBJBG	=	-ljbig
 BINPROGS=
 
 ifeq ($(UNAME),Linux)
@@ -482,34 +478,34 @@ all-done:
 	@echo "yourself."
 
 
-foo2ddst: foo2ddst.o $(LIBJBG)
+foo2ddst: foo2ddst.o
 	$(CC) $(CFLAGS) -o $@ foo2ddst.o $(LIBJBG)
 
-foo2hbpl2: foo2hbpl2.o $(LIBJBG)
+foo2hbpl2: foo2hbpl2.o
 	$(CC) $(CFLAGS) -o $@ foo2hbpl2.o $(LIBJBG)
 
-foo2hp: foo2hp.o $(LIBJBG)
+foo2hp: foo2hp.o
 	$(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG)
 
-foo2hiperc: foo2hiperc.o $(LIBJBG)
+foo2hiperc: foo2hiperc.o
 	$(CC) $(CFLAGS) -o $@ foo2hiperc.o $(LIBJBG)
 
-foo2lava: foo2lava.o $(LIBJBG)
+foo2lava: foo2lava.o
 	$(CC) $(CFLAGS) -o $@ foo2lava.o $(LIBJBG)
 
-foo2oak: foo2oak.o $(LIBJBG)
+foo2oak: foo2oak.o
 	$(CC) $(CFLAGS) -o $@ foo2oak.o $(LIBJBG)
 
-foo2qpdl: foo2qpdl.o $(LIBJBG)
+foo2qpdl: foo2qpdl.o
 	$(CC) $(CFLAGS) -o $@ foo2qpdl.o $(LIBJBG)
 
-foo2slx: foo2slx.o $(LIBJBG)
+foo2slx: foo2slx.o
 	$(CC) $(CFLAGS) -o $@ foo2slx.o $(LIBJBG)
 
-foo2xqx: foo2xqx.o $(LIBJBG)
+foo2xqx: foo2xqx.o
 	$(CC) $(CFLAGS) -o $@ foo2xqx.o $(LIBJBG)
 
-foo2zjs: foo2zjs.o $(LIBJBG)
+foo2zjs: foo2zjs.o
 	$(CC) $(CFLAGS) -o $@ foo2zjs.o $(LIBJBG)
 
 
@@ -595,43 +591,43 @@ ifeq ($(UNAME),Darwin)
 	cd osx-hotplug; $(MAKE) all
 endif
 
-ok: ok.o $(LIBJBG)
+ok: ok.o
 	$(CC) $(CFLAGS) ok.o $(LIBJBG) -o $@
 
-ddstdecode: ddstdecode.o $(LIBJBG)
+ddstdecode: ddstdecode.o
 	$(CC) $(CFLAGS) ddstdecode.o $(LIBJBG) -o $@
 
-gipddecode: gipddecode.o $(LIBJBG)
+gipddecode: gipddecode.o
 	$(CC) $(CFLAGS) gipddecode.o $(LIBJBG) -o $@
 
-hbpldecode: hbpldecode.o $(LIBJBG)
+hbpldecode: hbpldecode.o
 	$(CC) $(CFLAGS) hbpldecode.o $(LIBJBG) -o $@
 
-hipercdecode: hipercdecode.o $(LIBJBG)
+hipercdecode: hipercdecode.o
 	$(CC) $(CFLAGS) hipercdecode.o $(LIBJBG) -o $@
 
-lavadecode: lavadecode.o $(LIBJBG)
+lavadecode: lavadecode.o
 	$(CC) $(CFLAGS) lavadecode.o $(LIBJBG) -o $@
 
-oakdecode: oakdecode.o $(LIBJBG)
+oakdecode: oakdecode.o
 	$(CC) $(CFLAGS) -g oakdecode.o $(LIBJBG) -o $@
 
-opldecode: opldecode.o $(LIBJBG)
+opldecode: opldecode.o
 	$(CC) $(CFLAGS) -g opldecode.o $(LIBJBG) -o $@
 
-qpdldecode: qpdldecode.o $(LIBJBG)
+qpdldecode: qpdldecode.o
 	$(CC) $(CFLAGS) qpdldecode.o $(LIBJBG) -o $@
 
-splcdecode: splcdecode.o $(LIBJBG)
+splcdecode: splcdecode.o
 	$(CC) $(CFLAGS) splcdecode.o $(LIBJBG) -lz -o $@
 
-slxdecode: slxdecode.o $(LIBJBG)
+slxdecode: slxdecode.o
 	$(CC) $(CFLAGS) slxdecode.o $(LIBJBG) -o $@
 
-xqxdecode: xqxdecode.o $(LIBJBG)
+xqxdecode: xqxdecode.o
 	$(CC) $(CFLAGS) xqxdecode.o $(LIBJBG) -o $@
 
-zjsdecode: zjsdecode.o $(LIBJBG)
+zjsdecode: zjsdecode.o
 	$(CC) $(CFLAGS) zjsdecode.o $(LIBJBG) -o $@
 
 command2foo2lava-pjl: command2foo2lava-pjl.o
@@ -1258,7 +1254,7 @@ clean:
 	-rm -f $(PROGS) $(BINPROGS) $(SHELLS)
 	-rm -f *.zc *.zm *.zm1
 	-rm -f xxx.* xxxomatic
-	-rm -f foo2zjs.o jbig.o jbig_ar.o zjsdecode.o foo2hp.o
+	-rm -f foo2zjs.o zjsdecode.o foo2hp.o
 	-rm -f foo2oak.o oakdecode.o
 	-rm -f foo2xqx.o xqxdecode.o
 	-rm -f foo2lava.o lavadecode.o
@@ -1288,29 +1284,27 @@ clean:
 #
 # Header dependencies
 #
-jbig.o: jbig.h
-
-foo2ddst.o: jbig.h ddst.h
-foo2hiperc.o: jbig.h hiperc.h
-foo2hp.o: jbig.h zjs.h cups.h
-foo2hbpl2.o: jbig.h hbpl.h
-foo2lava.o: jbig.h
-foo2oak.o: jbig.h oak.h
-foo2qpdl.o: jbig.h qpdl.h
-foo2slx.o: jbig.h slx.h
-foo2xqx.o: jbig.h xqx.h
-foo2zjs.o: jbig.h zjs.h
-
-ddstdecode.o: ddst.h jbig.h
-gipddecode.o: slx.h jbig.h
-hbpldecode.o: jbig.h
-hipercdecode.o: hiperc.h jbig.h
-lavadecode.o: jbig.h
-opldecode.o: jbig.h
-qpdldecode.o: jbig.h
-slxdecode.o: slx.h jbig.h
-xqxdecode.o: xqx.h jbig.h
-zjsdecode.o: jbig.h zjs.h
+foo2ddst.o: ddst.h
+foo2hiperc.o: hiperc.h
+foo2hp.o: zjs.h cups.h
+foo2hbpl2.o: hbpl.h
+foo2lava.o:
+foo2oak.o: oak.h
+foo2qpdl.o: qpdl.h
+foo2slx.o: slx.h
+foo2xqx.o: xqx.h
+foo2zjs.o: zjs.h
+
+ddstdecode.o: ddst.h
+gipddecode.o: slx.h
+hbpldecode.o:
+hipercdecode.o: hiperc.h
+lavadecode.o:
+opldecode.o:
+qpdldecode.o:
+slxdecode.o: slx.h
+xqxdecode.o: xqx.h
+zjsdecode.o: zjs.h
 
 #
 # foo2* Regression tests
