diff -rupN --no-dereference qt-creator-opensource-src-20.0.0-beta1/cmake/Findelfutils.cmake qt-creator-opensource-src-20.0.0-beta1-new/cmake/Findelfutils.cmake
--- qt-creator-opensource-src-20.0.0-beta1/cmake/Findelfutils.cmake	2026-05-07 12:58:43.000000000 +0200
+++ qt-creator-opensource-src-20.0.0-beta1-new/cmake/Findelfutils.cmake	2026-05-12 14:30:14.575403349 +0200
@@ -32,7 +32,7 @@ find_path(ELFUTILS_INCLUDE_DIR
     "${ELFUTILS_INSTALL_DIR}" ENV ELFUTILS_INSTALL_DIR "${CMAKE_PREFIX_PATH}"
 )
 
-foreach(lib dw elf eu_compat)
+foreach(lib dw elf eu_compat debuginfod)
   find_library(ELFUTILS_LIB_${lib}
     NAMES ${lib}
     PATH_SUFFIXES lib
@@ -48,7 +48,7 @@ find_package_handle_standard_args(elfuti
 )
 
 if(elfutils_FOUND)
-  foreach(lib dw elf)
+  foreach(lib dw elf debuginfod)
     if (NOT TARGET elfutils::${lib})
       add_library(elfutils::${lib} UNKNOWN IMPORTED)
       set_target_properties(elfutils::${lib}
