diff -rupN --no-dereference qt-creator-opensource-src-20.0.0-beta1/share/applications/org.qt-project.qtcreator.desktop qt-creator-opensource-src-20.0.0-beta1-new/share/applications/org.qt-project.qtcreator.desktop
--- qt-creator-opensource-src-20.0.0-beta1/share/applications/org.qt-project.qtcreator.desktop	2026-05-07 12:58:49.000000000 +0200
+++ qt-creator-opensource-src-20.0.0-beta1-new/share/applications/org.qt-project.qtcreator.desktop	2026-05-12 14:30:12.620821101 +0200
@@ -8,5 +8,5 @@ Icon=QtProject-qtcreator
 StartupWMClass=qtcreator
 Terminal=false
 Categories=Development;IDE;Qt;
-MimeType= text/x-c++src;text/x-c++hdr;text/x-xsrc;application/x-designer;application/vnd.qt.qmakeprofile;application/vnd.qt.xml.resource;
+MimeType=text/x-c++src;text/x-c++hdr;text/x-xsrc;application/x-designer;application/vnd.qt.qmakeprofile;application/vnd.qt.xml.resource;
 
