From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 11 Jun 2024 19:06:15 +0200
Subject: [PATCH] Fix for i686

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/c_src/emmap.cpp b/c_src/emmap.cpp
index b97f0a3..aaa0024 100644
--- a/c_src/emmap.cpp
+++ b/c_src/emmap.cpp
@@ -318,7 +318,7 @@ static bool decode_flags(ErlNifEnv* env, ERL_NIF_TERM list, int* prot, int* flag
   int    f = MAP_FILE;
   int    p = PROT_READ;
   int    arity;
-  size_t tmp;
+  unsigned long int tmp;
   const  ERL_NIF_TERM* tuple;
 
   *auto_unlink  = false;
@@ -403,7 +403,7 @@ static bool decode_flags(ErlNifEnv* env, ERL_NIF_TERM list, int* prot, int* flag
       *auto_unlink = true;
     } else if (enif_get_tuple(env, head, &arity, &tuple) && arity == 2) {
       if (enif_is_identical  (tuple[0], ATOM_ADDRESS) &&
-          enif_get_ulong(env, tuple[1], address)) {
+          enif_get_ulong(env, tuple[1], (unsigned long*) address)) {
         // If address is given, set the "MAP_FIXED" option
         if (*address) f |= MAP_FIXED;
         continue;
@@ -413,7 +413,7 @@ static bool decode_flags(ErlNifEnv* env, ERL_NIF_TERM list, int* prot, int* flag
       } else if (enif_is_identical (tuple[0], ATOM_MAX_INC_SIZE) &&
                  enif_get_ulong(env,tuple[1], &tmp)) {
         if (tmp > 0)  // Otherwise use default size
-          *max_inc_size = tmp;
+          *max_inc_size = (size_t) tmp;
         continue;
       } else
         return false;
