diff -urp '--exclude=*~' rack-cors-1.1.1.rack3/lib/rack/cors.rb rack-cors-1.1.1/lib/rack/cors.rb
--- rack-cors-1.1.1.rack3/lib/rack/cors.rb	2025-11-14 00:08:29.632358049 +0900
+++ rack-cors-1.1.1/lib/rack/cors.rb	2025-11-14 12:06:16.181430914 +0900
@@ -17,7 +17,7 @@ module Rack
     ENV_KEY     = 'rack.cors'.freeze
 
     OPTIONS     = 'OPTIONS'.freeze
-    VARY        = 'Vary'.freeze
+    VARY        = 'vary'.freeze
 
     DEFAULT_VARY_HEADERS = ['Origin'].freeze
 
@@ -221,7 +221,7 @@ module Rack
       end
 
       class Result
-        HEADER_KEY = 'X-Rack-CORS'.freeze
+        HEADER_KEY = 'x-rack-cors'.freeze
 
         MISS_NO_ORIGIN = 'no-origin'.freeze
         MISS_NO_PATH   = 'no-path'.freeze
@@ -400,11 +400,11 @@ module Rack
 
         def to_headers(env)
           h = {
-            'Access-Control-Allow-Origin'     => origin_for_response_header(env[HTTP_ORIGIN]),
-            'Access-Control-Allow-Methods'    => methods.collect{|m| m.to_s.upcase}.join(', '),
-            'Access-Control-Expose-Headers'   => expose.nil? ? '' : expose.join(', '),
-            'Access-Control-Max-Age'          => max_age.to_s }
-          h['Access-Control-Allow-Credentials'] = 'true' if credentials
+            'access-control-allow-origin'     => origin_for_response_header(env[HTTP_ORIGIN]),
+            'access-control-allow-methods'    => methods.collect{|m| m.to_s.upcase}.join(', '),
+            'access-control-expose-headers'   => expose.nil? ? '' : expose.join(', '),
+            'access-control-max-age'          => max_age.to_s }
+          h['access-control-allow-credentials'] = 'true' if credentials
           h
         end
 
@@ -421,7 +421,7 @@ module Rack
           def to_preflight_headers(env)
             h = to_headers(env)
             if env[HTTP_ACCESS_CONTROL_REQUEST_HEADERS]
-              h.merge!('Access-Control-Allow-Headers' => env[HTTP_ACCESS_CONTROL_REQUEST_HEADERS])
+              h.merge!('access-control-allow-headers' => env[HTTP_ACCESS_CONTROL_REQUEST_HEADERS])
             end
             h
           end
diff -urp '--exclude=*~' rack-cors-1.1.1.rack3/test/unit/cors_test.rb rack-cors-1.1.1/test/unit/cors_test.rb
--- rack-cors-1.1.1.rack3/test/unit/cors_test.rb	2025-11-14 11:51:18.830945314 +0900
+++ rack-cors-1.1.1/test/unit/cors_test.rb	2025-11-14 12:04:18.290498499 +0900
@@ -1,3 +1,5 @@
+require 'rack/builder'
+require 'rack/lint'
 require 'minitest/autorun'
 require 'rack/test'
 require 'mocha/minitest'
@@ -47,7 +49,7 @@ class FakeProxy
 
   def call(env)
     status, headers, body = @app.call(env)
-    headers['Vary'] = %w(Origin User-Agent)
+    headers['vary'] = %w(Origin User-Agent)
     [status, headers, body]
   end
 end
@@ -68,7 +70,7 @@ describe Rack::Cors do
       use FakeProxy if options[:proxy]
       map('/') do
         run proc { |env|
-          [200, {'Content-Type' => 'text/html'}, ['success']]
+          [200, {'content-type' => 'text/html'}, ['success']]
         }
       end
     end
@@ -421,7 +423,7 @@ describe Rack::Cors do
       @app ||= Rack::Builder.new do
         use Rack::Cors
         use Rack::Lint
-        run ->(env) { [200, {'Content-Type' => 'text/html'}, ['hello']] }
+        run ->(env) { [200, {'content-type' => 'text/html'}, ['hello']] }
       end
     end
 
@@ -467,7 +469,7 @@ describe Rack::Cors do
           end
         end
         map('/') do
-          run ->(env) { [200, {'Content-Type' => 'text/html'}, ['hello']] }
+          run ->(env) { [200, {'content-type' => 'text/html'}, ['hello']] }
         end
       end
     end
@@ -488,7 +490,7 @@ describe Rack::Cors do
           end
         end
         map('/') do
-          run ->(env) { [200, {'Content-Type' => 'text/html'}, ['hello']] }
+          run ->(env) { [200, {'content-type' => 'text/html'}, ['hello']] }
         end
       end
     end
