diff -Naur a/Makefile b/Makefile
--- a/Makefile	2026-01-30 15:19:53.373271006 +0100
+++ b/Makefile	2026-01-30 15:20:08.956452996 +0100
@@ -133,7 +133,7 @@
 	$(call build,build,building retis (debug))
 
 release: ebpf
-	$(call build,build --release,building retis (release),$(RELEASE_FLAGS))
+	$(call build,build --profile rpm,building retis (release),$(RELEASE_FLAGS))
 
 test: ebpf
 ifeq ($(COV),1)
