diff --git a/configure b/configure
index 3df01fa..273c8d3 100755
--- a/configure
+++ b/configure
@@ -1707,7 +1707,7 @@ then
 	fi
 fi
 
-EXTRA_CFLAGS="-Werror -W -Wall -Wbad-function-cast -Wcast-align -Wcast-qual -DXTSTRINGDEFINES -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wmissing-format-attribute"
+EXTRA_CFLAGS="-W -Wall -Wbad-function-cast -Wcast-align -Wcast-qual -DXTSTRINGDEFINES -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wmissing-format-attribute"
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
diff --git a/configure.in b/configure.in
index 6533951..d0d0362 100644
--- a/configure.in
+++ b/configure.in
@@ -15,7 +15,7 @@ then
 	fi
 fi
 
-EXTRA_CFLAGS="-Werror -W -Wall -Wbad-function-cast -Wcast-align -Wcast-qual -DXTSTRINGDEFINES -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wmissing-format-attribute"
+EXTRA_CFLAGS="-W -Wall -Wbad-function-cast -Wcast-align -Wcast-qual -DXTSTRINGDEFINES -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wmissing-format-attribute"
 AC_SUBST(EXTRA_CFLAGS)
 dnl Checks for programs.
 AC_PROG_CC
diff --git a/src/cpio.c b/src/cpio.c
index 67a8256..6559395 100644
--- a/src/cpio.c
+++ b/src/cpio.c
@@ -37,6 +37,7 @@
 #endif
 
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <sys/stat.h>
 #include <sys/param.h>
diff --git a/src/gen_subs.c b/src/gen_subs.c
index e5db820..e1f58b3 100644
--- a/src/gen_subs.c
+++ b/src/gen_subs.c
@@ -38,6 +38,7 @@
 #endif
 
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <sys/stat.h>
 #include <sys/param.h>
diff --git a/src/tar.c b/src/tar.c
index 33904bc..95623d6 100644
--- a/src/tar.c
+++ b/src/tar.c
@@ -38,6 +38,7 @@
 #endif
 
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <sys/stat.h>
 #include <sys/param.h>
