From a7db2ccac83b4cf596ec6543014b6e56ce893d6a Mon Sep 17 00:00:00 2001
From: "Steven A. Falco" <stevenfalco@gmail.com>
Date: Mon, 5 Jan 2026 09:55:38 -0500
Subject: [PATCH] Fedora doesn't need the static library

---
 host/libhackrf/src/CMakeLists.txt | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/host/libhackrf/src/CMakeLists.txt b/host/libhackrf/src/CMakeLists.txt
index f10ce61c..633a726d 100644
--- a/host/libhackrf/src/CMakeLists.txt
+++ b/host/libhackrf/src/CMakeLists.txt
@@ -94,15 +94,15 @@ if(ENABLE_SHARED_LIB)
 endif()
 
 # Static library
-if(ENABLE_STATIC_LIB)
-  add_library(hackrf_static STATIC hackrf.c)
-  if(MSVC)
-    set_target_properties(hackrf_static PROPERTIES OUTPUT_NAME "hackrf_static")
-  else()
-    set_target_properties(hackrf_static PROPERTIES OUTPUT_NAME "hackrf")
-  endif()
-  libhackrf_common_settings(hackrf_static)
-endif()
+#if(ENABLE_STATIC_LIB)
+#  add_library(hackrf_static STATIC hackrf.c)
+#  if(MSVC)
+#    set_target_properties(hackrf_static PROPERTIES OUTPUT_NAME "hackrf_static")
+#  else()
+#    set_target_properties(hackrf_static PROPERTIES OUTPUT_NAME "hackrf")
+#  endif()
+#  libhackrf_common_settings(hackrf_static)
+#endif()
 
 if(${WIN32})
   install(
-- 
2.52.0

