diff -up itk4.1.0/configure.in.libdir itk4.1.0/configure.in
--- itk4.1.0/configure.in.libdir	2017-07-28 11:01:26.000000000 -0600
+++ itk4.1.0/configure.in	2019-09-18 21:20:01.450013803 -0600
@@ -211,6 +211,39 @@ AC_SUBST(itkstub_LIB_FILE)
 AC_SUBST(itk_LIB_FILE)
 
 #--------------------------------------------------------------------
+# These are for itkConfig.sh
+#--------------------------------------------------------------------
+
+# pkglibdir must be a fully qualified path and (not ${exec_prefix}/lib)
+eval pkglibdir="${libdir}/${PACKAGE_NAME}${PACKAGE_VERSION}"
+if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+    eval ITK_LIB_FLAG="-litk${PACKAGE_VERSION}${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub${PACKAGE_VERSION}${DBGX}"
+else
+    eval ITK_LIB_FLAG="-litk`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+fi
+ITK_BUILD_LIB_SPEC="-L`pwd` ${itk_LIB_FLAG}"
+ITK_LIB_SPEC="-L${libdir} ${itk_LIB_FLAG}"
+
+ITK_BUILD_STUB_LIB_SPEC="-L`pwd` ${ITK_STUB_LIB_FLAG}"
+ITK_STUB_LIB_SPEC="-L${pkglibdir} ${ITK_STUB_LIB_FLAG}"
+ITK_BUILD_STUB_LIB_PATH="`pwd`/${ITK_STUB_LIB_FILE}"
+ITK_STUB_LIB_PATH="${pkglibdir}/${ITK_STUB_LIB_FILE}"
+
+AC_SUBST(ITK_BUILD_LIB_SPEC)
+AC_SUBST(ITK_LIB_SPEC)
+AC_SUBST(ITK_BUILD_STUB_LIB_SPEC)
+AC_SUBST(ITK_STUB_LIB_SPEC)
+AC_SUBST(ITK_BUILD_STUB_LIB_PATH)
+AC_SUBST(ITK_STUB_LIB_PATH)
+
+# ITK_SRC_DIR must be a fully qualified path
+eval ITK_SRC_DIR="$srcdir"
+ITK_SRC_DIR=`cd "${ITK_SRC_DIR}"; pwd`
+AC_SUBST(ITK_SRC_DIR)
+
+#--------------------------------------------------------------------
 # Find tclsh so that we can run pkg_mkIndex to generate the pkgIndex.tcl
 # file during the install process.  Don't run the TCLSH_PROG through
 # ${CYGPATH} because it's being used directly by make.
diff -up itk4.1.0/configure.libdir itk4.1.0/configure
--- itk4.1.0/configure.libdir	2017-07-28 11:01:26.000000000 -0600
+++ itk4.1.0/configure	2019-09-18 21:20:01.453013824 -0600
@@ -9360,6 +9360,32 @@ rm -f conftest*
 #--------------------------------------------------------------------
 
 
+#--------------------------------------------------------------------
+# These are for itkConfig.sh
+#--------------------------------------------------------------------
+
+# pkglibdir must be a fully qualified path and (not ${exec_prefix}/lib)
+eval pkglibdir="${libdir}/${PACKAGE_NAME}${PACKAGE_VERSION}"
+if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+    eval ITK_LIB_FLAG="-litk${PACKAGE_VERSION}${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub${PACKAGE_VERSION}${DBGX}"
+else
+    eval ITK_LIB_FLAG="-litk`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+fi
+ITK_BUILD_LIB_SPEC="-L`pwd` ${itk_LIB_FLAG}"
+ITK_LIB_SPEC="-L${libdir} ${itk_LIB_FLAG}"
+
+ITK_BUILD_STUB_LIB_SPEC="-L`pwd` ${ITK_STUB_LIB_FLAG}"
+ITK_STUB_LIB_SPEC="-L${pkglibdir} ${ITK_STUB_LIB_FLAG}"
+ITK_BUILD_STUB_LIB_PATH="`pwd`/${ITK_STUB_LIB_FILE}"
+ITK_STUB_LIB_PATH="${pkglibdir}/${ITK_STUB_LIB_FILE}"
+
+
+# ITK_SRC_DIR must be a fully qualified path
+eval ITK_SRC_DIR="$srcdir"
+ITK_SRC_DIR=`cd "${ITK_SRC_DIR}"; pwd`
+
 
 
 #--------------------------------------------------------------------
diff -up itk4.1.0/Makefile.in.libdir itk4.1.0/Makefile.in
--- itk4.1.0/Makefile.in.libdir	2017-07-28 11:00:34.000000000 -0600
+++ itk4.1.0/Makefile.in	2019-09-18 21:20:01.453013824 -0600
@@ -94,8 +94,8 @@ includedir	= @includedir@
 DESTDIR		=
 
 PKG_DIR		= $(PACKAGE_NAME)$(PACKAGE_VERSION)
-pkgdatadir	= $(datadir)/$(PKG_DIR)
-pkglibdir	= $(libdir)/$(PKG_DIR)
+pkgdatadir	= $(datadir)/tcl@TCL_VERSION@/$(PKG_DIR)
+pkglibdir	= $(libdir)/tcl@TCL_VERSION@/$(PKG_DIR)
 pkgincludedir	= $(includedir)/$(PKG_DIR)
 
 top_builddir	= .
@@ -364,13 +364,15 @@ install-lib-binaries:
 	@mkdir -p $(DESTDIR)$(pkglibdir)
 	@list='$(lib_BINARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
-	    echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p"; \
-	    $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p; \
 	    stub=`echo $$p|sed -e "s/.*\(stub\).*/\1/"`; \
 	    if test "x$$stub" = "xstub"; then \
+		echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p"; \
+		$(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p; \
 		echo " $(RANLIB_STUB) $(DESTDIR)$(pkglibdir)/$$p"; \
 		$(RANLIB_STUB) $(DESTDIR)$(pkglibdir)/$$p; \
 	    else \
+		echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(libdir)/$$p"; \
+		$(INSTALL_PROGRAM) $$p $(DESTDIR)$(libdir)/$$p; \
 		echo " $(RANLIB) $(DESTDIR)$(pkglibdir)/$$p"; \
 		$(RANLIB) $(DESTDIR)$(pkglibdir)/$$p; \
 	    fi; \
diff -up itk4.1.0/pkgIndex.tcl.in.libdir itk4.1.0/pkgIndex.tcl.in
--- itk4.1.0/pkgIndex.tcl.in.libdir	2019-09-18 21:20:01.454013832 -0600
+++ itk4.1.0/pkgIndex.tcl.in	2019-09-18 21:20:45.225321448 -0600
@@ -2,5 +2,5 @@
 
 if {![package vsatisfies [package provide Tcl] 8.6]} return
 if {[string length [package provide Itcl]] && ![package vsatisfies [package provide Itcl] 4.1]} return
-package ifneeded itk @PACKAGE_VERSION@ [list load [file join $dir "@PKG_LIB_FILE@"] Itk]
-package ifneeded Itk @PACKAGE_VERSION@ [list load [file join $dir "@PKG_LIB_FILE@"] Itk]
+package ifneeded itk @PACKAGE_VERSION@ [list load [file join $dir .. .. "@PKG_LIB_FILE@"] Itk]
+package ifneeded Itk @PACKAGE_VERSION@ [list load [file join $dir .. .. "@PKG_LIB_FILE@"] Itk]
