From 02bb37946cafdb5892c04a2e9804566c3d36e8a0 Mon Sep 17 00:00:00 2001
From: "Ankur Sinha (Ankur Sinha Gmail)" <sanjay.ankur@gmail.com>
Date: Mon, 29 Aug 2022 17:22:11 +0100
Subject: [PATCH] Quote various cmake var values

---
 cmake/arbor-config.cmake.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/cmake/arbor-config.cmake.in b/cmake/arbor-config.cmake.in
index c9edecdc..7b6e1391 100644
--- a/cmake/arbor-config.cmake.in
+++ b/cmake/arbor-config.cmake.in
@@ -46,11 +46,11 @@ endfunction()
 
 set(ARB_VECTORIZE @ARB_VECTORIZE@)
 set(ARB_WITH_GPU @ARB_WITH_GPU@)
-set(ARB_ARCH @ARB_ARCH@)
-set(ARB_MODCC_FLAGS @ARB_MODCC_FLAGS@)
-set(ARB_CXX @CMAKE_CXX_COMPILER@)
-set(ARB_CXX_FLAGS @CMAKE_CXX_FLAGS@)
-set(ARB_CXX_FLAGS_TARGET @ARB_CXX_FLAGS_TARGET_FULL@)
+set(ARB_ARCH "@ARB_ARCH@")
+set(ARB_MODCC_FLAGS "@ARB_MODCC_FLAGS@")
+set(ARB_CXX "@CMAKE_CXX_COMPILER@")
+set(ARB_CXX_FLAGS "@CMAKE_CXX_FLAGS@")
+set(ARB_CXX_FLAGS_TARGET "@ARB_CXX_FLAGS_TARGET_FULL@")
 
 _append_property(arbor::arbor INTERFACE_LINK_LIBRARIES @arbor_add_import_libs@)
 _append_property(arbor::arborenv INTERFACE_LINK_LIBRARIES @arborenv_add_import_libs@)
-- 
2.37.2

