diff -Naur arbor-0.9.0-original/python/pyarb.hpp arbor-0.9.0/python/pyarb.hpp
--- arbor-0.9.0-original/python/pyarb.hpp	2023-08-10 10:47:38.000000000 +0100
+++ arbor-0.9.0/python/pyarb.hpp	2025-11-03 08:24:00.093256311 +0000
@@ -13,18 +13,6 @@
 
 #include <pybind11/pybind11.h>
 
-// Version check
-#define mk_tok(x) #x
-#define mk_ver(M, m, p) mk_tok(M) "." mk_tok(m) "." mk_tok(p)
-#define PB11_ERR(M, m, p) "Required version of pybind11 is 2.8.1 <= version < 3.0.0 Found " mk_ver(M, m, p)
-static_assert((PYBIND11_VERSION_HEX >= 0x02080100)
-              &&
-              (PYBIND11_VERSION_HEX  < 0x03000000),
-              PB11_ERR(PYBIND11_VERSION_MAJOR, PYBIND11_VERSION_MINOR, PYBIND11_VERSION_PATCH));
-#undef PB11_ERR
-#undef mk_ver
-#undef mk_tok
-
 namespace pyarb {
 
 // Sample recorder object interface.
