diff --git a/syntax_checkers/python/codec.py b/syntax_checkers/python/codec.py
index 6e980c7..850df25 100755
--- a/syntax_checkers/python/codec.py
+++ b/syntax_checkers/python/codec.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 from __future__ import print_function
 from sys import argv, exit
diff --git a/syntax_checkers/python/compile.py b/syntax_checkers/python/compile.py
index 32f1413..2171563 100755
--- a/syntax_checkers/python/compile.py
+++ b/syntax_checkers/python/compile.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 from __future__ import print_function
 from sys import argv, exit
