diff -rupN --no-dereference podofo-0.10.5/CMakeLists.txt podofo-0.10.5-new/CMakeLists.txt
--- podofo-0.10.5/CMakeLists.txt	2025-05-26 16:32:47.000000000 +0200
+++ podofo-0.10.5-new/CMakeLists.txt	2025-10-04 10:40:21.898853428 +0200
@@ -95,7 +95,7 @@ endif()
 
 if(WIN32)
     # We must explicitly link to the core win32 libraries
-    list(APPEND PLATFORM_SYSTEM_LIBRARIES kernel32 user32 gdi32 winspool comdlg32 advapi32 shell32 ole32 oleaut32 uuid ws2_32 Crypt32)
+    list(APPEND PLATFORM_SYSTEM_LIBRARIES kernel32 user32 gdi32 winspool comdlg32 advapi32 shell32 ole32 oleaut32 uuid ws2_32 crypt32)
 endif()
 
 # Set minimum required standard C++17, but also enforce it
diff -rupN --no-dereference podofo-0.10.5/src/podofo/private/WindowsLeanMean.h podofo-0.10.5-new/src/podofo/private/WindowsLeanMean.h
--- podofo-0.10.5/src/podofo/private/WindowsLeanMean.h	2025-05-26 16:32:47.000000000 +0200
+++ podofo-0.10.5-new/src/podofo/private/WindowsLeanMean.h	2025-10-04 10:40:21.899235557 +0200
@@ -4,7 +4,7 @@
 
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
-#include <Windows.h>
+#include <windows.h>
 #undef Yield
 #undef DrawText
 #undef GetObject
