diff -ur rakarrack-git-7dba0c4bccf88740c56867568d5b218b4a131499.orig/src/rakarrack.cxx rakarrack-git-7dba0c4bccf88740c56867568d5b218b4a131499/src/rakarrack.cxx
--- rakarrack-git-7dba0c4bccf88740c56867568d5b218b4a131499.orig/src/rakarrack.cxx	2014-07-23 05:15:00.000000000 +1000
+++ rakarrack-git-7dba0c4bccf88740c56867568d5b218b4a131499/src/rakarrack.cxx	2014-08-09 22:26:47.978707746 +1000
@@ -23583,7 +23583,7 @@
     Fl_Widget *m = fl_message_icon();
     m->parent()->copy_label(rkr->jackcliname);
     sprintf(temp2,"Overwrite \"%s\"?",w->label());
-    ok=fl_choice(temp2, "No","Yes", NULL);
+    ok=fl_choice("%s", temp2, "No", "Yes", NULL);
    if (!ok)
    { 
    o->value(0);
@@ -24850,7 +24850,7 @@
    w->parent()->copy_label(rkr->jackcliname);
    
   
-   ok=fl_choice("Bank was modified, but not saved", "Discard","Save",NULL);
+   ok=fl_choice("%s", "Bank was modified, but not saved", "Discard", "Save", NULL);
   
   
   
@@ -28008,7 +28008,7 @@
   } 
   
   sprintf(temp2,"Delete? \"%s\"",s->text());
-  ok=fl_choice("%d",temp2,"No","Yes",NULL);
+  ok=fl_choice("%s", temp2, "No", "Yes", NULL);
   if (!ok) return;
   memset(Rname,0,sizeof(Rname));
   sprintf(Rname,"%s",s->text());
