diff -up InsightToolkit-4.7.2/Modules/Core/Common/include/itkMathDetail.h.sse InsightToolkit-4.7.2/Modules/Core/Common/include/itkMathDetail.h
--- InsightToolkit-4.7.2/Modules/Core/Common/include/itkMathDetail.h.sse	2015-04-30 18:28:32.000000000 -0600
+++ InsightToolkit-4.7.2/Modules/Core/Common/include/itkMathDetail.h	2015-05-22 13:52:07.759541689 -0600
@@ -39,10 +39,6 @@
 #include <fenv.h> // should this be cfenv?
 #endif /* ITK_HAVE_FENV_H */
 
-#if defined( ITK_HAVE_EMMINTRIN_H ) && !defined( ITK_WRAPPING_PARSER )
-#include <emmintrin.h> // sse 2 intrinsics
-#endif /* ITK_HAVE_EMMINTRIN_H && ! ITK_WRAPPING_PARSER */
-
 // assume no SSE2:
 #define USE_SSE2_64IMPL 0
 #define USE_SSE2_32IMPL 0
@@ -104,6 +100,10 @@
 #define VC_USE_ASM_64IMPL 0
 #endif
 
+#if defined( ITK_HAVE_EMMINTRIN_H ) && !defined( ITK_WRAPPING_PARSER ) && ( USE_SSE2_32IMPL || USE_SSE2_64IMPL )
+#include <emmintrin.h> // sse 2 intrinsics
+#endif /* ITK_HAVE_EMMINTRIN_H && ! ITK_WRAPPING_PARSER && ( USE_SSE2_32IMPL || USE_SSE2_64IMPL ) */
+
 namespace itk
 {
 namespace Math
