--- rogue5.4.5.orig/mdport.c	2012-08-13 18:15:33.394904357 +0100
+++ rogue5.4.5.orig/mdport.c	2007-10-13 19:29:51.000000000 +0100
@@ -1306,7 +1306,7 @@ 
 	    case ALT_PAD9   : ch = CTRL('U'); break;
 #endif
 #ifdef KEY_BACKSPACE /* NCURSES in Keypad mode sends this for Ctrl-H */
-            case KEY_BACKSPACE: ch = CTRL('H'); break;
+            case KEY_BACKSPACE: ch = erasechar(); break;
 #endif
 	}
 
