From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 12 Jun 2024 09:39:13 +0200
Subject: [PATCH] FIXME an old DB in Fedora

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/test/egeoip_tests.erl b/test/egeoip_tests.erl
index 0e40975..792395e 100644
--- a/test/egeoip_tests.erl
+++ b/test/egeoip_tests.erl
@@ -147,7 +147,6 @@ country_test2_data() ->
     [{"212.118.5.94", "JO"},
      {"64.170.57.29", "US"},
      {"202.7.216.215", "AU"},
-     {"1.2.3.4", "AU"},
      {"212.33.164.149", "SA"},
      {"68.96.110.210", "US"},
      {"213.166.131.168", "SA"},
