diff -Naur a/cmake/FindModbus.cmake b/cmake/FindModbus.cmake
--- a/cmake/FindModbus.cmake	2022-05-30 10:26:37.000000000 -0700
+++ b/cmake/FindModbus.cmake	2023-10-28 08:57:58.065173100 -0700
@@ -47,16 +47,16 @@
 	"${MODBUS_ROOT_DIR}"
 	PATH_SUFFIXES
 	include
+	include/modbus
 	PATHS
 	"${_progfiles}/libmodbus"
 	C:/usr/local
-	/usr/local)
+	/usr/local
+	/usr)
 
 # Look for the library.
 find_library(MODBUS_LIBRARY
-	NAMES
-	libmodbus.lib
-	libmodbus.a
+	modbus
 	HINTS
 	"${MODBUS_ROOT_DIR}"
 	PATH_SUFFIXES
