diff -rupN --no-dereference flask_security_too-5.6.2/pyproject-too.toml flask_security_too-5.6.2-new/pyproject-too.toml
--- flask_security_too-5.6.2/pyproject-too.toml	2025-05-29 19:32:34.713156563 +0200
+++ flask_security_too-5.6.2-new/pyproject-too.toml	2025-05-29 19:32:34.717719029 +0200
@@ -53,13 +53,13 @@ dependencies = [
 
 [project.optional-dependencies]
 babel = ["babel>=2.12.1", "flask_babel>=4.0.0"]
-fsqla = ["flask_sqlalchemy>=3.1.0", "sqlalchemy>=2.0.18"]
+fsqla = ["flask_sqlalchemy>=3.0.0", "sqlalchemy>=2.0.18"]
 common = ["argon2_cffi>=21.3.0", "bcrypt>=4.0.1", "flask_mailman>=0.3.0", "bleach>=6.0.0"]
 mfa = ["cryptography>=40.0.2", "qrcode>=7.4.2", "phonenumbers>=8.13.11", "webauthn>=2.0.0"]
 low = [
     # Lowest supported versions
     "Flask==3.0.0",
-    "Flask-SQLAlchemy==3.1.0",
+    "Flask-SQLAlchemy==3.0.0",
     "Flask-SQLAlchemy-Lite==0.1.0;python_version>='3.10'",
     "Flask-Babel==4.0.0",
     "Flask-Mailman==0.3.0",
