From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 14 Dec 2020 13:09:11 +0100
Subject: [PATCH] Fix javadoc

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/lib/ic/doc/src/Makefile b/lib/ic/doc/src/Makefile
index 7d0048c..f809841 100644
--- a/lib/ic/doc/src/Makefile
+++ b/lib/ic/doc/src/Makefile
@@ -36,7 +36,7 @@ APPLICATION=ic
 # Java specific
 # ----------------------------------------------------
 JAVADOC=javadoc
-JAVA_INCL_ROOT = $(ERL_TOP)/lib/jinterface/priv/
+JAVA_INCL_ROOT = /usr/share/java/erlang/OtpErlang.jar
 JAVA_SRC_ROOT = $(CORBA_TOP)/lib/ic/java_src/
 JAVA_CLASS_SUBDIR = com/ericsson/otp/ic/
 
