diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index d3bc20cf4f..129d41d9f6 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -463,6 +463,8 @@ foreach(arch ${archs})
                 use_bluez=false
                 is_cfi=false
                 use_ozone=true
+                media_use_openh264=true
+                rtc_use_h264=true
                 ozone_auto_platforms=false
                 ozone_platform_headless=false
                 ozone_platform_external=true
