diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 129d41d9f6..28b5ff7cfc 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -590,6 +590,10 @@ foreach(arch ${archs})
                     list(APPEND gnArgArg libyuv_use_sve=false)
                 endif()
             endif()
+            if(cpu MATCHES "arm.*")
+                # https://bugreports.qt.io/browse/QTBUG-129985
+                list(APPEND gnArgArg build_webnn_with_xnnpack=false)
+            endif()
             unset(cpu)
         endif()
 
