--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/BUILD.gn	2026-04-11 03:32:34.058970679 -0700
@@ -57,7 +57,7 @@
     "../../include/perfetto/ext/trace_processor:export_json",
     "../base",
     "containers",
-    "importers/common",
+    "importers/common:tp_importer_common",
     "importers/json:minimal",
     "storage",
     "tables",
@@ -122,7 +122,7 @@
     "../base",
     "../protozero",
     "containers",
-    "importers/common",
+    "importers/common:tp_importer_common",
     "importers/common:parser_types",
     "importers/ftrace:minimal",
     "importers/fuchsia:fuchsia_record",
@@ -169,7 +169,7 @@
       "importers/android_bugreport",
       "importers/archive",
       "importers/art_method",
-      "importers/common",
+      "importers/common:tp_importer_common",
       "importers/etw:full",
       "importers/ftrace:full",
       "importers/fuchsia:full",
@@ -300,7 +300,7 @@
       "../../gn:jsoncpp",
       "../../include/perfetto/ext/trace_processor:export_json",
       "containers",
-      "importers/common",
+      "importers/common:tp_importer_common",
       "importers/proto:minimal",
       "storage",
       "tables",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/android_bugreport/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/android_bugreport/BUILD.gn	2026-04-11 03:32:34.059857890 -0700
@@ -72,7 +72,7 @@
     "../../types",
     "../../util:trace_type",
     "../../util:zip_reader",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
 
@@ -92,6 +92,6 @@
     "../../sorter",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/archive/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/archive/BUILD.gn	2026-04-11 03:32:34.059905421 -0700
@@ -39,7 +39,7 @@
     "../../util:trace_type",
     "../../util:zip_reader",
     "../android_bugreport",
-    "../common",
+    "../common:tp_importer_common",
     "../proto:minimal",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/art_method/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/art_method/BUILD.gn	2026-04-11 03:32:34.059946144 -0700
@@ -35,7 +35,7 @@
     "../../../../protos/perfetto/common:zero",
     "../../../base",
     "../../containers",
-    "../../importers/common",
+    "../../importers/common:tp_importer_common",
     "../../sorter",
     "../../storage",
     "../../types",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/common/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/common/BUILD.gn	2026-04-11 03:32:34.058891078 -0700
@@ -14,7 +14,7 @@
 
 import("../../../../gn/perfetto.gni")
 
-source_set("common") {
+source_set("tp_importer_common") {
   sources = [
     "address_range.h",
     "args_tracker.cc",
@@ -138,7 +138,7 @@
   ]
   testonly = true
   deps = [
-    ":common",
+    ":tp_importer_common",
     "../../../../gn:default_deps",
     "../../../../gn:gtest_and_gmock",
     "../../../../protos/perfetto/common:zero",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/etm/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/etm/BUILD.gn	2026-04-11 03:32:34.060029703 -0700
@@ -89,7 +89,7 @@
     "../../tables:tables_python",
     "../../types",
     "../../util:build_id",
-    "../common",
+    "../common:tp_importer_common",
     "../elf:binary_info",
     "../perf",
     "../perf:record",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/etw/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/etw/BUILD.gn	2026-04-11 03:32:34.060068910 -0700
@@ -47,7 +47,7 @@
     "../../sorter",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
     "../common:parser_types",
     "../i2c:full",
     "../proto:minimal",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/ftrace/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/ftrace/BUILD.gn	2026-04-11 03:32:34.060112745 -0700
@@ -76,7 +76,7 @@
     "../../storage",
     "../../tables",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
     "../common:parser_types",
     "../i2c:full",
     "../proto:minimal",
@@ -109,7 +109,7 @@
     "../../../../gn:gtest_and_gmock",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
     "../ftrace:full",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/fuchsia/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/fuchsia/BUILD.gn	2026-04-11 03:32:34.060162833 -0700
@@ -43,7 +43,7 @@
     "../../tables",
     "../../types",
     "../../util:trace_type",
-    "../common",
+    "../common:tp_importer_common",
     "../proto:minimal",
   ]
 }
@@ -88,7 +88,7 @@
     "../../tables",
     "../../types",
     "../../util:descriptors",
-    "../common",
+    "../common:tp_importer_common",
     "../ftrace:full",
     "../proto:full",
     "../proto:minimal",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gecko/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gecko/BUILD.gn	2026-04-11 03:32:34.060201616 -0700
@@ -38,7 +38,7 @@
       "../../../../include/perfetto/trace_processor:storage",
       "../../../../protos/perfetto/trace:zero",
       "../../../base",
-      "../../importers/common",
+      "../../importers/common:tp_importer_common",
       "../../sorter",
       "../../storage",
       "../../tables",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/i2c/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/i2c/BUILD.gn	2026-04-11 03:32:34.060237482 -0700
@@ -22,6 +22,6 @@
     "../../sorter",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/instruments/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/instruments/BUILD.gn	2026-04-11 03:32:34.060275457 -0700
@@ -44,7 +44,7 @@
       "../../sorter",
       "../../storage",
       "../../types",
-      "../common:common",
+      "../common:tp_importer_common",
     ]
   }
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/json/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/json/BUILD.gn	2026-04-11 03:32:34.060314357 -0700
@@ -23,7 +23,7 @@
   deps = [
     "../../../../gn:default_deps",
     "../../storage",
-    "../common",
+    "../common:tp_importer_common",
   ]
   if (enable_perfetto_trace_processor_json) {
     public_deps = [ "../../../../gn:jsoncpp" ]
@@ -45,7 +45,7 @@
       "../../storage",
       "../../tables",
       "../../types",
-      "../common",
+      "../common:tp_importer_common",
       "../common:parser_types",
       "../systrace:full",
       "../systrace:systrace_line",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/ninja/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/ninja/BUILD.gn	2026-04-11 03:32:34.060386658 -0700
@@ -19,7 +19,7 @@
   ]
   deps = [
     "../../../../gn:default_deps",
-    "../../importers/common",
+    "../../importers/common:tp_importer_common",
     "../../sorter",
     "../../storage",
   ]
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/perf/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/perf/BUILD.gn	2026-04-11 03:32:34.060431618 -0700
@@ -36,7 +36,7 @@
     "../../tables:tables_python",
     "../../types",
     "../../util:build_id",
-    "../common:common",
+    "../common:tp_importer_common",
     "../common:parser_types",
   ]
 }
@@ -94,7 +94,7 @@
     "../../util:build_id",
     "../../util:trace_blob_view_reader",
     "../../util:util",
-    "../common:common",
+    "../common:tp_importer_common",
     "../proto:minimal",
   ]
 
@@ -118,6 +118,6 @@
     "../../../base",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/perf_text/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/perf_text/BUILD.gn	2026-04-11 03:32:34.060469481 -0700
@@ -30,7 +30,7 @@
     "../../tables",
     "../../types",
     "../../util:trace_blob_view_reader",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
 
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/BUILD.gn	2026-04-11 03:32:34.060534729 -0700
@@ -100,7 +100,7 @@
     "../../util:gzip",
     "../../util:profiler_util",
     "../../util:trace_blob_view_reader",
-    "../common",
+    "../common:tp_importer_common",
     "../common:parser_types",
     "../etw:minimal",
     "../ftrace:minimal",
@@ -202,7 +202,7 @@
     "../../util:profiler_util",
     "../../util:proto_profiler",
     "../../util:proto_to_args_parser",
-    "../common",
+    "../common:tp_importer_common",
     "../common:parser_types",
     "../etw:full",
     "../ftrace:full",
@@ -314,7 +314,7 @@
     "../../util:build_id",
     "../../util:descriptors",
     "../../util:profiler_util",
-    "../common",
+    "../common:tp_importer_common",
     "../ftrace:full",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/syscalls/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/syscalls/BUILD.gn	2026-04-11 03:32:34.060575176 -0700
@@ -25,7 +25,7 @@
     "../../containers",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
 
@@ -36,6 +36,6 @@
     ":full",
     "../../../../gn:default_deps",
     "../../../../gn:gtest_and_gmock",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/systrace/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/systrace/BUILD.gn	2026-04-11 03:32:34.060614410 -0700
@@ -33,7 +33,7 @@
     "../../storage",
     "../../tables",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
   ]
 }
 
@@ -55,7 +55,7 @@
     "../../sorter",
     "../../storage",
     "../../types",
-    "../common",
+    "../common:tp_importer_common",
     "../ftrace:full",
   ]
 }
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/perfetto_sql/intrinsics/functions/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/perfetto_sql/intrinsics/functions/BUILD.gn	2026-04-11 03:32:34.059133543 -0700
@@ -73,7 +73,7 @@
     "../../../containers",
     "../../../db",
     "../../../db/column",
-    "../../../importers/common",
+    "../../../importers/common:tp_importer_common",
     "../../../importers/ftrace:ftrace_descriptors",
     "../../../perfetto_sql/intrinsics/table_functions",
     "../../../sqlite",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/perfetto_sql/intrinsics/table_functions/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/perfetto_sql/intrinsics/table_functions/BUILD.gn	2026-04-11 03:32:34.059082007 -0700
@@ -105,7 +105,7 @@
     "../../../containers",
     "../../../db",
     "../../../db/column",
-    "../../../importers/common",
+    "../../../importers/common:tp_importer_common",
     "../../../perfetto_sql/engine",
     "../../../sqlite",
     "../../../storage",
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/sqlite/BUILD.gn	2026-01-23 11:06:19.000000000 -0800
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/sqlite/BUILD.gn	2026-04-11 03:32:34.059027957 -0700
@@ -45,7 +45,7 @@
     "../containers",
     "../db",
     "../db/column",
-    "../importers/common",
+    "../importers/common:tp_importer_common",
     "../importers/ftrace:ftrace_descriptors",
     "../perfetto_sql/intrinsics/table_functions:interface",
     "../storage",
