diff -rupN --no-dereference qtactiveqt-everywhere-src-6.10/src/activeqt/container/qaxbase.cpp qtactiveqt-everywhere-src-6.10-new/src/activeqt/container/qaxbase.cpp
--- qtactiveqt-everywhere-src-6.10/src/activeqt/container/qaxbase.cpp	2025-08-17 21:32:08.000000000 +0200
+++ qtactiveqt-everywhere-src-6.10-new/src/activeqt/container/qaxbase.cpp	2025-09-03 08:48:04.408779241 +0200
@@ -2116,6 +2116,10 @@ void MetaObjectGenerator::readClassInfo(
             const QStringList versions = controls.childGroups();
             for (const QString &version : versions) {
                 tlfile = controls.value(QLatin1Char('/') + version + QLatin1String("/0/win32/.")).toString();
+                if(tlfile.isEmpty())
+                {
+                  tlfile = controls.value(QLatin1Char('/') + version + QLatin1String("/0/win64/.")).toString();
+                }
                 if (!tlfile.isEmpty())
                     break;
             }
diff -rupN --no-dereference qtactiveqt-everywhere-src-6.10/tools/dumpcpp/main.cpp qtactiveqt-everywhere-src-6.10-new/tools/dumpcpp/main.cpp
--- qtactiveqt-everywhere-src-6.10/tools/dumpcpp/main.cpp	2025-08-17 21:32:08.000000000 +0200
+++ qtactiveqt-everywhere-src-6.10-new/tools/dumpcpp/main.cpp	2025-09-03 08:48:04.409569870 +0200
@@ -1087,6 +1087,10 @@ int main(int argc, char **argv)
             typeLib = settings.value(keyPrefix + QLatin1String("win32/.")).toString();
             if (QFile::exists(typeLib))
                 break;
+            typeLib = settings.value(QLatin1String("/") + codes.at(c) + QLatin1String("/0/win64/.")).toByteArray();
+            if (QFile::exists(typeLib)) {
+                break;
+            }
         }
 
         if (!typeLib.isEmpty())
@@ -1151,6 +1155,10 @@ int main(int argc, char **argv)
             typeLib = settings.value(keyPrefix + QLatin1String("win32/.")).toString();
             if (QFile::exists(typeLib))
                 break;
+            typeLib = settings.value(key + QLatin1String("/") + codes.at(c) + QLatin1String("/win64/.")).toByteArray();
+            if (QFile::exists(typeLib)) {
+                break;
+            }
         }
     }
 
