--- futures-0.3.32/Cargo.toml
+++ futures-0.3.32/Cargo.toml
@@ -46,8 +46,6 @@
 features = [
     "std",
     "async-await",
-    "compat",
-    "io-compat",
     "executor",
     "thread-pool",
 ]
@@ -66,10 +64,6 @@
 ]
 bilock = ["futures-util/bilock"]
 cfg-target-has-atomic = []
-compat = [
-    "std",
-    "futures-util/compat",
-]
 default = [
     "std",
     "async-await",
@@ -79,10 +73,6 @@
     "std",
     "futures-executor/std",
 ]
-io-compat = [
-    "compat",
-    "futures-util/io-compat",
-]
 spin = ["futures-util/spin"]
 std = [
     "alloc",
@@ -128,10 +118,6 @@
 path = "tests/bilock.rs"
 
 [[test]]
-name = "compat"
-path = "tests/compat.rs"
-
-[[test]]
 name = "eager_drop"
 path = "tests/eager_drop.rs"
 
@@ -379,9 +365,9 @@
 [dev-dependencies.static_assertions]
 version = "1"
 
-[dev-dependencies.tokio]
-version = "0.1.11"
-
+[dev-dependencies.futures-test]
+version = "0.3.32"
+
 [lints.rust]
 missing_debug_implementations = "warn"
 rust_2018_idioms = "warn"
