--- futures-0.3.32/Cargo.toml
+++ futures-0.3.32/Cargo.toml
@@ -46,8 +46,6 @@
 features = [
     "std",
     "async-await",
-    "compat",
-    "io-compat",
     "executor",
     "thread-pool",
 ]
@@ -66,10 +64,6 @@
 ]
 bilock = ["futures-util/bilock"]
 cfg-target-has-atomic = []
-compat = [
-    "std",
-    "futures-util/compat",
-]
 default = [
     "std",
     "async-await",
@@ -79,10 +73,6 @@
     "std",
     "futures-executor/std",
 ]
-io-compat = [
-    "compat",
-    "futures-util/io-compat",
-]
 spin = ["futures-util/spin"]
 std = [
     "alloc",
@@ -120,18 +110,10 @@
 path = "tests/async_await_macros.rs"
 
 [[test]]
-name = "auto_traits"
-path = "tests/auto_traits.rs"
-
-[[test]]
 name = "bilock"
 path = "tests/bilock.rs"
 
 [[test]]
-name = "compat"
-path = "tests/compat.rs"
-
-[[test]]
 name = "eager_drop"
 path = "tests/eager_drop.rs"
 
@@ -379,8 +361,8 @@
 [dev-dependencies.static_assertions]
 version = "1"
 
-[dev-dependencies.tokio]
-version = "0.1.11"
+[dev-dependencies.futures-test]
+version = "0.3.32"
 
 [lints.rust]
 missing_debug_implementations = "warn"
--- futures-0.3.32/src/lib.rs
+++ futures-0.3.32/src/lib.rs
@@ -197,24 +197,6 @@
     pub use futures_executor::{ThreadPool, ThreadPoolBuilder};
 }
 
-#[cfg(feature = "compat")]
-#[cfg_attr(docsrs, doc(cfg(feature = "compat")))]
-pub mod compat {
-    //! Interop between `futures` 0.1 and 0.3.
-    //!
-    //! This module is only available when the `compat` feature of this
-    //! library is activated.
-
-    pub use futures_util::compat::{
-        Compat, Compat01As03, Compat01As03Sink, CompatSink, Executor01As03, Executor01CompatExt,
-        Executor01Future, Future01CompatExt, Sink01CompatExt, Stream01CompatExt,
-    };
-
-    #[cfg(feature = "io-compat")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "io-compat")))]
-    pub use futures_util::compat::{AsyncRead01CompatExt, AsyncWrite01CompatExt};
-}
-
 pub mod prelude {
     //! A "prelude" for crates using the `futures` crate.
     //!
