diff -up Kst-2.0.8/src/libkst/editablevector.cpp.qreal Kst-2.0.8/src/libkst/editablevector.cpp
--- Kst-2.0.8/src/libkst/editablevector.cpp.qreal	2014-07-20 03:22:38.000000000 -0600
+++ Kst-2.0.8/src/libkst/editablevector.cpp	2015-01-12 10:46:16.884628350 -0700
@@ -52,7 +52,7 @@ void EditableVector::setValue(const int
     _scalars["sumsquared"]->setValue(_sum*_sum);
     _scalars["max"]->setValue(qMax(_max,val));
     _scalars["min"]->setValue(qMin(_min,_min));
-    double b=(float)(qMax((float)0.0f,(float)_minPos));
+    double b=qMax(double(0.0),_minPos);
     _scalars["minpos"]->setValue(qMin(_min,b));
     _scalars["last"]->setValue(_v[_size-1]);
     _scalars["first"]->setValue(_v[0]);
diff -up Kst-2.0.8/src/libkstmath/curve.cpp.qreal Kst-2.0.8/src/libkstmath/curve.cpp
--- Kst-2.0.8/src/libkstmath/curve.cpp.qreal	2014-07-20 02:53:17.000000000 -0600
+++ Kst-2.0.8/src/libkstmath/curve.cpp	2015-01-14 16:18:19.898681604 -0700
@@ -684,11 +684,11 @@ RelationPtr Curve::makeDuplicate() const
 // and not just pulled out of the air by what looks ~good.
 // They are currently "about right" for printing to US Letter.
 double Curve::pointDim(QRectF w) const {
-  return qMax(qreal(1.01), PointSize * ((w.width() + w.height()) * (1.0 / 4000.0)));
+  return qMax(double(1.01), PointSize * ((w.width() + w.height()) * (1.0 / 4000.0)));
 }
 
 double Curve::lineDim(const QRectF &R, double linewidth) {
-  double lw = qMax(qreal(1.01), linewidth*(R.width()+R.height())*(1.0/2000.0));
+  qreal lw = qMax(qreal(1.01), qreal(linewidth)*(R.width()+R.height())*qreal(1.0/2000.0));
   // if you ask for a thicker line, make sure it is at least 2 px wide
   if (linewidth>1.9) {
     lw = qMax(qreal(2.0),lw);
diff -up Kst-2.0.8/src/libkst/vector.cpp.qreal Kst-2.0.8/src/libkst/vector.cpp
--- Kst-2.0.8/src/libkst/vector.cpp.qreal	2014-07-20 03:22:38.000000000 -0600
+++ Kst-2.0.8/src/libkst/vector.cpp	2015-01-12 11:17:04.046417015 -0700
@@ -595,11 +595,11 @@ void Vector::oldChange(QByteArray &data)
       qds >> _v[i];
       if(!i) {
           _min=_max=_minPos=sum=_v[i];
-          _minPos=qMax(_minPos,qreal(0.0));
+          _minPos=qMax(_minPos,double(0.0));
       } else {
           _min=qMin(_v[i],_min);
           _max=qMax(_v[i],_max);
-          _minPos=qMin(qMax(_v[i],qreal(0.0)),_minPos);
+          _minPos=qMin(qMax(_v[i],double(0.0)),_minPos);
           sum+=_v[i];
       }
     }
@@ -626,11 +626,11 @@ void Vector::change(QByteArray &data) {
       qds >> _v[i];
       if(!i) {
           _min=_max=_minPos=sum=_v[i];
-          _minPos=qMax(_minPos,qreal(0.0));
+          _minPos=qMax(_minPos,double(0.0));
       } else {
           _min=qMin(_v[i],_min);
           _max=qMax(_v[i],_max);
-          _minPos=qMin(qMax(_v[i],qreal(0.0)),_minPos);
+          _minPos=qMin(qMax(_v[i],double(0.0)),_minPos);
           sum+=_v[i];
       }
     }
diff -up Kst-2.0.8/src/plugins/dataobject/activitylevel/activitylevel.cpp.qreal Kst-2.0.8/src/plugins/dataobject/activitylevel/activitylevel.cpp
--- Kst-2.0.8/src/plugins/dataobject/activitylevel/activitylevel.cpp.qreal	2014-07-20 02:53:17.000000000 -0600
+++ Kst-2.0.8/src/plugins/dataobject/activitylevel/activitylevel.cpp	2015-01-12 10:46:16.885628344 -0700
@@ -323,7 +323,7 @@ bool ActivityLevelSource::algorithm() {
       incomingValueNext = outputVectorDenoised->value()[i+1];
     }
     if ( (outgoingValue-outgoingValuePrev)*(outgoingValueNext-outgoingValue) < 0) {
-      dNbReversals = qMax(dNbReversals - 1.0, qreal(0.0)); // Avoid getting negative values, which can happen
+      dNbReversals = qMax(dNbReversals - 1.0, double(0.0)); // Avoid getting negative values, which can happen
     }
     if ( (incomingValue-incomingValuePrev)*(incomingValueNext-incomingValue) < 0) {
       dNbReversals += 1.0;
