diff --git a/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.cc b/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.cc
index 7b8d915ae930..f9b2811cf58d 100644
--- a/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.cc
+++ b/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.cc
@@ -76,7 +76,9 @@ VideoType PipeWireRawFormatToVideoType(uint32_t id) {
 void PipeWireNode::PipeWireNodeDeleter::operator()(
     PipeWireNode* node) const noexcept {
   spa_hook_remove(&node->node_listener_);
+  spa_hook_remove(&node->proxy_listener_);
   pw_proxy_destroy(node->proxy_);
+  pw_node_info_free(node->info_);
 }
 
 // static
@@ -107,6 +109,13 @@ PipeWireNode::PipeWireNode(PipeWireSession* session,
 
   pw_node_add_listener(reinterpret_cast<pw_node*>(proxy_), &node_listener_,
                        &node_events, this);
+
+  static const pw_proxy_events proxy_events{
+      .version = PW_VERSION_PROXY_EVENTS,
+      .done = OnProxyDone,
+  };
+
+  pw_proxy_add_listener(proxy_, &proxy_listener_, &proxy_events, this);
 }
 
 // static
@@ -114,6 +123,12 @@ RTC_NO_SANITIZE("cfi-icall")
 void PipeWireNode::OnNodeInfo(void* data, const pw_node_info* info) {
   PipeWireNode* that = static_cast<PipeWireNode*>(data);
 
+  that->info_ = pw_node_info_update(that->info_, info);
+  if (!that->info_)
+    return;
+
+  info = that->info_;
+
   if (info->change_mask & PW_NODE_CHANGE_MASK_PROPS) {
     const char* vid_str;
     const char* pid_str;
@@ -134,20 +149,27 @@ void PipeWireNode::OnNodeInfo(void* data, const pw_node_info* info) {
   }
 
   if (info->change_mask & PW_NODE_CHANGE_MASK_PARAMS) {
+    bool changed = false;
     for (uint32_t i = 0; i < info->n_params; i++) {
-      uint32_t id = info->params[i].id;
-      if (id == SPA_PARAM_EnumFormat &&
+      if (that->info_->params[i].user == 0)
+        continue;
+      that->info_->params[i].user = 0;
+      changed = true;
+
+      if (info->params[i].id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        RTC_LOG(LS_INFO)
-            << "Clearing existing capabilities before re-enumeration for "
-            << that->display_name_;
-        that->capabilities_.clear();
-        pw_node_enum_params(reinterpret_cast<pw_node*>(that->proxy_), 0, id, 0,
-                            UINT32_MAX, nullptr);
+        that->pending_capabilities_.clear();
+        pw_node_enum_params(reinterpret_cast<pw_node*>(that->proxy_), 0,
+                            info->params[i].id, 0, UINT32_MAX, nullptr);
+        that->pending_sync_ =
+            pw_proxy_sync(that->proxy_, that->pending_sync_);
         break;
       }
     }
-    that->session_->PipeWireSync();
+
+    if (changed) {
+      that->session_->PipeWireSync();
+    }
   }
 }
 
@@ -160,6 +182,7 @@ void PipeWireNode::OnNodeParam(void* data,
                                uint32_t next,
                                const spa_pod* param) {
   PipeWireNode* that = static_cast<PipeWireNode*>(data);
+
   auto* obj = reinterpret_cast<const spa_pod_object*>(param);
   const spa_pod_prop* prop = nullptr;
   VideoCaptureCapability cap;
@@ -211,11 +234,22 @@ void PipeWireNode::OnNodeParam(void* data,
   cap.height = rect[0].height;
 
   RTC_LOG(LS_VERBOSE) << "Found Format(" << that->display_name_
-                      << "): " << static_cast<int>(cap.videoType) << "("
-                      << cap.width << "x" << cap.height << "@" << cap.maxFPS
-                      << ")";
+                     << "): videoType=" << static_cast<int>(cap.videoType)
+                     << " (" << cap.width << "x" << cap.height << "@"
+                     << cap.maxFPS << ")";
+
+  that->pending_capabilities_.push_back(cap);
+}
+
+// static
+void PipeWireNode::OnProxyDone(void* data, int seq) {
+  PipeWireNode* that = static_cast<PipeWireNode*>(data);
+
+  if (seq != that->pending_sync_)
+    return;
 
-  that->capabilities_.push_back(cap);
+  that->capabilities_ = std::move(that->pending_capabilities_);
+  that->pending_capabilities_.clear();
 }
 
 // static
diff --git a/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.h b/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.h
index c6bc394c88f8..ef1e038ba3f5 100644
--- a/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.h
+++ b/third_party/libwebrtc/modules/video_capture/linux/pipewire_session.h
@@ -73,15 +73,20 @@ class PipeWireNode {
                           uint32_t next,
                           const spa_pod* param);
   static bool ParseFormat(const spa_pod* param, VideoCaptureCapability* cap);
+  static void OnProxyDone(void* data, int seq);
 
   struct pw_proxy* proxy_;
   struct spa_hook node_listener_;
+  struct spa_hook proxy_listener_;
   PipeWireSession* session_;
   uint32_t id_;
   std::string display_name_;
   std::string unique_id_;
   std::string model_id_;
+  struct pw_node_info* info_ = nullptr;
+  int pending_sync_ = -1;
   std::vector<VideoCaptureCapability> capabilities_;
+  std::vector<VideoCaptureCapability> pending_capabilities_;
 };
 
 class CameraPortalNotifier : public CameraPortal::PortalNotifier {
