--- xt/03_podspell.t
+++ xt/03_podspell.t
@@ -8,11 +8,12 @@ plan skip_all => "set TEST_POD or TEST_A
 
 my $spell;
 for my $path (split /$Config::Config{path_sep}/ => $ENV{PATH}) {
-    -x File::Spec->catfile($path => 'spell')  and $spell = 'spell',       last;
-    -x File::Spec->catfile($path => 'ispell') and $spell = 'ispell -l',   last;
-    -x File::Spec->catfile($path => 'aspell') and $spell = 'aspell list', last;
+    -x File::Spec->catfile($path => 'hunspell') and $spell = 'hunspell -l', last;
+    -x File::Spec->catfile($path => 'spell')    and $spell = 'spell',       last;
+    -x File::Spec->catfile($path => 'ispell')   and $spell = 'ispell -l',   last;
+    -x File::Spec->catfile($path => 'aspell')   and $spell = 'aspell list', last;
 }
-plan skip_all => "spell/ispell/aspell are not installed." unless $spell;
+plan skip_all => "hunspell/spell/ispell/aspell are not installed." unless $spell;
 
 add_stopwords(map { split /[\s\:\-]/ } <DATA>);
 set_spell_cmd($spell) if $spell;
