diff --git a/source/mupen64plus-ui-console/src/osal_files_unix.c b/source/mupen64plus-ui-console/src/osal_files_unix.c
index 3485116..9fd2ac9 100644
--- a/source/mupen64plus-ui-console/src/osal_files_unix.c
+++ b/source/mupen64plus-ui-console/src/osal_files_unix.c
@@ -37,8 +37,8 @@
 
 /* definitions for system directories to search when looking for mupen64plus plugins */
 #if defined(PLUGINDIR)
-  const int   osal_libsearchdirs = 4;
-  const char *osal_libsearchpath[4] = { PLUGINDIR, "/usr/local/lib/mupen64plus",  "/usr/lib/mupen64plus", "./" };
+  const int   osal_libsearchdirs = 5;
+  const char *osal_libsearchpath[5] = { PLUGINDIR, "/usr/local/lib/mupen64plus",  "/usr/lib/mupen64plus", "/usr/lib64/mupen64plus/", "./" };
 #else
   const int   osal_libsearchdirs = 3;
   const char *osal_libsearchpath[3] = { "/usr/local/lib/mupen64plus",  "/usr/lib/mupen64plus", "./" };
