diff -Nrup a/eftest.c b/eftest.c
--- a/eftest.c	2019-12-15 18:00:58.872100174 -0700
+++ b/eftest.c	2019-12-15 18:01:19.466047998 -0700
@@ -60,8 +60,8 @@ gotSegmentationFault(int (*test)(void))
 }
 
 static char *	allocation;
-/* c is global so that assignments to it won't be optimized out. */
-char	c;
+/* c is volatile global so that assignments to it won't be optimized out. */
+volatile char	c;
 
 static int
 testSizes(void)
