--- rpassword-5.0.1/Cargo.toml	2021-01-19T22:15:25+00:00
+++ rpassword-5.0.1/Cargo.toml	2023-04-17T23:30:45.286434+00:00
@@ -23,6 +23,3 @@
 repository = "https://github.com/conradkleinespel/rpassword"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["std", "winnt", "fileapi", "processenv", "winbase", "handleapi", "consoleapi", "minwindef", "wincon"]
