--- auto_enums-0.8.10/Cargo.toml
+++ auto_enums-0.8.10/Cargo.toml
@@ -116,10 +116,6 @@
 ]
 default-features = false
 
-[dev-dependencies.futures01_crate]
-version = "0.1"
-package = "futures"
-
 [dev-dependencies.futures03_crate]
 version = "0.3"
 features = [
@@ -146,22 +142,6 @@
 version = "1"
 package = "serde"
 
-[dev-dependencies.tokio01_crate]
-version = "0.1"
-features = ["io"]
-default-features = false
-package = "tokio"
-
-[dev-dependencies.tokio02_crate]
-version = "0.2"
-default-features = false
-package = "tokio"
-
-[dev-dependencies.tokio03_crate]
-version = "0.3"
-default-features = false
-package = "tokio"
-
 [dev-dependencies.tokio1_crate]
 version = "1"
 default-features = false
