diff -up grafX2-v2.8/src/sdlscreen.c.ttf-fix grafX2-v2.8/src/sdlscreen.c
--- grafX2-v2.8/src/sdlscreen.c.ttf-fix	2022-12-29 13:28:11.897342192 -0500
+++ grafX2-v2.8/src/sdlscreen.c	2022-12-29 13:30:08.344515563 -0500
@@ -373,17 +373,11 @@ byte * Surface_to_bytefield(SDL_Surface
   byte *src;
   byte *dest_ptr;
   int y;
-  int remainder;
 
   // Support seulement des images 256 couleurs
   if (source->format->BytesPerPixel != 1)
     return NULL;
 
-  if (source->w & 3)
-    remainder=4-(source->w&3);
-  else
-    remainder=0;
-
   if (dest==NULL)
     dest=(byte *)malloc(source->w*source->h);
 
@@ -393,7 +387,7 @@ byte * Surface_to_bytefield(SDL_Surface
   {
     memcpy(dest_ptr, src,source->w);
     dest_ptr += source->w;
-    src += source->w + remainder;
+    src += source->pitch;
   }
   return dest;
 
