--- genromfs-0.5.2/Makefile.zp	2008-09-09 11:04:53.000000000 +0200
+++ genromfs-0.5.2/Makefile	2008-09-09 11:05:06.000000000 +0200
@@ -49,11 +49,6 @@ install-bin:
 	install -m 755 genromfs $(PREFIX)$(bindir)/
 
 install-man:
-	# genromfs 0.5 installed the man page in this file,
-	# remove it before someone notices :)
-	if [ -f $(PREFIX)$(bindir)/man8 ]; then \
-		rm -f $(PREFIX)$(bindir)/man8; \
-	fi
 	mkdir -p $(PREFIX)$(mandir)/man8
 	install -m 644 genromfs.8 $(PREFIX)$(mandir)/man8/
 
