From ac434535395eee46555e2640284710dac2c707b8 Mon Sep 17 00:00:00 2001
From: Luc Grosheintz <luc.grosheintz@gmail.com>
Date: Mon, 11 Nov 2024 11:06:30 +0100
Subject: [PATCH] Skip locale check if `std::locale` fails.

---
 tests/test_morphology_readers.cpp | 29 +++++++++++++++++++++++------
 1 file changed, 23 insertions(+), 6 deletions(-)

diff --git a/tests/test_morphology_readers.cpp b/tests/test_morphology_readers.cpp
index 3b42595b..38d70c15 100644
--- a/tests/test_morphology_readers.cpp
+++ b/tests/test_morphology_readers.cpp
@@ -12,6 +12,7 @@
 #include <morphio/morphology.h>
 #include <morphio/mut/morphology.h>
 #include <morphio/soma.h>
+#include <stdexcept>
 
 TEST_CASE("LoadH5Morphology", "[morphology]") {
     {
@@ -210,15 +211,31 @@ class LocaleGuard {
     std::locale previousLocale;
 };
 
-TEST_CASE("LoadSWCMorphologyLocale", "[morphology]") {
-    {
+static std::unique_ptr<LocaleGuard> make_locale() {
 #ifdef __APPLE__
-        const auto locale = LocaleGuard(std::locale("de_CH.UTF-8"));
+    const std::string locale_name = "de_CH.UTF-8";
 #else
-        const auto locale = LocaleGuard(std::locale("de_CH.UTF8"));
+    const std::string locale_name = "de_CH.UTF8";
 #endif
-        const morphio::Morphology m("data/simple.swc");
-        REQUIRE(m.diameters().size() == 12);
+    try {
+        return std::make_unique<LocaleGuard>(std::locale(locale_name));
+    } catch (const std::runtime_error&) {
+        return nullptr;
+    }
+}
+
+
+TEST_CASE("LoadSWCMorphologyLocale", "[morphology]") {
+    {
+        const auto locale = make_locale();
+        if (locale) {
+            const morphio::Morphology m("data/simple.swc");
+            REQUIRE(m.diameters().size() == 12);
+        } else {
+            // With Catch2 v3 this can be modernize using SKIP.
+            std::cout << "Skipping LoadSWCMorphologyLocale because `std::locale` failed."
+                      << std::endl;
+        }
     }
 }
 
