From ca71b6cde38d98a9df401b2f5a1ad0b684caa692 Mon Sep 17 00:00:00 2001
From: Tom Rix <Tom.Rix@amd.com>
Date: Sat, 27 Sep 2025 09:00:32 -0700
Subject: [PATCH] rocm-validation-suite do not download mxDataGenerator

Signed-off-by: Tom Rix <Tom.Rix@amd.com>
---
 CMakeLists.txt | 39 ++++++++++++++++++++-------------------
 1 file changed, 20 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e4abf1c35a3..47ae5d64d8a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -388,30 +388,31 @@ endif()
 
 ################################################################################
 # mxDataGenerator Download
-configure_file(CMakeMXDataGeneratorDownload.cmake mxDataGenerator-download/CMakeLists.txt)
+if (NOT DEFINED MXDATAGENERATOR_INC_DIR)
+  configure_file(CMakeMXDataGeneratorDownload.cmake mxDataGenerator-download/CMakeLists.txt)
   execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-      RESULT_VARIABLE result
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-download )
-if(result)
-  message(FATAL_ERROR "CMake step for mxDataGenerator failed: ${result}")
-endif()
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-download )
+  if(result)
+    message(FATAL_ERROR "CMake step for mxDataGenerator failed: ${result}")
+  endif()
   execute_process(COMMAND ${CMAKE_COMMAND} --build .
-      RESULT_VARIABLE result
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-download )
-if(result)
-  message(FATAL_ERROR "Build step for mxDataGenerator failed: ${result}")
-endif()
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-download )
+  if(result)
+    message(FATAL_ERROR "Build step for mxDataGenerator failed: ${result}")
+  endif()
   execute_process(COMMAND ${CMAKE_COMMAND} ${CMAKE_BINARY_DIR}/mxDataGenerator-src -B${CMAKE_BINARY_DIR}/mxDataGenerator-build
-      RESULT_VARIABLE result
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-src )
-if(result)
-  message(FATAL_ERROR "Config step for mxDataGenerator-src failed: ${result}")
-endif()
-
-if (NOT DEFINED MXDATAGENERATOR_INC_DIR)
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/mxDataGenerator-src )
+  if(result)
+    message(FATAL_ERROR "Config step for mxDataGenerator-src failed: ${result}")
+  endif()
+  
   set (MXDATAGENERATOR_INC_DIR "${CMAKE_BINARY_DIR}/mxDataGenerator-src/lib/include")
-  message ("MXDATAGENERATOR_INC_DIR ${MXDATAGENERATOR_INC_DIR}")
 endif ()
+message ("MXDATAGENERATOR_INC_DIR ${MXDATAGENERATOR_INC_DIR}")
+
 ################################################################################
 
 ## rocBLAS
-- 
2.51.0

