--- RestSharp.Tests/RestSharp.Tests.csproj.old	2015-10-14 15:29:46.256122931 +0200
+++ RestSharp.Tests/RestSharp.Tests.csproj	2015-10-14 15:30:50.314054204 +0200
@@ -61,10 +61,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.Build.Framework" />
     <Reference Include="Microsoft.Build.Utilities.v3.5" />
-    <Reference Include="nunit.framework, Version=2.6.4.14350, Culture=neutral, PublicKeyToken=96d09a1eb7f44a77, processorArchitecture=MSIL">
-      <HintPath>..\packages\NUnit.2.6.4\lib\nunit.framework.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <Reference Include="nunit.framework" />
     <Reference Include="System" />
     <Reference Include="System.Core">
     </Reference>
