--- RestSharp.Net45/RestSharp.Net45.Signed.csproj.old	2015-10-14 15:21:31.564607430 +0200
+++ RestSharp.Net45/RestSharp.Net45.Signed.csproj	2015-10-14 15:24:17.786448908 +0200
@@ -61,100 +61,100 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="..\restsharp\authenticators\HttpBasicAuthenticator.cs">
+    <Compile Include="..\RestSharp\Authenticators\HttpBasicAuthenticator.cs">
       <Link>Authenticators\HttpBasicAuthenticator.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\IAuthenticator.cs">
+    <Compile Include="..\RestSharp\Authenticators\IAuthenticator.cs">
       <Link>Authenticators\IAuthenticator.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\Authenticators\JwtAuthenticator.cs">
       <Link>Authenticators\JwtAuthenticator.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\NtlmAuthenticator.cs">
+    <Compile Include="..\RestSharp\Authenticators\NtlmAuthenticator.cs">
       <Link>Authenticators\NtlmAuthenticator.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\Authenticators\OAuth1Authenticator.cs">
       <Link>Authenticators\OAuth1Authenticator.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\OAuth2Authenticator.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth2Authenticator.cs">
       <Link>Authenticators\OAuth2Authenticator.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\extensions\CollectionExtensions.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\Extensions\CollectionExtensions.cs">
       <Link>Authenticators\OAuth\Extensions\CollectionExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\extensions\OAuthExtensions.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\Extensions\OAuthExtensions.cs">
       <Link>Authenticators\OAuth\Extensions\OAuthExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\extensions\StringExtensions.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\Extensions\StringExtensions.cs">
       <Link>Authenticators\OAuth\Extensions\StringExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\extensions\TimeExtensions.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\Extensions\TimeExtensions.cs">
       <Link>Authenticators\OAuth\Extensions\TimeExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\HttpPostParameter.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\HttpPostParameter.cs">
       <Link>Authenticators\OAuth\HttpPostParameter.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\HttpPostParameterType.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\HttpPostParameterType.cs">
       <Link>Authenticators\OAuth\HttpPostParameterType.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthParameterHandling.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthParameterHandling.cs">
       <Link>Authenticators\OAuth\OAuthParameterHandling.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthSignatureMethod.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthSignatureMethod.cs">
       <Link>Authenticators\OAuth\OAuthSignatureMethod.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthSignatureTreatment.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthSignatureTreatment.cs">
       <Link>Authenticators\OAuth\OAuthSignatureTreatment.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthTools.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthTools.cs">
       <Link>Authenticators\OAuth\OAuthTools.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthType.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthType.cs">
       <Link>Authenticators\OAuth\OAuthType.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthWebQueryInfo.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthWebQueryInfo.cs">
       <Link>Authenticators\OAuth\OAuthWebQueryInfo.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\OAuthWorkflow.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\OAuthWorkflow.cs">
       <Link>Authenticators\OAuth\OAuthWorkflow.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\WebPair.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\WebPair.cs">
       <Link>Authenticators\OAuth\WebPair.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\WebPairCollection.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\WebPairCollection.cs">
       <Link>Authenticators\OAuth\WebPairCollection.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\WebParameter.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\WebParameter.cs">
       <Link>Authenticators\OAuth\WebParameter.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\oauth\WebParameterCollection.cs">
+    <Compile Include="..\RestSharp\Authenticators\OAuth\WebParameterCollection.cs">
       <Link>Authenticators\OAuth\WebParameterCollection.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\authenticators\SimpleAuthenticator.cs">
+    <Compile Include="..\RestSharp\Authenticators\SimpleAuthenticator.cs">
       <Link>Authenticators\SimpleAuthenticator.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\DeserializeAsAttribute.cs">
+    <Compile Include="..\RestSharp\Deserializers\DeserializeAsAttribute.cs">
       <Link>Deserializers\DeserializeAsAttribute.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\DotNetXmlDeserializer.cs">
+    <Compile Include="..\RestSharp\Deserializers\DotNetXmlDeserializer.cs">
       <Link>Deserializers\DotNetXmlDeserializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\IDeserializer.cs">
+    <Compile Include="..\RestSharp\Deserializers\IDeserializer.cs">
       <Link>Deserializers\IDeserializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\JsonDeserializer.cs">
+    <Compile Include="..\RestSharp\Deserializers\JsonDeserializer.cs">
       <Link>Deserializers\JsonDeserializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\XmlAttributeDeserializer.cs">
+    <Compile Include="..\RestSharp\Deserializers\XmlAttributeDeserializer.cs">
       <Link>Deserializers\XmlAttributeDeserializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\deserializers\XmlDeserializer.cs">
+    <Compile Include="..\RestSharp\Deserializers\XmlDeserializer.cs">
       <Link>Deserializers\XmlDeserializer.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\Enum.cs">
       <Link>Enum.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\extensions\MiscExtensions.cs">
+    <Compile Include="..\RestSharp\Extensions\MiscExtensions.cs">
       <Link>Extensions\MiscExtensions.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\Extensions\MonoHttp\Helpers.cs">
@@ -166,19 +166,19 @@
     <Compile Include="..\RestSharp\Extensions\MonoHttp\HttpUtility.cs">
       <Link>Extensions\MonoHttp\HttpUtility.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\extensions\ReflectionExtensions.cs">
+    <Compile Include="..\RestSharp\Extensions\ReflectionExtensions.cs">
       <Link>Extensions\ReflectionExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\extensions\ResponseExtensions.cs">
+    <Compile Include="..\RestSharp\Extensions\ResponseExtensions.cs">
       <Link>Extensions\ResponseExtensions.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\Extensions\ResponseStatusExtensions.cs">
       <Link>Extensions\ResponseStatusExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\extensions\StringExtensions.cs">
+    <Compile Include="..\RestSharp\Extensions\StringExtensions.cs">
       <Link>Extensions\StringExtensions.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\extensions\XmlExtensions.cs">
+    <Compile Include="..\RestSharp\Extensions\XmlExtensions.cs">
       <Link>Extensions\XmlExtensions.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\FileParameter.cs">
@@ -253,31 +253,31 @@
     <Compile Include="..\RestSharp\RestResponseCookie.cs">
       <Link>RestResponseCookie.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\serializers\DotNetXmlSerializer.cs">
+    <Compile Include="..\RestSharp\Serializers\DotNetXmlSerializer.cs">
       <Link>Serializers\DotNetXmlSerializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\serializers\ISerializer.cs">
+    <Compile Include="..\RestSharp\Serializers\ISerializer.cs">
       <Link>Serializers\ISerializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\serializers\JsonSerializer.cs">
+    <Compile Include="..\RestSharp\Serializers\JsonSerializer.cs">
       <Link>Serializers\JsonSerializer.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\serializers\SerializeAsAttribute.cs">
+    <Compile Include="..\RestSharp\Serializers\SerializeAsAttribute.cs">
       <Link>Serializers\SerializeAsAttribute.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\serializers\XmlSerializer.cs">
+    <Compile Include="..\RestSharp\Serializers\XmlSerializer.cs">
       <Link>Serializers\XmlSerializer.cs</Link>
     </Compile>
     <Compile Include="..\RestSharp\SharedAssemblyInfo.cs">
       <Link>SharedAssemblyInfo.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\validation\Require.cs">
+    <Compile Include="..\RestSharp\Validation\Require.cs">
       <Link>Validation\Require.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\SimpleJson.cs">
+    <Compile Include="..\RestSharp\SimpleJson.cs">
       <Link>SimpleJson.cs</Link>
     </Compile>
-    <Compile Include="..\restsharp\validation\Validate.cs">
+    <Compile Include="..\RestSharp\Validation\Validate.cs">
       <Link>Validation\Validate.cs</Link>
     </Compile>
     <Compile Include="Properties\AssemblyInfo.cs" />
--- RestSharp.Tests/RestSharp.Tests.csproj.old	2015-10-14 15:25:42.008368588 +0200
+++ RestSharp.Tests/RestSharp.Tests.csproj	2015-10-14 15:26:00.106351328 +0200
@@ -102,7 +102,7 @@
     <Compile Include="SampleClasses\nullables.cs" />
     <Compile Include="SampleClasses\eventful.cs" />
     <Compile Include="SampleClasses\Oddball.cs" />
-    <Compile Include="SampleClasses\SoUser.cs" />
+    <Compile Include="SampleClasses\SOUser.cs" />
     <Compile Include="SampleClasses\TwilioCallList.cs" />
     <Compile Include="SampleClasses\twitter.cs" />
     <Compile Include="SerializerTests.cs" />
