--- mlapronidl/apron_caml.c.orig	2024-06-07 09:36:08.000000000 -0600
+++ mlapronidl/apron_caml.c	2024-12-10 13:10:35.917848446 -0700
@@ -149,7 +149,7 @@ value camlidl_apron_lincons0_c2ml(ap_lin
 {
   value vres;
   value _v[3];
-  _v[0] = _v[1] = _v[2] = 0;
+  _v[0] = _v[1] = _v[2] = Val_unit;
 
   Begin_roots_block(_v, 3)
     _v[0] = camlidl_apron_linexpr0_ptr_c2ml(&cons->linexpr0);
@@ -519,7 +519,7 @@ value camlidl_apron_policy_optr_c2ml(ap_
   if (p==NULL){
     return Val_int(0);
   } else {
-    value v,v2=0;
+    value v,v2=Val_unit;
     Begin_roots1(v2);
     v2 = camlidl_apron_policy_ptr_c2ml(p);
     v = caml_alloc_small(1,0);
--- mlapronidl/tcons0.idl.orig	2024-06-07 09:36:08.000000000 -0600
+++ mlapronidl/tcons0.idl	2024-12-10 13:10:54.629608314 -0700
@@ -73,7 +73,7 @@ value camlidl_apron_tcons0_c2ml(ap_tcons
 {\n\
   value vres;\n\
   value _v[3];\n\
-  _v[0] = _v[1] = _v[2] = 0;\n\
+  _v[0] = _v[1] = _v[2] = Val_unit;\n\
 \n\
   Begin_roots_block(_v, 3)\n\
     _v[0] = camlidl_apron_texpr0_ptr_c2ml(&cons->texpr0);\n\
