diff -up sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTFont.i~ sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTFont.i
--- sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTFont.i~	2005-02-20 23:19:13.000000000 +0100
+++ sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTFont.i	2009-05-22 12:52:37.000000000 +0200
@@ -1,5 +1,3 @@
-%feature("notabstract") FTFont;
-
 class FTFont
 {
     public:
diff -up sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTGL.i~ sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTGL.i
--- sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTGL.i~	2005-02-20 23:19:13.000000000 +0100
+++ sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/FTGL.i	2009-05-22 12:55:12.000000000 +0200
@@ -1,13 +1,6 @@
 %module SWIG_FTGL
 %{
-    #include "FTFont.h"
-    #include "FTFace.h"
-    #include "FTGLBitmapFont.h"
-    #include "FTGLExtrdFont.h"
-    #include "FTGLOutlineFont.h"
-    #include "FTGLPixmapFont.h"
-    #include "FTGLPolygonFont.h"
-    #include "FTGLTextureFont.h"
+    #include <FTGL/ftgl.h>
 
   void SWIG_BBox(const char* string, FTFont* font, float* llx, float* lly, float* llz, float* urx, float* ury, float* urz) {
     font->BBox(string, *llx, *lly, *llz, *urx, *ury, *urz);
@@ -35,7 +28,6 @@
 typedef int FT_Error;
 
 %include "FTFont.i"
-%include "FTFace.i"
 %include "FTGLBitmapFont.i"
 %include "FTGLExtrdFont.i"
 %include "FTGLOutlineFont.i"
