diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2019-07-05 14:29:48.486775637 +0200
+++ ./tests/bootstrap.php	2019-07-05 14:32:02.171555393 +0200
@@ -2,15 +2,13 @@
 
 declare(strict_types=1);
 
-set_include_path(__DIR__.'/../lib/'.PATH_SEPARATOR.__DIR__.PATH_SEPARATOR.get_include_path());
-
-$autoLoader = include __DIR__.'/../vendor/autoload.php';
+require '@BUILDROOT@/usr/share/php/Sabre/DAV4/autoload.php';
+if (file_exists('/usr/share/php/Monolog2/autoload.php'))
+     require '/usr/share/php/Monolog2/autoload.php';
+else require '/usr/share/php/Monolog/autoload.php';
 
 // SabreDAV tests auto loading
-$autoLoader->add('Sabre\\', __DIR__);
-// VObject tests auto loading
-$autoLoader->addPsr4('Sabre\\VObject\\', __DIR__.'/../vendor/sabre/vobject/tests/VObject');
-$autoLoader->addPsr4('Sabre\\Xml\\', __DIR__.'/../vendor/sabre/xml/tests/Sabre/Xml');
+\Fedora\Autoloader\Autoload::addPsr0('Sabre\\', __DIR__);
 
 date_default_timezone_set('UTC');
 
@@ -35,8 +33,8 @@ foreach ($environmentVars as $var) {
 $config = [
     'SABRE_TEMPDIR' => dirname(__FILE__).'/temp/',
     'SABRE_HASSQLITE' => in_array('sqlite', PDO::getAvailableDrivers()),
-    'SABRE_HASMYSQL' => in_array('mysql', PDO::getAvailableDrivers()),
-    'SABRE_HASPGSQL' => in_array('pgsql', PDO::getAvailableDrivers()),
+    'SABRE_HASMYSQL' => false,
+    'SABRE_HASPGSQL' => false,
     'SABRE_MYSQLDSN' => 'mysql:host=127.0.0.1;dbname=sabredav_test',
     'SABRE_MYSQLUSER' => 'sabredav',
     'SABRE_MYSQLPASS' => '',
