--- regorus-0.2.6/Cargo.toml	1970-01-01T00:00:01+00:00
+++ regorus-0.2.6/Cargo.toml	2026-01-10T08:44:03.129179+00:00
@@ -27,6 +27,7 @@
     "policy-as-code",
     "rego",
 ]
+license = "MIT AND Apache-2.0 AND BSD-3-Clause"
 license-file = "LICENSE"
 repository = "https://github.com/microsoft/regorus"
 
@@ -92,7 +93,7 @@
 optional = true
 
 [dependencies.constant_time_eq]
-version = "0.3.0"
+version = "0.4.2"
 optional = true
 default-features = false
 
@@ -115,11 +116,6 @@
 
 [dependencies.itertools]
 version = "0.13.0"
-optional = true
-default-features = false
-
-[dependencies.jsonschema]
-version = "0.22.3"
 optional = true
 default-features = false
 
@@ -257,7 +253,6 @@
     "hex",
     "http",
     "jwt",
-    "jsonschema",
     "opa-runtime",
     "regex",
     "semver",
@@ -271,7 +266,6 @@
 graph = []
 hex = ["dep:data-encoding"]
 http = []
-jsonschema = ["dep:jsonschema"]
 jwt = [
     "dep:jsonwebtoken",
     "dep:data-encoding",
