diff --git a/tests/TESTrun b/tests/TESTrun
index e934b69..9b6e0da 100755
--- a/tests/TESTrun
+++ b/tests/TESTrun
@@ -120,7 +120,7 @@ sub runtest {
         # all non-flag arguments.
         $test_args = "-tttt $test_args";
     }
-    $r = system "$TCPDUMP -# -n -r $input $test_args >${newdir}/${outputbase} 2>${rawstderrlog}";
+    $r = system "$TCPDUMP -# -nn -r $input $test_args >${newdir}/${outputbase} 2>${rawstderrlog}";
 
     if($r != 0) {
         #
