diff -Naur Coin-2.5.0.doxygen/include/Inventor/SbBasic.h Coin-2.5.0.gcc-4.7.0/include/Inventor/SbBasic.h
--- Coin-2.5.0.doxygen/include/Inventor/SbBasic.h	2007-10-01 03:23:36.000000000 +0200
+++ Coin-2.5.0.gcc-4.7.0/include/Inventor/SbBasic.h	2012-01-05 05:10:21.817879424 +0100
@@ -27,6 +27,9 @@
 /* *********************************************************************** */
 
 #include <Inventor/C/basic.h>
+#ifndef NDEBUG
+#include <Inventor/C/errors/debugerror.h>
+#endif // !NDEBUG
 
 /* *********************************************************************** */
 /* Trap people trying to use Inventor headers while compiling C source code.
