--- ghc-9.10.3/rts/rts.cabal~	2025-09-11 00:02:37.000000000 +0800
+++ ghc-9.10.3/rts/rts.cabal	2026-02-03 01:52:56.432806708 +0800
@@ -362,14 +362,14 @@
           else
             asm-sources: adjustor/NativeAmd64Asm.S
             c-sources: adjustor/NativeAmd64.c
-        if arch(ppc) || arch(ppc64)
+        if arch(ppc) || arch(ppc64) || arch(ppc64le)
           asm-sources: AdjustorAsm.S
           c-sources: adjustor/NativePowerPC.c
         if arch(ia64)
           c-sources: adjustor/NativeIA64.c
 
       -- Use assembler STG entrypoint on architectures where it is used
-      if arch(ppc) || arch(ppc64) || arch(s390x) || arch(riscv64) || arch(loongarch64)
+      if arch(ppc) || arch(ppc64) || arch(ppc64le) || arch(s390x) || arch(riscv64) || arch(loongarch64)
         asm-sources: StgCRunAsm.S
 
       c-sources: Adjustor.c
