From 768366c350b6a7c9bf22090a3f98c526c5e79184 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 29 May 2012 20:44:18 +0100
Subject: [PATCH 2/2] configure: Allow user defined C compiler flags.

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 94c4f114d3..f06347a5b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3243,7 +3243,7 @@ AC_CONFIG_COMMANDS_PRE([
       [mkexedebugflag="${mkexe_ldflags_prefix}${mkexedebugflag}"])
     mkdll_ldflags=""
     AS_IF([test -n "${LDFLAGS}"],
-      [for flag in ${LDFLAGS}; do
+      [for flag in "${LDFLAGS}"; do
         mkdll_ldflags="${mkdll_ldflags} ${mkexe_ldflags_prefix}${flag}"
       done
       mkdll_ldflags_exp="$mkdll_ldflags"])
-- 
2.55.0

