--- kcheckers-0.8.1/common.h~	2005-12-29 20:53:16 +0300
+++ kcheckers-0.8.1/common.h	2007-10-02 16:28:17 +0400
@@ -35,7 +35,7 @@
 
 /* !!! Do not change PREFIX variable name, please. !!! */
 /* !!! It is used in kcheckers.pro. !!! */
-#define PREFIX		"/usr/local"
+#define PREFIX		"/usr"
 #define USER_PATH	".kcheckers"		// in $HOME
 #define THEME_DIR	"themes/"
 
--- kcheckers-0.8.1/kcheckers.pro~	2005-12-29 20:55:55 +0300
+++ kcheckers-0.8.1/kcheckers.pro	2007-10-02 16:33:22 +0400
@@ -21,8 +21,8 @@ RESOURCES = kcheckers.qrc
 
 
 TARGET		= kcheckers
-PREFIX		= $$system(grep 'define PREFIX' common.h | cut -d'"' -f2)
-SHARE_PATH	= $$system(grep 'define SHARE_PATH' common.h | cut -d'"' -f2)
+PREFIX		= "/usr"
+SHARE_PATH	= "/usr/share"
 
 TRANSLATIONS	= i18n/kcheckers_de.ts i18n/kcheckers_fr.ts
 #		i18n/kcheckers_ru.ts
