diff -rupN --no-dereference gcc-16.1.1-20260501/libgcc/config/i386/enable-execute-stack-mingw32.c gcc-16.1.1-20260501-new/libgcc/config/i386/enable-execute-stack-mingw32.c
--- gcc-16.1.1-20260501/libgcc/config/i386/enable-execute-stack-mingw32.c	2026-05-01 17:17:42.000000000 +0200
+++ gcc-16.1.1-20260501-new/libgcc/config/i386/enable-execute-stack-mingw32.c	2026-05-06 23:06:09.637637643 +0200
@@ -25,6 +25,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <stdlib.h>
 #include <windows.h>
+#include <stdlib.h>
 
 extern void __enable_execute_stack (void *);
 
