From 19cfcb106e45e1bed51f449f08277f3349079f37 Mon Sep 17 00:00:00 2001
From: Pavel Odintsov <pavel.odintsov@gmail.com>
Date: Wed, 19 Feb 2025 14:58:13 +0300
Subject: [PATCH] Boost migration patch by Marek Zarychta. Closes #1027

---
 src/fast_library.cpp | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/src/fast_library.cpp b/src/fast_library.cpp
index 4ac76d7b..3643cf5f 100644
--- a/src/fast_library.cpp
+++ b/src/fast_library.cpp
@@ -1162,15 +1162,13 @@ std::string serialize_statistic_counters_about_attack(attack_details_t& current_
 
 std::string dns_lookup(std::string domain_name) {
     try {
-        boost::asio::io_service io_service;
-        boost::asio::ip::tcp::resolver resolver(io_service);
+        boost::asio::io_context io_context;
+        boost::asio::ip::tcp::resolver resolver(io_context);
 
-        boost::asio::ip::tcp::resolver::query query(domain_name, "");
+        auto results = resolver.resolve(domain_name, "");
 
-        for (boost::asio::ip::tcp::resolver::iterator i = resolver.resolve(query);
-             i != boost::asio::ip::tcp::resolver::iterator(); ++i) {
-            boost::asio::ip::tcp::endpoint end = *i;
-            return end.address().to_string();
+        for (const auto& entry : results) {
+            return entry.endpoint().address().to_string();
         }
     } catch (std::exception& e) {
         return "";
@@ -1487,7 +1485,7 @@ bool validate_ipv6_or_ipv4_host(const std::string host) {
     boost::system::error_code ec;
 
     // Try to build it from string representation
-    boost::asio::ip::address::from_string(host, ec);
+    boost::asio::ip::make_address(host, ec);
 
     // If we failed to parse it
     if (ec) {
@@ -1595,7 +1593,7 @@ bool execute_web_request_secure(std::string address,
             return false;
         }
 
-        auto end_point = r.resolve(boost::asio::ip::tcp::resolver::query{ host, port }, ec);
+        auto end_point = r.resolve(host, port, ec);
 
         if (ec) {
             logger << log4cpp::Priority::ERROR << "Could not resolve peer address in execute_web_request " << ec;
@@ -1749,11 +1747,11 @@ bool execute_web_request(std::string address,
 
         // Normal boost::asio setup
         // std::string const host = "178.62.227.110";
-        boost::asio::io_service ios;
+        boost::asio::io_context ios;
         boost::asio::ip::tcp::resolver r(ios);
         boost::asio::ip::tcp::socket sock(ios);
 
-        auto end_point = r.resolve(boost::asio::ip::tcp::resolver::query{ host, port }, ec);
+        auto end_point = r.resolve(host, port, ec);
 
         if (ec) {
             error_text = "Could not resolve peer address in execute_web_request " + ec.message();
-- 
2.52.0

