diff -up notify-sharp/configure.ac.dbus2 notify-sharp/configure.ac
--- notify-sharp/configure.ac.dbus2	2016-10-04 10:37:28.169593565 -0400
+++ notify-sharp/configure.ac	2016-10-04 10:37:44.417473413 -0400
@@ -68,8 +68,8 @@ AC_SUBST(GTK_SHARP_LIBS)
 
 DBUS_SHARP_REQ_VERSION=0.7
 DBUS_SHARP_GLIB_REQ_VERSION=0.5
-PKG_CHECK_MODULES(DBUS_SHARP, dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
-                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION)
+PKG_CHECK_MODULES(DBUS_SHARP, dbus-sharp-2.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-2.0 >= $DBUS_SHARP_GLIB_REQ_VERSION)
 AC_SUBST(DBUS_SHARP_LIBS)
 
 required_assemblies="Mono.Posix"
