--- io-uring-0.5.13/Cargo.toml	1970-01-01T00:00:01+00:00
+++ io-uring-0.5.13/Cargo.toml	2023-05-10T15:38:21.546427+00:00
@@ -33,10 +33,6 @@
 version = "0.2.98"
 default-features = false
 
-[dependencies.sc]
-version = "0.2"
-optional = true
-
 [dev-dependencies.anyhow]
 version = "1"
 
@@ -46,12 +42,6 @@
 [dev-dependencies.socket2]
 version = "0.4"
 
-[build-dependencies.bindgen]
-version = "0.61"
-optional = true
-
 [features]
-direct-syscall = ["sc"]
 io_safety = []
-overwrite = ["bindgen"]
 unstable = []
