--- fs_at-0.2.1/Cargo.toml
+++ fs_at-0.2.1/Cargo.toml
@@ -69,18 +69,3 @@
 version = "1.19.0"
 optional = true
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Wdk_Foundation",
-    "Wdk_Storage_FileSystem",
-    "Wdk_System_SystemServices",
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_Ioctl",
-    "Win32_System_Kernel",
-    "Win32_System_SystemServices",
-    "Win32_System_WindowsProgramming",
-]
