diff -up CMakeLists.txt.orig CMakeLists.txt
--- CMakeLists.txt.orig	2021-02-23 07:27:10.611067241 -0300
+++ CMakeLists.txt	2021-02-23 07:27:35.953203458 -0300
@@ -38,10 +38,10 @@ ADD_SUBDIRECTORY(extern)
 ADD_SUBDIRECTORY(src)
 
 # These files should only be installed when creating packages.
-INSTALL(FILES 
-	LICENSE
-	README.md
-	DESTINATION share/doc/nvtt)
+# INSTALL(FILES 
+#	LICENSE
+#	README.md
+#	DESTINATION share/doc/nvtt)
 
 # Add packaging support
 INCLUDE(InstallRequiredSystemLibraries)
