diff -up src/nvtt/tests/CMakeLists.txt.orig src/nvtt/tests/CMakeLists.txt
--- src/nvtt/tests/CMakeLists.txt.orig	2021-02-23 06:12:25.272491130 -0300
+++ src/nvtt/tests/CMakeLists.txt	2021-02-23 06:12:47.525609378 -0300
@@ -4,12 +4,12 @@ TARGET_LINK_LIBRARIES(filtertest nvcore
 
 ADD_EXECUTABLE(nvtestsuite testsuite.cpp)
 TARGET_LINK_LIBRARIES(nvtestsuite nvcore nvmath nvimage nvtt)
-ADD_TEST(NVTT.TestSuite.Kodak.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 0 -out output-cuda-kodak)
-ADD_TEST(NVTT.TestSuite.Waterloo.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 1 -out output-cuda-waterloo)
-ADD_TEST(NVTT.TestSuite.Epic.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 2 -out output-cuda-epic)
-ADD_TEST(NVTT.TestSuite.Kodak.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 0 -nocuda -out output-nocuda-kodak)
-ADD_TEST(NVTT.TestSuite.Waterloo.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 1 -nocuda -out output-nocuda-waterloo)
-ADD_TEST(NVTT.TestSuite.Epic.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite -set 2 -nocuda -out output-nocuda-epic)
+ADD_TEST(NVTT.TestSuite.Kodak.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 0 -out output-cuda-kodak)
+ADD_TEST(NVTT.TestSuite.Waterloo.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 1 -out output-cuda-waterloo)
+ADD_TEST(NVTT.TestSuite.Epic.cuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 2 -out output-cuda-epic)
+ADD_TEST(NVTT.TestSuite.Kodak.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 0 -nocuda -out output-nocuda-kodak)
+ADD_TEST(NVTT.TestSuite.Waterloo.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 1 -nocuda -out output-nocuda-waterloo)
+ADD_TEST(NVTT.TestSuite.Epic.nocuda nvtestsuite -path ${NV_SOURCE_DIR}/data/testsuite/ -set 2 -nocuda -out output-nocuda-epic)
 
 IF (CUDA_FOUND)
     ADD_EXECUTABLE(driverapitest driverapi.cpp)
@@ -31,7 +31,7 @@ TARGET_LINK_LIBRARIES(nvhdrtest nvcore n
 #ADD_EXECUTABLE(bc1enc bc1enc.cpp)
 #TARGET_LINK_LIBRARIES(bc1enc nvcore nvimage nvmath squish CMP_Core)
 
-INSTALL(TARGETS nvtestsuite nvhdrtest DESTINATION bin)
+#INSTALL(TARGETS nvtestsuite nvhdrtest DESTINATION bin)
  
 #include_directories("/usr/include/ffmpeg/")
 #ADD_EXECUTABLE(nvmpegenc tools/mpegenc.cpp tools/cmdline.h)
