diff -up src/nvtt/icbc.h.orig src/nvtt/icbc.h
--- src/nvtt/icbc.h.orig	2021-02-25 14:58:15.331333196 -0300
+++ src/nvtt/icbc.h	2021-02-25 14:59:57.648990370 -0300
@@ -91,7 +91,7 @@ namespace icbc {
         #endif
     #endif
 
-    #if ICBC_PPC
+    #if ICBC_PPC && BUILD_SQUISH_WITH_ALTIVEC
         #define ICBC_SIMD ICBC_VMX
     #endif
 #endif
@@ -1195,17 +1195,17 @@ ICBC_FORCEINLINE void vtranspose4(VFloat
 constexpr int VEC_SIZE = 4;
 
 union VFloat {
-    vectro float v;
+    __vector float v;
     float e[4];
     VFloat() {}
-    VFloat(vector float v) : v(v) {}
-    operator vector float & () { return v; }
+    VFloat(__vector float v) : v(v) {}
+    operator __vector float & () { return v; }
 };
 struct VMask {
-    vector unsigned int v;
+    __vector unsigned int v;
     VMask() {}
-    VMask(vector unsigned int v) : v(v) {}
-    operator vector unsigned int & () { return v; }
+    VMask(__vector unsigned int v) : v(v) {}
+    operator __vector unsigned int & () { return v; }
 };
 
 ICBC_FORCEINLINE float & lane(VFloat & v, int i) {
@@ -1213,15 +1213,15 @@ ICBC_FORCEINLINE float & lane(VFloat & v
 }
 
 ICBC_FORCEINLINE VFloat vzero() {
-    return vec_splats(0.0f);
+    return ( __vector float ){ 0.0f };
 }
 
 ICBC_FORCEINLINE VFloat vbroadcast(float a) {
-    return vec_splats(a);
+    return ( __vector float ){ a };
 }
 
 ICBC_FORCEINLINE VFloat vload(const float * ptr) {
-    return vec_ld(ptr)
+    return ( __vector float ){ ptr[0], ptr[1], ptr[2], ptr[3] };
 }
 
 ICBC_FORCEINLINE VFloat operator+(VFloat a, VFloat b) {
@@ -1233,16 +1233,16 @@ ICBC_FORCEINLINE VFloat operator-(VFloat
 }
 
 ICBC_FORCEINLINE VFloat operator*(VFloat a, VFloat b) {
-    return vec_madd(a, b, vec_splats(-0.0f));
+    return vec_madd(a, b, ( __vector float ){ -0.0f });
 }
 
 ICBC_FORCEINLINE VFloat vrcp(VFloat a) {
 #if ICBC_USE_RCP
     // get the reciprocal estimate
-    vector float estimate = vec_re( v.vec );
+    __vector float estimate = vec_re( v.vec );
 
     // one round of Newton-Rhaphson refinement
-    vector float diff = vec_nmsub( estimate, v.vec, vec_splats( 1.0f ) );
+    __vector float diff = vec_nmsub( estimate, v.vec, ( __vector float ){ 1.0f } );
     return vec_madd(diff, estimate, estimate );
 #else
     return vec_div(vec_splats(1),a);
diff -up src/nvtt/squish/simd_ve.h.orig src/nvtt/squish/simd_ve.h
--- src/nvtt/squish/simd_ve.h.orig	2021-02-25 14:37:32.907095799 -0300
+++ src/nvtt/squish/simd_ve.h	2021-02-25 14:37:35.891108297 -0300
@@ -27,14 +27,12 @@
 #ifndef SQUISH_SIMD_VE_H
 #define SQUISH_SIMD_VE_H
 
-#if !defined(__APPLE_ALTIVEC__) || defined(__PPC64__)
 #include <altivec.h>
 #undef bool
-#endif
 
 namespace nvsquish {
 
-#define VEC4_CONST( X ) Vec4( vec_splats( (float)X ) )
+#define VEC4_CONST( X ) Vec4( ( __vector float ) { X } )
 
 class Vec4
 {
@@ -42,8 +40,8 @@ public:
 	typedef Vec4 Arg;
 
 	Vec4() {}
-		
-	explicit Vec4( vector float v ) : m_v( v ) {}
+
+	explicit Vec4( __vector float v ) : m_v( v ) {}
 	
 	Vec4( Vec4 const& arg ) : m_v( arg.m_v ) {}
 	
@@ -55,7 +53,7 @@ public:
 	
 	Vec4( const float * v )
 	{
-		union { vector float v; float c[4]; } u;
+		union { __vector float v; float c[4]; } u;
 		u.c[0] = v[0];
 		u.c[1] = v[1];
 		u.c[2] = v[2];
@@ -65,7 +63,7 @@ public:
 
 	Vec4( float x, float y, float z, float w )
 	{
-		union { vector float v; float c[4]; } u;
+		union { __vector float v; float c[4]; } u;
 		u.c[0] = x;
 		u.c[1] = y;
 		u.c[2] = z;
@@ -75,14 +73,14 @@ public:
 	
 	Vec3 GetVec3() const
 	{
-		union { vector float v; float c[4]; } u;
+		union { __vector float v; float c[4]; } u;
 		u.v = m_v;
 		return Vec3( u.c[0], u.c[1], u.c[2] );
 	}
 
 	float GetX() const
 	{
-		union { vector float v; float c[4]; } u;
+		union { __vector float v; float c[4]; } u;
 		u.v = m_v;
 		return u.c[0];
 	}
@@ -106,7 +104,7 @@ public:
 	
 	Vec4& operator*=( Arg v )
 	{
-		m_v = vec_madd( m_v, v.m_v, vec_splats( -0.0f ) );
+		m_v = vec_madd( m_v, v.m_v, ( __vector float ){ -0.0f } );
 		return *this;
 	}
 	
@@ -122,7 +120,7 @@ public:
 	
 	friend Vec4 operator*( Vec4::Arg left, Vec4::Arg right  )
 	{
-		return Vec4( vec_madd( left.m_v, right.m_v, vec_splats( -0.0f ) ) );
+		return Vec4( vec_madd( left.m_v, right.m_v, ( __vector float ){ -0.0f } ) );
 	}
 	
 	//! Returns a*b + c
@@ -140,10 +138,10 @@ public:
 	friend Vec4 Reciprocal( Vec4::Arg v )
 	{
 		// get the reciprocal estimate
-		vector float estimate = vec_re( v.m_v );
+		__vector float estimate = vec_re( v.m_v );
 		
 		// one round of Newton-Rhaphson refinement
-		vector float diff = vec_nmsub( estimate, v.m_v, vec_splats( 1.0f ) );
+		__vector float diff = vec_nmsub( estimate, v.m_v, ( __vector float ){ 1.0f } );
 		return Vec4( vec_madd( diff, estimate, estimate ) );
 	}
 	
@@ -164,12 +162,12 @@ public:
 	
 	friend Vec4 CompareEqual( Vec4::Arg left, Vec4::Arg right )
 	{
-		return Vec4( ( vector float )vec_cmpeq( left.m_v, right.m_v ) );
+		return Vec4( ( __vector float )vec_cmpeq( left.m_v, right.m_v ) );
 	}
 	
 	friend Vec4 Select( Vec4::Arg off, Vec4::Arg on, Vec4::Arg bits )
 	{
-		return Vec4( vec_sel( off.m_v, on.m_v, ( vector unsigned int )bits.m_v ) );
+		return Vec4( vec_sel( off.m_v, on.m_v, ( __vector unsigned int )bits.m_v ) );
 	}
 	
 	friend bool CompareAnyLessThan( Vec4::Arg left, Vec4::Arg right ) 
@@ -178,7 +176,7 @@ public:
 	}
 	
 private:
-	vector float m_v;
+	__vector float m_v;
 };
 
 } // namespace squish
diff -up extern/libsquish-1.15/simd_ve.h.orig extern/libsquish-1.15/simd_ve.h
--- extern/libsquish-1.15/simd_ve.h.orig	2021-02-25 14:37:32.910095811 -0300
+++ extern/libsquish-1.15/simd_ve.h	2021-02-25 14:37:35.892108302 -0300
@@ -31,7 +31,7 @@
 
 namespace squish {
 
-#define VEC4_CONST( X ) Vec4( ( vector float ){ X } )
+#define VEC4_CONST( X ) Vec4( ( __vector float ){ X } )
 
 class Vec4
 {
@@ -40,7 +40,7 @@ public:
 
     Vec4() {}
 
-    explicit Vec4( vector float v ) : m_v( v ) {}
+    explicit Vec4( __vector float v ) : m_v( v ) {}
 
     Vec4( Vec4 const& arg ) : m_v( arg.m_v ) {}
 
@@ -52,7 +52,7 @@ public:
 
     explicit Vec4( float s )
     {
-        union { vector float v; float c[4]; } u;
+        union { __vector float v; float c[4]; } u;
         u.c[0] = s;
         u.c[1] = s;
         u.c[2] = s;
@@ -62,7 +62,7 @@ public:
 
     Vec4( float x, float y, float z, float w )
     {
-        union { vector float v; float c[4]; } u;
+        union { __vector float v; float c[4]; } u;
         u.c[0] = x;
         u.c[1] = y;
         u.c[2] = z;
@@ -72,7 +72,7 @@ public:
 
     Vec3 GetVec3() const
     {
-        union { vector float v; float c[4]; } u;
+        union { __vector float v; float c[4]; } u;
         u.v = m_v;
         return Vec3( u.c[0], u.c[1], u.c[2] );
     }
@@ -96,7 +96,7 @@ public:
 
     Vec4& operator*=( Arg v )
     {
-        m_v = vec_madd( m_v, v.m_v, ( vector float ){ -0.0f } );
+        m_v = vec_madd( m_v, v.m_v, ( __vector float ){ -0.0f } );
         return *this;
     }
 
@@ -112,7 +112,7 @@ public:
 
     friend Vec4 operator*( Vec4::Arg left, Vec4::Arg right  )
     {
-        return Vec4( vec_madd( left.m_v, right.m_v, ( vector float ){ -0.0f } ) );
+        return Vec4( vec_madd( left.m_v, right.m_v, ( __vector float ){ -0.0f } ) );
     }
 
     //! Returns a*b + c
@@ -130,10 +130,10 @@ public:
     friend Vec4 Reciprocal( Vec4::Arg v )
     {
         // get the reciprocal estimate
-        vector float estimate = vec_re( v.m_v );
+        __vector float estimate = vec_re( v.m_v );
 
         // one round of Newton-Rhaphson refinement
-        vector float diff = vec_nmsub( estimate, v.m_v, ( vector float ){ 1.0f } );
+        __vector float diff = vec_nmsub( estimate, v.m_v, ( __vector float ){ 1.0f } );
         return Vec4( vec_madd( diff, estimate, estimate ) );
     }
 
@@ -158,7 +158,7 @@ public:
     }
 
 private:
-    vector float m_v;
+    __vector float m_v;
 };
 
 } // namespace squish
