diff -rupN SDL_mixer-1.2.12/dynamic_ogg.c SDL_mixer-1.2.12-new/dynamic_ogg.c
--- SDL_mixer-1.2.12/dynamic_ogg.c	2012-01-15 23:01:05.000000000 +0100
+++ SDL_mixer-1.2.12-new/dynamic_ogg.c	2024-07-24 22:19:07.182975481 +0200
@@ -108,7 +108,7 @@ int Mix_InitOgg()
 	if ( vorbis.loaded == 0 ) {
 		vorbis.ov_clear = ov_clear;
 		vorbis.ov_info = ov_info;
-		vorbis.ov_open_callbacks = ov_open_callbacks;
+		vorbis.ov_open_callbacks = (int (*)(void *, OggVorbis_File *, char *, long, ov_callbacks)) ov_open_callbacks;
 		vorbis.ov_pcm_total = ov_pcm_total;
 		vorbis.ov_read = ov_read;
 		vorbis.ov_time_seek = ov_time_seek;
