--- rd-util-2.2.5/Cargo.toml	1970-01-01T00:00:01+00:00
+++ rd-util-2.2.5/Cargo.toml	2025-07-26T12:55:31.806426+00:00
@@ -30,7 +30,7 @@
 version = "2.33"
 
 [dependencies.console]
-version = "0.15"
+version = ">=0.15, <0.17"
 
 [dependencies.crossbeam]
 version = "0.8"
@@ -92,8 +92,6 @@
 [dependencies.sysinfo]
 version = "0.30"
 
-[build-dependencies.anyhow]
-version = "1.0"
-
-[build-dependencies.vergen]
-version = "7.0"
+[build-dependencies]
+anyhow = "1.0"
+vergen = { version = "8.3", features = ["git", "gitcl", "cargo"] }
