diff --git a/build.rs b/build.rs
index 63685704..0f3ff96e 100644
--- a/build.rs
+++ b/build.rs
@@ -1,16 +1,8 @@
 // Copyright (c) Facebook, Inc. and its affiliates.
-use vergen::{vergen, Config, SemverKind};
-
 fn main() -> anyhow::Result<()> {
-    let mut config = Config::default();
-    *config.git_mut().semver_kind_mut() = SemverKind::Lightweight;
-    *config.git_mut().semver_dirty_mut() = Some("-dirty");
-    match vergen(config) {
-        Ok(()) => Ok(()),
-        Err(_) => {
-            let mut config = Config::default();
-            *config.git_mut().enabled_mut() = false;
-            vergen(config)
-        }
-    }
+    vergen::EmitBuilder::builder()
+        .git_sha(true)
+        .git_dirty(true)
+        .cargo_target_triple()
+        .emit()
 }
diff --git a/src/lib.rs b/src/lib.rs
index 844d03bf..861e6d8e 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -49,14 +49,19 @@ pub const READ: usize = 0;
 pub const WRITE: usize = 1;
 
 lazy_static::lazy_static! {
-    static ref GIT_VERSION: &'static str = {
-        match option_env!("VERGEN_GIT_SEMVER_LIGHTWEIGHT") {
-            Some(v) => split_git_version(v).1,
-            None => ""
+    static ref GIT_VERSION: String = {
+	let mut ver = String::new();
+        if let Some(v) = option_env!("VERGEN_GIT_SHA") {
+            ver += "g";
+	    ver += v;
+	    if let Some("true") = option_env!("VERGEN_GIT_DIRTY") {
+		ver += "-dirty";
+	    }
         }
+	ver
     };
     static ref BUILD_TAG: String = {
-        let mut tag = env!("VERGEN_RUSTC_HOST_TRIPLE").to_string();
+        let mut tag = env!("VERGEN_CARGO_TARGET_TRIPLE").to_string();
         if cfg!(debug_assertions) {
             write!(tag, "/debug").unwrap();
         }
