diff --git a/src/reportlab/lib/utils.py b/src/reportlab/lib/utils.py
index 29ca08a..1024634 100644
--- a/src/reportlab/lib/utils.py
+++ b/src/reportlab/lib/utils.py
@@ -604,7 +604,7 @@ def rl_get_module(name,dir):
                 dir = _startswith_rl(dir)
                 dir = (dir=='.' or not dir) and _archive or os.path.join(_archive,dir.replace('/',os.sep))
                 zi = zipimport.zipimporter(dir)
-                return zi.load_module(name)
+                return zi.exec_module(name)
             raise ImportError('%s[%s]' % (name,dir))
     finally:
         if om: sys.modules[name] = om
