diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4a24e0c6..8ba95e95 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -170,26 +170,22 @@ set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
 set(BUILD_STATIC_LIBS ON)
 set(BUILD_SHARED_LIBS OFF)
 
-include(FetchContent)
 
 # Bundled liblzma
-include(dependencies/xz.cmake)
+find_package(LibLZMA 5.8.2 REQUIRED)
 
 # Bundled zstd
-include(dependencies/zstd.cmake)
-
-# Remote nghttp2
-include(dependencies/nghttp2.cmake)
+find_package(zstd 1.5.7 REQUIRED)
 
 # Bundled yescrypt
 include(dependencies/yescrypt.cmake)
 
 
 # Bundled zlib
-include(dependencies/zlib.cmake)
+find_package(ZLIB REQUIRED)
 
 # Bundled libarchive
-include(dependencies/libarchive.cmake)
+find_package(LibArchive 3.8.4 REQUIRED)
 
 # libcurl
 if(APPLE)
@@ -203,7 +199,7 @@ if(APPLE)
     # Note: No need to find OpenSSL on macOS - we use native Security.framework
     # for RSA operations and CommonCrypto for hashing
 else()
-    include(dependencies/curl.cmake)
+    find_package(CURL 8.17.0 REQUIRED)
 endif()
 
 
@@ -537,7 +533,7 @@ else()
     include(linux/PlatformPackaging.cmake)
 endif()
 
-add_dependencies(${PROJECT_NAME} zlibstatic yescrypt)
+add_dependencies(${PROJECT_NAME} yescrypt)
 include_directories(${CURL_INCLUDE_DIR} ${LibArchive_INCLUDE_DIR} ${LIBLZMA_INCLUDE_DIRS} ${LIBDRM_INCLUDE_DIRS} ${ZLIB_INCLUDE_DIRS} ${ZSTD_INCLUDE_DIR} ${YESCRYPT_INCLUDE_DIR})
 
 # Link different Qt components based on build type
