commit 54da7c8206e38d0bf08a250835553754a1235360
Author: Christos Zoulas <christos@zoulas.com>
Date:   Wed Jan 21 23:28:33 2015 -0500

    make sure variables are initialized

diff --git a/amd/nfs_subr.c b/amd/nfs_subr.c
index 7192a6a..30effba 100644
--- a/amd/nfs_subr.c
+++ b/amd/nfs_subr.c
@@ -1711,6 +1711,9 @@ am_nfs3_fsstat_3_svc(am_FSSTAT3args *argp, struct svc_req *rqstp)
     if ((gopt.flags & CFM_SHOW_STATFS_ENTRIES) &&
 	mp->am_al->al_mnt && mp->am_al->al_mnt->mf_mopts) {
       mnt.mnt_opts = mp->am_al->al_mnt->mf_mopts;
+      blocks = 0;
+      bfree = 0;
+      bavail = 0;
       if (amu_hasmntopt(&mnt, "browsable")) {
 	count_map_entries(mp, &blocks, &bfree, &bavail);
       }
