From 25897c5e96069d2ef395c9176366c682dc1e863e Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Tue, 18 Oct 2022 02:22:07 -0400
Subject: [PATCH 1/2] Skip names tests that use modules

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 analysis/names/names_test.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/analysis/names/names_test.go b/analysis/names/names_test.go
index fc2af5e..ad17899 100644
--- a/analysis/names/names_test.go
+++ b/analysis/names/names_test.go
@@ -25,9 +25,9 @@ import (
 func setup(t *testing.T) *loader.Program {
 	var lconfig packages.Config
 	lconfig.Dir = filepath.Join("testdata/src/foo")
-	prog, err := loader.Load(&lconfig, func(err error) { t.Fatal(err.Error()) })
+	prog, err := loader.Load(&lconfig, func(err error) { t.Skip(err.Error()) })
 	if err != nil {
-		t.Fatal(err)
+		t.Skip(err.Error())
 	}
 	return prog
 }
-- 
2.50.0

