diff -up transmission-4.0.0/gtk/Application.cc.orig transmission-4.0.0/gtk/Application.cc
--- transmission-4.0.0/gtk/Application.cc.orig	2023-01-30 06:23:05.000000000 +0100
+++ transmission-4.0.0/gtk/Application.cc	2023-02-15 20:49:40.137166535 +0100
@@ -700,7 +700,7 @@ std::string get_application_id(std::stri
 {
     struct stat sb = {};
     (void)::stat(config_dir.c_str(), &sb);
-    return fmt::format("com.transmissionbt.transmission_{}_{}", sb.st_dev, sb.st_ino);
+    return fmt::format("com.transmissionbt.Transmission._{}_{}", sb.st_dev, sb.st_ino);
 }
 
 } // namespace
