diff --git a/targetd/block.py b/targetd/block.py
index b7b03f0..edd47ec 100644
--- a/targetd/block.py
+++ b/targetd/block.py
@@ -26,10 +26,17 @@ from rtslib_fb import (
     LUN,
     MappedLUN,
     RTSLibError,
-    RTSLibNotInCFS,
     NodeACLGroup,
 )
 
+# This commit https://github.com/open-iscsi/rtslib-fb/commit/fdd69b1984bac6a6c3f08890bc22267b848a26ac
+# renamed a number of error classes which causes us to do the following
+try:
+    from rtslib_fb import RTSLibNotInCFS
+except ImportError:
+    from rtslib_fb import RTSLibNotInCFSError as RTSLibNotInCFS
+
+
 from targetd.backends import lvm, zfs
 from targetd.main import TargetdError
 from targetd.utils import ignored, name_check
