From 622e62e0d597c84bf23d9171bda955baf6fdab89 Mon Sep 17 00:00:00 2001
From: Antoine Damhet <antoine.damhet@lse.epita.fr>
Date: Sat, 14 Oct 2023 11:54:13 +0200
Subject: [PATCH] remove shebang from non executable `__main__.py` file

Signed-off-by: Antoine Damhet <antoine.damhet@lse.epita.fr>
---
 pass_audit/__main__.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pass_audit/__main__.py b/pass_audit/__main__.py
index 2fcb871c690d..1067ccc93aee 100644
--- a/pass_audit/__main__.py
+++ b/pass_audit/__main__.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*
 # pass audit - Password Store Extension (https://www.passwordstore.org/)
 # Copyright (C) 2018-2022 Alexandre PUJOL <alexandre@pujol.io>.
-- 
2.41.0

