--- a/pyproject.orig.toml	2024-12-20 03:16:32.000000000 +0100
+++ b/pyproject.toml	2025-07-13 20:43:43.104246647 +0200
@@ -48,7 +48,7 @@
     packages = ["periodictable"]
 
 [tool.pytest.ini_options]
-    addopts = ["--doctest-modules", "--doctest-glob=*.rst", "--cov=periodictable"]
+    addopts = ["--doctest-modules", "--doctest-glob=*.rst"]
     doctest_optionflags = "ELLIPSIS"
     pythonpath = ["doc/sphinx"]
     testpaths = ["periodictable", "test", "doc/sphinx/guide"]
