--- cap-tempfile-3.4.5/Cargo.toml	1970-01-01T00:00:01+00:00
+++ cap-tempfile-3.4.5/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -60,8 +60,6 @@
 [target.'cfg(any(target_os = "android", target_os = "linux"))'.dependencies.rustix-linux-procfs]
 version = "0.1.1"
 
-[target."cfg(not(windows))".dependencies.rustix]
-version = "1.0.0"
 
 [target.'cfg(target_os = "emscripten")'.dependencies.rand]
 version = "0.8.1"
