Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.orig.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.3)
-project(CombBLAS VERSION 1.16.0 LANGUAGES C CXX)
+project(CombBLAS VERSION 2.0.0 LANGUAGES C CXX)
 
 # require c++14
 set(CMAKE_CXX_STANDARD 14)
@@ -45,9 +45,11 @@
 
 add_subdirectory(usort)
 target_link_libraries(CombBLAS PUBLIC Usortlib)
+set_property(TARGET Usortlib PROPERTY VERSION "${CMAKE_PROJECT_VERSION}")
 
 add_subdirectory(graph500-1.2/generator)
 target_link_libraries(CombBLAS PUBLIC GraphGenlib)
+set_property(TARGET GraphGenlib PROPERTY VERSION 1.2)
 
 # Set up exported configuration
 # This allows CombBLAS to be installed in two ways:
