From 81c3efde71359f6fade75f53864fc2174f5f327a Mon Sep 17 00:00:00 2001
From: Ariel Rokem <arokem@gmail.com>
Date: Wed, 20 Dec 2023 07:24:58 -0800
Subject: [PATCH] Move setuptools_scm to be a setup requirement, instead of
 installation.

Closes #73.
---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 3622ae7..42cd474 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -25,8 +25,8 @@ python_requires = >=3.8
 setup_requires =
     packaging >= 19.0
     cython >= 0.29
-install_requires =
     setuptools_scm
+install_requires =
     deepdiff
     nibabel >= 5
     numpy >= 1.22
