--- a/CMakeModules/FindQCP.orig.cmake	2023-11-21 00:26:09.000000000 +0100
+++ b/CMakeModules/FindQCP.cmake	2023-11-30 18:03:57.036873503 +0100
@@ -16,7 +16,7 @@
 find_path(QCP_INCLUDE_DIR qcustomplot.h
     PATHS $ENV{QCP_DIR}/include
           $ENV{QCP_DIR}
-          ${${_PROJECT_DEPENDENCY_DIR}}/include
+          ${INCLUDE_INSTALL_DIR}
           ~/Library/Frameworks
           /Library/Frameworks
           /sw/include        # Fink
@@ -31,8 +31,8 @@
 endif ()
 
 find_library(QCP_LIBRARY
-    NAMES qcustomplot qcustomplot.lib
-    PATHS ${${_PROJECT_DEPENDENCY_DIR}}/lib
+    NAMES qcustomplot-qt6 qcustomplot.lib
+    PATHS ${LIB_INSTALL_DIR}
           ${${_PROJECT_DEPENDENCY_DIR}}/lib64
           $ENV{QCP_DIR}/lib
           $ENV{QCP_DIR}
