--- a/copasi/bindings/python/CMakeLists.txt.orig	2020-08-07 00:20:53.000000000 +0200
+++ b/copasi/bindings/python/CMakeLists.txt	2020-11-12 14:39:19.557471979 +0100
@@ -153,7 +153,7 @@
 #
 set(PYTHON_PACKAGE_INSTALL_DIR)
 if (UNIX OR CYGWIN) 
-  execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; sys.stdout.write('{}.{}'.format(*sys.version_info[:2]))"
+  execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c "import sys;import platform; sys.stdout.write('{}.{}'.format(*platform.python_version_tuple()[:2]))"
     OUTPUT_VARIABLE PYTHON_VERSION)
   set(PYTHON_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION}/site-packages)
 else()
