--- CMakeModules/FindLIBCOMBINE.orig.cmake	2018-07-30 20:03:33.000000000 +0200
+++ CMakeModules/FindLIBCOMBINE.cmake	2018-07-31 22:08:43.379406760 +0200
@@ -26,7 +26,7 @@
 
 set(LIBCOMBINE_LIBRARY_NAME)
 if (UNIX)
-  set(LIBCOMBINE_LIBRARY_NAME Combine-static)
+  set(LIBCOMBINE_LIBRARY_NAME Combine)
 else()
   set(LIBCOMBINE_LIBRARY_NAME libCombine-static)
 endif()
@@ -116,7 +116,7 @@
   # Fallback if no CONFIG is found
 
   find_path(LIBCOMBINE_INCLUDE_DIR combine/combinearchive.h
-    PATHS $ENV{LIBCOMBINE_DIR}/include
+    PATHS ${INCLUDE_INSTALL_DIR}
           $ENV{LIBCOMBINE_DIR}
           ${${_PROJECT_DEPENDENCY_DIR}}/include
           ${${_PROJECT_DEPENDENCY_DIR}}
@@ -122,7 +121,7 @@
     NAMES ${LIBCOMBINE_LIBRARY_NAME}
     PATHS $ENV{LIBCOMBINE_DIR}/lib
           $ENV{LIBCOMBINE_DIR}
-          ${${_PROJECT_DEPENDENCY_DIR}}/${CMAKE_INSTALL_LIBDIR}
+          ${CMAKE_INSTALL_LIBDIR}
           ${${_PROJECT_DEPENDENCY_DIR}}/lib
           ${${_PROJECT_DEPENDENCY_DIR}}
           ${CONAN_LIB_DIRS_LIBCOMBINE}
