--- CMakeModules/FindQWTPLOT3D.orig.cmake	2015-06-17 07:55:18.000000000 +0200
+++ CMakeModules/FindQWTPLOT3D.cmake	2015-06-23 18:16:42.954635914 +0200
@@ -21,18 +21,17 @@
           /opt/csw/include # Blastwave
           /opt/include
           /usr/freeware/include
-    PATH_SUFFIXES qwtplot3d qwtplot3d-qt4 
+    @@qtplot3d_includedir@@
+    PATH_SUFFIXES qwtplot3d-qt6 
     NO_DEFAULT_PATH)
 
 if (NOT QWTPLOT3D_INCLUDE_DIR)
     find_path(QWTPLOT3D_INCLUDE_DIR qwt3d_plot.h
-        PATH_SUFFIXES qwtplot3d qwtplot3d-qt4)
+        PATH_SUFFIXES qwtplot3d-qt6)
 endif (NOT QWTPLOT3D_INCLUDE_DIR)
 
 find_library(QWTPLOT3D_LIBRARY
-    NAMES qwtplot3d-static
-          qwtplot3d-qt4
-          qwtplot3d-qt5
+    NAMES qwtplot3d-qt6
           qwtplot3d
     PATHS $ENV{QWTPLOT3D_DIR}/lib
           $ENV{QWTPLOT3D_DIR}
@@ -45,6 +44,7 @@
           /opt/csw/lib   # Blastwave
           /opt/lib
           /usr/freeware/lib64
+          @@qtplot3d_libdir@@
     NO_DEFAULT_PATH)
 
 if (NOT QWTPLOT3D_LIBRARY)
