--- sysinfo-0.29.11/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sysinfo-0.29.11/Cargo.toml	2024-03-07T03:15:17.273743+00:00
@@ -68,52 +68,9 @@
 [target."cfg(all(target_os = \"linux\", not(target_os = \"android\")))".dev-dependencies.tempfile]
 version = "3.2"
 
-[target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dependencies.core-foundation-sys]
-version = "0.8"
-
 [target."cfg(any(windows, target_os = \"linux\", target_os = \"android\"))".dependencies.once_cell]
 version = "1.0"
 
 [target."cfg(not(any(target_os = \"unknown\", target_arch = \"wasm32\")))".dependencies.libc]
 version = "^0.2.144"
 
-[target."cfg(windows)".dependencies.ntapi]
-version = "0.4"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.9"
-features = [
-    "errhandlingapi",
-    "fileapi",
-    "handleapi",
-    "heapapi",
-    "ifdef",
-    "ioapiset",
-    "minwindef",
-    "pdh",
-    "psapi",
-    "synchapi",
-    "sysinfoapi",
-    "winbase",
-    "winerror",
-    "winioctl",
-    "winnt",
-    "oleauto",
-    "wbemcli",
-    "rpcdce",
-    "combaseapi",
-    "objidl",
-    "powerbase",
-    "netioapi",
-    "lmcons",
-    "lmaccess",
-    "lmapibuf",
-    "memoryapi",
-    "ntlsa",
-    "securitybaseapi",
-    "shellapi",
-    "std",
-    "iphlpapi",
-    "winsock2",
-    "sddl",
-]
