diff -up ophcrack-3.8.0/LICENSE.friend ophcrack-3.8.0/LICENSE
--- ophcrack-3.8.0/LICENSE.friend	2019-12-14 22:42:54.625232949 +0100
+++ ophcrack-3.8.0/LICENSE	2019-12-14 22:43:24.336351424 +0100
@@ -19,9 +19,9 @@
 
     You should have received a copy of the GNU General Public License
     along with Ophcrack; if not, write to the Free Software
-    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
     This program is released under the GPL with the additional exemption 
     that compiling, linking, and/or using OpenSSL is allowed.
 
-*/
\ No newline at end of file
+*/
