diff -up artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-de-sources-1.3/makepcf.sh.BAD artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-de-sources-1.3/makepcf.sh
--- artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-de-sources-1.3/makepcf.sh.BAD	2010-01-21 15:58:19.237307604 -0500
+++ artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-de-sources-1.3/makepcf.sh	2010-01-21 15:58:29.738317722 -0500
@@ -6,7 +6,7 @@
 LIST=`ls -la *.bdf | awk '{print $9}'`
 for font in $LIST
 do
-	pcf=`echo $font | cut -d'.' -f1`
+	pcf=`basename $font .bdf`
 	pcf="$pcf.pcf"
 	bdftopcf -o $pcf $font
 done;
diff -up artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-en-sources-1.3/makepcf.sh.BAD artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-en-sources-1.3/makepcf.sh
--- artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-en-sources-1.3/makepcf.sh.BAD	2010-01-21 15:58:52.018332523 -0500
+++ artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-en-sources-1.3/makepcf.sh	2010-01-21 15:58:55.857314725 -0500
@@ -6,7 +6,7 @@
 LIST=`ls -la *.bdf | awk '{print $9}'`
 for font in $LIST
 do
-	pcf=`echo $font | cut -d'.' -f1`
+	pcf=`basename $font .bdf`
 	pcf="$pcf.pcf"
 	bdftopcf -o $pcf $font
 done;
diff -up artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-se-sources-1.3/makepcf.sh.BAD artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-se-sources-1.3/makepcf.sh
--- artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-se-sources-1.3/makepcf.sh.BAD	2010-01-21 15:59:07.485332588 -0500
+++ artwiz-aleczapka-fonts-1.3/artwiz-aleczapka-se-sources-1.3/makepcf.sh	2010-01-21 15:59:15.277308049 -0500
@@ -6,7 +6,7 @@
 LIST=`ls -la *.bdf | awk '{print $9}'`
 for font in $LIST
 do
-	pcf=`echo $font | cut -d'.' -f1`
+	pcf=`basename $font .bdf`
 	pcf="$pcf.pcf"
 	bdftopcf -o $pcf $font
 done;
