diff -rupN --no-dereference ZRTPCPP-4.6.6/zrtp/libzrtpcpp/ZrtpCWrapper.h ZRTPCPP-4.6.6-new/zrtp/libzrtpcpp/ZrtpCWrapper.h
--- ZRTPCPP-4.6.6/zrtp/libzrtpcpp/ZrtpCWrapper.h	2017-06-02 09:05:34.000000000 +0200
+++ ZRTPCPP-4.6.6-new/zrtp/libzrtpcpp/ZrtpCWrapper.h	2025-01-24 00:02:07.370693629 +0100
@@ -983,7 +983,7 @@ extern "C"
      */
     const char* zrtp_getSasType(ZrtpContext* zrtpContext);
 #ifdef  __GNUC__ 
-#warning zrtp_getSasType(...) API changed - caller shall free() returned data
+/* warning zrtp_getSasType(...) API changed - caller shall free() returned data */
 #endif
 
     /**
