diff --git a/third_party/webrtc/src/webrtc/typedefs.h b/third_party/webrtc/src/webrtc/typedefs.h
index a59412c..464c9ae 100644
--- a/third_party/webrtc/src/webrtc/typedefs.h
+++ b/third_party/webrtc/src/webrtc/typedefs.h
@@ -61,6 +61,13 @@
 #elif defined(__s390__)
 # define WEBRTC_ARCH_32_BITS
 # define WEBRTC_ARCH_BIG_ENDIAN
+#elif defined(__riscv) || defined(__riscv__)
+#define WEBRTC_ARCH_LITTLE_ENDIAN
+#if __riscv_xlen == 64
+#define WEBRTC_ARCH_64_BITS
+#else
+#define WEBRTC_ARCH_32_BITS
+#endif
 #else
 #error Please add support for your architecture in typedefs.h
 #endif
