diff -uNr a/src/pwsafe/PW_BlowFish.cpp b/src/pwsafe/PW_BlowFish.cpp
--- a/src/pwsafe/PW_BlowFish.cpp	2013-12-21 12:53:16.877044394 +0100
+++ b/src/pwsafe/PW_BlowFish.cpp	2013-12-21 12:54:05.581963998 +0100
@@ -419,9 +419,8 @@
 {
   //  trashMemory((unsigned char*)tempbf_P, 18*4);
   // trashMemory((unsigned char*)tempbf_S, 256*4);
-  trashMemory((unsigned char*)bf_P, 18*sizeof(long));
-  for(int i = 0; i < 4; i++)
-    trashMemory((unsigned char*)bf_S[i], 256);
+  trashMemory((unsigned char*)bf_P, sizeof(bf_P));
+  trashMemory((unsigned char*)bf_S, sizeof(bf_S));
 }
 
 void
