diff -uNr MyPasswordSafe-20061216.orig/src/encryptedstring.cpp MyPasswordSafe-20061216/src/encryptedstring.cpp
--- MyPasswordSafe-20061216.orig/src/encryptedstring.cpp	2010-11-13 18:16:06.000000000 +0100
+++ MyPasswordSafe-20061216/src/encryptedstring.cpp	2010-11-13 18:16:33.000000000 +0100
@@ -98,7 +98,7 @@
     unsigned char cbcbuffer[8];
     memcpy(cbcbuffer, m_cbc, 8);
 
-    unsigned char *buffer = new unsigned char[BlockLength]; // so we lie a little...
+    unsigned char *buffer = new unsigned char[BlockLength+1]; // so we lie a little...
 
     unsigned char tempcbc[8];
     for (int x=0;x<BlockLength;x+=8)
