diff -uNr MyPasswordSafe-20061216-orig/src/crypto/bfproxy.cpp MyPasswordSafe-20061216/src/crypto/bfproxy.cpp
--- MyPasswordSafe-20061216-orig/src/crypto/bfproxy.cpp	2004-12-06 14:07:03.000000000 +0100
+++ MyPasswordSafe-20061216/src/crypto/bfproxy.cpp	2008-01-04 14:48:06.000000000 +0100
@@ -17,6 +17,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 #include "bfproxy.hpp"
+#include <cstring>
 
 BFProxy::BFProxy(const unsigned char *key, int len)
 {
diff -uNr MyPasswordSafe-20061216-orig/src/encryptedstring.cpp MyPasswordSafe-20061216/src/encryptedstring.cpp
--- MyPasswordSafe-20061216-orig/src/encryptedstring.cpp	2004-12-06 13:32:05.000000000 +0100
+++ MyPasswordSafe-20061216/src/encryptedstring.cpp	2008-01-04 14:20:16.000000000 +0100
@@ -17,6 +17,7 @@
  */
 #include <time.h>
 #include <iostream>
+#include <cstdlib>
 #include "pwsafe/Util.h"
 #include "crypto/bfproxy.hpp"
 #include "crypto/sha1.h"
diff -uNr MyPasswordSafe-20061216-orig/src/pwsafe/Util.h MyPasswordSafe-20061216/src/pwsafe/Util.h
--- MyPasswordSafe-20061216-orig/src/pwsafe/Util.h	2004-11-01 22:54:35.000000000 +0100
+++ MyPasswordSafe-20061216/src/pwsafe/Util.h	2008-01-04 14:37:08.000000000 +0100
@@ -9,6 +9,9 @@
 #include "securedstring.hpp"
 #include <assert.h>
 
+#include <cstring>
+#include <cstdlib>
+#include <climits>
 #include <string>
 using std::string;
 #include "PW_sha1.h"
diff -uNr MyPasswordSafe-20061216-orig/src/safe.hpp MyPasswordSafe-20061216/src/safe.hpp
--- MyPasswordSafe-20061216-orig/src/safe.hpp	2005-11-23 14:21:29.000000000 +0100
+++ MyPasswordSafe-20061216/src/safe.hpp	2008-01-04 14:21:36.000000000 +0100
@@ -129,7 +129,7 @@
   SafeEntry(SafeGroup *parent);
   SafeEntry(SafeGroup *parent,
 	    const QString &n, const QString &u,
-	    const EncryptedString &p, const QString &n);
+	    const EncryptedString &p, const QString &note);
   SafeEntry(const SafeEntry &item);
   ~SafeEntry();
 
diff -uNr MyPasswordSafe-20061216-orig/src/safeserializer.cpp MyPasswordSafe-20061216/src/safeserializer.cpp
--- MyPasswordSafe-20061216-orig/src/safeserializer.cpp	2004-12-06 13:34:28.000000000 +0100
+++ MyPasswordSafe-20061216/src/safeserializer.cpp	2008-01-04 14:24:57.000000000 +0100
@@ -22,6 +22,7 @@
 #include "plaintextlizer.hpp"
 #include "serializers.hpp"
 #include "xmlserializer.hpp"
+#include <algorithm>
 
 using namespace std;
 
diff -uNr MyPasswordSafe-20061216-orig/src/xmlserializer.hpp MyPasswordSafe-20061216/src/xmlserializer.hpp
--- MyPasswordSafe-20061216-orig/src/xmlserializer.hpp	2004-11-01 22:34:58.000000000 +0100
+++ MyPasswordSafe-20061216/src/xmlserializer.hpp	2008-01-04 14:22:59.000000000 +0100
@@ -22,7 +22,7 @@
 #include "safeserializer.hpp"
 
 class Safe;
-enum Safe::Error;
+// enum Safe::Error;
 class EncryptedString;
 class SecuredString;
 
