diff -uNr MyPasswordSafe-20061216-orig/Makefile MyPasswordSafe-20061216/Makefile
--- MyPasswordSafe-20061216-orig/Makefile	2004-12-06 14:14:53.000000000 +0100
+++ MyPasswordSafe-20061216/Makefile	2007-07-06 20:55:45.000000000 +0200
@@ -27,7 +27,7 @@
 MyPasswordSafe.mak: MyPasswordSafe.pro
 	$(QMAKE) -o $@ $<
 
-MyPasswordSafe: config uuid-1.0.0/.libs/libuuid.a MyPasswordSafe.mak
+MyPasswordSafe: config MyPasswordSafe.mak
 	${MAKE} -f MyPasswordSafe.mak
 	lrelease locale/*.ts
 
diff -uNr MyPasswordSafe-20061216-orig/MyPasswordSafe.pro MyPasswordSafe-20061216/MyPasswordSafe.pro
--- MyPasswordSafe-20061216-orig/MyPasswordSafe.pro	2006-04-06 13:04:38.000000000 +0200
+++ MyPasswordSafe-20061216/MyPasswordSafe.pro	2007-07-06 21:04:34.000000000 +0200
@@ -3,8 +3,8 @@
 
 CONFIG	+= qt warn_on
 
-LIBS	+= uuid-1.0.0/.libs/libuuid.a
-INCLUDEPATH	+= uuid-1.0.0
+LIBS	+= -lossp-uuid
+#INCLUDEPATH	+= uuid-1.0.0
 
 # idle
 CONFIG += idle
