--- src/grobot.c~	2008-08-03 09:09:54.000000000 -0500
+++ src/grobot.c	2024-02-05 16:17:51.606480422 -0600
@@ -206,7 +206,7 @@
 		obj = g_value_get_object(value);
 		if (obj != NULL)
 		{
-			robot->ui = g_object_ref(obj);
+			robot->ui = (UIArena *)g_object_ref(obj);
 		}
 
 		else
@@ -223,7 +223,7 @@
 		obj = g_value_get_object(value);
 		if (obj != NULL)
 		{
-			robot->map = g_object_ref(obj);
+			robot->map = (Map *)g_object_ref(obj);
 		}
 
 		else
