--- ogre-1.9.0/OgreMain/src/OgrePlatformInformation.cpp~	2020-05-28 22:26:51.798696996 +0100
+++ ogre-1.9.0/OgreMain/src/OgrePlatformInformation.cpp	2020-05-28 22:27:24.901691518 +0100
@@ -42,7 +42,7 @@
 
     #if OGRE_PLATFORM == OGRE_PLATFORM_ANDROID
         #include <cpu-features.h>
-    #elif OGRE_CPU == OGRE_CPU_ARM 
+    #elif OGRE_CPU == OGRE_CPU_ARM && OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS
         #include <sys/sysctl.h>
         #if __MACH__
             #include <mach/machine.h>
