diff -uNr ogre-1.9.0/Samples/Browser/CMakeLists.txt ogre-1.9.0.cmake_build-fix/Samples/Browser/CMakeLists.txt
--- ogre-1.9.0/Samples/Browser/CMakeLists.txt	2013-12-01 21:28:12.000000000 +0400
+++ ogre-1.9.0.cmake_build-fix/Samples/Browser/CMakeLists.txt	2014-06-07 22:59:19.423900980 +0400
@@ -339,8 +339,8 @@
 
     foreach(FWK ${FRAMEWORKS})
 	add_custom_command(TARGET SampleBrowser POST_BUILD
-        COMMAND ARGS if [ -d ${OGRE_BINARY_DIR}/lib/${OGRE_OSX_BUILD_CONFIGURATION}/${FWK}.framework ]\; then ln -s -f ${OGRE_BINARY_DIR}/lib/${OGRE_OSX_BUILD_CONFIGURATION}/${FWK}.framework ${OGRE_SAMPLE_CONTENTS_PATH}/Frameworks/${FWK}.framework\; fi
-		)
+        COMMAND ARGS if test -d ${OGRE_BINARY_DIR}/lib/${OGRE_OSX_BUILD_CONFIGURATION}/${FWK}.framework\; then ln -s -f ${OGRE_BINARY_DIR}/lib/${OGRE_OSX_BUILD_CONFIGURATION}/${FWK}.framework ${OGRE_SAMPLE_CONTENTS_PATH}/Frameworks/${FWK}.framework\; fi
+    )
     endforeach(FWK)
 
 	# Add links for the sample plugins
