diff -Nurp kBuild.orig/src/kmk/glob/fnmatch.c kBuild/src/kmk/glob/fnmatch.c
--- kBuild.orig/src/kmk/glob/fnmatch.c	2025-04-25 21:54:33.000000000 +0100
+++ kBuild/src/kmk/glob/fnmatch.c	2025-04-25 23:20:55.477680324 +0100
@@ -120,10 +120,6 @@ USA.  */
 /* Avoid depending on library functions or files
    whose names are inconsistent.  */
 
-# if !defined _LIBC && !defined getenv && !defined _MSC_VER
-extern char *getenv ();
-# endif
-
 # ifndef errno
 extern int errno;
 # endif
