diff -up glob2-0.9.4.4/SConstruct\~ glob2-0.9.4.4/SConstruct
--- glob2-0.9.4.4/SConstruct~	2009-08-30 21:23:30.000000000 +0200
+++ glob2-0.9.4.4/SConstruct	2013-07-28 19:05:52.979943549 +0200
@@ -120,11 +120,22 @@ def configure(env):
         missing.append("libboost_thread")
     env.Append(LIBS=[boost_thread])
 
+    # Boost.Thread needs Boost.System DSO's.
+    boost_system = ''
+    if conf.CheckLib("boost_system"):
+	boost_system="boost_system"
+    elif conf.CheckLib("boost_system-mt"):
+	boost_system="boost_system-mt"
+    else:
+        print "Could not find libboost_system or libboost_system-mt"
+        missing.append("libboost_system")
+    env.Append(LIBS=[boost_system])
+
     boost_date_time = ''
     if conf.CheckLib("boost_date_time") and conf.CheckCXXHeader("boost/date_time/posix_time/posix_time.hpp"):
-        boost_thread="boost_thread"
+        boost_date_time="boost_date_time"
     elif conf.CheckLib("boost_date_time-mt") and conf.CheckCXXHeader("boost/date_time/posix_time/posix_time.hpp"):
-        boost_thread="boost_thread-mt"
+        boost_date_time="boost_date_time-mt"
     else:
         print "Could not find libboost_date_time or libboost_date_time-mt or boost/thread/thread.hpp"
         missing.append("libboost_date_time")

Diff finished.  Sun Jul 28 19:06:06 2013
