diff -up ./com/jgoodies/looks/common/MenuSelectionProcessor.java.orig ./com/jgoodies/looks/common/MenuSelectionProcessor.java
--- ./com/jgoodies/looks/common/MenuSelectionProcessor.java.orig	2014-01-10 12:09:26.000000000 +0100
+++ ./com/jgoodies/looks/common/MenuSelectionProcessor.java	2021-07-27 17:30:14.286523615 +0200
@@ -43,8 +43,6 @@ import javax.swing.MenuSelectionManager;
 import javax.swing.SwingUtilities;
 import javax.swing.plaf.basic.ComboPopup;
 
-import com.sun.java.swing.plaf.windows.WindowsRootPaneUI;
-
 /**
  * Handles the Alt key to select the first menu in the menu bar - if any.
  * Useful to let non-Windows L&amp;fs like Plastic feel more like Windows.
diff -up ./com/jgoodies/looks/Options.java.orig ./com/jgoodies/looks/Options.java
--- ./com/jgoodies/looks/Options.java.orig	2014-12-08 10:47:22.000000000 +0100
+++ ./com/jgoodies/looks/Options.java	2021-07-27 17:30:32.286793624 +0200
@@ -39,7 +39,6 @@ import javax.swing.text.JTextComponent;
 import com.jgoodies.common.base.SystemUtils;
 import com.jgoodies.looks.common.ShadowPopup;
 import com.jgoodies.looks.plastic.PlasticLookAndFeel;
-import com.jgoodies.looks.windows.WindowsLookAndFeel;
 
 /**
  * Provides access to optional features of the JGoodies L&amp;Fs
