diff -up kdebase-3.5.10/admin/cvs.sh.autoconf2.7x kdebase-3.5.10/admin/cvs.sh
--- kdebase-3.5.10/admin/cvs.sh.autoconf2.7x	2021-08-31 15:50:51.033160910 +0200
+++ kdebase-3.5.10/admin/cvs.sh	2021-08-31 15:54:10.664792794 +0200
@@ -32,7 +32,7 @@ check_autotool_versions()
 required_autoconf_version="2.53 or newer"
 AUTOCONF_VERSION=`$AUTOCONF --version | head -n 1`
 case $AUTOCONF_VERSION in
-  Autoconf*2.5* | autoconf*2.5* | autoconf*2.6* ) : ;;
+  Autoconf* | autoconf* ) : ;;
   "" )
     echo "*** AUTOCONF NOT FOUND!."
     echo "*** KDE requires autoconf $required_autoconf_version"
@@ -47,7 +47,7 @@ esac
 
 AUTOHEADER_VERSION=`$AUTOHEADER --version | head -n 1`
 case $AUTOHEADER_VERSION in
-  Autoconf*2.5* | autoheader*2.5* | autoheader*2.6* ) : ;;
+  Autoconf* | autoheader* ) : ;;
   "" )
     echo "*** AUTOHEADER NOT FOUND!."
     echo "*** KDE requires autoheader $required_autoconf_version"
