diff -up xtb-6.7.1/src/optimizer.f90.formatting xtb-6.7.1/src/optimizer.f90
--- xtb-6.7.1/src/optimizer.f90.formatting	2024-07-23 20:08:02.000000000 +0000
+++ xtb-6.7.1/src/optimizer.f90	2025-09-21 13:40:52.804499013 +0000
@@ -849,7 +849,7 @@ subroutine relax(env,iter,mol,anc,restar
       write(env%unit,'(5x,"change   ",e18.7,1x,"Eh")')                      echng
       write(env%unit,'(3x,"gradient norm :",f14.7,1x,"Eh/α")',advance='no') gnorm
       write(env%unit,'(3x,"predicted",e18.7)',advance='no')                 depred
-      write(env%unit,'(1x,"("f7.2"%)")')         (depred-echng)/echng*100
+      write(env%unit,'(1x,"(",f7.2,"%)")')         (depred-echng)/echng*100
    endif
    
    ! check 0 energy case !
