diff -up gtkimageview-1.6.4/aclocal.m4.no-werror gtkimageview-1.6.4/aclocal.m4
diff -up gtkimageview-1.6.4/configure.in.no-werror gtkimageview-1.6.4/configure.in
--- gtkimageview-1.6.4/configure.in.no-werror	2016-10-14 13:15:30.504411843 +0200
+++ gtkimageview-1.6.4/configure.in	2016-10-14 13:15:59.126089943 +0200
@@ -39,7 +39,7 @@ GTK_DOC_CHECK([1.8])
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+CFLAGS="${CFLAGS} -Wall -std=c99 -Wmissing-prototypes"
 
 
 ######################################################################
diff -up gtkimageview-1.6.4/configure.no-werror gtkimageview-1.6.4/configure
--- gtkimageview-1.6.4/configure.no-werror	2016-10-14 13:15:30.506411820 +0200
+++ gtkimageview-1.6.4/configure	2016-10-14 13:16:24.937799646 +0200
@@ -20740,7 +20740,7 @@ fi
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+CFLAGS="${CFLAGS} -Wall -std=c99 -Wmissing-prototypes"
 
 
 ######################################################################
diff -up gtkimageview-1.6.4/wscript.no-werror gtkimageview-1.6.4/wscript
--- gtkimageview-1.6.4/wscript.no-werror	2016-10-14 13:15:30.507411809 +0200
+++ gtkimageview-1.6.4/wscript	2016-10-14 13:16:45.245571250 +0200
@@ -142,7 +142,7 @@ def configure(conf):
     # here.
     if not conf.env['CCFLAGS']:
         conf.env['CCFLAGS'] = ['-g', '-O2']
-    flags = ['-std=c99', '-Wall', '-Werror', '-Wmissing-prototypes']
+    flags = ['-std=c99', '-Wall', '-Wmissing-prototypes']
     conf.env.append_value('CCFLAGS', flags)
 
 def build(bld):
