diff --git a/src/spawn.c b/src/spawn.c
index 99e81f1..e69e257 100644
--- a/src/spawn.c
+++ b/src/spawn.c
@@ -938,8 +938,8 @@ int run_def_action(int action, char *sname, char *dname, int delay)
 
 #else /* Linux */
 
-#define HANDBOOK_BROWSER "seamonkey"
-#define HANDBOOK_LOCATION "/usr/doc/mtpaint/index.html"
+#define HANDBOOK_BROWSER "xdg-open"
+#define HANDBOOK_LOCATION "/usr/share/doc/mtpaint-handbook/index.html"
 #define HANDBOOK_LOCATION2 "/usr/share/doc/mtpaint/index.html"
 
 #endif
