diff -urp '--exclude=*~' erubi-1.12.0.orig/test/test.rb erubi-1.12.0/test/test.rb
--- erubi-1.12.0.orig/test/test.rb	2025-12-28 14:50:02.357261861 +0900
+++ erubi-1.12.0/test/test.rb	2025-12-28 14:44:22.536991591 +0900
@@ -28,7 +28,7 @@ require 'erubi/capture_end'
 
 ENV['MT_NO_PLUGINS'] = '1' # Work around stupid autoloading of plugins
 gem 'minitest'
-require 'minitest/global_expectations/autorun'
+require 'minitest/autorun'
 
 describe Erubi::Engine do
   before do
@@ -38,10 +38,10 @@ describe Erubi::Engine do
   def check_output(input, src, result, &block)
     t = (@options[:engine] || Erubi::Engine).new(input, @options)
     tsrc = t.src
-    eval(tsrc, block.binding).must_equal result
+    _(eval(tsrc, block.binding)).must_equal result
     strip_freeze = defined?(@strip_freeze) ? @strip_freeze : RUBY_VERSION >= '2.1'
     tsrc = tsrc.gsub(/\.freeze/, '') if strip_freeze
-    tsrc.must_equal src
+    _(tsrc).must_equal src
   end
 
   def setup_foo
@@ -290,7 +290,7 @@ END2
 </table>
 1
 END3
-    @a.must_equal 'bar'
+    _(@a).must_equal 'bar'
   end
 
   it "should handle chain_appends option" do
@@ -475,27 +475,27 @@ END3
   end
 
   it "should handle trailing rspace with - modifier in <%|= and <%|" do
-    eval(::Erubi::CaptureEndEngine.new("<%|= '&' -%>\n<%| -%>\n").src).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new("<%|= '&' -%>\n<%| -%>\n").src)).must_equal '&'
   end
 
   it "should handle lspace in <%|=" do
-    eval(::Erubi::CaptureEndEngine.new("<%|= %><%| %><%|= %><%| %>").src).must_equal ''
+    _(eval(::Erubi::CaptureEndEngine.new("<%|= %><%| %><%|= %><%| %>").src)).must_equal ''
   end
 
   it "should have <%|= with CaptureEndEngine not escape by default" do
-    eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>').src).must_equal '&'
-    eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape=>false).src).must_equal '&'
-    eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape_capture=>false).src).must_equal '&'
-    eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape=>true).src).must_equal '&amp;'
-    eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape_capture=>true).src).must_equal '&amp;'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>').src)).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape=>false).src)).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape_capture=>false).src)).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape=>true).src)).must_equal '&amp;'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|= "&" %><%| %>', :escape_capture=>true).src)).must_equal '&amp;'
   end
 
   it "should have <%|== with CaptureEndEngine escape by default" do
-    eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>').src).must_equal '&amp;'
-    eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape=>true).src).must_equal '&'
-    eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape_capture=>true).src).must_equal '&'
-    eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape=>false).src).must_equal '&amp;'
-    eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape_capture=>false).src).must_equal '&amp;'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>').src)).must_equal '&amp;'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape=>true).src)).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape_capture=>true).src)).must_equal '&'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape=>false).src)).must_equal '&amp;'
+    _(eval(::Erubi::CaptureEndEngine.new('<%|== "&" %><%| %>', :escape_capture=>false).src)).must_equal '&amp;'
   end
 
   [['', false], ['=', true]].each do |ind, escape|
@@ -929,31 +929,31 @@ END3
   end
 
   it "should have working filename accessor" do
-    Erubi::Engine.new('', :filename=>'foo.rb').filename.must_equal 'foo.rb'
+    _(Erubi::Engine.new('', :filename=>'foo.rb').filename).must_equal 'foo.rb'
   end
 
   it "should have working bufvar accessor" do
-    Erubi::Engine.new('', :bufvar=>'foo').bufvar.must_equal 'foo'
-    Erubi::Engine.new('', :outvar=>'foo').bufvar.must_equal 'foo'
+    _(Erubi::Engine.new('', :bufvar=>'foo').bufvar).must_equal 'foo'
+    _(Erubi::Engine.new('', :outvar=>'foo').bufvar).must_equal 'foo'
   end
 
   it "should work with BasicObject methods" do
     c = Class.new(BasicObject)
     c.class_eval("def a; #{Erubi::Engine.new('2').src} end")
-    c.new.a.must_equal '2'
+    _(c.new.a).must_equal '2'
   end if defined?(BasicObject)
 
   it "should return frozen object" do
-    Erubi::Engine.new('').frozen?.must_equal true
+    _(Erubi::Engine.new('').frozen?).must_equal true
   end
 
   it "should have frozen src" do
-    Erubi::Engine.new('').src.frozen?.must_equal true
+    _(Erubi::Engine.new('').src.frozen?).must_equal true
   end
 
   it "should raise an error if a tag is not handled when a custom regexp is used" do
-    proc{Erubi::Engine.new('<%] %>', :regexp =>/<%(={1,2}|\]|-|\#|%)?(.*?)([-=])?%>([ \t]*\r?\n)?/m)}.must_raise ArgumentError
-    proc{Erubi::CaptureEndEngine.new('<%] %>', :regexp =>/<%(={1,2}|\]|-|\#|%)?(.*?)([-=])?%>([ \t]*\r?\n)?/m)}.must_raise ArgumentError
+    _(proc{Erubi::Engine.new('<%] %>', :regexp =>/<%(={1,2}|\]|-|\#|%)?(.*?)([-=])?%>([ \t]*\r?\n)?/m)}).must_raise ArgumentError
+    _(proc{Erubi::CaptureEndEngine.new('<%] %>', :regexp =>/<%(={1,2}|\]|-|\#|%)?(.*?)([-=])?%>([ \t]*\r?\n)?/m)}).must_raise ArgumentError
   end
 
   it "should respect the :yield_returns_buffer option for making templates return the (potentially modified) buffer" do
