--- a/setup.cfg
+++ b/setup.cfg
@@ -44,7 +44,7 @@ zip_safe = false
 tests_require =
     Pillow >= 9.0
     pytest >= 7.4
-    pytest-asyncio == 0.23.8
+    pytest-asyncio >= 0.23.6
     pytest-django >= 4.5.2
 install_requires =
     Django >= 4.2.16
--- a/tox.ini
+++ b/tox.ini
@@ -26,7 +26,7 @@ deps =
     coverage
     Pillow>=9.0
     pytest>=7.4
-    pytest-asyncio == 0.23.8
+    pytest-asyncio >= 0.23.6
     pytest-django>=4.5.2
     django42: Django==4.2.*
     django50: Django==5.0.*
