--- jpegxl-sys-0.12.1+libjxl-0.11.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ jpegxl-sys-0.12.1+libjxl-0.11.2/Cargo.toml	2026-10-03T06:32:57.739054+00:00
@@ -13,7 +13,7 @@
 edition = "2021"
 rust-version = "1.85.0"
 name = "jpegxl-sys"
-version = "0.12.1+libjxl-0.11.2"
+version = "0.12.1"
 build = "build.rs"
 links = "jxl"
 autolib = false
@@ -73,3 +73,4 @@
 
 [lints.rustdoc]
 broken_intra_doc_links = "deny"
+
