diff -U 3 -dHrN a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	1970-01-01 01:00:00.000000000 +0100
+++ b/pyproject.toml	2025-12-19 09:51:55.127666571 +0100
@@ -65,7 +65,7 @@
 falcon = {version = ">=3.0", optional = true}
 flask = {version = "*", optional = true}
 aiohttp = {version = ">=3.0", optional = true}
-starlette = {version = ">=0.26.1,<0.50.0", optional = true}
+starlette = {version = ">=0.26.1,<0.53.0", optional = true}
 isodate = "*"
 more-itertools = "*"
 openapi-schema-validator = "^0.6.0"
