--- fs-set-times-0.20.3/Cargo.toml	1970-01-01T00:00:01+00:00
+++ fs-set-times-0.20.3/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -40,16 +40,4 @@
 version = "2.0.0"
 default-features = false
 
-[target."cfg(not(windows))".dependencies.rustix]
-version = "1.0.0"
-features = [
-    "fs",
-    "time",
-]
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.52, <0.60"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-]
