--- test/serializationinfo-test.cpp.orig	2021-02-05 14:37:23.016038640 +0100
+++ test/serializationinfo-test.cpp	2021-02-05 14:38:27.232046272 +0100
@@ -554,36 +554,6 @@
             si.setValue(static_cast<long>(std::numeric_limits<short>::max()) + 1);
             CXXTOOLS_UNIT_ASSERT_THROW(siValue<short>(si), std::range_error);
             CXXTOOLS_UNIT_ASSERT_NOTHROW(siValue<long>(si));
-
-            if (std::numeric_limits<long double>::max() > static_cast<long double>(std::numeric_limits<double>::max()))
-            {
-                si.setValue(static_cast<long double>(std::numeric_limits<double>::max()) * 1.01);
-                CXXTOOLS_UNIT_ASSERT_THROW(siValue<double>(si), std::range_error);
-                CXXTOOLS_UNIT_ASSERT_NOTHROW(siValue<long double>(si));
-
-                si.setValue(static_cast<long double>(-std::numeric_limits<double>::max()) * 1.01);
-                CXXTOOLS_UNIT_ASSERT_THROW(siValue<double>(si), std::range_error);
-                CXXTOOLS_UNIT_ASSERT_NOTHROW(siValue<long double>(si));
-            }
-            else
-            {
-                log_info("range error for double skipped since long double is not larger than double");
-            }
-
-            if (std::numeric_limits<long double>::max() > static_cast<long double>(std::numeric_limits<float>::max()))
-            {
-                si.setValue(static_cast<long double>(std::numeric_limits<float>::max()) * 1.01);
-                CXXTOOLS_UNIT_ASSERT_THROW(siValue<float>(si), std::range_error);
-                CXXTOOLS_UNIT_ASSERT_NOTHROW(siValue<double>(si));
-
-                si.setValue(static_cast<long double>(-std::numeric_limits<float>::max()) * 1.01);
-                CXXTOOLS_UNIT_ASSERT_THROW(siValue<float>(si), std::range_error);
-                CXXTOOLS_UNIT_ASSERT_NOTHROW(siValue<double>(si));
-            }
-            else
-            {
-                log_info("range error for float skipped since long double is not larger than double");
-            }
         }
 
         void testMember()
