--- signal-hook-0.1.17/Cargo.toml	1970-01-01T00:00:00+00:00
+++ signal-hook-0.1.17/Cargo.toml	2025-02-13T19:44:52.366185+00:00
@@ -23,33 +23,12 @@
 repository = "https://github.com/vorner/signal-hook"
 [package.metadata.docs.rs]
 all-features = true
-[dependencies.futures]
-version = "^0.1"
-optional = true
 
 [dependencies.libc]
 version = "^0.2"
 
-[dependencies.mio]
-version = "^0.6"
-optional = true
-
-[dependencies.mio_0_7]
-version = "^0.7"
-features = ["os-util"]
-optional = true
-package = "mio"
-
 [dependencies.signal-hook-registry]
 version = "^1.2"
-
-[dependencies.tokio-reactor]
-version = "^0.1"
-optional = true
-[dev-dependencies.mio_0_7]
-version = "^0.7"
-features = ["os-poll", "os-util"]
-package = "mio"
 
 [dev-dependencies.tokio]
 version = "^0.1"
@@ -57,10 +36,6 @@
 [dev-dependencies.version-sync]
 version = "^0.8"
 
-[features]
-mio-0_7-support = ["mio_0_7"]
-mio-support = ["mio"]
-tokio-support = ["futures", "mio-support", "tokio-reactor"]
 [badges.maintenance]
 status = "actively-developed"
 
