diff --git a/src/pydicom/pixel_data_handlers/pillow_handler.py b/src/pydicom/pixel_data_handlers/pillow_handler.py
index 5c85c4925..f387acfa8 100644
--- a/src/pydicom/pixel_data_handlers/pillow_handler.py
+++ b/src/pydicom/pixel_data_handlers/pillow_handler.py
@@ -226,7 +226,7 @@ def get_pixeldata(ds: "Dataset") -> "numpy.ndarray":
             shift = bits_allocated - j2k_precision
             if not j2k_sign and j2k_sign != ds.PixelRepresentation:
                 # Convert unsigned J2K data to 2's complement
-                arr = numpy.right_shift(arr, shift)
+                numpy.right_shift(arr, shift, out=arr)
             else:
                 if ds.PixelRepresentation == 1:
                     # Pillow converts signed data to unsigned
@@ -234,14 +234,14 @@ def get_pixeldata(ds: "Dataset") -> "numpy.ndarray":
                     arr -= numpy.uint32(2 ** (bits_allocated - 1))
 
                 if shift:
-                    arr = numpy.right_shift(arr, shift)
+                    numpy.right_shift(arr, shift, out=arr)
         else:
             # Corrections based on dataset elements
             if ds.PixelRepresentation == 1:
                 arr -= numpy.uint32(2 ** (bits_allocated - 1))
 
             if shift:
-                arr = numpy.right_shift(arr, shift)
+                numpy.right_shift(arr, shift, out=arr)
 
     if should_change_PhotometricInterpretation_to_RGB(ds):
         ds.PhotometricInterpretation = "RGB"
diff --git a/src/pydicom/pixels/decoders/pillow.py b/src/pydicom/pixels/decoders/pillow.py
index 6ccc064dd..103e661b6 100644
--- a/src/pydicom/pixels/decoders/pillow.py
+++ b/src/pydicom/pixels/decoders/pillow.py
@@ -102,7 +102,7 @@ def _decode_frame(src: bytes, runner: DecodeRunner) -> bytes:
     buffer = bytearray(image.tobytes())  # so the array is writeable
     del image
     dtype = runner.pixel_dtype
-    arr = np.frombuffer(buffer, dtype=f"u{dtype.itemsize}")
+    arr = np.frombuffer(buffer, dtype=f"<u{dtype.itemsize}")
 
     is_signed = runner.pixel_representation
     if runner.get_option("apply_j2k_sign_correction", False):
diff --git a/tests/test_pillow_pixel_data.py b/tests/test_pillow_pixel_data.py
index a9ee1a36b..8e320848c 100644
--- a/tests/test_pillow_pixel_data.py
+++ b/tests/test_pillow_pixel_data.py
@@ -240,19 +240,19 @@ REFERENCE_DATA = [
     (JPGB_08_08_3_0_1F_RGB, (JPGB, 8, 3, 0, 1, (256, 256, 3), "uint8")),
     (JPGB_08_08_3_0_1F_RGB_APP14, (JPGB, 8, 3, 0, 1, (256, 256, 3), "uint8")),
     (J2KR_08_08_3_0_1F_YBR_ICT, (J2KR, 8, 3, 0, 1, (480, 640, 3), "uint8")),
-    (J2KR_16_10_1_0_1F_M1, (J2KR, 16, 1, 0, 1, (1760, 1760), "uint16")),
-    (J2KR_16_12_1_0_1F_M2, (J2KR, 16, 1, 0, 1, (1024, 1024), "uint16")),
-    (J2KR_16_15_1_0_1F_M1, (J2KR, 16, 1, 0, 1, (1955, 1841), "uint16")),
-    (J2KR_16_16_1_0_10F_M2, (J2KR, 16, 1, 0, 10, (10, 64, 64), "uint16")),
-    (J2KR_16_14_1_1_1F_M2, (J2KR, 16, 1, 1, 1, (512, 512), "int16")),
-    (J2KR_16_16_1_1_1F_M2, (J2KR, 16, 1, 1, 1, (64, 64), "int16")),
+    (J2KR_16_10_1_0_1F_M1, (J2KR, 16, 1, 0, 1, (1760, 1760), "<u2")),
+    (J2KR_16_12_1_0_1F_M2, (J2KR, 16, 1, 0, 1, (1024, 1024), "<u2")),
+    (J2KR_16_15_1_0_1F_M1, (J2KR, 16, 1, 0, 1, (1955, 1841), "<u2")),
+    (J2KR_16_16_1_0_10F_M2, (J2KR, 16, 1, 0, 10, (10, 64, 64), "<u2")),
+    (J2KR_16_14_1_1_1F_M2, (J2KR, 16, 1, 1, 1, (512, 512), "<i2")),
+    (J2KR_16_16_1_1_1F_M2, (J2KR, 16, 1, 1, 1, (64, 64), "<i2")),
     (J2KI_08_08_3_0_1F_RGB, (J2KI, 8, 3, 0, 1, (100, 100, 3), "uint8")),
     (J2KI_08_08_3_0_1F_YBR_ICT, (J2KI, 8, 3, 0, 1, (480, 640, 3), "uint8")),
-    (J2KI_16_10_1_0_1F_M1, (J2KI, 16, 1, 0, 1, (1760, 1760), "uint16")),
-    (J2KI_16_12_1_0_1F_M2, (J2KI, 16, 1, 0, 1, (1024, 1024), "uint16")),
-    (J2KI_16_15_1_0_1F_M1, (J2KI, 16, 1, 0, 1, (1955, 1841), "uint16")),
-    (J2KI_16_14_1_1_1F_M2, (J2KI, 16, 1, 1, 1, (512, 512), "int16")),
-    (J2KI_16_16_1_1_1F_M2, (J2KI, 16, 1, 1, 1, (1024, 256), "int16")),
+    (J2KI_16_10_1_0_1F_M1, (J2KI, 16, 1, 0, 1, (1760, 1760), "<u2")),
+    (J2KI_16_12_1_0_1F_M2, (J2KI, 16, 1, 0, 1, (1024, 1024), "<u2")),
+    (J2KI_16_15_1_0_1F_M1, (J2KI, 16, 1, 0, 1, (1955, 1841), "<u2")),
+    (J2KI_16_14_1_1_1F_M2, (J2KI, 16, 1, 1, 1, (512, 512), "<i2")),
+    (J2KI_16_16_1_1_1F_M2, (J2KI, 16, 1, 1, 1, (1024, 256), "<i2")),
 ]
 
 JPEG_MATCHING_DATASETS = [
@@ -594,7 +594,7 @@ class TestPillowHandler_JPEG2K:
         assert not params["is_signed"]
         arr = ds.pixel_array
 
-        assert "int16" == arr.dtype
+        assert "<i2" == arr.dtype
         assert (512, 512) == arr.shape
         assert arr.flags.writeable
