diff -Naur colored_json-2.1.0-original/src/lib.rs colored_json-2.1.0/src/lib.rs
--- colored_json-2.1.0-original/src/lib.rs	2019-11-04 10:00:21.000000000 +0000
+++ colored_json-2.1.0/src/lib.rs	2026-02-21 10:36:10.700074924 +0000
@@ -136,20 +136,18 @@
 //!```
 
 extern crate ansi_term;
-extern crate atty;
 #[cfg(unix)]
 extern crate libc;
 extern crate serde;
 extern crate serde_json;
 
-use std::io;
+use std::io::{self, IsTerminal};
 
 #[cfg(windows)]
 pub use ansi_term::enable_ansi_support;
 pub use ansi_term::Colour;
 pub use ansi_term::Colour as Color;
 pub use ansi_term::Style;
-use atty::Stream;
 use serde::Serialize;
 use serde_json::ser::Formatter;
 pub use serde_json::ser::{CompactFormatter, PrettyFormatter};
@@ -744,8 +742,8 @@
 impl ColorMode {
     fn is_tty(output: Output) -> bool {
         match output {
-            Output::StdOut => atty::is(Stream::Stdout),
-            Output::StdErr => atty::is(Stream::Stderr),
+            Output::StdOut => io::stdout().is_terminal(),
+            Output::StdErr => io::stderr().is_terminal(),
         }
     }
 
