--- dxpc-3.9.0/Makefile.in.mandir	2006-02-28 15:46:52.000000000 -0500
+++ dxpc-3.9.0/Makefile.in	2007-01-22 11:02:08.000000000 -0500
@@ -9,7 +9,7 @@
 prefix      = @prefix@
 exec_prefix = @exec_prefix@
 bindir      = @bindir@
-man1dir     = @mandir@/man1
+mandir     = @mandir@
 VPATH       = @srcdir@
 
 INSTALL         = @INSTALL@
@@ -76,8 +76,8 @@
 	$(INSTALL) $(TARGET) $(bindir)/$(TARGET)
 
 install.man:
-	$(srcdir)/mkinstalldirs $(man1dir)
-	$(INSTALL_DATA) $(TARGET).man $(man1dir)/$(TARGET).1
+	$(srcdir)/mkinstalldirs $(mandir)/man1
+	$(INSTALL_DATA) $(TARGET).man $(mandir)/man1/$(TARGET).1
 
 clean:
 	-rm -f *~ *.o $(TARGET) pure_* quant_* *.bak core gmon.out
