*** Makefile.in.orig	2007-02-02 14:04:40.000000000 -0500
--- Makefile.in	2007-02-04 21:31:57.000000000 -0500
***************
*** 73,88 ****
  install: install.bin install.man install.scripts
  
  install.bin:	$(TARGET)
! 	$(srcdir)/mkinstalldirs $(bindir)
! 	$(INSTALL) $(TARGET) $(bindir)/$(TARGET)
  
  install.man:
! 	$(srcdir)/mkinstalldirs $(mandir)/man1
! 	$(INSTALL_DATA) $(TARGET).man $(mandir)/man1/$(TARGET).1
  
  install.scripts:
! 	$(srcdir)/mkinstalldirs $(mandir)/man1
! 	$(INSTALL) $(TARGETSSH) $(bindir)/$(TARGETSSH)
  
  clean:
  	-rm -f *~ *.o $(TARGET) pure_* quant_* *.bak core gmon.out
--- 73,88 ----
  install: install.bin install.man install.scripts
  
  install.bin:	$(TARGET)
! 	$(srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
! 	$(INSTALL) $(TARGET) $(DESTDIR)$(bindir)/$(TARGET)
  
  install.man:
! 	$(srcdir)/mkinstalldirs $(DESTDIR)$(mandir)/man1
! 	$(INSTALL_DATA) $(TARGET).man $(DESTDIR)$(mandir)/man1/$(TARGET).1
  
  install.scripts:
! 	$(srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
! 	$(INSTALL) $(TARGETSSH) $(DESTDIR)$(bindir)/$(TARGETSSH)
  
  clean:
  	-rm -f *~ *.o $(TARGET) pure_* quant_* *.bak core gmon.out
