--- src/bootloaders/grub.c.orig	2025-04-09 22:10:09.326873332 +0200
+++ src/bootloaders/grub.c	2025-04-09 22:10:12.448882276 +0200
@@ -3,7 +3,7 @@
 #include "context.h"
 #include "utils.h"
 
-#define GRUB_EDITENV "grub-editenv"
+#define GRUB_EDITENV "grub2-editenv"
 
 static gboolean grub_env_get(const gchar *key, GString **value, GError **error)
 {
--- test/conftest.py.orig	2025-04-09 21:56:42.928471901 +0200
+++ test/conftest.py	2025-04-09 22:06:04.312171376 +0200
@@ -74,11 +74,11 @@
 
 
 def _have_grub():
-    out, err, exitcode = run("grub-editenv -V")
+    out, err, exitcode = run("grub2-editenv -V")
     return exitcode == 0
 
 
-have_grub = pytest.mark.skipif(not _have_grub(), reason="Have no grub-editenv")
+have_grub = pytest.mark.skipif(not _have_grub(), reason="Have no grub2-editenv")
 
 
 def _have_openssl():
@@ -383,7 +383,7 @@
     os.symlink(os.path.abspath("openssl-ca"), tmp_path / "openssl-ca")
     os.symlink(os.path.abspath("openssl-enc"), tmp_path / "openssl-enc")
 
-    run(f'grub-editenv {tmp_path}/grubenv.test set ORDER="A B" A_TRY="0" B_TRY="0" A_OK="1" B_OK="0"')
+    run(f'grub2-editenv {tmp_path}/grubenv.test set ORDER="A B" A_TRY="0" B_TRY="0" A_OK="1" B_OK="0"')
 
 
 @pytest.fixture
@@ -604,7 +604,7 @@
         open(self.tmp_path / "images/appfs-2", mode="w").close()
         # prepare grub env for 3 slots
         run(
-            f'grub-editenv {self.tmp_path}/grubenv.test set ORDER="A B C" A_TRY="0" B_TRY="0" C_TRY="0" A_OK="1" B_OK="1" C_OK="1"'
+            f'grub2-editenv {self.tmp_path}/grubenv.test set ORDER="A B C" A_TRY="0" B_TRY="0" C_TRY="0" A_OK="1" B_OK="1" C_OK="1"'
         )
 
     def prepare_composefs_config(self):
