From 2a29d5cea055bb54e2269dc4e2c4e6916fb57dff Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Mon, 20 Jan 2025 23:39:25 +0100
Subject: [PATCH] Port tests from tempdir to tempfile

---
 src/lib.rs | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/src/lib.rs b/src/lib.rs
index 1b75ecd..6124a8f 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -905,13 +905,13 @@ impl IntoRawFd for UnixDatagram {
 
 #[cfg(test)]
 mod test {
-    extern crate tempdir;
+    extern crate tempfile;
 
     use std::thread;
     use std::io;
     use std::io::prelude::*;
     use std::time::Duration;
-    use self::tempdir::TempDir;
+    use self::tempfile::TempDir;
 
     use super::*;
 
@@ -926,7 +926,7 @@ mod test {
 
     #[test]
     fn basic() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
         let msg1 = b"hello";
         let msg2 = b"world!";
@@ -1007,7 +1007,7 @@ mod test {
 
     #[test]
     fn try_clone() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
         let msg1 = b"hello";
         let msg2 = b"world";
@@ -1033,7 +1033,7 @@ mod test {
 
     #[test]
     fn iter() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
 
         let listener = or_panic!(UnixListener::bind(&socket_path));
@@ -1055,7 +1055,7 @@ mod test {
 
     #[test]
     fn long_path() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path()
                              .join("asdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfa\
                                     sasdfasdfasdasdfasdfasdfadfasdfasdfasdfasdfasdf");
@@ -1080,7 +1080,7 @@ mod test {
 
     #[test]
     fn timeouts() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
 
         let _listener = or_panic!(UnixListener::bind(&socket_path));
@@ -1107,7 +1107,7 @@ mod test {
 
     #[test]
     fn test_read_timeout() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
 
         let _listener = or_panic!(UnixListener::bind(&socket_path));
@@ -1122,7 +1122,7 @@ mod test {
 
     #[test]
     fn test_read_with_timeout() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let socket_path = dir.path().join("sock");
 
         let listener = or_panic!(UnixListener::bind(&socket_path));
@@ -1143,7 +1143,7 @@ mod test {
 
     #[test]
     fn test_unix_datagram() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let path1 = dir.path().join("sock1");
         let path2 = dir.path().join("sock2");
 
@@ -1159,7 +1159,7 @@ mod test {
 
     #[test]
     fn test_unnamed_unix_datagram() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let path1 = dir.path().join("sock1");
 
         let sock1 = or_panic!(UnixDatagram::bind(&path1));
@@ -1176,7 +1176,7 @@ mod test {
 
     #[test]
     fn test_connect_unix_datagram() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let path1 = dir.path().join("sock1");
         let path2 = dir.path().join("sock2");
 
@@ -1202,7 +1202,7 @@ mod test {
 
     #[test]
     fn test_unix_datagram_recv() {
-        let dir = or_panic!(TempDir::new("unix_socket"));
+        let dir = or_panic!(TempDir::with_prefix("unix_socket"));
         let path1 = dir.path().join("sock1");
 
         let sock1 = or_panic!(UnixDatagram::bind(&path1));
-- 
2.48.1

