--- plugins/ad3dsfilter/Makefile.orig	2008-02-16 10:48:40.000000000 +0100
+++ plugins/ad3dsfilter/Makefile	2008-02-16 10:52:34.000000000 +0100
@@ -2,28 +2,20 @@
 			  ad3dsfilter.so \
 
 LFLAGS = -g -Wall
-CFLAGS = -g -Wall -DMM3D_EDIT -I../../src -I../../src/libmm3d -I../../src/mm3dcore -I../.. -Ilib3ds -DPLUGIN
+CPPFLAGS = -fPIC -DMM3D_EDIT -I../../src -I../../src/libmm3d -I../../src/mm3dcore -I../.. -DPLUGIN
 
 # Mac OS X flags. Uncomment the following line to compile on Mac OS X
 #LFLAGS = -g -Wall -undefined suppress -bundle -flat_namespace
 
-all: lib3ds_lib ${PLUGINS}
+all: ${PLUGINS}
 	
-lib3ds_lib: lib3ds/Makefile
-	make -C lib3ds all
-
-lib3ds/Makefile: lib3ds/configure.in
-	cd lib3ds && ./configure && cd ..
-
 %.o: %.cc
-	g++ ${CFLAGS} -c $<
+	g++ ${CPPFLAGS} ${CXXFLAGS} -c $<
 
 %.so: %.o
-	g++ ${LFLAGS} -shared -o $*.so $< lib3ds/lib3ds/lib3ds.a
+	g++ ${LFLAGS} -shared -o $*.so $< -l3ds
 
 distclean: clean
-	make -C lib3ds distclean
 
 clean:
 	rm -rf *.o *.so core core.*
-	make -C lib3ds clean
