diff --git a/src/input.c b/src/input.c
index 79db485..92bad36 100644
--- a/src/input.c
+++ b/src/input.c
@@ -463,7 +463,7 @@ static gboolean get_file_to_load(gchar* file, gchar* pal_file, gboolean show_err
 	    return FALSE;
 	}
 	else
-	    sprintf(pal_file, file);
+	    sprintf(pal_file, "%s", file);
     }
 
     else
@@ -618,7 +618,7 @@ GHashTable* load_files()
 	FILE* pal_file_handle = NULL;
 
 	if(!get_file_to_load(settings->pal_file, pal_file, FALSE))
-	    sprintf(pal_file, settings->pal_file);
+	    sprintf(pal_file, "%s", settings->pal_file);
 
 	pal_file_handle = get_file_handle(pal_file, TRUE);
 	if(pal_file_handle != NULL)
diff --git a/src/output.c b/src/output.c
index 97aaa0f..9806089 100644
--- a/src/output.c
+++ b/src/output.c
@@ -249,7 +249,7 @@ static void pal_output_text_week(GDate* date, gboolean force_month_label,
 		pal_output_fg(BRIGHT, color, utf8_buf);
 	}
 	else
-	    g_print(utf8_buf);
+	    g_print("%s", utf8_buf);
 
 
 	if(g_date_compare(date,today) == 0)	/* make today bright */
@@ -270,7 +270,7 @@ static void pal_output_text_week(GDate* date, gboolean force_month_label,
 
 	}
 	else
-	    g_print(utf8_buf);
+	    g_print("%s", utf8_buf);
 
 
 	/* print extra space between days */
