--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,10 +41,6 @@
 
 endif()
 
-IF(NOT EXISTS ${PROJECT_SOURCE_DIR}/src/ipcgull)
-    message(FATAL_ERROR "Missing ipcgull submodule")
-endif()
-
 if(NOT LOGIOPS_VERSION)
     set(LOGIOPS_VERSION "null")
 endif()
@@ -56,5 +52,4 @@
     add_definitions(-DUSE_USER_BUS)
 endif()
 
-add_subdirectory(src/ipcgull)
 add_subdirectory(src/logid)

--- a/src/logid/CMakeLists.txt
+++ b/src/logid/CMakeLists.txt
@@ -81,13 +81,25 @@
 find_library(EVDEV_LIBRARY
         NAMES evdev libevdev)
 
-set(IPCGULL_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../ipcgull/src/include)
-message(${IPCGULL_INCLUDE_DIRS})
+find_path(IPCGULL_INCLUDE_DIRS 
+		ipcgull/connection.h  
+		ipcgull/exception.h  
+		ipcgull/function.h  
+		ipcgull/interface.h  
+		ipcgull/node.h  
+		ipcgull/property.h  
+		ipcgull/server.h  
+		ipcgull/signal.h  
+		ipcgull/variant.h  
+		${IPCGULL_INCLUDE_DIRS})
+find_library(IPCGULL_LIBRARY
+        NAMES ipcgull ipcgull_shared libipcgull libipcgull_shared
+        HINTS "${CMAKE_PREFIX_PATH}/ipcgull")
 
 include_directories(. ${EVDEV_INCLUDE_DIR} ${LIBUDEV_INCLUDE_DIRECTORIES} ${IPCGULL_INCLUDE_DIRS})
 
 target_link_libraries(logid ${CMAKE_THREAD_LIBS_INIT} ${EVDEV_LIBRARY} config++
-        ${LIBUDEV_LIBRARIES} ipcgull)
+        ${LIBUDEV_LIBRARIES} ${IPCGULL_LIBRARY})
 
 install(TARGETS logid DESTINATION bin)
 

