From c14947308ff6851cbbbad098e9758f6baf06dc0a Mon Sep 17 00:00:00 2001
From: Vojtech Trefny <vtrefny@redhat.com>
Date: Fri, 28 Jun 2019 11:18:26 +0200
Subject: [PATCH] Skip test that needs internet connection

---
 tests/test_style.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/test_style.py b/tests/test_style.py
index 1480d0c..b16cc36 100644
--- a/tests/test_style.py
+++ b/tests/test_style.py
@@ -55,6 +55,7 @@ class TestScript(unittest.TestCase):
         self.assertEqual(result, "https://fonts.gstatic.com/s/indieflower/v9/m8JVjfNVeKWVnh3QMuKkFcZVaUuH.woff2")
 
     def test_embed_google_web_font(self):
+        self.skipTest("Internet connection is not available in Koji")
         dwg = Drawing()
         dwg.embed_google_web_font(name="Indie Flower", uri="http://fonts.googleapis.com/css?family=Indie+Flower")
         style = dwg.defs.elements[0]
-- 
2.20.1

