diff -up meshlab-MeshLab-2023.12/src/vcglib/wrap/io_trimesh/import_ply.h.no-LoadCamera meshlab-MeshLab-2023.12/src/vcglib/wrap/io_trimesh/import_ply.h
--- meshlab-MeshLab-2023.12/src/vcglib/wrap/io_trimesh/import_ply.h.no-LoadCamera	2025-03-24 09:03:16.976556627 -0400
+++ meshlab-MeshLab-2023.12/src/vcglib/wrap/io_trimesh/import_ply.h	2025-03-24 09:04:16.334655273 -0400
@@ -1084,77 +1084,77 @@ public:
 
 
 	// Caricamento camera da un ply
-	int LoadCamera(const char * filename)
-	{
-		vcg::ply::PlyFile pf;
-		if( pf.Open(filename,vcg::ply::PlyFile::MODE_READ)==-1 )
-		{
-			this->pi.status = pf.GetError();
-			return this->pi.status;
-		}
-
-
-		bool found = true;
-		for(int i=0;i<23;++i)
-		{
-			if( pf.AddToRead(CameraDesc(i))==-1 )
-			{
-				found = false;
-				break;
-			}
-		}
-
-		if(!found)
-			return this->pi.status;
-
-		for(size_t i=0;i<pf.elements.size();i++)
-		{
-			int n = pf.ElemNumber(i);
-
-			if( !strcmp( pf.ElemName(i),"camera" ) )
-			{
-				pf.SetCurElement(i);
-
-				LoadPly_Camera ca;
-
-				for(int j=0;j<n;++j)
-				{
-					if( pf.Read( (void *)&(ca) )==-1 )
-					{
-						this->pi.status = PlyInfo::E_SHORTFILE;
-						return this->pi.status;
-					}
-					this->camera.valid     = true;
-					this->camera.view_p[0] = ca.view_px;
-					this->camera.view_p[1] = ca.view_py;
-					this->camera.view_p[2] = ca.view_pz;
-					this->camera.x_axis[0] = ca.x_axisx;
-					this->camera.x_axis[1] = ca.x_axisy;
-					this->camera.x_axis[2] = ca.x_axisz;
-					this->camera.y_axis[0] = ca.y_axisx;
-					this->camera.y_axis[1] = ca.y_axisy;
-					this->camera.y_axis[2] = ca.y_axisz;
-					this->camera.z_axis[0] = ca.z_axisx;
-					this->camera.z_axis[1] = ca.z_axisy;
-					this->camera.z_axis[2] = ca.z_axisz;
-					this->camera.f         = ca.focal;
-					this->camera.s[0]      = ca.scalex;
-					this->camera.s[1]      = ca.scaley;
-					this->camera.c[0]      = ca.centerx;
-					this->camera.c[1]      = ca.centery;
-					this->camera.viewport[0] = ca.viewportx;
-					this->camera.viewport[1] = ca.viewporty;
-					this->camera.k[0]      = ca.k1;
-					this->camera.k[1]      = ca.k2;
-					this->camera.k[2]      = ca.k3;
-					this->camera.k[3]      = ca.k4;
-				}
-				break;
-			}
-		}
-
-		return 0;
-	}
+//	int LoadCamera(const char * filename)
+//	{
+//		vcg::ply::PlyFile pf;
+//		if( pf.Open(filename,vcg::ply::PlyFile::MODE_READ)==-1 )
+//		{
+//			this->pi.status = pf.GetError();
+//			return this->pi.status;
+//		}
+//
+//
+//		bool found = true;
+//		for(int i=0;i<23;++i)
+//		{
+//			if( pf.AddToRead(CameraDesc(i))==-1 )
+//			{
+//				found = false;
+//				break;
+//			}
+//		}
+//
+//		if(!found)
+//			return this->pi.status;
+//
+//		for(size_t i=0;i<pf.elements.size();i++)
+//		{
+//			int n = pf.ElemNumber(i);
+//
+//			if( !strcmp( pf.ElemName(i),"camera" ) )
+//			{
+//				pf.SetCurElement(i);
+//
+//				LoadPly_Camera ca;
+//
+//				for(int j=0;j<n;++j)
+//				{
+//					if( pf.Read( (void *)&(ca) )==-1 )
+//					{
+//						this->pi.status = PlyInfo::E_SHORTFILE;
+//						return this->pi.status;
+//					}
+//					this->camera.valid     = true;
+//					this->camera.view_p[0] = ca.view_px;
+//					this->camera.view_p[1] = ca.view_py;
+//					this->camera.view_p[2] = ca.view_pz;
+//					this->camera.x_axis[0] = ca.x_axisx;
+//					this->camera.x_axis[1] = ca.x_axisy;
+//					this->camera.x_axis[2] = ca.x_axisz;
+//					this->camera.y_axis[0] = ca.y_axisx;
+//					this->camera.y_axis[1] = ca.y_axisy;
+//					this->camera.y_axis[2] = ca.y_axisz;
+//					this->camera.z_axis[0] = ca.z_axisx;
+//					this->camera.z_axis[1] = ca.z_axisy;
+//					this->camera.z_axis[2] = ca.z_axisz;
+//					this->camera.f         = ca.focal;
+//					this->camera.s[0]      = ca.scalex;
+//					this->camera.s[1]      = ca.scaley;
+//					this->camera.c[0]      = ca.centerx;
+//					this->camera.c[1]      = ca.centery;
+//					this->camera.viewport[0] = ca.viewportx;
+//					this->camera.viewport[1] = ca.viewporty;
+//					this->camera.k[0]      = ca.k1;
+//					this->camera.k[1]      = ca.k2;
+//					this->camera.k[2]      = ca.k3;
+//					this->camera.k[3]      = ca.k4;
+//				}
+//				break;
+//			}
+//		}
+//
+//		return 0;
+//	}
 
 
 	static bool LoadMask(const char * filename, int &mask)
