diff -up meshlab-MeshLab-2023.12/src/CMakeLists.txt.cmakefix meshlab-MeshLab-2023.12/src/CMakeLists.txt
--- meshlab-MeshLab-2023.12/src/CMakeLists.txt.cmakefix	2025-03-24 08:27:33.821542185 -0400
+++ meshlab-MeshLab-2023.12/src/CMakeLists.txt	2025-03-24 08:27:57.908599870 -0400
@@ -2,6 +2,8 @@
 # Copyright 2019, 2021, Visual Computing Lab, ISTI - Italian National Research Council
 # SPDX-License-Identifier: BSL-1.0
 
+cmake_minimum_required(VERSION 3.18)
+
 ### Dependencies
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/meshlab_global_settings.cmake" NO_POLICY_SCOPE)
