From 0c961becebb52b9e7077e4217b7dfdbcd6730b1b Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Sat, 8 Mar 2025 18:43:41 +0100
Subject: [PATCH 1/5] fix invalid crate manifest for tests workspace member

---
 tests/Cargo.toml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/Cargo.toml b/tests/Cargo.toml
index 35d15cb..4f9fe76 100644
--- a/tests/Cargo.toml
+++ b/tests/Cargo.toml
@@ -56,4 +56,9 @@ name = "processor-loader"
 path = "processor-loader.rs"
 
 [dependencies]
-gufo-common.workspace = true
+glycin = { workspace = true }
+glycin-core = { workspace = true }
+glycin-utils = { workspace = true }
+gufo-common = { workspace = true }
+zbus = { workspace = true }
+
-- 
2.55.0

