diff -up Zasx/src/weapon.c~ Zasx/src/weapon.c
--- Zasx/src/weapon.c~	2018-10-15 11:15:27.000000000 +0200
+++ Zasx/src/weapon.c	2018-10-15 11:17:02.507744263 +0200
@@ -10,6 +10,7 @@
 #include <math.h>
 #include <string.h>
 #include <stdio.h>
+#include <locale.h>
 
 #include "allegro.h"
 #include "particle.h"
@@ -119,6 +120,7 @@ WEAPON *load_weapon(char *filename) {
 	PACKFILE *p;
 	WEAPON *w = NULL;
 	WEAPON_CANNON *c = NULL;
+	char *old_locale;
 	char buf[520];
 	char buf2[520];
 
@@ -142,6 +144,12 @@ WEAPON *load_weapon(char *filename) {
 	GamePal[0].r =  63; GamePal[0].g =  0; GamePal[0].b =  63;
 	select_palette(GamePal);
 
+	/*
+	 * Ensure sscanf uses '.' as decimal separator, as we use this in
+	 * our weapon configs.
+	 */
+	old_locale = setlocale(LC_NUMERIC, "C");
+
 	while(pack_fgets(buf, 512, p)) {
 		int r, i;
 		float x, y;
@@ -393,6 +401,9 @@ WEAPON *load_weapon(char *filename) {
 
 	pack_fclose(p);
 
+	if (old_locale)
+		setlocale(LC_NUMERIC, old_locale);
+
 	unselect_palette();
 	GamePal[0].r = 0; GamePal[0].g = 0;  GamePal[0].b = 0; 
 
