diff --git a/src/tpm2_pytss/cryptography.py b/src/tpm2_pytss/cryptography.py
--- a/src/tpm2_pytss/cryptography.py
+++ b/src/tpm2_pytss/cryptography.py
@@ -257,6 +257,12 @@ class tpm_rsa_private_key(rsa.RSAPrivateKey):
         """Always raises a NotImplementedError."""
         raise NotImplementedError()
 
+    def __copy__(self) -> "tpm_rsa_private_key":
+        return tpm_rsa_private_key(
+            ectx=self._ectx, handle=self._handle, session=self._session
+        )
+
+    def __deepcopy__(self, memo) -> "tpm_rsa_private_key":
+        return self.__copy__()
+
 
 class tpm_ecc_private_key(ec.EllipticCurvePrivateKey):
@@ -428,3 +434,12 @@ class tpm_ecc_private_key(ec.EllipticCurvePrivateKey):
     ) -> None:
         """Always raises a NotImplementedError."""
         raise NotImplementedError()
+
+    def __copy__(self) -> "tpm_ecc_private_key":
+        return tpm_ecc_private_key(
+            ectx=self._ectx, handle=self._handle, session=self._session
+        )
+
+    def __deepcopy__(self, memo) -> "tpm_ecc_private_key":
+        return self.__copy__()
