diff --git a/configure.in b/configure.in
index f15ef06..0f2adfc 100644
--- a/configure.in
+++ b/configure.in
@@ -22,6 +22,7 @@ AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(libast, 0.7.1)
 AC_DEFINE_UNQUOTED(LIBAST_VERSION, "$VERSION", [Version])
 AM_CONFIG_HEADER(config.h include/libast/sysdefs.h)
+m4_include([libast.m4])
 
 LIBTOOLIZE="${SAVE_LIBTOOLIZE:-libtoolize}"
 ACLOCAL="${SAVE_ACLOCAL:-aclocal}"
