--- histo-1.0.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ histo-1.0.0/Cargo.toml	2024-03-19T22:45:39.807935+00:00
@@ -25,17 +25,14 @@
 license = "Apache-2.0/MIT"
 repository = "https://github.com/fitzgen/histo"
 resolver = "2"
+exclude = ["/ci/"]
 
 [[bin]]
 name = "histo"
 doc = false
 
-[dependencies.quickcheck]
-version = "0.4.1"
-optional = true
-
 [dependencies.streaming-stats]
-version = "0.1.28"
+version = "0.2"
 
 [badges.travis-ci]
 repository = "fitzgen/histo"
