diff --git a/configure b/configure
index 0452e96..0a23391 100755
--- a/configure
+++ b/configure
@@ -2692,8 +2692,8 @@ main ()
   for (i = 0; i < 256; i++)
     if (XOR (islower (i), ISLOWER (i))
 	|| toupper (i) != TOUPPER (i))
-      exit(2);
-  exit (0);
+      return 2;
+  return 0;
 }
 _ACEOF
 rm -f conftest$ac_exeext
@@ -4460,6 +4460,7 @@ else
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* By Ruediger Kuhlmann. */
+      #include "confdefs.h"
       #include <sys/types.h>
       #if HAVE_UNISTD_H
       # include <unistd.h>
@@ -4468,8 +4469,8 @@ else
       int main ()
       {
 	if (fork() < 0)
-	  exit (1);
-	exit (0);
+	  return 1;
+	retutn 0;
       }
 _ACEOF
 rm -f conftest$ac_exeext
