--- custom/conf/app.example.ini	2026-01-29 08:37:32.000000000 +0100
+++ app.ini.tmpl	2026-02-17 18:03:17.788649399 +0100
@@ -1,16 +1,3 @@
-; This file lists the default values used by Forgejo
-;; Copy required sections to your own app.ini (default is custom/conf/app.ini)
-;; and modify as needed.
-;; Do not copy the whole file as-is, as it contains some invalid sections for illustrative purposes.
-;; If you don't know what a setting is you should not set it.
-;;
-;; see https://forgejo.org/docs/next/admin/config-cheat-sheet for additional documentation.
-
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Default Configuration (non-`app.ini` configuration)
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;
 ;; These values are environment-dependent but form the basis of a lot of values. They will be
 ;; reported as part of the default configuration when running `forgejo help` or on start-up. The order they are emitted there is slightly different but we will list them here in the order they are set-up.
 ;;
@@ -51,14 +38,14 @@
 ;APP_DISPLAY_NAME_FORMAT = {APP_NAME}: {APP_SLOGAN}
 ;;
 ;; RUN_USER will automatically detect the current user - but you can set it here change it if you run locally
-RUN_USER = ; git
+RUN_USER = forgejo
 ;;
 ;; Application run mode, affects performance and debugging: "dev" or "prod", default is "prod"
 ;; Mode "dev" makes Forgejo easier to develop and debug, values other than "dev" are treated as "prod" which is for production use.
-;RUN_MODE = prod
+RUN_MODE = prod
 ;;
 ;; The working directory, see the comment of AppWorkPath above
-;WORK_PATH =
+WORK_PATH = /var/lib/forgejo
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
@@ -140,7 +127,7 @@
 ;LOCAL_USE_PROXY_PROTOCOL = %(USE_PROXY_PROTOCOL)s
 ;;
 ;; Disable SSH feature when not available
-;DISABLE_SSH = false
+DISABLE_SSH = true
 ;;
 ;; Whether to use the builtin SSH server or not.
 ;START_SSH_SERVER = false
@@ -161,7 +148,7 @@
 ;SSH_LISTEN_HOST =
 ;;
 ;; Port number to be exposed in clone URL
-;SSH_PORT = 22
+SSH_PORT =
 ;;
 ;; The port number the builtin SSH server should listen on
 ;SSH_LISTEN_PORT = %(SSH_PORT)s
@@ -307,14 +294,14 @@
 ;;
 ;; Landing page, can be "home", "explore", "organizations", "login", or any URL such as "/org/repo" or even "https://anotherwebsite.com"
 ;; The "login" choice is not a security measure but just a UI flow change, use REQUIRE_SIGNIN_VIEW to force users to log in.
-;LANDING_PAGE = home
+LANDING_PAGE = explore
 ;;
 ;; Enables git-lfs support. true or false, default is false.
 ;LFS_START_SERVER = false
 ;;
 ;;
 ;; LFS authentication secret, change this yourself
-;LFS_JWT_SECRET =
+LFS_JWT_SECRET = @LFS_JWT_SECRET@
 ;;
 ;; Alternative location to specify LFS authentication secret. You cannot specify both this and LFS_JWT_SECRET, and must pick one
 ;LFS_JWT_SECRET_URI = file:/etc/gitea/lfs_jwt_secret
@@ -434,18 +421,18 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
 ;; Whether the installer is disabled (set to true to disable the installer)
-INSTALL_LOCK = false
+INSTALL_LOCK = true
 ;;
 ;; Global secret key that will be used
 ;; This key is VERY IMPORTANT. If you lose it, the data encrypted by it (like 2FA secret) can't be decrypted anymore.
-SECRET_KEY =
+SECRET_KEY = @SECRET_KEY@
 ;;
 ;; Alternative location to specify secret key, instead of this file; you cannot specify both this and SECRET_KEY, and must pick one
 ;; This key is VERY IMPORTANT. If you lose it, the data encrypted by it (like 2FA secret) can't be decrypted anymore.
 ;SECRET_KEY_URI = file:/etc/gitea/secret_key
 ;;
 ;; Secret used to validate communication within Forgejo binary.
-INTERNAL_TOKEN =
+INTERNAL_TOKEN = @INTERNAL_TOKEN@
 ;;
 ;; Alternative location to specify internal token, instead of this file; you cannot specify both this and INTERNAL_TOKEN, and must pick one
 ;INTERNAL_TOKEN_URI = file:/etc/gitea/internal_token
@@ -536,7 +523,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
 ;; Enables OAuth2 provider
-ENABLED = true
+ENABLED = false
 ;;
 ;; Algorithm used to sign OAuth2 tokens. Valid values: HS256, HS384, HS512, RS256, RS384, RS512, ES256, ES384, ES512, EdDSA
 ;JWT_SIGNING_ALGORITHM = RS256
@@ -548,7 +535,7 @@
 ;;
 ;; OAuth2 authentication secret for access and refresh tokens, change this yourself to a unique string. CLI generate option is helpful in this case. https://forgejo.org/docs/latest/admin/command-line/#generate-secret
 ;; This setting is only needed if JWT_SIGNING_ALGORITHM is set to HS256, HS384 or HS512.
-;JWT_SECRET =
+JWT_SECRET = @JWT_SECRET@
 ;;
 ;; Alternative location to specify OAuth2 authentication secret. You cannot specify both this and JWT_SECRET, and must pick one
 ;JWT_SECRET_URI = file:/etc/gitea/oauth2_jwt_secret
@@ -759,7 +746,7 @@
 ;EMAIL_DOMAIN_BLOCKLIST =
 ;;
 ;; Disallow registration, only allow admins to create accounts.
-;DISABLE_REGISTRATION = false
+DISABLE_REGISTRATION = true
 ;;
 ;; Allow registration only using Forgejo itself, it works only when DISABLE_REGISTRATION is false
 ;ALLOW_ONLY_INTERNAL_REGISTRATION = false
@@ -837,7 +824,7 @@
 ;; Users will be able to use dots when choosing their username. Disabling this is
 ;; helpful if your usersare having issues with e.g. RSS feeds or advanced third-party
 ;; extensions that use strange regex patterns.
-; ALLOW_DOTS_IN_USERNAMES = true
+ALLOW_DOTS_IN_USERNAMES = true
 ;;
 ;; Either "public", "limited" or "private", default is "public"
 ;; Limited is for users visible only to signed users
@@ -920,12 +907,12 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;[service.explore]
+[service.explore]
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
 ;; Only allow signed in users to view the explore pages.
-;REQUIRE_SIGNIN_VIEW = false
+REQUIRE_SIGNIN_VIEW = true
 ;;
 ;; Disable the users explore page.
 ;DISABLE_USERS_PAGE = false
@@ -1588,7 +1575,7 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;[openid]
+[openid]
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
@@ -1609,11 +1596,11 @@
 ;;  - <username>.livejournal.com
 ;;
 ;; Whether to allow signin in via OpenID
-;ENABLE_OPENID_SIGNIN = true
+ENABLE_OPENID_SIGNIN = false
 ;;
 ;; Whether to allow registering via OpenID
 ;; Do not include to rely on rhw DISABLE_REGISTRATION setting
-;ENABLE_OPENID_SIGNUP = true
+ENABLE_OPENID_SIGNUP = false
 ;;
 ;; Allowed URI patterns (POSIX regexp).
 ;; Space separated.
@@ -2346,10 +2333,10 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Check for new Forgejo versions
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;[cron.update_checker]
+[cron.update_checker]
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;ENABLED = true
+ENABLED = false
 ;RUN_AT_START = false
 ;ENABLE_SUCCESS_NOTICE = false
 ;SCHEDULE = @every 168h
@@ -2442,15 +2429,15 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;[other]
+[other]
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Show version information about Forgejo and Go in the footer
-;SHOW_FOOTER_VERSION = true
+SHOW_FOOTER_VERSION = false
 ;; Show template execution time in the footer
-;SHOW_FOOTER_TEMPLATE_LOAD_TIME = true
+SHOW_FOOTER_TEMPLATE_LOAD_TIME = false
 ;; Show the "powered by" text in the footer
-;SHOW_FOOTER_POWERED_BY = true
+SHOW_FOOTER_POWERED_BY = false
 ;; Generate sitemap. Defaults to `true`.
 ;ENABLE_SITEMAP = true
 ;; Enable/Disable RSS/Atom feed
