--- tree-sitter-lua-0.2.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ tree-sitter-lua-0.2.0/Cargo.toml	2025-03-12T04:19:02.205243+00:00
@@ -16,6 +16,7 @@
 authors = ["Munif Tanjim <hello@muniftanjim.dev>"]
 build = "bindings/rust/build.rs"
 include = [
+    "LICENSE.md",
     "bindings/rust/*",
     "grammar.js",
     "queries/*",
@@ -48,7 +49,7 @@
 version = "0.1.0"
 
 [dev-dependencies.tree-sitter]
-version = "0.23"
+version = ">=0.23,<0.25"
 
 [build-dependencies.cc]
 version = "1.1.18"
