--- hf-hub-0.4.3/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hf-hub-0.4.3/Cargo.toml	2025-12-03T21:42:54.244337+00:00
@@ -58,7 +58,6 @@
     "dep:reqwest",
     "reqwest/charset",
     "reqwest/http2",
-    "reqwest/macos-system-configuration",
     "dep:serde",
     "dep:serde_json",
     "dep:thiserror",
@@ -86,7 +85,7 @@
 path = "examples/download.rs"
 
 [dependencies.dirs]
-version = "6"
+version = ">=5, <7"
 
 [dependencies.futures]
 version = "0.3.28"
@@ -97,7 +96,7 @@
 optional = true
 
 [dependencies.indicatif]
-version = "0.17.5"
+version = "0.18"
 optional = true
 
 [dependencies.log]
@@ -154,7 +153,7 @@
 optional = true
 
 [dev-dependencies.hex-literal]
-version = "1.0"
+version = ">=0.4.1, <2.0"
 
 [dev-dependencies.sha2]
 version = "0.10"
