diff -r -U4 blueman-2.3.beta1--orig/configure.ac blueman-2.3.beta1--patched/configure.ac
--- blueman-2.3.beta1--orig/configure.ac	2022-06-28 15:57:48.000000000 +0200
+++ blueman-2.3.beta1--patched/configure.ac	2022-06-29 11:59:29.593483779 +0200
@@ -154,33 +154,21 @@
 AC_ARG_ENABLE([caja-sendto],
   AS_HELP_STRING([--enable-caja-sendto], [Enable caja-sendto installation]),
   [enable_caja=$enableval], [enable_caja=yes])
 
-AS_IF([test "x$enable_caja" = "xyes"],
-  [PKG_CHECK_EXISTS([caja-python], [have_caja_python=yes], [have_caja_python=no])],
-  [have_caja_python=no])
-
-AM_CONDITIONAL([HAVE_CAJA_PYTHON], [test "x$have_caja_python" = "xyes"])
+AM_CONDITIONAL([HAVE_CAJA], [test "x$enable_caja" = "xyes"])
 
 AC_ARG_ENABLE([nemo-sendto],
   AS_HELP_STRING([--enable-nemo-sendto], [Enable nemo-sendto installation]),
   [enable_nemo=$enableval], [enable_nemo=yes])
 
-AS_IF([test "x$enable_nemo" = "xyes"],
-  [PKG_CHECK_EXISTS([nemo-python], [have_nemo_python=yes], [have_nemo_python=no])],
-  [have_nemo_python=no])
-
-AM_CONDITIONAL([HAVE_NEMO_PYTHON], [test "x$have_nemo_python" = "xyes"])
+AM_CONDITIONAL([HAVE_NEMO], [test "x$enable_nemo" = "xyes"])
 
 AC_ARG_ENABLE([nautilus-sendto],
   AS_HELP_STRING([--enable-nautilus-sendto], [Enable nautilus-sendto installation]),
   [enable_nautilus=$enableval], [enable_nautilus=yes])
 
-AS_IF([test "x$enable_nautilus" = "xyes"],
-  [PKG_CHECK_EXISTS([nautilus-python], [have_nautilus_python=yes], [have_nautilus_python=no])],
-  [have_nautilus_python=no])
-
-AM_CONDITIONAL([HAVE_NAUTILUS_PYTHON], [test "x$have_nautilus_python" = "xyes"])
+AM_CONDITIONAL([HAVE_NAUTILUS], [test "x$enable_nautilus" = "xyes"])
 
 AC_ARG_ENABLE([thunar-sendto],
   AS_HELP_STRING([--enable-thunar-sendto], [Enable thunar-sendto installation]),
   [have_thunar=$enableval], [have_thunar=yes])
diff -r -U4 blueman-2.3.beta1--orig/sendto/Makefile.am blueman-2.3.beta1--patched/sendto/Makefile.am
--- blueman-2.3.beta1--orig/sendto/Makefile.am	2022-06-28 15:57:48.000000000 +0200
+++ blueman-2.3.beta1--patched/sendto/Makefile.am	2022-06-29 12:00:00.749738623 +0200
@@ -6,19 +6,19 @@
 
 nautilus_blueman_sendto.py: blueman_sendto.py.in
 	$(SED) -e "s|@FILEMANAGER@|Nautilus|" $< >$@
 
-if HAVE_CAJA_PYTHON
+if HAVE_CAJA
 cajasendtodir = $(datadir)/caja-python/extensions
 cajasendto_DATA = caja_blueman_sendto.py
 endif
 
-if HAVE_NEMO_PYTHON
+if HAVE_NEMO
 nemosendtodir = $(datadir)/nemo-python/extensions
 nemosendto_DATA = nemo_blueman_sendto.py
 endif
 
-if HAVE_NAUTILUS_PYTHON
+if HAVE_NAUTILUS
 nautilussendtodir = $(datadir)/nautilus-python/extensions
 nautilussendto_DATA = nautilus_blueman_sendto.py
 endif
 
