--- a/src/occt_wrapper/CMakeLists.txt.orig	2023-12-27 21:08:35.865374600 +0000
+++ b/src/occt_wrapper/CMakeLists.txt	2023-12-27 21:08:43.610496996 +0000
@@ -19,14 +19,10 @@ include(GenerateExportHeader)
 
 generate_export_header(OCCTWrapper)
 
-find_package(OpenCASCADE 7.6.1 REQUIRED)
+find_package(OpenCASCADE REQUIRED)
 
 set(OCCT_LIBS
-    TKXDESTEP
-    TKSTEP
-    TKSTEP209
-    TKSTEPAttr
-    TKSTEPBase
+    TKDESTEP
     TKXCAF
     TKXSBase
     TKVCAF
@@ -58,5 +58,5 @@ target_link_libraries(OCCTWrapper ${OCCT
 
 include(GNUInstallDirs)
 
-install(TARGETS OCCTWrapper DESTINATION "${CMAKE_INSTALL_BINDIR}")
+install(TARGETS OCCTWrapper DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
