--- a/src/lib.rs
+++ b/src/lib.rs
@@ -22,7 +22,7 @@
 /// Marks async entry-point function to be executed by Actix system.
 ///
 /// # Examples
-/// ```
+/// ```ignore
 /// #[actix_rt::main]
 /// async fn main() {
 ///     println!("Hello world");
@@ -112,7 +112,7 @@
 /// Marks async test function to be executed in an Actix system.
 ///
 /// # Examples
-/// ```
+/// ```ignore
 /// #[actix_rt::test]
 /// async fn my_test() {
 ///     assert!(true);
