From 444bae5e1e73ecdc71f7506cbfafe2520444f15b Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Thu, 28 Aug 2014 15:34:14 +0200
Subject: [PATCH 2/3] Look up evolution-addressbook-export in libexec rather
 than lib

---
 m_evolution.sh.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m_evolution.sh.in b/m_evolution.sh.in
index e0c02cb..2a8350f 100644
--- a/m_evolution.sh.in
+++ b/m_evolution.sh.in
@@ -31,7 +31,7 @@ m_evolution_query ()
     EVOLUTION_ADDRESSBOOK_EXPORT=@EVOLUTION_ADDRESSBOOK_EXPORT@
 
     if [ "$EVOLUTION_ADDRESSBOOK_EXPORT" = "auto" ]; then
-	for e in /usr/lib/evolution/*/evolution-addressbook-export
+	for e in /usr/libexec/evolution/*/evolution-addressbook-export
 	do
 	    if [ -x $e ]; then
 		EVOLUTION_ADDRESSBOOK_EXPORT=$e
-- 
1.9.3

