diff -up OpenColorIO-2.3.2/src/apps/ocioarchive/main.cpp.mzver OpenColorIO-2.3.2/src/apps/ocioarchive/main.cpp
--- OpenColorIO-2.3.2/src/apps/ocioarchive/main.cpp.mzver	2024-02-14 19:52:58.342003757 -0700
+++ OpenColorIO-2.3.2/src/apps/ocioarchive/main.cpp	2024-03-03 18:07:39.405984145 -0700
@@ -235,7 +235,7 @@ int main(int argc, const char **argv)
         }
 
         std::string path = args[0];
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
         reader = mz_zip_reader_create();
 #else
         mz_zip_reader_create(&reader);
diff -up OpenColorIO-2.3.2/src/OpenColorIO/OCIOZArchive.cpp.mzver OpenColorIO-2.3.2/src/OpenColorIO/OCIOZArchive.cpp
--- OpenColorIO-2.3.2/src/OpenColorIO/OCIOZArchive.cpp.mzver	2024-02-14 19:52:58.342003757 -0700
+++ OpenColorIO-2.3.2/src/OpenColorIO/OCIOZArchive.cpp	2024-03-03 18:08:27.921308017 -0700
@@ -226,7 +226,7 @@ void archiveConfig(std::ostream & ostrea
     std::string configStr = ss.str();
 
     // Write zip to memory stream.
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
     write_mem_stream = mz_stream_mem_create();
 #else
     mz_stream_mem_create(&write_mem_stream);
@@ -242,7 +242,7 @@ void archiveConfig(std::ostream & ostrea
     options.compress_level  = ArchiveCompressionLevels::BEST;
 
     // Create the writer handle.
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
     archiver = mz_zip_writer_create();
 #else
     mz_zip_writer_create(&archiver);
@@ -341,7 +341,7 @@ void ExtractOCIOZArchive(const char * ar
     std::string outputDestination = pystring::os::path::normpath(destination);
 
     // Create zip reader.
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
     extracter = mz_zip_reader_create();
 #else
     mz_zip_reader_create(&extracter);
@@ -463,7 +463,7 @@ std::vector<uint8_t> getFileStringFromAr
     std::vector<uint8_t> buffer;
 
     // Create the reader object.
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
     reader = mz_zip_reader_create();
 #else
     mz_zip_reader_create(&reader);
@@ -527,7 +527,7 @@ void getEntriesMappingFromArchiveFile(co
     void *reader = NULL;
 
     // Create the reader object.
-#if MZ_VERSION_BUILD >= 040000
+#if MZ_VERSION_BUILD >= 0x40000
     reader = mz_zip_reader_create();
 #else
     mz_zip_reader_create(&reader);
