From c30511d03ee593bbd3e8d82681d89684be844d60 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ren=C3=A9=20Reucher?= <rene.reucher@batcom-it.net>
Date: Sat, 30 Apr 2022 10:40:44 +0200
Subject: [PATCH] fix: MachineListDatabase wasn't saved to qmc2.ini

---
 src/options.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/options.cpp b/src/options.cpp
index c51e69aa2..0e0ee627a 100644
--- a/src/options.cpp
+++ b/src/options.cpp
@@ -1426,6 +1426,7 @@ void Options::on_pushButtonApply_clicked()
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/XmlCacheDatabase", lineEditXmlCacheDatabase->text());
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/UserDataDatabase", lineEditUserDataDatabase->text());
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/MachineListCacheFile", lineEditMachineListCacheFile->text());
+	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/MachineListDatabase", lineEditMachineListDatabase->text());
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/ROMStateCacheFile", lineEditROMStateCacheFile->text());
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/SoftwareListCacheDatabase", lineEditSoftwareListCacheDb->text());
 	config->setValue(QMC2_EMULATOR_PREFIX + "FilesAndDirectories/SoftwareStateCache", lineEditSoftwareStateCache->text());
-- 
2.43.0

