diff -rupN --no-dereference apitrace-13.0/frametrim/ft_dependecyobject.hpp apitrace-13.0-new/frametrim/ft_dependecyobject.hpp
--- apitrace-13.0/frametrim/ft_dependecyobject.hpp	2025-06-26 14:43:44.000000000 +0200
+++ apitrace-13.0-new/frametrim/ft_dependecyobject.hpp	2025-07-18 10:35:58.077078472 +0200
@@ -27,6 +27,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include "ft_tracecall.hpp"
 #include <stdint.h>
 
diff -rupN --no-dereference apitrace-13.0/retrace/metric_writer.cpp apitrace-13.0-new/retrace/metric_writer.cpp
--- apitrace-13.0/retrace/metric_writer.cpp	2025-06-26 14:43:44.000000000 +0200
+++ apitrace-13.0-new/retrace/metric_writer.cpp	2025-07-18 10:35:58.077393930 +0200
@@ -23,6 +23,7 @@
  *
  **************************************************************************/
 
+#include <cstdint>
 #include <iostream>
 #include <stdint.h>
 
