diff -up itzam-core-2.1.1/libitzam/itzam.h.itzam32 itzam-core-2.1.1/libitzam/itzam.h
--- itzam-core-2.1.1/libitzam/itzam.h.itzam32	2008-09-06 22:23:02.000000000 +0200
+++ itzam-core-2.1.1/libitzam/itzam.h	2008-09-06 22:24:58.000000000 +0200
@@ -67,6 +67,10 @@ typedef signed char      int8_t;
 #include <stdint.h>
 #endif
 
+#if __WORDSIZE == 32
+#define ITZAM32
+#endif
+
 //------------------------------------------------------------
 // enumerated types
 typedef enum
diff -up itzam-core-2.1.1/libitzam/itzam_btree.c.patch itzam-core-2.1.1/libitzam/itzam_btree.c
--- itzam-core-2.1.1/libitzam/itzam_btree.c.patch	2008-09-06 03:20:32.000000000 +0200
+++ itzam-core-2.1.1/libitzam/itzam_btree.c	2008-09-06 03:21:06.000000000 +0200
@@ -285,11 +285,7 @@ static itzam_btree_page * read_page_dire
 }
 
 // cen't be static because it's used by debug routines
-#if defined(_MSC_VER)
 itzam_btree_page * read_page(itzam_btree * btree, itzam_ref where)
-#else
-inline itzam_btree_page * read_page(itzam_btree * btree, itzam_ref where)
-#endif
 {
     // what we return
     itzam_btree_page * page = NULL;
