diff -up oneko-1.2.sakura.5/oneko.man.BAD oneko-1.2.sakura.5/oneko.man
--- oneko-1.2.sakura.5/oneko.man.BAD	2010-12-15 13:26:34.501930058 -0500
+++ oneko-1.2.sakura.5/oneko.man	2010-12-15 13:27:08.243929980 -0500
@@ -14,8 +14,8 @@ The program oneko creates a cute cat cha
 .SH DESCRIPTION
 .I oneko
 changes your mouse cursor into mouse and creates a little cute cat and the
-cat start chasing around your mouse cursor.  If the cat catchup the ``mouse'',
-start sleeping.
+cat starts chasing around your mouse cursor.  If the cat catches the ``mouse'',
+it starts sleeping.
 .SS Options
 .TP
 .B -help
@@ -119,7 +119,7 @@ Foreground color.
 Background color.
 .SH Notes
 While this program uses XGetDefault, be sure to use "neko.resouce" form.
-If you run this program as "tora", by hard of soft link, the \fB-tora\fP
+If you run this program as "tora", by hard or soft link, the \fB-tora\fP
 option is enabled by default.
 .PP
 BSD Daemon Copyright 1988 by Marshall Kirk McKusick. All Rights Reserved.
