diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4598197..b83b9ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,7 @@ option(BUNDLE_ICONS "Bundle breeze icons" FALSE)
 set(AUTOMOC_MOC_OPTIONS -Muri=${APPLICATION_ID})
 
 # Find packages
-find_package(Qt6 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Concurrent Qml Quick Svg Sql QuickControls2 Xml Multimedia Positioning Location)
+find_package(Qt6 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Concurrent GuiPrivate Qml Quick Svg Sql QuickControls2 Xml Multimedia Positioning Location)
 find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS KIO Kirigami Prison)
 if (NOT ANDROID)
    find_package(KF6 ${KF_MIN_VERSION} OPTIONAL_COMPONENTS Crash)
