--- webvfx-master/webvfx/webvfx.pro.orig	2019-11-29 20:23:21.128757752 +0100
+++ webvfx-master/webvfx/webvfx.pro	2019-11-29 20:25:48.160623092 +0100
@@ -47,5 +47,5 @@
 
 TARGET = webvfx
 
-target.path = $$PREFIX/lib
+target.path = $$PREFIX/$$LIB_SUFFIX
 INSTALLS += target
--- webvfx-master/tools/render/render.pro.orig	2019-11-29 20:26:37.350247148 +0100
+++ webvfx-master/tools/render/render.pro	2019-11-29 20:28:10.348435675 +0100
@@ -12,7 +12,7 @@
 TARGET = webvfx_render
 LIBS += -L$$DESTDIR -lwebvfx
 
-QMAKE_RPATHDIR += $$PREFIX/lib
+QMAKE_RPATHDIR += $$PREFIX/$$LIB_SUFFIX
 
 target.path = $$PREFIX/bin
 INSTALLS += target
--- webvfx-master/mlt/mlt.pro.orig	2019-11-29 20:28:53.301986870 +0100
+++ webvfx-master/mlt/mlt.pro	2019-11-29 20:30:26.938188441 +0100
@@ -31,7 +31,7 @@
     }
     INCLUDEPATH += $$MLT_PREFIX/include
     INCLUDEPATH += $$MLT_PREFIX/include/mlt
-    LIBS += -L$$MLT_PREFIX/lib -lmlt
+    LIBS += -L$$MLT_PREFIX/$$LIB_SUFFIX -lmlt
 } else {
     CONFIG += link_pkgconfig
     PKGCONFIG += mlt-framework
@@ -46,7 +46,7 @@
     LIBS += -lglu32 -lopengl32 -lpthread
 }
 
-QMAKE_RPATHDIR += $$PREFIX/lib
+QMAKE_RPATHDIR += $$PREFIX/$$LIB_SUFFIX
 
 # Install in mlt plugins directory
 target.path = $$system(pkg-config --variable=libdir mlt-framework)/mlt
--- webvfx-master/mlt/qmelt/qmelt.pro.orig	2019-11-29 20:31:14.557799506 +0100
+++ webvfx-master/mlt/qmelt/qmelt.pro	2019-11-29 20:32:26.668717343 +0100
@@ -24,7 +24,7 @@
     }
     INCLUDEPATH += $$MLT_PREFIX/include
     INCLUDEPATH += $$MLT_PREFIX/include/mlt
-    LIBS += -L$$MLT_PREFIX/lib -lmlt
+    LIBS += -L$$MLT_PREFIX/$$LIB_SUFFIX -lmlt
 } else {
     CONFIG += link_pkgconfig
     PKGCONFIG += mlt-framework
@@ -40,7 +40,7 @@
 }
 
 TARGET = qmelt
-QMAKE_RPATHDIR += $$PREFIX/lib
+QMAKE_RPATHDIR += $$PREFIX/$$LIB_SUFFIX
 
 target.path = $$PREFIX/bin
 INSTALLS += target
--- webvfx-master/viewer/viewer.pro.orig	2019-11-29 20:32:55.169079175 +0100
+++ webvfx-master/viewer/viewer.pro	2019-11-29 20:33:25.085458980 +0100
@@ -26,7 +26,7 @@
 mac:TARGET = "WebVfx Viewer"
 LIBS += -L$$DESTDIR -lwebvfx
 
-QMAKE_RPATHDIR += $$PREFIX/lib
+QMAKE_RPATHDIR += $$PREFIX/$$LIB_SUFFIX
 
 target.path = $$PREFIX/bin
 INSTALLS += target
