From 6c014c5171a941da8a51c691b75b486c16c10833 Mon Sep 17 00:00:00 2001
From: Tatsuhiko Miyagawa <miyagawa@bulknews.net>
Date: Mon, 29 Jul 2013 11:09:28 -0700
Subject: [PATCH 1/2] Fix warnings counting for 5.18

---
 t/charnames.t | 1 +
 1 file changed, 1 insertion(+)

diff --git a/t/charnames.t b/t/charnames.t
index 4fe6fa5..7242fb3 100644
--- a/t/charnames.t
+++ b/t/charnames.t
@@ -49,6 +49,7 @@ BEGIN {
     is "\N{DIGIT SIX}", "6";
 
     {
+        @WARN = (); 
         my $v_charnames_version = pack "C*", split /\./, $charnames::VERSION;
         my $backslash_N_fatal = $v_charnames_version ge v1.33;
 
-- 
1.8.3.1

