diff -up ffms2-5.0/test/indexer.cpp.orig ffms2-5.0/test/indexer.cpp
--- ffms2-5.0/test/indexer.cpp.orig	2024-05-28 16:52:48.000000000 +0200
+++ ffms2-5.0/test/indexer.cpp	2024-09-06 23:04:28.689484139 +0200
@@ -7,34 +7,14 @@
 #include <ffms.h>
 #include <gtest/gtest.h>
 
-#include "data/test.mp4.cpp"
 #include "data/vp9_audfirst.webm.cpp"
-#include "data/qrvideo_24fps_1elist_1ctts.mov.cpp"
-#include "data/qrvideo_24fps_1elist_ends_last_bframe.mov.cpp"
-#include "data/qrvideo_24fps_1elist_noctts.mov.cpp"
-#include "data/qrvideo_24fps_2elist_elist1_dur_zero.mov.cpp"
-#include "data/qrvideo_24fps_2elist_elist1_ends_bframe.mov.cpp"
-#include "data/qrvideo_24fps_2s_3elist.mov.cpp"
-#include "data/qrvideo_24fps_3elist_1ctts.mov.cpp"
-#include "data/qrvideo_24fps_elist_starts_ctts_2ndsample.mov.cpp"
-#include "data/qrvideo_stream_shorter_than_movie.mov.cpp"
 #include "tests.h"
 
 
 namespace {
 
 const TestDataMap TestFiles[] = {
-    TEST_ENTRY("test.mp4", testmp4data),
     TEST_ENTRY("vp9_audfirst.webm", vp9audfirst),
-    TEST_ENTRY("qrvideo_24fps_1elist_1ctts.mov", qrvideo_24fps_1elist_1ctts),
-    TEST_ENTRY("qrvideo_24fps_1elist_ends_last_bframe.mov", qrvideo_24fps_1elist_ends_last_bframe),
-    TEST_ENTRY("qrvideo_24fps_1elist_noctts.mov", qrvideo_24fps_1elist_noctts),
-    TEST_ENTRY("qrvideo_24fps_2elist_elist1_dur_zero.mov", qrvideo_24fps_2elist_elist1_dur_zero),
-    TEST_ENTRY("qrvideo_24fps_2elist_elist1_ends_bframe.mov", qrvideo_24fps_2elist_elist1_ends_bframe),
-    TEST_ENTRY("qrvideo_24fps_2s_3elist.mov", qrvideo_24fps_2s_3elist),
-    TEST_ENTRY("qrvideo_24fps_3elist_1ctts.mov", qrvideo_24fps_3elist_1ctts),
-    TEST_ENTRY("qrvideo_24fps_elist_starts_ctts_2ndsample.mov", qrvideo_24fps_elist_starts_ctts_2ndsample),
-    TEST_ENTRY("qrvideo_stream_shorter_than_movie.mov", qrvideo_stream_shorter_than_movie),
 };
 
 class IndexerTest : public ::testing::TestWithParam<TestDataMap> {
