From 4d7f7a113920e001e0a11b6f5bd4eb3075f39d7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomi=20L=C3=A4hteenm=C3=A4ki?= <lihis@lihis.net>
Date: Sun, 10 Aug 2025 21:16:39 +0300
Subject: [PATCH] Require appstreamcli to validate appstream file

---
 data/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index a7e61b6..455db9b 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -39,7 +39,7 @@ appstream_file = i18n.merge_file(
   install_dir: join_paths(get_option('datadir'), 'metainfo'),
 )
 
-appstream_util = find_program('appstreamcli', required: false)
+appstream_util = find_program('appstreamcli', required: true)
 if appstream_util.found()
   test('Validate appstream file', appstream_util, args: ['validate', '--no-net', appstream_file])
 endif
-- 
2.50.1

