From 532391c6d31080dc5b7b8af81eb89e3885ceacea Mon Sep 17 00:00:00 2001
Message-ID: <532391c6d31080dc5b7b8af81eb89e3885ceacea.1769800794.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Sat, 1 Nov 2025 22:32:10 +0100
Subject: [PATCH] tests/conftest: do not call pip

Oh well ...
---
 tests/conftest.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index 44c5e4d0..371038eb 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -36,7 +36,6 @@ def install_required_packages():
     print(f'{__file__}:install_required_packages)(): Running: {command}', flush=1)
     subprocess.run(command, shell=1, check=1)
 
-install_required_packages()
 
 # Need to import pymupdf only after we've installed pymupdf-fonts above,
 # because pymupdf imports pymupdf_fonts, and copes with import failure.
-- 
2.53.0.rc1.285.g12662c75d7

