diff -up dehydrated-0.7.0/dehydrated.autowash dehydrated-0.7.0/dehydrated
--- dehydrated-0.7.0/dehydrated.autowash	2021-01-22 13:11:47.018517599 +0200
+++ dehydrated-0.7.0/dehydrated	2021-01-22 13:12:44.673042823 +0200
@@ -1734,7 +1734,7 @@ command_sign_domains() {
 
   [[ -n "${HOOK}" ]] && ("${HOOK}" "exit_hook" || echo 'exit_hook returned with non-zero exit code!' >&2)
   if [[ "${AUTO_CLEANUP}" == "yes" ]]; then
-    echo "+ Running automatic cleanup"
+    echo " + Running automatic cleanup"
     command_cleanup noinit
   fi
 
