--- futures-util-0.3.32/Cargo.toml	2026-07-06 13:40:27.242706535 -0700
+++ futures-util-0.3.32/Cargo.toml	2026-07-06 13:40:27.259482564 -0700
@@ -52,11 +52,6 @@
     "std",
     "futures-channel",
 ]
-compat = [
-    "std",
-    "futures_01",
-    "libc",
-]
 default = [
     "std",
     "async-await",
@@ -67,13 +62,6 @@
     "futures-io",
     "memchr",
 ]
-io-compat = [
-    "io",
-    "compat",
-    "tokio-io",
-    "libc",
-]
-portable-atomic = ["futures-core/portable-atomic"]
 sink = ["futures-sink"]
 std = [
     "alloc",
@@ -137,10 +125,6 @@
 version = "0.3.32"
 default-features = false
 
-[dependencies.futures_01]
-version = "0.1.25"
-optional = true
-package = "futures"
 
 [dependencies.libc]
 version = "0.2.26"
@@ -162,9 +146,6 @@
 version = "0.10.0"
 optional = true
 
-[dependencies.tokio-io]
-version = "0.1.9"
-optional = true
 
 [dev-dependencies.tokio]
 version = "0.1.11"
