From 25d9adb743c8483c0f2ae41f56c8872fdd44977f Mon Sep 17 00:00:00 2001
From: firefly-cpp <iztok@iztok-jr-fister.eu>
Date: Thu, 24 Nov 2022 19:25:37 +0100
Subject: [PATCH] Fix warning (description_file)

---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 0e63a02..a233ba3 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -6,4 +6,4 @@ extend-ignore = E203, E402, E501, E722, E741, F401, F407, W605
 profile = black
 
 [metadata]
-description-file = README.md
\ No newline at end of file
+description_file = README.md
