From: Sandro <devel@penguinpee.nl>
Date: Thu, 23 Oct 2025 17:22:42 +0200
Subject: [PATCH] Adjust test to accomodate changes in Python

TypeError wording on Python 3.14.

--- a/pandas/tests/computation/test_eval.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/computation/test_eval.py	2026-09-12 14:55:24.280607240 -0700
@@ -177,7 +177,7 @@
                 r"only list-like( or dict-like)? objects are allowed to be "
                 r"passed to (DataFrame\.)?isin\(\), you passed a "
                 r"(`|')bool(`|')",
-                "argument of type 'bool' is not iterable",
+                "argument of type 'bool' is not (a container or )?iterable",
             ]
         )
         if cmp_op in ("in", "not in") and not is_list_like(rhs):
@@ -222,7 +222,7 @@
                 r"only list-like( or dict-like)? objects are allowed to be "
                 r"passed to (DataFrame\.)?isin\(\), you passed a "
                 r"(`|')float(`|')",
-                "argument of type 'float' is not iterable",
+                "argument of type 'float' is not (a container or )?iterable",
             ]
         )
         if is_scalar(rhs) and op in skip_these:
@@ -1092,7 +1092,7 @@
             ex3 = f"1 {op} (x + 1)"
 
             if op in ("in", "not in"):
-                msg = "argument of type 'int' is not iterable"
+                msg = "argument of type 'int' is not (a container or )?iterable"
                 with pytest.raises(TypeError, match=msg):
                     pd.eval(ex, engine=engine, parser=parser)
             else:
