diff -u a/pandas/tests/plotting/frame/test_frame_color.py b/pandas/tests/plotting/frame/test_frame_color.py
--- a/pandas/tests/plotting/frame/test_frame_color.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/frame/test_frame_color.py	2026-09-13 20:45:08.095110537 -0700
@@ -3,6 +3,7 @@
 
 import numpy as np
 import pytest
+import warnings
 
 import pandas as pd
 from pandas import DataFrame
@@ -201,7 +202,8 @@
         )
         df["species"] = ["r", "r", "g", "g", "b"]
         if cmap is not None:
-            with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", UserWarning)
                 ax = df.plot.scatter(x=0, y=1, cmap=cmap, c="species")
         else:
             ax = df.plot.scatter(x=0, y=1, c="species", cmap=cmap)
diff -u a/pandas/tests/plotting/frame/test_hist_box_by.py b/pandas/tests/plotting/frame/test_hist_box_by.py
--- a/pandas/tests/plotting/frame/test_hist_box_by.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/frame/test_hist_box_by.py	2026-09-13 20:45:08.095436705 -0700
@@ -2,6 +2,7 @@
 
 import numpy as np
 import pytest
+import warnings
 
 from pandas import DataFrame
 import pandas._testing as tm
@@ -146,7 +147,8 @@
     def test_hist_plot_layout_with_by(self, by, column, layout, axes_num, hist_df):
         # GH 15079
         # _check_plot_works adds an ax so catch warning. see GH #13188
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(
                 hist_df.plot.hist, column=column, by=by, layout=layout
             )
diff -u a/pandas/tests/plotting/test_boxplot_method.py b/pandas/tests/plotting/test_boxplot_method.py
--- a/pandas/tests/plotting/test_boxplot_method.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/test_boxplot_method.py	2026-09-13 20:45:08.094584085 -0700
@@ -7,6 +7,7 @@
 
 import numpy as np
 import pytest
+import warnings
 
 from pandas import (
     DataFrame,
@@ -108,7 +109,8 @@
         )
         df["X"] = Series(["A", "A", "A", "A", "A", "B", "B", "B", "B", "B"])
         df["Y"] = Series(["A"] * 10)
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(df.boxplot, by="X")
 
     def test_boxplot_legacy2_with_ax(self):
@@ -143,7 +145,8 @@
         df["Y"] = Series(["A"] * 10)
         # Multiple columns with an ax argument should use same figure
         fig, ax = mpl.pyplot.subplots()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.boxplot(
                 column=["Col1", "Col2"], by="X", ax=ax, return_type="axes"
             )
@@ -419,7 +422,8 @@
 class TestDataFrameGroupByPlots:
     def test_boxplot_legacy1(self, hist_df):
         grouped = hist_df.groupby(by="gender")
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(grouped.boxplot, return_type="axes")
         _check_axes_shape(list(axes.values), axes_num=2, layout=(1, 2))
 
@@ -436,7 +440,8 @@
             index=MultiIndex.from_tuples(tuples),
         )
         grouped = df.groupby(level=1)
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(grouped.boxplot, return_type="axes")
         _check_axes_shape(list(axes.values), axes_num=10, layout=(4, 3))
 
@@ -579,7 +584,8 @@
     ):
         df = hist_df
         # _check_plot_works adds an ax so catch warning. see GH #13188 GH 6769
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(
                 df.groupby(gb_key).boxplot, column="height", return_type="dict"
             )
@@ -612,7 +618,8 @@
     @pytest.mark.parametrize("cols", [2, -1])
     def test_grouped_box_layout_works(self, hist_df, cols):
         df = hist_df
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(
                 df.groupby("category").boxplot,
                 column="height",
@@ -651,7 +658,8 @@
         # passes multiple axes to plot, hist or boxplot
         # location should be changed if other test is added
         # which has earlier alphabetical order
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _, axes = mpl.pyplot.subplots(2, 2)
             df.groupby("category").boxplot(column="height", return_type="axes", ax=axes)
             _check_axes_shape(mpl.pyplot.gcf().axes, axes_num=4, layout=(2, 2))
@@ -661,7 +669,8 @@
         # GH 6970, GH 7069
         df = hist_df
         fig, axes = mpl.pyplot.subplots(2, 3)
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             returned = df.boxplot(
                 column=["height", "weight", "category"],
                 by="gender",
@@ -674,7 +683,8 @@
         assert returned[0].figure is fig
 
         # draw on second row
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             returned = df.groupby("classroom").boxplot(
                 column=["height", "weight", "category"], return_type="axes", ax=axes[1]
             )
@@ -691,7 +701,8 @@
         with pytest.raises(ValueError, match=msg):
             fig, axes = mpl.pyplot.subplots(2, 3)
             # pass different number of axes from required
-            with tm.assert_produces_warning(UserWarning):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", UserWarning)
                 axes = df.groupby("classroom").boxplot(ax=axes)
 
     def test_fontsize(self):
diff -u a/pandas/tests/plotting/test_hist_method.py b/pandas/tests/plotting/test_hist_method.py
--- a/pandas/tests/plotting/test_hist_method.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/test_hist_method.py	2026-09-13 20:45:08.093393002 -0700
@@ -3,6 +3,7 @@
 
 import numpy as np
 import pytest
+import warnings
 
 from pandas import (
     DataFrame,
@@ -45,7 +46,8 @@
     @pytest.mark.parametrize("kwargs", [{}, {"bins": 5}])
     def test_hist_legacy_kwargs_warning(self, ts, kwargs):
         # _check_plot_works adds an ax so catch warning. see GH #13188
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(ts.hist, by=ts.index.month, **kwargs)
 
     def test_hist_legacy_ax(self, ts):
@@ -107,7 +109,8 @@
         # _check_plot_works adds an `ax` kwarg to the method call
         # so we get a warning about an axis being cleared, even
         # though we don't explicing pass one, see GH #13188
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(df.height.hist, by=getattr(df, by), layout=layout)
         _check_axes_shape(axes, axes_num=axes_num, layout=res_layout)
 
@@ -256,7 +259,8 @@
 class TestDataFramePlots:
     @pytest.mark.slow
     def test_hist_df_legacy(self, hist_df):
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(hist_df.hist)
 
     @pytest.mark.slow
@@ -271,7 +275,8 @@
                 dtype=np.int64,
             )
         )
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(df.hist, grid=False)
         _check_axes_shape(axes, axes_num=3, layout=(2, 2))
         assert not axes[1, 1].get_visible()
@@ -295,7 +300,8 @@
                 dtype=np.int64,
             )
         )
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(df.hist, layout=(4, 2))
         _check_axes_shape(axes, axes_num=6, layout=(4, 2))
 
@@ -316,7 +322,8 @@
         # make sure sharex, sharey is handled
         # handle figsize arg
         # check bins argument
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             _check_plot_works(df.hist, **kwargs)
 
     @pytest.mark.slow
@@ -850,7 +857,8 @@
     @pytest.mark.slow
     def test_grouped_hist_layout_warning(self, hist_df):
         df = hist_df
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(
                 df.hist, column="height", by=df.gender, layout=(2, 1)
             )
@@ -871,7 +879,8 @@
     def test_grouped_hist_layout_by_warning(self, hist_df, kwargs):
         df = hist_df
         # GH 6769
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(df.hist, by="classroom", **kwargs)
         _check_axes_shape(axes, axes_num=3, layout=(2, 2))
 
diff -u a/pandas/tests/plotting/test_misc.py b/pandas/tests/plotting/test_misc.py
--- a/pandas/tests/plotting/test_misc.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/test_misc.py	2026-09-13 20:45:08.093981484 -0700
@@ -3,6 +3,7 @@
 
 import numpy as np
 import pytest
+import warnings
 
 import pandas.util._test_decorators as td
 
@@ -158,7 +159,8 @@
         df = DataFrame(np.random.default_rng(2).standard_normal((100, 3)))
 
         # we are plotting multiples on a sub-plot
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(
                 scatter_matrix,
                 frame=df,
@@ -184,7 +186,8 @@
         df[0] = (df[0] - 2) / 3
 
         # we are plotting multiples on a sub-plot
-        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = _check_plot_works(
                 scatter_matrix,
                 frame=df,
