--- a/src/solvers/cdcl/delegate.c
+++ b/src/solvers/cdcl/delegate.c
@@ -21,11 +21,11 @@
 #include <stdio.h>
 
 #ifdef HAVE_CADICAL
-#include "ccadical.h"
+#include <ccadical.h>
 #endif
 
 #ifdef HAVE_CRYPTOMINISAT
-#include "cryptominisat5/cmsat_c.h"
+#include <cryptominisat5/cryptominisat_c.h>
 #endif
 
 #ifdef HAVE_KISSAT
@@ -308,44 +308,44 @@ static void cadical_as_delegate(delegate
 
 #if HAVE_CRYPTOMINISAT
 static void cryptominisat_add_empty_clause(void *solver) {
-  cmsat_add_clause(solver, NULL, 0);
+  cmsat_add_clause((SATSolver *)solver, NULL, 0);
 }
 
 static void cryptominisat_add_unit_clause(void *solver, literal_t l) {
-  uint32_t c[1];
+  c_Lit c[1];
 
   assert(l >= 0);
-  c[0] = l;
-  cmsat_add_clause(solver, c, 1);
+  c[0].x = l;
+  cmsat_add_clause((SATSolver *)solver, c, 1);
 }
 
 static void cryptominisat_add_binary_clause(void *solver, literal_t l1, literal_t l2) {
-  uint32_t c[2];
+  c_Lit c[2];
 
   assert(l1 >= 0 && l2 >= 0);
-  c[0] = l1;
-  c[1] = l2;
-  cmsat_add_clause(solver, c, 2);
+  c[0].x = l1;
+  c[1].x = l2;
+  cmsat_add_clause((SATSolver *)solver, c, 2);
 }
 
 static void cryptominisat_add_ternary_clause(void *solver, literal_t l1, literal_t l2, literal_t l3) {
-  uint32_t c[3];
+  c_Lit c[3];
 
   assert(l1 >= 0 && l2 >= 0 && l3 >= 0);
-  c[0] = l1;
-  c[1] = l2;
-  c[2] = l3;
-  cmsat_add_clause(solver, c, 3);
+  c[0].x = l1;
+  c[1].x = l2;
+  c[2].x = l3;
+  cmsat_add_clause((SATSolver *)solver, c, 3);
 }
 
 static void cryptominisat_add_clause(void *solver, uint32_t n, literal_t *a) {
-  cmsat_add_clause(solver, (uint32_t*) a, n);
+  cmsat_add_clause((SATSolver *)solver, (c_Lit *)a, n);
 }
 
 static smt_status_t cryptominisat_check(void *solver) {
-  switch (cmsat_solve(solver)) {
-  case CMSAT_SAT: return YICES_STATUS_SAT;
-  case CMSAT_UNSAT: return YICES_STATUS_UNSAT;
+  switch (cmsat_solve((SATSolver *)solver).x) {
+  case L_TRUE: return YICES_STATUS_SAT;
+  case L_FALSE: return YICES_STATUS_UNSAT;
   default: return YICES_STATUS_UNKNOWN;
   }
 }
@@ -356,28 +356,23 @@ static smt_status_t cryptominisat_check(
  * that cryptominisat always produces a full truth assignment.
  */
 static bval_t cryptominisat_get_value(void *solver, bvar_t x) {
-  switch (cmsat_var_value(solver, x)) {
-  case CMSAT_VAL_TRUE:
+  slice_lbool model = cmsat_get_model((SATSolver *)solver);
+  if ((size_t)x < model.num_vals && model.vals[x].x == L_TRUE)
     return VAL_TRUE;
-
-  case CMSAT_VAL_FALSE:
-  default:
-    return VAL_FALSE;
-  }
+  return VAL_FALSE;
 }
 
-static void cryptominisat_set_verbosity(void *solver, uint32_t level) {
+static void cryptominisat_set_verbosity(void *solver __attribute__((unused)), uint32_t level __attribute__((unused))) {
   // verbosity 0 --> quiet (this is the default)
-  cmsat_set_verbosity(solver, level);
 }
 
 static void cryptominisat_delete(void *solver) {
-  cmsat_free_solver(solver);
+  cmsat_free((SATSolver *)solver);
 }
 
 static void cryptominisat_as_delegate(delegate_t *d, uint32_t nvars) {
-  d->solver = cmsat_new_solver();
-  cmsat_new_vars(d->solver, nvars);
+  d->solver = cmsat_new();
+  cmsat_new_vars((SATSolver *)d->solver, nvars);
   init_ivector(&d->buffer, 0); // not used
   d->add_empty_clause = cryptominisat_add_empty_clause;
   d->add_unit_clause = cryptominisat_add_unit_clause;
