--- a/jsmin/test.py	2022-06-24 14:04:01.074567690 +0300
+++ b/jsmin/test.py	2022-06-24 14:11:32.777776459 +0300
@@ -337,8 +337,8 @@
         self.assertMinified("return//comment...\r\na", "return\na")
 
     def testSingleComment2(self):
-        self.assertMinified('x.replace(/\//, "_")// slash to underscore',
-                'x.replace(/\//,"_")')
+        self.assertMinified('x.replace(/\\//, "_")// slash to underscore',
+                'x.replace(/\\//,"_")')
 
     def testSlashesNearComments(self):
         original = '''
@@ -383,8 +383,8 @@
         self.assertMinified(original, original)
 
     def test_brackets_around_slashed_regex(self):
-        original = 'function a() { /\//.test("a") }'
-        expected = 'function a(){/\//.test("a")}'
+        original = 'function a() { /\\//.test("a") }'
+        expected = 'function a(){/\\//.test("a")}'
         self.assertMinified(original, expected)
 
     def test_angular_1(self):
@@ -455,12 +455,12 @@
     def test_issue_bitbucket_16(self):
         original = """
             f = function() {
-                return /DataTree\/(.*)\//.exec(this._url)[1];
+                return /DataTree\\/(.*)\\//.exec(this._url)[1];
             }
         """
         self.assertMinified(
             original,
-            'f=function(){return /DataTree\/(.*)\//.exec(this._url)[1];}')
+            'f=function(){return /DataTree\\/(.*)\\//.exec(this._url)[1];}')
 
     def test_issue_bitbucket_17(self):
         original = "// hi\n/^(get|post|head|put)$/i.test('POST')"
@@ -470,7 +470,7 @@
     def test_issue_6(self):
         original = '''
             respond.regex = {
-                comments: /\/\*[^*]*\*+([^/][^*]*\*+)*\//gi,
+                comments: /\\/\\*[^*]*\\*+([^/][^*]*\\*+)*\\//gi,
                 urls: 'whatever'
             };
         '''
@@ -548,16 +548,16 @@
 
     def test_issue_12_re_nl_if(self):
         original = '''
-            var re = /\d{4}/
+            var re = /\\d{4}/
             if (1) { console.log(2); }'''
         self.assertMinified(
-            original, 'var re=/\d{4}/\nif(1){console.log(2);}')
+            original, 'var re=/\\d{4}/\nif(1){console.log(2);}')
 
     def test_issue_12_re_nl_other(self):
         original = '''
-            var re = /\d{4}/
+            var re = /\\d{4}/
             g = 10'''
-        self.assertMinified(original , 'var re=/\d{4}/\ng=10')
+        self.assertMinified(original , 'var re=/\\d{4}/\ng=10')
 
     def test_preserve_copyright(self):
         original = '''
