--- eccodes-2.45.0.unchanged/setup.py	2026-01-15 17:26:54.000000000 +0100
+++ eccodes-2.45.0.p1/setup.py	2026-01-28 10:34:45.210872341 +0100
@@ -110,6 +110,8 @@ setuptools.setup(
     license="Apache License Version 2.0",
     url="https://github.com/ecmwf/eccodes-python",
     packages=setuptools.find_packages(),
+    setup_requires=["cffi"],
+    cffi_modules=["builder.py:ffibuilder"],
     include_package_data=True,
     package_data={"": shared_files},
     install_requires=install_requires,
