Index: neovim-0.12.2/src/nvim/ex_cmds.lua
===================================================================
--- neovim-0.12.2.orig/src/nvim/ex_cmds.lua	2026-04-22 16:05:55.000000000 +0200
+++ neovim-0.12.2/src/nvim/ex_cmds.lua	2026-05-12 07:38:24.458771680 +0200
@@ -1,4 +1,4 @@
-local bit = require 'bit'
+local bit = require 'bit32'
 
 local M = {}
 
Index: neovim-0.12.2/runtime/lua/vim/lsp/_watchfiles.lua
===================================================================
--- neovim-0.12.2.orig/runtime/lua/vim/lsp/_watchfiles.lua	2026-04-22 16:05:55.000000000 +0200
+++ neovim-0.12.2/runtime/lua/vim/lsp/_watchfiles.lua	2026-05-12 07:38:24.459095537 +0200
@@ -1,4 +1,4 @@
-local bit = require('bit')
+local bit = require('bit32')
 local glob = vim.glob
 local watch = vim._watch
 local protocol = require('vim.lsp.protocol')
Index: neovim-0.12.2/runtime/lua/vim/lsp/semantic_tokens.lua
===================================================================
--- neovim-0.12.2.orig/runtime/lua/vim/lsp/semantic_tokens.lua	2026-04-22 16:05:55.000000000 +0200
+++ neovim-0.12.2/runtime/lua/vim/lsp/semantic_tokens.lua	2026-05-12 07:38:40.731504227 +0200
@@ -1,5 +1,5 @@
 local api = vim.api
-local bit = require('bit')
+local bit = require('bit32')
 local util = require('vim.lsp.util')
 local Range = require('vim.treesitter._range')
 local uv = vim.uv
