--- livecd-tools-31.0/Makefile.py312	2022-06-15 20:49:24.000000000 +0900
+++ livecd-tools-31.0/Makefile	2023-07-27 14:10:19.460042745 +0900
@@ -14,7 +14,7 @@ define COMPILE_PYTHON
 	$(PYTHON_PROGRAM) -c "import compileall as c; c.compile_dir('$(1)', force=1)"
 	$(PYTHON_PROGRAM) -O -c "import compileall as c; c.compile_dir('$(1)', force=1)"
 endef
-PYTHONDIR := $(shell $(PYTHON_PROGRAM) -c "from __future__ import print_function; from distutils.sysconfig import get_python_lib; print(get_python_lib())")
+PYTHONDIR := $(shell $(PYTHON_PROGRAM) -c "import sysconfig; print(sysconfig.get_path('purelib', vars={'platbase': '/usr', 'base': '/usr'}))")
 
 all: 
 
