From ec139833eba2781507cf32d9df30772138a76829 Mon Sep 17 00:00:00 2001
From: Simon Larsen <simonhffh@gmail.com>
Date: Wed, 11 May 2016 01:08:29 +0200
Subject: [PATCH 1/2] Fixed love.conf being doubly defined

---
 conf.lua | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/conf.lua b/conf.lua
index db46228..8776da4 100644
--- a/conf.lua
+++ b/conf.lua
@@ -1,26 +1,3 @@
-function love.conf(t)
-    t.identity = "mrrescue"
-    t.version = "0.10.0"
-    t.console = false
-
-    t.window.title = "Mr. Rescue"
-    t.window.icon = nil
-    t.window.width = 256*3
-    t.window.height = 200*3
-    t.window.borderless = false
-    t.window.resizable = false
-    t.window.fullscreen = false
-    t.window.fullscreentype = "exclusive"
-    t.window.vsync = true
-    t.window.fsaa = 0
-    t.window.display = 1
-    t.window.highdpi = false
-    t.window.srgb = false
-
-    t.modules.physics = false
-    t.modules.mouse = false
-end
-
 function love.conf(t)
     t.identity = "mrrescue"
     t.version = "0.10.0"
-- 
2.14.4

