diff -up IQmol3-3.2.0/src/Amber/ConfigDialog.C.qmsgbox IQmol3-3.2.0/src/Amber/ConfigDialog.C
--- IQmol3-3.2.0/src/Amber/ConfigDialog.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Amber/ConfigDialog.C	2025-08-05 09:10:15.999639213 +0000
@@ -20,7 +20,7 @@
    
 ********************************************************************************/
 
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Preferences.h"
 #include "Amber/ConfigDialog.h"
 
diff -up IQmol3-3.2.0/src/Amber/ParametrizeMoleculeDialog.C.qmsgbox IQmol3-3.2.0/src/Amber/ParametrizeMoleculeDialog.C
--- IQmol3-3.2.0/src/Amber/ParametrizeMoleculeDialog.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Amber/ParametrizeMoleculeDialog.C	2025-08-05 09:10:18.283598889 +0000
@@ -22,7 +22,7 @@
 
 #include "Layer/MoleculeLayer.h"
 #include "Layer/SystemLayer.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Preferences.h"
 #include "Amber/ConfigDialog.h"
 #include "Amber/ParametrizeMoleculeDialog.h"
diff -up IQmol3-3.2.0/src/Amber/SystemBuilderDialog.C.qmsgbox IQmol3-3.2.0/src/Amber/SystemBuilderDialog.C
--- IQmol3-3.2.0/src/Amber/SystemBuilderDialog.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Amber/SystemBuilderDialog.C	2025-08-05 09:10:11.932711018 +0000
@@ -23,7 +23,7 @@
 #include "Layer/SystemLayer.h"
 #include "Layer/MoleculeLayer.h"
 #include "Layer/AtomLayer.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Preferences.h"
 #include "Amber/ConfigDialog.h"
 #include "Amber/SystemBuilderDialog.h"
diff -up IQmol3-3.2.0/src/Layer/EfpFragmentLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/EfpFragmentLayer.C
--- IQmol3-3.2.0/src/Layer/EfpFragmentLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/EfpFragmentLayer.C	2025-08-05 09:03:22.166429084 +0000
@@ -26,7 +26,7 @@
 #include "LayerFactory.h"
 #include "Util/Preferences.h"
 #include "Math/EulerAngles.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include <QFileInfo>
 #include <QFile>
 #include <QSet>
diff -up IQmol3-3.2.0/src/Layer/GeminalOrbitalsLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/GeminalOrbitalsLayer.C
--- IQmol3-3.2.0/src/Layer/GeminalOrbitalsLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/GeminalOrbitalsLayer.C	2025-08-05 09:03:22.166723893 +0000
@@ -30,7 +30,7 @@
 #include "Grid/MarchingCubes.h"
 #include "Grid/MeshDecimator.h"
 #include "Grid/BoundingBoxDialog.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/QsLog.h"
 
 #include "QGLViewer/vec.h"
diff -up IQmol3-3.2.0/src/Layer/GroupLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/GroupLayer.C
--- IQmol3-3.2.0/src/Layer/GroupLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/GroupLayer.C	2025-08-05 09:03:22.167015487 +0000
@@ -27,7 +27,7 @@
 #include "Parser/ParseFile.h"
 #include "LayerFactory.h"
 #include "Math/Align.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/QsLog.h"
 #include "Viewer/UndoCommands.h"
 
diff -up IQmol3-3.2.0/src/Layer/MoleculeLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/MoleculeLayer.C
--- IQmol3-3.2.0/src/Layer/MoleculeLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/MoleculeLayer.C	2025-08-05 09:03:22.167456529 +0000
@@ -60,7 +60,7 @@
 #include "Viewer/UndoCommands.h"
 
 #include "Util/QsLog.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Constants.h"
 #include "Util/Preferences.h"
 #include "Process/JobInfo.h" 
diff -up IQmol3-3.2.0/src/Layer/OrbitalsLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/OrbitalsLayer.C
--- IQmol3-3.2.0/src/Layer/OrbitalsLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/OrbitalsLayer.C	2025-08-05 09:03:22.167919389 +0000
@@ -30,7 +30,7 @@
 #include "Grid/BoundingBoxDialog.h"
 #include "Data/SurfaceType.h"
 #include "Data/SurfaceInfo.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Data/Density.h"
 #include "QGLViewer/vec.h"
 #include "Data/Surface.h"
diff -up IQmol3-3.2.0/src/Layer/SurfaceLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/SurfaceLayer.C
--- IQmol3-3.2.0/src/Layer/SurfaceLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/SurfaceLayer.C	2025-08-05 09:03:22.168179210 +0000
@@ -28,7 +28,7 @@
 #include "Util/QsLog.h"
 #include "QGLViewer/vec.h"
 #include "Grid/MeshDecimator.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Color.h"
 #include <QColorDialog>
 #include <cmath>
diff -up IQmol3-3.2.0/src/Layer/SystemLayer.C.qmsgbox IQmol3-3.2.0/src/Layer/SystemLayer.C
--- IQmol3-3.2.0/src/Layer/SystemLayer.C.qmsgbox	2025-06-10 11:23:43.000000000 +0000
+++ IQmol3-3.2.0/src/Layer/SystemLayer.C	2025-08-05 09:03:22.168447206 +0000
@@ -31,7 +31,7 @@
 #include "Data/Bank.h"
 #include "Viewer/UndoCommands.h"
 #include "Util/QsLog.h"
-#include "Util/QMsgBox.h"
+#include "QMsgBox.h"
 #include "Util/Preferences.h"
 #include "Process/JobInfo.h"
 
