--- runtime/Jlib/foreign.java.orig	2021-03-22 13:41:14.652144132 -0600
+++ runtime/Jlib/foreign.java	2021-03-22 13:46:10.822676017 -0600
@@ -5597,13 +5597,28 @@ public final class foreign
    public static final int SIGBUS = 7;
    public static final int SIGFPE = 8;
    public static final int SIGKILL = 9;
+   public static final int SIGUSR1 = 10;
    public static final int SIGSEGV = 11;
+   public static final int SIGUSR2 = 12;
    public static final int SIGPIPE = 13;
    public static final int SIGALRM = 14;
    public static final int SIGTERM = 15;
-   public static final int SIGUSR1 = 16;
-   public static final int SIGUSR2 = 17;
-   public static final int SIGWINCH = 20;
+   public static final int SIGSTKFLT = 16;
+   public static final int SIGCHLD = 17;
+   public static final int SIGCONT = 18;
+   public static final int SIGSTOP = 19;
+   public static final int SIGTSTP = 20;
+   public static final int SIGTTIN = 21;
+   public static final int SIGTTOU = 22;
+   public static final int SIGURG = 23;
+   public static final int SIGXCPU = 24;
+   public static final int SIGXFSZ = 25;
+   public static final int SIGVTALRM = 26;
+   public static final int SIGPROF = 27;
+   public static final int SIGWINCH = 28;
+   public static final int SIGPOLL = 29;
+   public static final int SIGPWR = 30;
+   public static final int SIGSYS = 31;
 
    public static int sigsetmask(int n)
       {
