diff -rupN qsynth-0.6.2.org/src/qsynthOptions.cpp qsynth-0.6.2/src/qsynthOptions.cpp
--- qsynth-0.6.2.org/src/qsynthOptions.cpp	2020-03-24 08:13:11.479573266 -0400
+++ qsynth-0.6.2/src/qsynthOptions.cpp	2020-06-21 11:48:28.598470716 -0400
@@ -66,7 +66,7 @@ void qsynthOptions::loadOptions (void)
 {
 	// Load defaults...
 	m_settings.beginGroup("/Defaults");
-	sSoundFontDir  = m_settings.value("/SoundFontDir").toString();
+	sSoundFontDir  = m_settings.value("/SoundFontDir", "@DATADIR@/soundfonts").toString();
 	bPresetPreview = m_settings.value("/PresetPreview", false).toBool();
 	m_settings.endGroup();
 
@@ -525,7 +525,7 @@ void qsynthOptions::loadSetup ( qsynthSe
 	pSetup->iAudioBufSize    = m_settings.value("/AudioBufSize", 512).toInt();
 	pSetup->iAudioBufCount   = m_settings.value("/AudioBufCount", 8).toInt();
 #else
-	pSetup->iAudioBufSize    = m_settings.value("/AudioBufSize", 64).toInt();
+	pSetup->iAudioBufSize    = m_settings.value("/AudioBufSize", 1024).toInt();
 	pSetup->iAudioBufCount   = m_settings.value("/AudioBufCount", 2).toInt();
 #endif
 	pSetup->sMidiName        = m_settings.value("/AlsaName", "pid").toString();
@@ -567,7 +567,9 @@ void qsynthOptions::loadSetup ( qsynthSe
 	m_settings.beginGroup("/SoundFonts");
 	const QString sSoundFontPrefix  = "/SoundFont%1";
 	const QString sBankOffsetPrefix = "/BankOffset%1";
+	const QString defaultSoundfont = "@DATADIR@/soundfonts/default.sf2";
 	int i = 0;
+	int f = 0;
 	for (;;) {
 		++i;
 		QString sSoundFont  = m_settings.value(sSoundFontPrefix.arg(i)).toString();
@@ -576,6 +578,11 @@ void qsynthOptions::loadSetup ( qsynthSe
 				break;
 		pSetup->soundfonts.append(sSoundFont);
 		pSetup->bankoffsets.append(sBankOffset);
+		f++;
+	}
+	if (f == 0) {
+		pSetup->soundfonts.append(defaultSoundfont);
+		pSetup->bankoffsets.append("0");
 	}
 	m_settings.endGroup();
 
