diff -pruN kasumi-2.5.orig/KasumiMainWindow.cxx kasumi-2.5/KasumiMainWindow.cxx
--- kasumi-2.5.orig/KasumiMainWindow.cxx	2024-09-05 15:30:41.601889850 +0900
+++ kasumi-2.5/KasumiMainWindow.cxx	2024-09-05 15:42:09.527087743 +0900
@@ -376,7 +376,22 @@ void KasumiMainWindow::quit(){
                                        "before Kasumi quits?"));
     switch(gtk_dialog_run (GTK_DIALOG (dialog))){
     case GTK_RESPONSE_YES:
-      dictionary->store();
+      try {
+        dictionary->store();
+      } catch (KasumiException e) {
+        /* Do not call gtk_message_dialog_new without parent window as handleException() does.
+	 * It causes a crash in GObject.
+	 * https://bugzilla.redhat.com/show_bug.cgi?id=1936817
+	 */
+        GtkWidget *dlg = gtk_message_dialog_new (GTK_WINDOW(mWindow),
+                                                 GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                 GTK_MESSAGE_ERROR,
+                                                 GTK_BUTTONS_CLOSE,
+                                                 "%s",
+                                                 e.getMessage().c_str());
+        gtk_dialog_run (GTK_DIALOG (dlg));
+        gtk_widget_destroy (dlg);
+      }
       break;
     case GTK_RESPONSE_NO:
       break;
