diff -ruN tinyexr-1.0.7-orig/tinyexr.h tinyexr-1.0.7/tinyexr.h
--- tinyexr-1.0.7-orig/tinyexr.h	2023-08-03 12:35:04.000000000 -0400
+++ tinyexr-1.0.7/tinyexr.h	2023-09-08 22:01:01.125646644 -0300
@@ -604,6 +604,12 @@
 #ifndef TINYEXR_IMPLEMENTATION_DEFINED
 #define TINYEXR_IMPLEMENTATION_DEFINED
 
+#ifdef tinyexr_EXPORTS
+#define TINYEXR_PUBLIC __attribute__((visibility ("default")))
+#else
+#define TINYEXR_PUBLIC
+#endif
+
 #ifdef _WIN32
 
 #ifndef WIN32_LEAN_AND_MEAN
@@ -6136,7 +6142,7 @@
 
 }  // namespace tinyexr
 
-int EXRLayers(const char *filename, const char **layer_names[], int *num_layers,
+TINYEXR_PUBLIC int EXRLayers(const char *filename, const char **layer_names[], int *num_layers,
               const char **err) {
   EXRVersion exr_version;
   EXRHeader exr_header;
@@ -6181,13 +6187,13 @@
   return TINYEXR_SUCCESS;
 }
 
-int LoadEXR(float **out_rgba, int *width, int *height, const char *filename,
+TINYEXR_PUBLIC int LoadEXR(float **out_rgba, int *width, int *height, const char *filename,
             const char **err) {
   return LoadEXRWithLayer(out_rgba, width, height, filename,
                           /* layername */ NULL, err);
 }
 
-int LoadEXRWithLayer(float **out_rgba, int *width, int *height,
+TINYEXR_PUBLIC int LoadEXRWithLayer(float **out_rgba, int *width, int *height,
                      const char *filename, const char *layername,
                      const char **err) {
   if (out_rgba == NULL) {
@@ -6435,7 +6441,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int IsEXR(const char *filename) {
+TINYEXR_PUBLIC int IsEXR(const char *filename) {
   EXRVersion exr_version;
 
   int ret = ParseEXRVersionFromFile(&exr_version, filename);
@@ -6446,7 +6452,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int IsEXRFromMemory(const unsigned char *memory, size_t size) {
+TINYEXR_PUBLIC int IsEXRFromMemory(const unsigned char *memory, size_t size) {
   EXRVersion exr_version;
 
   int ret = ParseEXRVersionFromMemory(&exr_version, memory, size);
@@ -6457,7 +6463,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int ParseEXRHeaderFromMemory(EXRHeader *exr_header, const EXRVersion *version,
+TINYEXR_PUBLIC int ParseEXRHeaderFromMemory(EXRHeader *exr_header, const EXRVersion *version,
                              const unsigned char *memory, size_t size,
                              const char **err) {
   if (memory == NULL || exr_header == NULL) {
@@ -6517,7 +6523,7 @@
   return ret;
 }
 
-int LoadEXRFromMemory(float **out_rgba, int *width, int *height,
+TINYEXR_PUBLIC int LoadEXRFromMemory(float **out_rgba, int *width, int *height,
                       const unsigned char *memory, size_t size,
                       const char **err) {
   if (out_rgba == NULL || memory == NULL) {
@@ -6918,7 +6924,7 @@
   bool valid() const { return data; }
 };
 
-int LoadEXRImageFromFile(EXRImage *exr_image, const EXRHeader *exr_header,
+TINYEXR_PUBLIC int LoadEXRImageFromFile(EXRImage *exr_image, const EXRHeader *exr_header,
                          const char *filename, const char **err) {
   if (exr_image == NULL) {
     tinyexr::SetErrorMessage("Invalid argument for LoadEXRImageFromFile", err);
@@ -6941,7 +6947,7 @@
                                 err);
 }
 
-int LoadEXRImageFromMemory(EXRImage *exr_image, const EXRHeader *exr_header,
+TINYEXR_PUBLIC int LoadEXRImageFromMemory(EXRImage *exr_image, const EXRHeader *exr_header,
                            const unsigned char *memory, const size_t size,
                            const char **err) {
   if (exr_image == NULL || memory == NULL ||
@@ -7986,13 +7992,13 @@
 
 } // tinyexr
 
-size_t SaveEXRImageToMemory(const EXRImage* exr_image,
+TINYEXR_PUBLIC size_t SaveEXRImageToMemory(const EXRImage* exr_image,
                              const EXRHeader* exr_header,
                              unsigned char** memory_out, const char** err) {
   return tinyexr::SaveEXRNPartImageToMemory(exr_image, &exr_header, 1, memory_out, err);
 }
 
-int SaveEXRImageToFile(const EXRImage *exr_image, const EXRHeader *exr_header,
+TINYEXR_PUBLIC int SaveEXRImageToFile(const EXRImage *exr_image, const EXRHeader *exr_header,
                        const char *filename, const char **err) {
   if (exr_image == NULL || filename == NULL ||
       exr_header->compression_type < 0) {
@@ -8062,7 +8068,7 @@
   return TINYEXR_SUCCESS;
 }
 
-size_t SaveEXRMultipartImageToMemory(const EXRImage* exr_images,
+TINYEXR_PUBLIC size_t SaveEXRMultipartImageToMemory(const EXRImage* exr_images,
                                      const EXRHeader** exr_headers,
                                      unsigned int num_parts,
                                      unsigned char** memory_out, const char** err) {
@@ -8075,7 +8081,7 @@
   return tinyexr::SaveEXRNPartImageToMemory(exr_images, exr_headers, num_parts, memory_out, err);
 }
 
-int SaveEXRMultipartImageToFile(const EXRImage* exr_images,
+TINYEXR_PUBLIC int SaveEXRMultipartImageToFile(const EXRImage* exr_images,
                                 const EXRHeader** exr_headers,
                                 unsigned int num_parts,
                                 const char* filename,
@@ -8132,7 +8138,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int LoadDeepEXR(DeepImage *deep_image, const char *filename, const char **err) {
+TINYEXR_PUBLIC int LoadDeepEXR(DeepImage *deep_image, const char *filename, const char **err) {
   if (deep_image == NULL) {
     tinyexr::SetErrorMessage("Invalid argument for LoadDeepEXR", err);
     return TINYEXR_ERROR_INVALID_ARGUMENT;
@@ -8486,7 +8492,7 @@
   return TINYEXR_SUCCESS;
 }
 
-void InitEXRImage(EXRImage *exr_image) {
+TINYEXR_PUBLIC void InitEXRImage(EXRImage *exr_image) {
   if (exr_image == NULL) {
     return;
   }
@@ -8504,14 +8510,14 @@
   exr_image->num_tiles = 0;
 }
 
-void FreeEXRErrorMessage(const char *msg) {
+TINYEXR_PUBLIC void FreeEXRErrorMessage(const char *msg) {
   if (msg) {
     free(reinterpret_cast<void *>(const_cast<char *>(msg)));
   }
   return;
 }
 
-void InitEXRHeader(EXRHeader *exr_header) {
+TINYEXR_PUBLIC void InitEXRHeader(EXRHeader *exr_header) {
   if (exr_header == NULL) {
     return;
   }
@@ -8519,7 +8525,7 @@
   memset(exr_header, 0, sizeof(EXRHeader));
 }
 
-int FreeEXRHeader(EXRHeader *exr_header) {
+TINYEXR_PUBLIC int FreeEXRHeader(EXRHeader *exr_header) {
   if (exr_header == NULL) {
     return TINYEXR_ERROR_INVALID_ARGUMENT;
   }
@@ -8551,7 +8557,7 @@
   return TINYEXR_SUCCESS;
 }
 
-void EXRSetNameAttr(EXRHeader* exr_header, const char* name) {
+TINYEXR_PUBLIC void EXRSetNameAttr(EXRHeader* exr_header, const char* name) {
   if (exr_header == NULL) {
     return;
   }
@@ -8564,7 +8570,7 @@
   }
 }
 
-int EXRNumLevels(const EXRImage* exr_image) {
+TINYEXR_PUBLIC int EXRNumLevels(const EXRImage* exr_image) {
   if (exr_image == NULL) return 0;
   if(exr_image->images) return 1; // scanlines
   int levels = 1;
@@ -8573,7 +8579,7 @@
   return levels;
 }
 
-int FreeEXRImage(EXRImage *exr_image) {
+TINYEXR_PUBLIC int FreeEXRImage(EXRImage *exr_image) {
   if (exr_image == NULL) {
     return TINYEXR_ERROR_INVALID_ARGUMENT;
   }
@@ -8610,7 +8616,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int ParseEXRHeaderFromFile(EXRHeader *exr_header, const EXRVersion *exr_version,
+TINYEXR_PUBLIC int ParseEXRHeaderFromFile(EXRHeader *exr_header, const EXRVersion *exr_version,
                            const char *filename, const char **err) {
   if (exr_header == NULL || exr_version == NULL || filename == NULL) {
     tinyexr::SetErrorMessage("Invalid argument for ParseEXRHeaderFromFile",
@@ -8628,7 +8634,7 @@
                                   err);
 }
 
-int ParseEXRMultipartHeaderFromMemory(EXRHeader ***exr_headers,
+TINYEXR_PUBLIC int ParseEXRMultipartHeaderFromMemory(EXRHeader ***exr_headers,
                                       int *num_headers,
                                       const EXRVersion *exr_version,
                                       const unsigned char *memory, size_t size,
@@ -8740,7 +8746,7 @@
   return retcode;
 }
 
-int ParseEXRMultipartHeaderFromFile(EXRHeader ***exr_headers, int *num_headers,
+TINYEXR_PUBLIC int ParseEXRMultipartHeaderFromFile(EXRHeader ***exr_headers, int *num_headers,
                                     const EXRVersion *exr_version,
                                     const char *filename, const char **err) {
   if (exr_headers == NULL || num_headers == NULL || exr_version == NULL ||
@@ -8760,7 +8766,7 @@
       exr_headers, num_headers, exr_version, file.data, file.size, err);
 }
 
-int ParseEXRVersionFromMemory(EXRVersion *version, const unsigned char *memory,
+TINYEXR_PUBLIC int ParseEXRVersionFromMemory(EXRVersion *version, const unsigned char *memory,
                               size_t size) {
   if (version == NULL || memory == NULL) {
     return TINYEXR_ERROR_INVALID_ARGUMENT;
@@ -8817,7 +8823,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int ParseEXRVersionFromFile(EXRVersion *version, const char *filename) {
+TINYEXR_PUBLIC int ParseEXRVersionFromFile(EXRVersion *version, const char *filename) {
   if (filename == NULL) {
     return TINYEXR_ERROR_INVALID_ARGUMENT;
   }
@@ -8856,7 +8862,7 @@
   return ParseEXRVersionFromMemory(version, buf, tinyexr::kEXRVersionSize);
 }
 
-int LoadEXRMultipartImageFromMemory(EXRImage *exr_images,
+TINYEXR_PUBLIC int LoadEXRMultipartImageFromMemory(EXRImage *exr_images,
                                     const EXRHeader **exr_headers,
                                     unsigned int num_parts,
                                     const unsigned char *memory,
@@ -8987,7 +8993,7 @@
   return TINYEXR_SUCCESS;
 }
 
-int LoadEXRMultipartImageFromFile(EXRImage *exr_images,
+TINYEXR_PUBLIC int LoadEXRMultipartImageFromFile(EXRImage *exr_images,
                                   const EXRHeader **exr_headers,
                                   unsigned int num_parts, const char *filename,
                                   const char **err) {
@@ -9007,7 +9013,7 @@
                                          file.data, file.size, err);
 }
 
-int SaveEXRToMemory(const float *data, int width, int height, int components,
+TINYEXR_PUBLIC int SaveEXRToMemory(const float *data, int width, int height, int components,
             const int save_as_fp16, const unsigned char **outbuf, const char **err) {
 
   if ((components == 1) || components == 3 || components == 4) {
@@ -9157,7 +9163,7 @@
   return int(mem_size);
 }
 
-int SaveEXR(const float *data, int width, int height, int components,
+TINYEXR_PUBLIC int SaveEXR(const float *data, int width, int height, int components,
             const int save_as_fp16, const char *outfilename, const char **err) {
   if ((components == 1) || components == 3 || components == 4) {
     // OK
