diff -up tinyexr-1.0.7/CMakeLists.txt.orig tinyexr-1.0.7/CMakeLists.txt
--- tinyexr-1.0.7/CMakeLists.txt.orig	2025-07-23 00:43:54.903658087 +0200
+++ tinyexr-1.0.7/CMakeLists.txt	2025-07-23 03:09:57.358989138 +0200
@@ -32,7 +32,8 @@ set(TINYEXR_SOURCES
     )
 
 if(TINYEXR_USE_MINIZ)
-  find_library(MINIZ miniz REQUIRED)
+  find_package(miniz REQUIRED)
+  list(APPEND TINYEXR_EXT_INCLUDES /usr/include/miniz)
   list(APPEND TINYEXR_EXT_LIBRARIES -lminiz)  
 endif()
 
@@ -40,7 +41,7 @@ add_library(${BUILD_TARGET} SHARED ${TIN
 add_sanitizers(${BUILD_TARGET})
 target_compile_features(${BUILD_TARGET} PRIVATE cxx_auto_type)
 
-target_include_directories(${BUILD_TARGET} PRIVATE ${PROJECT_SOURCE_DIR})
+target_include_directories(${BUILD_TARGET} PRIVATE ${PROJECT_SOURCE_DIR} ${TINYEXR_EXT_INCLUDES})
 target_link_libraries(${BUILD_TARGET} ${TINYEXR_EXT_LIBRARIES} ${CMAKE_DL_LIBS})
 set_target_properties(${BUILD_TARGET} PROPERTIES
   VERSION ${PROJECT_VERSION} SOVERSION 1
