--- hyperfine-1.20.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hyperfine-1.20.0/Cargo.toml	2025-12-16T20:48:21.763134+00:00
@@ -116,25 +116,12 @@
 [build-dependencies.clap_complete]
 version = "4.2.1"
 
-[target."cfg(all(windows, not(windows_process_extensions_main_thread_handle)))".dependencies.once_cell]
-version = "1.19"
-
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2"
 
 [target.'cfg(target_os="linux")'.dependencies.nix]
 version = "0.29"
 features = ["zerocopy"]
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_JobObjects",
-    "Win32_System_LibraryLoader",
-    "Win32_System_Threading",
-]
 
 [profile.dev.package.insta]
 opt-level = 3
@@ -146,3 +133,4 @@
 lto = true
 codegen-units = 1
 strip = true
+
