diff -up khal-0.8.2/doc/source/conf.py.doc khal-0.8.2/doc/source/conf.py
--- khal-0.8.2/doc/source/conf.py.doc	2016-05-15 23:50:18.000000000 +0200
+++ khal-0.8.2/doc/source/conf.py	2016-05-22 21:39:14.443637245 +0200
@@ -31,7 +31,7 @@ extensions = [
     'sphinx.ext.autodoc',
     'sphinx.ext.intersphinx',
     'sphinx.ext.todo',
-    'sphinxcontrib.newsfeed',
+#    'sphinxcontrib.newsfeed',
 ]
 
 # Add any paths that contain templates here, relative to this directory.
