diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 90e20d5..dd0f623 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -187,7 +187,7 @@ if (CONFIG_LV2)
   target_link_directories (${PROJECT_NAME}_lv2 PRIVATE ${LV2_LIBRARY_DIRS})
   target_link_libraries (${PROJECT_NAME}_lv2 PRIVATE ${LV2_LIBRARIES})
   if (UNIX AND NOT APPLE)
-    if (NOT CONFIG_DEBUG)
+    if (NOT CONFIG_DEBUG AND CMAKE_INSTALL_DO_STRIP)
       add_custom_command(TARGET ${PROJECT_NAME}_lv2  POST_BUILD
         COMMAND strip lib${PROJECT_NAME}_lv2.so)
     endif ()
@@ -206,7 +206,7 @@ if (CONFIG_LV2)
   endif ()
   if (WIN32)
     target_link_options (${PROJECT_NAME}_lv2 PRIVATE -static-libgcc -static-libstdc++)
-    if (NOT CONFIG_DEBUG)
+    if (NOT CONFIG_DEBUG AND CMAKE_INSTALL_DO_STRIP)
       add_custom_command(TARGET ${PROJECT_NAME}_lv2  POST_BUILD
         COMMAND strip lib${PROJECT_NAME}_lv2.dll)
     endif ()
@@ -236,7 +236,7 @@ if (CONFIG_JACK)
     target_link_libraries (${PROJECT_NAME}_jack PRIVATE PkgConfig::LIBLO)
   endif ()
   if (UNIX AND NOT APPLE)
-    if (NOT CONFIG_DEBUG)
+    if (NOT CONFIG_DEBUG AND CMAKE_INSTALL_DO_STRIP)
       add_custom_command(TARGET ${PROJECT_NAME}_jack POST_BUILD
         COMMAND strip ${PROJECT_NAME}_jack)
     endif ()
