--- fs4-1.1.0/Cargo.toml
+++ fs4-1.1.0/Cargo.toml
@@ -52,19 +52,10 @@
 [features]
 async-std = ["dep:async-std"]
 default = ["sync"]
-fs-err2 = [
-    "dep:fs-err2",
-    "fs-err2/io_safety",
-]
-fs-err2-tokio = [
-    "fs-err2",
-    "fs-err2/tokio",
-]
+fs-err2 = ["dep:fs-err2", "fs-err2/io_safety"]
+fs-err2-tokio = ["fs-err2", "fs-err2/tokio"]
 fs-err3 = ["dep:fs-err3"]
-fs-err3-tokio = [
-    "fs-err3",
-    "fs-err3/tokio",
-]
+fs-err3-tokio = ["fs-err3", "fs-err3/tokio"]
 smol = ["dep:smol"]
 sync = []
 tokio = ["dep:tokio"]
@@ -128,10 +119,3 @@
 version = "1"
 features = ["fs"]
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-]
