--- ./abcde.conf~	2019-02-05 03:19:03.000000000 -0700
+++ ./abcde.conf	2021-01-15 18:07:27.125346596 -0700
@@ -15,19 +15,24 @@
 # If you wish to use a different CDDB server, edit this line.
 # If you just wanted to use a proxy server, just set your http_proxy
 # environment variable - wget will use it correctly.
-#CDDBURL="http://freedb.freedb.org/~cddb/cddb.cgi"
+# In the past freedb.org was a popular choice but this service
+# was acquired and discontinued.  The following may serve as
+# a replacemet
+#CDDBURL="http://gnudb.gnudb.org/~cddb/cddb.cgi"
 
 # The CDDB protocol level.
 # Right now 5 is latin1 output and 6 is UTF8 encoding.
 #CDDBPROTO=6
 
 # The CDDB protocol requires hello information, including a valid username
-# and hostname. If you feel paranoid about giving away such info, edit this
-# line - the format is username@hostname.
+# and hostname. If you feel paranoid about giving away such info, edit
+# "HELLOINFO" line - the format is username@hostname.
+# Submissions to cddb actually require a valid email address which may, or may
+# not be like the following:
 #HELLOINFO="`whoami`@`hostname`"
 
 # This controls the email address CDDB changes are submitted to.
-#CDDBSUBMIT=freedb-submit@freedb.org
+#CDDBSUBMIT=submit@gnudb.org
 
 # The following options control whether or not fetched CDDB entries
 # are cached locally in $CDDBLOCALDIR
