--- picky-asn1-der-0.5.5/Cargo.toml	1970-01-01T00:00:01+00:00
+++ picky-asn1-der-0.5.5/Cargo.toml	2026-03-05T14:50:26.993360+00:00
@@ -63,21 +63,5 @@
 [dependencies.serde_bytes]
 version = "0.11"
 
-[dev-dependencies.base64]
-version = "0.22"
-
-[dev-dependencies.crypto-bigint]
-version = "0.7.0-rc.10"
-features = ["alloc"]
-default-features = false
-
-[dev-dependencies.oid]
-version = "0.2"
-features = ["serde_support"]
-default-features = false
-
 [dev-dependencies.pretty_assertions]
 version = "1.4"
-
-[dev-dependencies.serde_bytes]
-version = "0.11"
