From c78748826adfde23a17aabc8175d1bd9ef3865c8 Mon Sep 17 00:00:00 2001
Message-ID: <c78748826adfde23a17aabc8175d1bd9ef3865c8.1770841363.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Fri, 2 Feb 2024 22:42:52 +0100
Subject: [PATCH] configure for shared mupdf build

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 182b979..c935fb4 100644
--- a/meson.build
+++ b/meson.build
@@ -35,7 +35,6 @@ build_dependencies = [
 if not mupdf.found()
   # normal build of mupdf
   mupdf = cc.find_library('mupdf', has_headers: ['mupdf/fitz/version.h', 'mupdf/fitz.h', 'mupdf/pdf.h'], required: true)
-  mupdfthird = cc.find_library('mupdf-third')
   version_check = '''
 #include <mupdf/fitz/version.h>
 
@@ -47,7 +46,7 @@ if not mupdf.found()
     error('mupdf @0@.@1@ or newer is required'.format(mupdf_required_version_major, mupdf_required_version_minor))
   endif
 
-  build_dependencies += [mupdf, mupdfthird]
+  build_dependencies += [mupdf, ]
 else
   # build from Debian's libmupdf-dev
   build_dependencies += [mupdf]
-- 
2.53.0.247.g99c804103f

