From 19bb84aca1b89fd4b76b11c4025cf4011c0cae43 Mon Sep 17 00:00:00 2001
Message-ID: <19bb84aca1b89fd4b76b11c4025cf4011c0cae43.1770841345.git.mjg@fedoraproject.org>
From: Sebastian Ramacher <sebastian@ramacher.at>
Date: Mon, 7 Apr 2025 11:27:31 +0200
Subject: [PATCH] Update Debian build branch recent packaging changes (fixes
 #83)

---
 meson.build | 22 ++--------------------
 1 file changed, 2 insertions(+), 20 deletions(-)

diff --git a/meson.build b/meson.build
index eda954e..861f6e7 100644
--- a/meson.build
+++ b/meson.build
@@ -24,7 +24,6 @@ girara = dependency('girara-gtk3')
 glib = dependency('glib-2.0')
 cairo = dependency('cairo')
 mupdf = dependency('mupdf', required: false, version: '>=@0@.@1@'.format(mupdf_required_version_major, mupdf_required_version_minor))
-mupdfthird = cc.find_library('mupdf-third')
 
 build_dependencies = [
   zathura,
@@ -36,6 +35,7 @@ build_dependencies = [
 if not mupdf.found()
   # normal build of mupdf
   mupdf = cc.find_library('mupdf', has_headers: ['mupdf/fitz/version.h', 'mupdf/fitz.h', 'mupdf/pdf.h'], required: true)
+  mupdfthird = cc.find_library('mupdf-third')
   version_check = '''
 #include <mupdf/fitz/version.h>
 
@@ -50,25 +50,7 @@ if not mupdf.found()
   build_dependencies += [mupdf, mupdfthird]
 else
   # build from Debian's libmupdf-dev
-  build_dependencies += [mupdf, mupdfthird]
-
-  libjpeg = dependency('libjpeg')
-  libjbig2dec = cc.find_library('jbig2dec')
-  libopenjp2 = dependency('libopenjp2')
-  gumbo = dependency('gumbo')
-  tesseract = dependency('tesseract')
-  leptonica = dependency('lept')
-  mujs = dependency('mujs')
-
-  build_dependencies += [
-    libjpeg,
-    libjbig2dec,
-    libopenjp2,
-    gumbo,
-    tesseract,
-    leptonica,
-    mujs
-  ]
+  build_dependencies += [mupdf]
 endif
 
 if get_option('plugindir') == ''
-- 
2.53.0.247.g99c804103f

