diff -up iaxclient-2.1beta3/configure.ac.BAD iaxclient-2.1beta3/configure.ac
--- iaxclient-2.1beta3/configure.ac.BAD	2008-11-03 09:22:20.000000000 -0500
+++ iaxclient-2.1beta3/configure.ac	2008-11-03 09:22:28.000000000 -0500
@@ -193,7 +193,7 @@ PKG_CHECK_MODULES(GDK2, [gdk-2.0 >= 2.0.
 PKG_CHECK_MODULES(ALSA, [alsa >= 1.0], has_alsa=yes, has_alsa=no)
 PKG_CHECK_MODULES(OGG, [ogg >= 1.1.3], has_ogg=yes, has_ogg=no)
 PKG_CHECK_MODULES(OGGZ, [oggz >= 0.9.5], has_oggz=yes, has_oggz=no)
-PKG_CHECK_MODULES(THEORA, [theora >= 1.0alpha7], has_theora=yes, has_theora=no)
+PKG_CHECK_MODULES(THEORA, [theora >= 1.0], has_theora=yes, has_theora=no)
 PKG_CHECK_MODULES(VIDCAP, [vidcap >= 0.1], has_vidcap=yes, has_vidcap=no)
 PKG_CHECK_MODULES(FFMPEG, [libavcodec >= 51.40.3], has_ffmpeg=yes, has_ffmpeg=no)
 
diff -up iaxclient-2.1beta3/configure.BAD iaxclient-2.1beta3/configure
--- iaxclient-2.1beta3/configure.BAD	2008-11-03 09:20:58.000000000 -0500
+++ iaxclient-2.1beta3/configure	2008-11-03 09:21:31.000000000 -0500
@@ -23115,12 +23115,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_THEORA_CFLAGS="$THEORA_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"theora >= 1.0alpha7\"") >&5
-  ($PKG_CONFIG --exists --print-errors "theora >= 1.0alpha7") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"theora >= 1.0\"") >&5
+  ($PKG_CONFIG --exists --print-errors "theora >= 1.0") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_THEORA_CFLAGS=`$PKG_CONFIG --cflags "theora >= 1.0alpha7" 2>/dev/null`
+  pkg_cv_THEORA_CFLAGS=`$PKG_CONFIG --cflags "theora >= 1.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -23133,12 +23133,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_THEORA_LIBS="$THEORA_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"theora >= 1.0alpha7\"") >&5
-  ($PKG_CONFIG --exists --print-errors "theora >= 1.0alpha7") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"theora >= 1.0\"") >&5
+  ($PKG_CONFIG --exists --print-errors "theora >= 1.0") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_THEORA_LIBS=`$PKG_CONFIG --libs "theora >= 1.0alpha7" 2>/dev/null`
+  pkg_cv_THEORA_LIBS=`$PKG_CONFIG --libs "theora >= 1.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -23157,9 +23157,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        THEORA_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "theora >= 1.0alpha7"`
+	        THEORA_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "theora >= 1.0"`
         else
-	        THEORA_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "theora >= 1.0alpha7"`
+	        THEORA_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "theora >= 1.0"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$THEORA_PKG_ERRORS" >&5
