diff -rupN iaxclient-2.1beta3/configure.ac iaxclient-2.1beta3-new/configure.ac
--- iaxclient-2.1beta3/configure.ac	2008-04-07 18:07:13.000000000 +0200
+++ iaxclient-2.1beta3-new/configure.ac	2015-01-07 18:08:26.743078051 +0100
@@ -71,17 +71,6 @@ AC_ARG_ENABLE(local-gsm,
          [Use local gsm library [default=auto]])],,
       [enable_local_gsm="auto"])
 
-AC_ARG_ENABLE(local-iax,
-      [AS_HELP_STRING([--disable-local-iax],
-         [Don't use local iax library])],,
-      [enable_local_iax="yes"])
-
-AC_ARG_ENABLE(debug-iax,
-      [AS_HELP_STRING([--enable-debug-iax],
-         [Enable debug support in local iax library])],
-      enable_debug_iax2=$enableval,
-      enable_debug_iax2="no")
-
 AC_ARG_ENABLE(video,
       [AS_HELP_STRING([--enable-video],
          [Enable video support [default=yes]])],,
@@ -258,19 +247,6 @@ if test x$enable_video = xyes; then
    fi
 fi
 
-has_iax2=no
-if test x$enable_local_iax != xyes; then
-   AM_PATH_IAX2(0.2.3,has_iax2=yes,has_iax2=no)
-fi
-
-if test x$has_iax2 = xno && test x$enable_local_iax != xno; then
-   has_iax2=yes
-   enable_local_iax2=yes
-   IAX2_CFLAGS='-I$(top_srcdir)/lib/libiax2/src -DLIBIAX'
-   IAX2_LIBS=""
-   IAX2_CONFIG=""
-fi
-
 AM_OPTIONS_WXCONFIG
 
 has_wx="no"
@@ -357,8 +333,6 @@ AM_CONDITIONAL(FFMPEG, test x$has_ffmpeg
 AM_CONDITIONAL(SPAN_EC, test x$use_echo_can = xspan)
 AM_CONDITIONAL(USE_CODEC_GSM, test x$has_gsm = xyes && test x$with_gsm != xno)
 AM_CONDITIONAL(USE_LOCAL_GSM, test x$enable_local_gsm = xyes)
-AM_CONDITIONAL(USE_LOCAL_IAX2, test x$enable_local_iax2 = xyes)
-AM_CONDITIONAL(USE_DEBUG_IAX2, test x$enable_debug_iax2 = xyes)
 AM_CONDITIONAL(USE_LOCAL_ILBC, test x$enable_local_ilbc = xyes)
 AM_CONDITIONAL(LINUX, test x$OSTYPE = xLINUX)
 AM_CONDITIONAL(WIN32, test x$OSTYPE = xWIN32)
diff -rupN iaxclient-2.1beta3/lib/Makefile.am iaxclient-2.1beta3-new/lib/Makefile.am
--- iaxclient-2.1beta3/lib/Makefile.am	2008-04-07 18:05:42.000000000 +0200
+++ iaxclient-2.1beta3-new/lib/Makefile.am	2015-01-07 18:11:48.203579953 +0100
@@ -1,6 +1,9 @@
 # Autoconf etc. macros
 AUTOMAKE_OPTIONS = subdir-objects
 
+IAX2_CFLAGS=-I$(top_srcdir)/lib/libiax2/src
+IAX2_LIBS=$(top_srcdir)/lib/libiax2/src/libiax.la
+
 # Targets
 lib_LTLIBRARIES = libiaxclient.la
 libiaxclient_la_SOURCES = $(SRCS)
@@ -65,13 +68,6 @@ if USE_LOCAL_GSM
 SRCS += $(SRCS_LIBGSM)
 endif
 
-if USE_LOCAL_IAX2
-SRCS += $(SRCS_IAX2)
-if USE_DEBUG_IAX2
-AM_CFLAGS += -DDEBUG_SUPPORT 
-endif USE_DEBUG_IAX2
-endif USE_LOCAL_IAX2
-
 if USE_LOCAL_ILBC
 SRCS += codec_ilbc.c
 NODIST_SRCS += $(SRCS_ILBC)
@@ -188,18 +184,6 @@ SRCS_MACOSX=					\
 	portmixer/px_mac_core/px_mac_core.c	\
 	unixfuncs.c
 
-SRCS_IAX2=					\
-	libiax2/src/md5.c			\
-	libiax2/src/iax.c			\
-	libiax2/src/iax2-parser.c		\
-	libiax2/src/jitterbuf.c			\
-	libiax2/src/frame.h			\
-	libiax2/src/iax2.h			\
-	libiax2/src/iax2-parser.h		\
-	libiax2/src/iax-client.h		\
-	libiax2/src/jitterbuf.h			\
-	libiax2/src/md5.h
-
 EXTRA_DIST = \
 	audio_alsa.c				\
 	audio_alsa.h				\
