From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sun, 2 Jun 2024 17:06:08 +0200
Subject: [PATCH] Disable Rebar3 plugins

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/rebar.config.script b/rebar.config.script
index bc83197..ea5ea6e 100644
--- a/rebar.config.script
+++ b/rebar.config.script
@@ -220,17 +220,9 @@ Rules = [
 	  AppendStr("--coverage"), ""},
 	 {[deps], IsRebar3,
 	  Rebar3DepsFilter, []},
-	 {[plugins], IsRebar3,
-	  AppendList([pc]), []},
 	 {[provider_hooks], IsRebar3,
 	  AppendList([{pre, [
-			     {compile, {pc, compile}},
-			     {clean, {pc, clean}}
 			    ]}]), []},
-	 {[plugins], os:getenv("COVERALLS") == "true",
-	  AppendList([{coveralls, {git,
-	                           "https://github.com/processone/coveralls-erl.git",
-	                           {branch, "addjsonfile"}}} ]), []},
 	 {[deps], os:getenv("USE_GLOBAL_DEPS") /= false,
 	  GlobalDepsFilter, []}
 	],
