 displaybase.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/displaybase.c b/displaybase.c
index 969c55e..61170ed 100644
--- a/displaybase.c
+++ b/displaybase.c
@@ -1079,7 +1079,7 @@ void cDisplay::DrawMessage(const char *txt1, const char *txt2, const cString *tx
     };
 
     // Draw text array in frame color
-    if ((txtArray != NULL) && (txtArrayEntries > 0) && (txtArrayColumns > 0)) {
+    if (txtArrayActive) {
         DEBUG_OT_MSG("txtArray draw txtArrayEntries=%d txtArrayColumns=%d ca=%d cl=%d", txtArrayEntries, txtArrayColumns, ca, cl);
         int c = 0;
         int xa = 0;
