--- bin/varnishtest/vtc_process.c.orig	2018-04-26 14:12:29.539178105 +0100
+++ bin/varnishtest/vtc_process.c	2018-04-26 15:27:49.851948252 +0100
@@ -216,7 +216,7 @@
 		vtc_dump(p->vl, 4, "stdout", buf, i);
 	else if (p->log == 3)
 		vtc_hexdump(p->vl, 4, "stdout", buf, i);
-	(void)write(p->f_stdout, buf, i);
+	assert(write(p->f_stdout, buf, i) == i);
 	Term_Feed(p->term, buf, buf + i);
 	return (0);
 }
@@ -239,7 +239,7 @@
 	p->stderr_bytes += i;
 	AZ(pthread_mutex_unlock(&p->mtx));
 	vtc_dump(p->vl, 4, "stderr", buf, i);
-	(void)write(p->f_stderr, buf, i);
+	assert(write(p->f_stdout, buf, i) == i);
 	return (0);
 }
 
