Subject: [PATCH] Remove `LIB_SUFFIX` reference
---
diff --git a/cmake/Macros.cmake b/cmake/Macros.cmake
--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -81,9 +81,9 @@
 
     # add the install rule
     install(TARGETS ${target}
-            RUNTIME DESTINATION bin COMPONENT bin
-            LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT bin
-            ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT devel)
+            RUNTIME COMPONENT bin
+            LIBRARY COMPONENT bin
+            ARCHIVE COMPONENT devel)
 
     # define CSFML_STATIC if the build type is not set to 'shared'
     if(NOT BUILD_SHARED_LIBS)
