diff -urp gphotoframe-2.0.2-hg2084299dffb6.old/lib/menu.py gphotoframe-2.0.2-hg2084299dffb6.new/lib/menu.py
--- gphotoframe-2.0.2-hg2084299dffb6.old/lib/menu.py	2021-03-28 23:25:37.911241100 +0900
+++ gphotoframe-2.0.2-hg2084299dffb6.new/lib/menu.py	2021-10-13 16:36:37.838160965 +0900
@@ -53,7 +53,8 @@ class PopUpMenu(object):
 
     def on_menuitem5_activate(self, *args):
         "open_photo"
-        self.photoimage.photo.open()
+        photo = self.photoimage.photo
+        photo.open() if photo else None
 
     def on_next_photo(self, *args):
         self.photolist.next_photo(*args)
