--- spf-engine-3.0.4/data/lib/systemd/system/pyspf-milter.service.o	2023-07-06 10:48:06.476483887 +1000
+++ spf-engine-3.0.4/data/lib/systemd/system/pyspf-milter.service	2023-07-06 10:48:38.293088703 +1000
@@ -1,12 +1,15 @@
 [Unit]
 Description=pySPF Milter
 Documentation=man:pyspf-milter(8) man:pyspf-milter.conf(5) 
-After=network.target
+After=network.target nss-lookup.target syslog.target
 
 [Service]
 Type=simple
 PIDFile=/run/pyspf-milter/pyspf-milter.pid
-ExecStart=/usr/local/bin/pyspf-milter /usr/local/etc/pyspf-milter/pyspf-milter.conf 
+EnvironmentFile=-/etc/sysconfig/pyspf-milter
+ExecStart=/usr/bin/pyspf-milter /etc/python-policyd-spf/pyspf-milter.conf 
+User=pyspf-milter
+Group=pyspf-milter
 
 [Install]
 WantedBy=multi-user.target
