From a6f1e2d08ed7a081e8906591f7da8bfe93db8222 Mon Sep 17 00:00:00 2001
From: Carl George <carlwgeorge@gmail.com>
Date: Tue, 24 Jun 2025 12:26:54 -0500
Subject: [PATCH] Avoid build requirement on pytest-runner

---
 setup.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/setup.py b/setup.py
index 938fb35..fa0c589 100644
--- a/setup.py
+++ b/setup.py
@@ -16,11 +16,6 @@ requirements = [
     'Pygments>=2.2.0',
 ]
 
-setup_requirements = [
-    'pytest-runner==3.0',
-    # TODO(tommikaikkonen): put setup requirements (distutils extensions, etc.) here
-]
-
 test_requirements = [
     'pytest==4.3.0',
     'hypothesis==3.33.0',
@@ -60,5 +55,4 @@ setup(
     ],
     test_suite='tests',
     tests_require=test_requirements,
-    setup_requires=setup_requirements,
 )
-- 
2.49.0

