--- a/share/wallpapoz/lib/wallpapoz_system.py
+++ b/share/wallpapoz/lib/wallpapoz_system.py
@@ -115,6 +115,8 @@ class WallpapozSystem:
         pass
     if wm_name=='Xfwm4':
       self.window_manager = 'XFCE4'
+    elif wm_name=='Marco':
+      self.window_manager = 'MATE'
     else:
      try:
       output = os.popen("gnome-session --version")
@@ -161,6 +163,9 @@ class WallpapozSystem:
     elif self.window_manager == "Gnome3":
       os.system("gsettings set org.gnome.desktop.background picture-uri 'file://" + wallpaper + "'")
       os.system("gsettings set org.gnome.desktop.background picture-options " + self.wallpaper_style)
+    elif self.window_manager == "MATE":
+      os.system('gsettings set org.mate.background picture-filename ' + '"' + wallpaper + '"')
+      os.system('gsettings set org.mate.background picture-options ' + self.wallpaper_style)
     elif self.window_manager == "XFCE4":
       os.system("xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitor0/image-path -s " +
         '"' + wallpaper + '"')
@@ -176,6 +181,8 @@ class WallpapozSystem:
       return os.popen("gconftool-2 -g /desktop/gnome/background/picture_filename").read()[:-1]
     elif self.window_manager == "Gnome3":
       return os.popen("gsettings get org.gnome.desktop.background picture-uri").read()[8:-2]
+    elif self.window_manager == "MATE":
+      return os.popen("gsettings get org.mate.background picture-filename").read()[:-1]
     elif self.window_manager == "XFCE4":
       return os.popen("xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitor0/image-path").read()[:-1]
     else:
--- a/src/daemon_wallpapoz
+++ b/src/daemon_wallpapoz
@@ -111,6 +111,7 @@ if __name__ == "__main__":
   available_style = {}
   available_style['Gnome'] = { '3' : 'zoom', '2' : 'scaled', '1' : 'stretched',
           '0' : 'centered', '4' : 'wallpaper' }
+  available_style['MATE'] = available_style['Gnome']
   available_style['Gnome3'] = available_style['Gnome']
   available_style['XFCE4'] = { '3' : '5', '2' : '4', '1' : '3', '0' : '1',
           '4' : '2' }
