--- src-common/ogl.cxx~	2008-10-08 22:03:53.000000000 -0500
+++ src-common/ogl.cxx	2019-09-19 12:16:39.573835931 -0500
@@ -16,7 +16,7 @@
 #include <GL/glxext.h>	// for the opengl extensions we use for hardware accelerated shadowing
 
 #include <GL/gl.h>
-#include <GL/glut.h>
+#include <GL/freeglut.h>
 
 #include <plib/ssg.h>	// For ssgTexture
 
--- src-stormbaancoureur/main.cxx~	2019-09-19 12:18:04.000000000 -0500
+++ src-stormbaancoureur/main.cxx	2019-09-19 12:19:50.265572047 -0500
@@ -15,7 +15,7 @@
 #include <GL/glxext.h>	// for the opengl extensions we use for hardware accelerated shadowing
 
 #include <GL/gl.h>
-#include <GL/glut.h>
+#include <GL/freeglut.h>
 
 #include <plib/ssg.h>
 #include <plib/ul.h>
