--- slashem/util/lev_comp.y.orig	2024-02-27 18:14:00.007162249 -0600
+++ slashem/util/lev_comp.y	2024-02-27 18:14:23.522863157 -0600
@@ -597,7 +597,7 @@
 			/* ERR means random here */
 			if ($7 == ERR && $9 != ERR) {
 		     yyerror("If the door wall is random, so must be its pos!");
-			    tmprdoor[ndoor] = (struct room_door *)0;
+			    tmprdoor[ndoor] = (room_door *)0;
 			} else {
 			    tmprdoor[ndoor] = New(room_door);
 			    tmprdoor[ndoor]->secret = $3;
