From dda273992cf528258646302bdcea2bc3004dc3b3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@corsepiu.home>
Date: Fri, 4 May 2012 13:09:10 +0200
Subject: [PATCH 02/10] openal soname

---
 src/glue/openal_wrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/glue/openal_wrapper.cpp b/src/glue/openal_wrapper.cpp
index 1d47ecc..2a4ec08 100644
--- a/src/glue/openal_wrapper.cpp
+++ b/src/glue/openal_wrapper.cpp
@@ -108,7 +108,7 @@ openal_wrapper(void)
       int idx;
       const char * possiblelibnames[] = {
         NULL, /* is set below */
-        "openal", "openal32", "libopenal.so", "libopenal.dylib",
+        @LIBOPENAL_SONAME@,
         NULL
       };
 
-- 
1.8.1.4

