From 920601828a12a4802b8f5d4cb77ef913793c6b17 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@fedoraproject.org>
Date: Tue, 7 Feb 2017 23:49:32 +0100
Subject: [PATCH 13/13] Use NULL instead of '\0'

---
 src/3ds/SoStream.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/3ds/SoStream.cpp b/src/3ds/SoStream.cpp
index 1b5f96d..7f07a27 100644
--- a/src/3ds/SoStream.cpp
+++ b/src/3ds/SoStream.cpp
@@ -176,7 +176,7 @@ do { \
   if (!gotNum) { setBadBit(); return FALSE; } \
  \
   char *ce; \
-  s = '\0'; \
+  s = NULL; \
   _convertType_ tempVal = _convertFunc_(buf, &ce, 0); \
  \
   if (ce != s) \
@@ -273,7 +273,7 @@ do { \
 gotAll: \
   \
   char *ce; \
-  s = '\0'; \
+  s = NULL; \
   double tempVal = _convertFunc_(buf, &ce); \
  \
   if (ce != s) \
-- 
2.9.3

