--- wxWidgets-2.8.12/include/wx/wxchar.h.orig	2015-02-19 13:34:06.799313625 +0100
+++ wxWidgets-2.8.12/include/wx/wxchar.h	2015-02-19 13:34:33.317724044 +0100
@@ -352,12 +352,8 @@
     #define  wxStrtod    _tcstod
     #define  wxStrtol    _tcstol
     #define  wxStrtoul   _tcstoul
-    #ifdef __VISUALC__
-        #if __VISUALC__ >= 1300 && !defined(__WXWINCE__)
             #define wxStrtoll  _tcstoi64
             #define wxStrtoull _tcstoui64
-        #endif /* VC++ 7+ */
-    #endif
     #define  wxStrxfrm   _tcsxfrm
 
     /* stdio.h functions */
