--- a/libjcat/jcat-self-test.c
+++ b/libjcat/jcat-self-test.c
@@ -667,7 +667,10 @@ jcat_pkcs7_engine_self_signed_pq_func(gconstpointer test_data)
 					  payload,
 					  JCAT_SIGN_FLAG_ADD_TIMESTAMP | JCAT_SIGN_FLAG_USE_PQ,
 					  &error);
-	if (signature == NULL && g_error_matches(error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED)) {
+	if (signature == NULL &&
+	    (g_error_matches(error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED) ||
+	     g_error_matches(error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA) ||
+	     g_error_matches(error, G_IO_ERROR, G_IO_ERROR_FAILED))) {
 		g_test_skip("ML-MDA cannot be enabled at runtime, skipping");
 		return;
 	}
@@ -678,7 +681,10 @@ jcat_pkcs7_engine_self_signed_pq_func(gconstpointer test_data)
 					 jcat_blob_get_data(signature),
 					 JCAT_VERIFY_FLAG_ONLY_PQ,
 					 &error);
-	if (result == NULL && g_error_matches(error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED)) {
+	if (result == NULL &&
+	    (g_error_matches(error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED) ||
+	     g_error_matches(error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA) ||
+	     g_error_matches(error, G_IO_ERROR, G_IO_ERROR_FAILED))) {
 		g_test_skip("ML-MDA cannot be enabled at runtime, skipping");
 		return;
 	}
