diff -up mrcpp-1.4.0/tests/CMakeLists.txt.rpath mrcpp-1.4.0/tests/CMakeLists.txt
--- mrcpp-1.4.0/tests/CMakeLists.txt.rpath	2021-10-13 14:42:00.000000000 +0000
+++ mrcpp-1.4.0/tests/CMakeLists.txt	2021-12-27 16:25:47.121966095 +0000
@@ -19,24 +19,6 @@ target_link_libraries(mrcpp-tests
   Catch
   )
 
-# RPATH fixing
-file(RELATIVE_PATH _rel ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR} ${CMAKE_INSTALL_PREFIX})
-if(APPLE)
-  set(_rpath "@loader_path/${_rel}")
-else()
-  set(_rpath "\$ORIGIN/${_rel}")
-endif()
-file(TO_NATIVE_PATH "${_rpath}/${CMAKE_INSTALL_LIBDIR}" _RPATH)
-
-set_target_properties(mrcpp-tests
-  PROPERTIES
-  MACOSX_RPATH ON
-  SKIP_BUILD_RPATH OFF
-  BUILD_WITH_INSTALL_RPATH OFF
-  INSTALL_RPATH "${_RPATH}"
-  INSTALL_RPATH_USE_LINK_PATH ON
-  )
-
 install(
   TARGETS
   mrcpp-tests
