diff -up aqualung-2.0/src/decoder/dec_lavc.c.orig aqualung-2.0/src/decoder/dec_lavc.c
--- aqualung-2.0/src/decoder/dec_lavc.c.orig	2025-02-14 09:17:58.000000000 +0100
+++ aqualung-2.0/src/decoder/dec_lavc.c	2025-10-01 15:32:54.764878270 +0200
@@ -360,7 +360,7 @@ lavc_decoder_close(decoder_t * dec) {
 
 	lavc_pdata_t * pd = (lavc_pdata_t *)dec->pdata;
 
-	avcodec_close(pd->avCodecCtx);
+	avcodec_free_context(&pd->avCodecCtx);
 
 	avformat_close_input(&pd->avFormatCtx);
 
