diff -rupN --no-dereference numpy-2.4.2/numpy/_core/src/npysort/x86-simd-sort/src/xss-common-comparators.hpp numpy-2.4.2-new/numpy/_core/src/npysort/x86-simd-sort/src/xss-common-comparators.hpp
--- numpy-2.4.2/numpy/_core/src/npysort/x86-simd-sort/src/xss-common-comparators.hpp	2025-09-11 21:13:48.000000000 +0200
+++ numpy-2.4.2-new/numpy/_core/src/npysort/x86-simd-sort/src/xss-common-comparators.hpp	2026-02-03 13:35:42.836666670 +0100
@@ -40,8 +40,8 @@ struct Comparator {
     using opmask_t = typename vtype::opmask_t;
     using type_t = typename vtype::type_t;
 
-    X86_SIMD_SORT_FINLINE bool STDSortComparator(const type_t &a,
-                                                 const type_t &b)
+    X86_SIMD_SORT_INLINE bool STDSortComparator(const type_t &a,
+                                                const type_t &b)
     {
         if constexpr (descend) { return comparison_func<vtype>(b, a); }
         else {
