From adca73ffa30eef4ab67a1a6d636e790cab93d902 Mon Sep 17 00:00:00 2001
From: Dominique Martinet <dominique.martinet@atmark-techno.com>
Date: Wed, 5 Nov 2025 13:43:51 +0900
Subject: [PATCH 02/23] Fix typo: uncompess -> uncompress

Signed-off-by: Lianbo Jiang <lijiang@redhat.com>
---
 diskdump.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/diskdump.c b/diskdump.c
index ce3cbb7b12dd..b1ca0a73be23 100644
--- a/diskdump.c
+++ b/diskdump.c
@@ -1350,7 +1350,7 @@ cache_page(physaddr_t paddr)
 	} else if (pd.flags & DUMP_DH_COMPRESSED_ZSTD) {
 
 		if (!(dd->flags & ZSTD_SUPPORTED)) {
-			error(INFO, "%s: uncompess failed: no zstd compression support\n",
+			error(INFO, "%s: uncompress failed: no zstd compression support\n",
 				DISKDUMP_VALID() ? "diskdump" : "compressed kdump");
 			return READ_ERROR;
 		}
@@ -1358,7 +1358,7 @@ cache_page(physaddr_t paddr)
 		if (!dctx) {
 			dctx = ZSTD_createDCtx();
 			if (!dctx) {
-				error(INFO, "%s: uncompess failed: cannot create ZSTD_DCtx\n",
+				error(INFO, "%s: uncompress failed: cannot create ZSTD_DCtx\n",
 					DISKDUMP_VALID() ? "diskdump" : "compressed kdump");
 				return READ_ERROR;
 			}
-- 
2.51.1

