diff -ruNw a/src/chessgamestate.cpp b/src/chessgamestate.cpp
--- a/src/chessgamestate.cpp	2006-12-24 07:39:04.000000000 +0100
+++ b/src/chessgamestate.cpp	2009-01-23 22:11:07.000000000 +0100
@@ -141,7 +141,7 @@
 		m_threefold_count.push_back(make_pair(sb, 1));
 	}
 
-	cout << m_turn_number << ". " << bm.origin() << " " << bm.dest() << endl;
+	cout << m_turn_number << ". " << (m_white_turn?"...":"") << bm.origin() << bm.dest() << endl;
 
 	if(m_white_turn) {
 		m_turn_number++;

