diff -up freewrl-6.7-20240420gitb3254b1/src/libeai/EAI_C_Advise.c.fixindent freewrl-6.7-20240420gitb3254b1/src/libeai/EAI_C_Advise.c
--- freewrl-6.7-20240420gitb3254b1/src/libeai/EAI_C_Advise.c.fixindent	2025-03-26 10:45:28.000000000 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/libeai/EAI_C_Advise.c	2025-03-26 10:51:22.568109351 -0400
@@ -95,11 +95,13 @@ void _handleFreeWRLcallback (char *line)
 		sscanf (line, "%lf",&evTime);
 
 		/* get the event number */
-		while (!iscntrl(*line)) line++; while (iscntrl(*line)) line++;
+		while (!iscntrl(*line)) line++;
+		while (iscntrl(*line)) line++;
 		sscanf (line,"%d",&evIndex);
 
 		/* get to the data */
-		while (!iscntrl(*line)) line++; while (iscntrl(*line)) line++;
+		while (!iscntrl(*line)) line++;
+		while (iscntrl(*line)) line++;
 
 		#ifdef VERBOSE
 		printf ("event time %lf index %d data :%s:\n",evTime, evIndex, line);
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/input/EAIEventsIn.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/input/EAIEventsIn.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/input/EAIEventsIn.c.fixindent	2025-03-26 10:45:38.000000000 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/input/EAIEventsIn.c	2025-03-26 10:51:22.568331854 -0400
@@ -1248,7 +1248,8 @@ static void handleRoute (char command, c
         x = fieldTemp; ftlen = 0;
 
 	/* skip to the beginning of the field name */
-        while (*bufptr != ' ') bufptr++; while (*bufptr == ' ') bufptr++;
+        while (*bufptr != ' ') bufptr++;
+	while (*bufptr == ' ') bufptr++;
 
 	/* copy the field over */
         while ((*bufptr > ' ') && (ftlen <1000)) { *x = *bufptr;      x++; bufptr++; ftlen++;}
@@ -1259,7 +1260,8 @@ static void handleRoute (char command, c
 
 
 	/* skip past the first field, to get ready for the next one */
-	while (*bufptr != ' ') bufptr++; while (*bufptr == ' ') bufptr++;
+	while (*bufptr != ' ') bufptr++;
+	while (*bufptr == ' ') bufptr++;
 
 	/* ------- now, the route to section -------- */
 	/* get the to Node */
@@ -1270,7 +1272,8 @@ static void handleRoute (char command, c
         x = fieldTemp; ftlen = 0;
 
 	/* skip to the beginning of the field name */
-        while (*bufptr != ' ') bufptr++; while (*bufptr == ' ') bufptr++;
+        while (*bufptr != ' ') bufptr++;
+	while (*bufptr == ' ') bufptr++;
 
 	/* copy the field over */
         while ((*bufptr > ' ') && (ftlen <1000)) { *x = *bufptr;      x++; bufptr++; ftlen++;}
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c.fixindent	2025-03-26 10:45:38.000000000 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c	2025-03-26 10:51:22.568881892 -0400
@@ -2270,26 +2270,26 @@ contenttype *new_contenttype_stagefbo(in
 	if ( !haveFrameBufferObject() ) // GLEW_ARB_framebuffer_object 
 		return _self;
 	
-		glGenTextures(1, &self->itexturebuffer);
-		//bind to set some parameters
-		glBindTexture(GL_TEXTURE_2D, self->itexturebuffer);
-		glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MAG_FILTER,GL_LINEAR);
-		useMip = 0;
-		if(useMip){
-			glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
-			glTexParameteri(GL_TEXTURE_2D, GL_GENERATE_MIPMAP, GL_TRUE); // automatic mipmap generation included in OpenGL v1.4
-		}else{
-			glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MIN_FILTER,GL_LINEAR);
-		}
-		glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
-		glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-		glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, self->ivport.W, self->ivport.H, 0, GL_RGBA , GL_UNSIGNED_BYTE, 0);
-		//glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8, TEXTURE_WIDTH, TEXTURE_HEIGHT, 0, GL_RGBA, GL_UNSIGNED_BYTE, 0);
-		//unbind - will rebind during render to reset width, height as needed
-		glBindTexture(GL_TEXTURE_2D, 0); 
+	glGenTextures(1, &self->itexturebuffer);
+	//bind to set some parameters
+	glBindTexture(GL_TEXTURE_2D, self->itexturebuffer);
+	glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MAG_FILTER,GL_LINEAR);
+	useMip = 0;
+	if(useMip){
+		glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+		glTexParameteri(GL_TEXTURE_2D, GL_GENERATE_MIPMAP, GL_TRUE); // automatic mipmap generation included in OpenGL v1.4
+	}else{
+		glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MIN_FILTER,GL_LINEAR);
+	}
+	glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+	glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+	glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, self->ivport.W, self->ivport.H, 0, GL_RGBA , GL_UNSIGNED_BYTE, 0);
+	//glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8, TEXTURE_WIDTH, TEXTURE_HEIGHT, 0, GL_RGBA, GL_UNSIGNED_BYTE, 0);
+	//unbind - will rebind during render to reset width, height as needed
+	glBindTexture(GL_TEXTURE_2D, 0); 
 
 	glGenFramebuffers(1, &self->ibuffer);
-		glBindFramebuffer(GL_FRAMEBUFFER, self->ibuffer);
+	glBindFramebuffer(GL_FRAMEBUFFER, self->ibuffer);
 
 	// create a renderbuffer object to store depth info
 	// NOTE: A depth renderable image should be attached the FBO for depth test.
@@ -2298,11 +2298,11 @@ contenttype *new_contenttype_stagefbo(in
 	// If you also need stencil test for your rendering, then you must
 	// attach additional image to the stencil attachement point, too.
 	glGenRenderbuffers(1, &self->idepthbuffer);
-		//bind to set some parameters
-		glBindRenderbuffer(GL_RENDERBUFFER, self->idepthbuffer);
-		glRenderbufferStorage(GL_RENDERBUFFER, FW_GL_DEPTH_COMPONENT, self->ivport.W, self->ivport.H);
-		//unbind
-		glBindRenderbuffer(GL_RENDERBUFFER, 0);
+	//bind to set some parameters
+	glBindRenderbuffer(GL_RENDERBUFFER, self->idepthbuffer);
+	glRenderbufferStorage(GL_RENDERBUFFER, FW_GL_DEPTH_COMPONENT, self->ivport.W, self->ivport.H);
+	//unbind
+	glBindRenderbuffer(GL_RENDERBUFFER, 0);
 
 	// attach a texture to FBO color attachement point
 	glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, self->itexturebuffer, 0);
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/main/ProdCon.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/main/ProdCon.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/main/ProdCon.c.fixindent	2025-03-26 10:51:22.561793593 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/main/ProdCon.c	2025-03-26 10:51:22.569434347 -0400
@@ -682,8 +682,8 @@ bool parser_process_res_VRML_X3D(resourc
 	int offsetInNode;
 	int shouldBind;
 	//int shouldUnBind;
-    int parsedOk = FALSE; // results from parser
-    bool fromEAI_SAI = FALSE;
+	int parsedOk = FALSE; // results from parser
+	bool fromEAI_SAI = FALSE;
 	/* we only bind to new nodes, if we are adding via Inlines, etc */
 	//int origFogNodes, origBackgroundNodes, origNavigationNodes, origViewpointNodes;
 	struct X3D_Node *oldFogBindInRender, *oldBackgroundBindInRender, *oldNavigationBindInRender, *oldViewpointBindInRender;
@@ -696,7 +696,7 @@ bool parser_process_res_VRML_X3D(resourc
 
 	UNUSED(parsedOk); // compiler warning mitigation
 
-    //printf ("entering parser_process_res_VRML_X3D\n");
+	//printf ("entering parser_process_res_VRML_X3D\n");
 	//fwl_lockTestMutex();
 
 	/* printf("processing VRML/X3D resource: %s\n", res->URLrequest);  */
@@ -720,11 +720,11 @@ bool parser_process_res_VRML_X3D(resourc
 
     //ConsoleMessage ("parser_process_res_VRML_X3D, url %s",res->parsed_request);
 	/* save the current URL so that any local-url gets are relative to this */
-    if (res->parsed_request != NULL)
-        if (strncmp(res->parsed_request,EAI_Flag,strlen(EAI_Flag)) == 0) {
-            //ConsoleMessage("parser_process_res_VRML_X3D, from EAI, ignoring");
-            fromEAI_SAI = TRUE;
-        }
+	if (res->parsed_request != NULL)
+        	if (strncmp(res->parsed_request,EAI_Flag,strlen(EAI_Flag)) == 0) {
+			//ConsoleMessage("parser_process_res_VRML_X3D, from EAI, ignoring");
+			fromEAI_SAI = TRUE;
+        	}
 
 	if (!fromEAI_SAI){
 		pushInputResource(res);
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/opengl/Textures.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/opengl/Textures.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/opengl/Textures.c.fixindent	2025-03-26 10:51:22.569745288 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/opengl/Textures.c	2025-03-26 10:53:58.790103644 -0400
@@ -1728,7 +1728,8 @@ void move_texture_to_opengl(textureTable
 			}
 
 			borderWidth = tpNode->borderWidth;
-			if (borderWidth < 0) borderWidth = 0; if (borderWidth > 1) borderWidth = 1;
+			if (borderWidth < 0) borderWidth = 0;
+			if (borderWidth > 1) borderWidth = 1;
 
 			// http://www.web3d.org/documents/specifications/19775-1/V3.3/Part01/components/texturing.html#t-TextureMagnificationModes
 
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_DIS.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_DIS.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_DIS.c.fixindent	2025-03-26 10:45:38.000000000 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_DIS.c	2025-03-26 10:51:22.570292979 -0400
@@ -4337,24 +4337,48 @@ void compile_EspduTransform0(struct X3D_
 			n = node->articulationParameterArray.n;
 			for(i=0;i<8;i++){
 				switch(i){
-					case 0: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue0) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue0; n=max(n,i); break;
-					case 1: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue1) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue1; n=max(n,i); break;
-					case 2: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue2) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue2; n=max(n,i); break;
-					case 3: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue3) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue3; n=max(n,i); break;
-					case 4: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue4) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue4; n=max(n,i); break;
-					case 5: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue5) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue5; n=max(n,i); break;
-					case 6: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue6) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue6; n=max(n,i); break;
-					case 7: if(node->set_articulationParameterValue0 != old->set_articulationParameterValue7) 
-						node->articulationParameterArray.p[i] = node->set_articulationParameterValue7; n=max(n,i); break;
+					case 0:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue0)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue0;
+						n=max(n,i);
+						break;
+					case 1:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue1)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue1;
+						n=max(n,i);
+						break;
+					case 2:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue2)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue2;
+						n=max(n,i);
+						break;
+					case 3:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue3)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue3;
+						n=max(n,i);
+						break;
+					case 4:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue4)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue4;
+						n=max(n,i);
+						break;
+					case 5:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue5)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue5;
+						n=max(n,i);
+						break;
+					case 6:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue6)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue6;
+						n=max(n,i);
+						break;
+					case 7:
+						if(node->set_articulationParameterValue0 != old->set_articulationParameterValue7)
+							node->articulationParameterArray.p[i] = node->set_articulationParameterValue7;
+						n=max(n,i);
+						break;
 					default:
-					break;
+						break;
 				}
 			}
 			node->articulationParameterCount = node->articulationParameterArray.n;
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Geospatial.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Geospatial.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Geospatial.c.fixindent	2025-03-26 10:45:38.000000000 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Geospatial.c	2025-03-26 10:51:22.570909437 -0400
@@ -1690,8 +1690,8 @@ static void gccToGdc_fw (Geosys *geoSyst
                 gdc->c[elevation] = q-rnn;
             else
                 gdc->c[elevation] = GCC_Z / s1 + (g->Eps21 * rnn);
-                gdc->c[latitude] = atan(top / w);
-                gdc->c[longitude] = atan2(GCC_Y,GCC_X);
+            gdc->c[latitude] = atan(top / w);
+            gdc->c[longitude] = atan2(GCC_Y,GCC_X);
         }
               /* POINT ABOVE 50 KILOMETERS OR BELOW -10 KILOMETERS  */
         else /* Do Exact Solution  ************ */
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/vrml_parser/CParseParser.c.fixindent freewrl-6.7-20240420gitb3254b1/src/lib/vrml_parser/CParseParser.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/vrml_parser/CParseParser.c.fixindent	2025-03-26 10:51:22.565393311 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/vrml_parser/CParseParser.c	2025-03-26 10:56:57.815340361 -0400
@@ -1132,11 +1132,11 @@ static BOOL parser_componentStatement(st
 				clevel = &cfullname[i+1];
 				break;
 			}
-			char *cname1 = cname;
-			if(!strcmp(cname,"H-Anim")) cname1 = "HAnim";
-			myComponent = findFieldInCOMPONENTS(cname1);
-			myLevel = 0;
-			if(clevel) myLevel = atoi(clevel);
+		char *cname1 = cname;
+		if(!strcmp(cname,"H-Anim")) cname1 = "HAnim";
+		myComponent = findFieldInCOMPONENTS(cname1);
+		myLevel = 0;
+		if(clevel) myLevel = atoi(clevel);
 	}
     handleComponent(myComponent,myLevel);
 
@@ -3502,14 +3502,14 @@ static BOOL parser_interfaceDeclarationB
     }
 
     /* Script can not take inputOutputs */
-   if(0) // change in post-DUK era - we allow inputoutputs now
-    if (script != NULL) {
-		if(script->ShaderScriptNode->_nodeType==NODE_Script && mode==PKW_inputOutput)
-		{
-			PARSE_ERROR("Scripts must not have inputOutputs!")
-			//printf("dug9: maybe scripts can have inputOutputs\n");
+    if(0) // change in post-DUK era - we allow inputoutputs now
+		if (script != NULL) {
+			if(script->ShaderScriptNode->_nodeType==NODE_Script && mode==PKW_inputOutput)
+			{
+				PARSE_ERROR("Scripts must not have inputOutputs!")
+				//printf("dug9: maybe scripts can have inputOutputs\n");
+			}
 		}
-    }
   
     /* lexer_fieldType is #defined as lexer_specialID(me, r, NULL, FIELDTYPES, FIELDTYPES_COUNT, NULL) */
     /* Looks for the next token in the array FIELDTYPES and returns the index in type */
