diff -up freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Sound.c.fix-sound-stubs freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Sound.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Sound.c.fix-sound-stubs	2025-03-27 17:10:38.068137837 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/scenegraph/Component_Sound.c	2025-03-27 17:11:14.447775469 -0400
@@ -1605,6 +1605,8 @@ void render_ListenerPointSource(struct X
 
 #else //HAVE_LIBSOUND
 //HAVE_OPENAL
+void render_AudioBuffer(struct X3D_AudioBuffer* node) {}
+void render_ListenerPoint(struct X3D_ListenerPoint* node) {}
 void render_OscillatorSource(struct X3D_OscillatorSource* node) {}
 void render_BufferAudioSource(struct X3D_BufferAudioSource* node) {}
 void render_StreamAudioSource(struct X3D_StreamAudioSource* node) {}
