diff -up freewrl-6.7-20240420gitb3254b1/src/lib/io_files.c.fix-headers freewrl-6.7-20240420gitb3254b1/src/lib/io_files.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/io_files.c.fix-headers	2025-03-26 11:21:22.640111110 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/io_files.c	2025-03-26 11:21:49.414646808 -0400
@@ -64,6 +64,7 @@
 #include <fcntl.h>
 #if defined (INCLUDE_STL_FILES)
 #include "input/convertSTL.h"
+#include <errno.h>
 #endif //INCLUDE_STL_FILES
 
 #define UNUSED(v) ((void) v)
diff -up freewrl-6.7-20240420gitb3254b1/src/bin/options.c.fix-defines freewrl-6.7-20240420gitb3254b1/src/bin/options.c
--- freewrl-6.7-20240420gitb3254b1/src/bin/options.c.fix-defines	2025-03-27 16:25:31.680185468 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/bin/options.c	2025-03-27 16:30:43.749013633 -0400
@@ -35,6 +35,7 @@
 #include "main.h"
 #include "options.h"
 #include "../sound/internal.h"
+#include "../lib/ui/common.h"
 
 #if HAVE_GETOPT_H
 #include <getopt.h>
