diff -up freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c.fix-lookup_brotoDefname freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c
--- freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c.fix-lookup_brotoDefname	2025-03-27 17:22:19.982882453 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/main/MainLoop.c	2025-03-27 17:23:42.822431295 -0400
@@ -6856,7 +6856,22 @@ void setSensitive(struct X3D_Node *paren
 	se->interpptr = (void *)myp;
 	vector_pushBack(struct SensStruct *,p->SensorEvents,se);
 }
-char* lookup_brotoDefname(struct X3D_Proto* ec, struct X3D_Node* node);
+
+char* lookup_brotoDefname(struct X3D_Proto* ec, struct X3D_Node* node) {
+	int n = vectorSize(ec->__DEFnames);
+	char* name = NULL;
+	struct brotoDefpair def;
+	for (int i = 0; i < n; i++) {
+		def = vector_get(struct brotoDefpair, ec->__DEFnames, i);
+		//printf("%x %x %s\n",node,def.node,def.name);
+		if (def.node == node) {
+			name = def.name;
+			break;
+		}
+	}
+	return name;
+}
+
 char* getNodeDescription(struct X3D_Node* node) {
 	//not all nodetypes have description field, and those that do not all set, so expect some will return null.
 	int type, kind, iifield;
diff -up freewrl-6.7-20240420gitb3254b1/src/lib/world_script/jsVRMLBrowser_sm.cpp.fix-lookup_brotoDefname freewrl-6.7-20240420gitb3254b1/src/lib/world_script/jsVRMLBrowser_sm.cpp
--- freewrl-6.7-20240420gitb3254b1/src/lib/world_script/jsVRMLBrowser_sm.cpp.fix-lookup_brotoDefname	2025-03-27 17:21:39.053753198 -0400
+++ freewrl-6.7-20240420gitb3254b1/src/lib/world_script/jsVRMLBrowser_sm.cpp	2025-03-27 17:24:13.067102231 -0400
@@ -1264,23 +1264,6 @@ static JSClass ProfileInfoArrayClass = {
     JS_FinalizeStub
 };
 
-extern "C" {
-	char* lookup_brotoDefname(struct X3D_Proto* ec, struct X3D_Node* node) {
-		int n = vectorSize(ec->__DEFnames);
-		char* name = NULL;
-		struct brotoDefpair def;
-		for (int i = 0; i < n; i++) {
-			def = vector_get(struct brotoDefpair, ec->__DEFnames, i);
-			//printf("%x %x %s\n",node,def.node,def.name);
-			if (def.node == node) {
-				name = def.name;
-				break;
-			}
-		}
-		return name;
-	}
-} //extern C
-
 char* lookup_brotoDefname(struct X3D_Proto* ec, struct X3D_Node* node);
 JSBool
 X3DRouteToString(JSContext* cx, uintN argc, jsval* vp) {
