--- a/tests/test_climat2.py	2025-12-21 19:21:01.824123757 +0300
+++ a/tests/test_climat2.py	2025-12-21 19:29:20.178045761 +0300
@@ -24,10 +24,11 @@
         self.assertIn(b'mat2 [-h] [-V]', stdout)
         self.assertIn(b'[--unknown-members policy]', stdout)
         self.assertIn(b'[--inplace]', stdout)
-        self.assertIn(b' [-v] [-l]', stdout)
-        self.assertIn(b'[--check-dependencies]', stdout)
+        self.assertIn(b'-v', stdout)
+        self.assertIn(b'-l', stdout)
+        self.assertIn(b'--check-dependencies', stdout)
         self.assertIn(b'[-L | -s]', stdout)
-        self.assertIn(b'[files ...]', stdout)
+        self.assertIn(b'files', stdout)
 
     def test_no_arg(self):
         proc = subprocess.Popen(mat2_binary, stdout=subprocess.PIPE)
@@ -35,12 +36,11 @@
         self.assertIn(b'mat2 [-h] [-V]', stdout)
         self.assertIn(b'[--unknown-members policy]', stdout)
         self.assertIn(b'[--inplace]', stdout)
-        self.assertIn(b' [-v]', stdout)
-        self.assertIn(b'[-l]', stdout)
-        self.assertIn(b'[--check-dependencies]', stdout)
+        self.assertIn(b'-v', stdout)
+        self.assertIn(b'-l', stdout)
+        self.assertIn(b'--check-dependencies', stdout)
         self.assertIn(b'[-L | -s]', stdout)
-        self.assertIn(b'[files ...]', stdout)
-
+        self.assertIn(b'files', stdout)
 
 class TestVersion(unittest.TestCase):
     def test_version(self):
