diff -up ./dump1090.c.share ./dump1090.c
--- ./dump1090.c.share	2021-07-27 13:24:25.338014426 -0400
+++ ./dump1090.c	2021-07-27 13:26:55.515461586 -0400
@@ -46,6 +46,7 @@
 #include "rtl-sdr.h"
 #include "anet.h"
 
+#define GMAP_HTML	"/usr/share/dump1090/gmap.html"
 #define MODES_DEFAULT_RATE         2000000
 #define MODES_DEFAULT_FREQ         1090000000
 #define MODES_DEFAULT_WIDTH        1000
@@ -2253,8 +2254,8 @@ int handleHTTPRequest(struct client *c)
         struct stat sbuf;
         int fd = -1;
 
-        if (stat("gmap.html",&sbuf) != -1 &&
-            (fd = open("gmap.html",O_RDONLY)) != -1)
+        if (stat(GMAP_HTML,&sbuf) != -1 &&
+            (fd = open(GMAP_HTML,O_RDONLY)) != -1)
         {
             content = malloc(sbuf.st_size);
             if (read(fd,content,sbuf.st_size) == -1) {
