--- bwrap-1.3.0/Cargo.toml	2023-07-31T13:58:40+00:00
+++ bwrap-1.3.0/Cargo.toml	2023-11-13T14:21:30.484789+00:00
@@ -15,7 +15,7 @@
 name = "bwrap"
 version = "1.3.0"
 authors = ["Michael Lee <micl2e2@proton.me>"]
-exclude = ["*.bak"]
+exclude = ["*.bak", "/LICENSE-DEPS/", "/README.org", "/README.txt", "/*.png"]
 description = "A fast, lightweight, embedded systems-friendly library for\nwrapping text."
 homepage = "https://github.com/micl2e2/bwrap"
 documentation = "https://docs.rs/bwrap"
