--- stun.cxx	2012-01-26 16:32:46.000000000 +0530
+++ stun.cxx.huzaifas	2014-08-08 21:51:29.532523382 +0530
@@ -669,7 +669,7 @@
       tick = hightick;
       tick <<= 32;
       tick |= lowtick;
-#elif defined(__GNUC__) && ( defined(__i686__) || defined(__i386__) )
+#elif defined(__GNUC__) && ( defined(__i686__) || defined(__i386__)  || defined(__x86_64__) )
       asm("rdtsc" : "=A" (tick));
 #elif defined (__SUNPRO_CC) || defined( __sparc__ )	
       tick = gethrtime();
