diff -up libxsmm-1.17/Makefile.inc.rpath libxsmm-1.17/Makefile.inc
--- libxsmm-1.17/Makefile.inc.rpath	2021-12-02 13:45:19.000000000 -0700
+++ libxsmm-1.17/Makefile.inc	2023-09-10 20:04:37.070956645 -0600
@@ -917,8 +917,6 @@ ifneq (Darwin,$(UNAME))
   endif
   XLNKVERBOSE := --verbose
   linkopt = $(if $1,$(XLNKOPT)$(if $2,$1=$(call quote,$2),$1))
-  abslibrpath = $(strip $(if $(findstring .$(ILIBEXT),$1)$(wildcard $1/), \
-    $(call linkopt,--rpath,$(call qxdir,$(call qapath,$1)))))
   XGROUP_BEGIN := $(call linkopt,--start-group)
   XGROUP_END := $(call linkopt,--end-group)
   ifneq (0,$(ASNEEDED))
@@ -931,8 +929,6 @@ else # OSX
   endif
   XLNKVERBOSE := -t
   linkopt = $(if $1,$(XLNKOPT) $(if $2,$1 $(XLNKOPT) $(call quote,$2),$1))
-  abslibrpath = $(strip $(if $(findstring .$(ILIBEXT),$1)$(wildcard $1/), \
-    $(call linkopt,-rpath,$(call qxdir,$(call qapath,$1)))))
 endif
 XLIB_BEGIN ?= $(NULL)
 XLIB_END ?= $(NULL)
@@ -941,7 +937,7 @@ absliblpath = $(strip $(if $1,$(if $(fin
   -L$(call qxdir,$(call qapath,$1)))))
 
 ifneq (0,$(ABSLIBS))
-  abslibpath = $(strip $(call abslibrpath,$1) $(call absliblpath,$1))
+  abslibpath = $(strip $(call absliblpath,$1))
 endif
 abslibpath ?= $(call absliblpath,$1)
 
@@ -2338,22 +2334,16 @@ ifneq (,$(GFC)$(filter-out 0,$(XSMM_GCC)
   ifneq (,$(LIBGFORTRAN))
     ifneq (0,$(DEPSTATIC))
       BLAS_CLDFLAGS += $(XLIB_BEGIN) $(LIBGFORTRAN) $(XLIB_END)
-    else
-      BLAS_LDFLAGS += $(call abslibrpath,$(LIBGFORTRAN))
     endif
   endif
   ifneq (,$(QUADMATH))
     ifneq (0,$(DEPSTATIC))
       BLAS_CLDFLAGS += $(XLIB_BEGIN) $(QUADMATH) $(XLIB_END)
-    else
-      BLAS_LDFLAGS += $(call abslibrpath,$(QUADMATH))
     endif
   endif
   ifneq (,$(LIBM))
     ifneq (0,$(DEPSTATIC))
       BLAS_CLDFLAGS += $(XLIB_BEGIN) $(LIBM) $(XLIB_END)
-    else
-      BLAS_LDFLAGS += $(call abslibrpath,$(LIBM))
     endif
   endif
 endif
