diff -urp -x Makefile -x 'config.*' -x '*~' xcircuit-3.9.40.old/svg.c xcircuit-3.9.40.new/svg.c
--- xcircuit-3.9.40.old/svg.c	2015-10-07 00:04:49.000000000 +0900
+++ xcircuit-3.9.40.new/svg.c	2016-06-25 00:13:28.468986578 +0900
@@ -1108,12 +1108,12 @@ int xctcl_svg(ClientData clientData, Tcl
 
    if (locobjc >= 2) {
       /* If there is a non-option argument, use it for the output filename */
-      sprintf(filename, Tcl_GetString(objv[1]));
+      sprintf(filename, "%s", Tcl_GetString(objv[1]));
    }
    else if (xobjs.pagelist[areawin->page]->pageinst->thisobject->name == NULL)
-      sprintf(filename, xobjs.pagelist[areawin->page]->filename);
+      sprintf(filename, "%s", xobjs.pagelist[areawin->page]->filename);
    else
-      sprintf(filename, xobjs.pagelist[areawin->page]->pageinst->thisobject->name);
+      sprintf(filename, "%s", xobjs.pagelist[areawin->page]->pageinst->thisobject->name);
 
    pptr = strrchr(filename, '.');
    if (pptr != NULL)
diff -urp -x Makefile -x 'config.*' -x '*~' xcircuit-3.9.40.old/tclxcircuit.c xcircuit-3.9.40.new/tclxcircuit.c
--- xcircuit-3.9.40.old/tclxcircuit.c	2015-11-14 02:27:30.000000000 +0900
+++ xcircuit-3.9.40.new/tclxcircuit.c	2016-06-25 00:15:07.012510579 +0900
@@ -3876,7 +3876,7 @@ int xctcl_object(ClientData clientData,
       case NameIdx:
 	 if (nidx == 1 || areawin->selects == 0) {
 	    if (objc == 3) {
-	       sprintf(thisinst->thisobject->name, Tcl_GetString(objv[nidx + 2]));
+	       sprintf(thisinst->thisobject->name, "%s", Tcl_GetString(objv[nidx + 2]));
 	       checkname(thisinst->thisobject);
 	    }
 	    Tcl_AppendElement(interp, thisinst->thisobject->name);
@@ -7294,7 +7294,7 @@ int xctcl_page(ClientData clientData, Tc
 
       case LoadIdx:
 	 TechReplaceSave();
-	 sprintf(_STR2, Tcl_GetString(objv[2 + nidx]));
+	 sprintf(_STR2, "%s", Tcl_GetString(objv[2 + nidx]));
 	 for (i = 3 + nidx; i < objc; i++) {
 	    argv = Tcl_GetString(objv[i]);
 	    if ((*argv == '-') && !strncmp(argv, "-repl", 5)) {
@@ -7380,7 +7380,7 @@ int xctcl_page(ClientData clientData, Tc
 
 	 switch (importtype) {
 	    case XCircuitIdx:
-	       sprintf(_STR2, Tcl_GetString(objv[3 + nidx]));
+	       sprintf(_STR2, "%s", Tcl_GetString(objv[3 + nidx]));
 	       for (i = 4; i < objc; i++) {
 		  strcat(_STR2, ",");
 		  strcat(_STR2, Tcl_GetString(objv[i + nidx]));
@@ -7397,7 +7397,7 @@ int xctcl_page(ClientData clientData, Tc
 		  Tcl_SetObjResult(interp, objPtr);
 		  return XcTagCallback(interp, objc, objv);
 	       }
-	       sprintf(_STR2, Tcl_GetString(objv[3 + nidx]));
+	       sprintf(_STR2, "%s", Tcl_GetString(objv[3 + nidx]));
 	       if (savepage != pageno) newpage(pageno);
 	       loadbackground();
 	       if (savepage != pageno) newpage(savepage);
