--- gimli-0.34.0/Cargo.toml
+++ gimli-0.34.0/Cargo.toml
@@ -47,14 +47,8 @@
 resolver = "2"
 
 [features]
-default = [
-    "read-all",
-    "write",
-]
-endian-reader = [
-    "read",
-    "dep:stable_deref_trait",
-]
+default = ["read-all", "write"]
+endian-reader = ["read", "dep:stable_deref_trait"]
 fallible-iterator = ["dep:fallible-iterator"]
 read = ["read-core"]
 read-all = [
@@ -63,10 +57,6 @@
     "endian-reader",
 ]
 read-core = []
-rustc-dep-of-std = [
-    "dep:core",
-    "dep:alloc",
-]
 std = ["stable_deref_trait?/std"]
 write = [
     "dep:fnv",
@@ -78,16 +68,6 @@
 name = "gimli"
 path = "src/lib.rs"
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dependencies.fallible-iterator]
 version = "0.3.0"
 optional = true
