diff --git a/configure.ac b/configure.ac
index afe7c6a..438e030 100644
--- configure.ac
+++ configure.ac
@@ -2,8 +2,8 @@ AC_INIT([system-switch-java], [1.1.8.1pre], [dbhole@redhat.com], ,
         [https://fedorahosted.org/system-switch-java/])
 
 AM_INIT_AUTOMAKE([no-dist-gzip dist-xz -Wall])
-
-AM_GNU_GETTEXT_VERSION([0.19.7])
+AM_GNU_GETTEXT_VERSION([0.19.7])
+AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.7])
 AM_GNU_GETTEXT([external])
 GETTEXT_PACKAGE=system-switch-java
 AC_SUBST([GETTEXT_PACKAGE])
