--- e-antic-2.0.2/libeantic/configure.ac.orig	2024-01-10 16:20:21.000000000 -0700
+++ e-antic-2.0.2/libeantic/configure.ac	2024-03-04 09:22:28.195994422 -0700
@@ -31,8 +31,6 @@ AC_PROG_CXX
 dnl Set INSTALL or use install-sh.
 AC_PROG_INSTALL
 
-AC_CONFIG_LIBOBJ_DIR([upstream/patched])
-
 ## Test for valgrind test suite
 AX_VALGRIND_DFLT([sgcheck], [off])
 AX_VALGRIND_DFLT([drd], [off])
@@ -72,15 +70,6 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
   AC_CHECK_LIB(antic, nf_init, [], [AC_MSG_ERROR([libantic not found])])
   AM_CONDITIONAL([HAVE_LIBANTIC], [true])
   
-  # Check if we need to provide our own nf_init_randtest which was introduced in https://github.com/wbhart/antic/pull/47
-  AC_REPLACE_FUNCS(nf_init_randtest)
-  # Unconditionally patch nf_elem_scalar_div_fmpq which has aliasing issues in ANTIC, https://github.com/wbhart/antic/pull/48
-  AC_LIBOBJ(EANTIC_nf_elem_scalar_div_fmpq)
-  # Unconditionally patch fmpq_poly_add_fmpq which was fixed in https://github.com/wbhart/flint2/commit/17d26d4c957828f3c98c15ddec85108f841d3438
-  AC_LIBOBJ(EANTIC_fmpq_poly_add_fmpq)
-  # Unconditionally patch nf_elem_add_fmpq so it uses our patched fmpq_poly_add_fmpq
-  AC_LIBOBJ(EANTIC_nf_elem_add_fmpq)
-  
   AC_CHECK_HEADERS(arb.h arb_poly.h arf.h, , [AC_MSG_ERROR([Arb headers not found. If your arb headers are in a non-standard location, you might need to adapt your CPPFLAGS, e.g., by adding -I/usr/include/arb to the CPPFLAGS environment variable.])])
   AC_SEARCH_LIBS(arb_init, [arb flint-arb],[], [AC_MSG_ERROR([libarb not found])])
   AM_CONDITIONAL([HAVE_LIBARB], [true])
--- e-antic-2.0.2/libeantic/Makefile.am.orig	2024-01-10 16:20:21.000000000 -0700
+++ e-antic-2.0.2/libeantic/Makefile.am	2024-03-04 09:17:43.832665581 -0700
@@ -5,20 +5,3 @@ endif
 SUBDIRS = src srcxx test $(MAYBE_BENCHMARK)
 
 ACLOCAL_AMFLAGS = -I m4
-
-# Distribute files that we might need to fill/replace from our dependencies
-EXTRA_DIST =                                          \
-  upstream/patched/EANTIC_fmpq_poly_add_fmpq.c        \
-  upstream/patched/EANTIC_nf_elem_add_fmpq.c          \
-  upstream/patched/EANTIC_nf_elem_scalar_div_fmpq.c   \
-  upstream/patched/fmpq_poly_add_fmpq.c               \
-  upstream/patched/nf.h                               \
-  upstream/patched/nf_elem.h                          \
-  upstream/patched/nf_elem_add_fmpq.c                 \
-  upstream/patched/nf_elem_scalar_div_fmpq.c          \
-  upstream/patched/nf_init_randtest.c
-
-# Probably a bug in autotools:
-# .libs in AC_CONFIG_LIBOBJ_DIR is not cleaned automatically.
-mostlyclean-local:
-	-rm -rf upstream/patched/.libs
