From d95cd5a17c3192b905e654b121a53aa2da4d5666 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lubo.rintel@gooddata.com>
Date: Tue, 10 Jul 2012 15:03:14 +0200
Subject: [PATCH] Skip broken POD from coverage testing

---
 t/pod-coverage.t |   19 ++++++++++++++++++-
 1 files changed, 18 insertions(+), 1 deletions(-)

diff --git a/t/pod-coverage.t b/t/pod-coverage.t
index 6548525..54e6c04 100644
--- a/t/pod-coverage.t
+++ b/t/pod-coverage.t
@@ -17,4 +17,21 @@ eval "use Pod::Coverage $min_pc";
 plan skip_all => "Pod::Coverage $min_pc required for testing POD coverage"
     if $@;
 
-all_pod_coverage_ok();
+pod_coverage_ok('ServiceNow', { also_private => [ 'createNotification' ]});
+pod_coverage_ok('ServiceNow::Attachment');
+pod_coverage_ok('ServiceNow::GlideRecord', { also_private => [qw/delete getConfig getRecord getTableName/] });
+pod_coverage_ok('ServiceNow::Connection', { also_private => [qw/close new open send/] });
+pod_coverage_ok('ServiceNow::ITIL::Ticket');
+pod_coverage_ok('ServiceNow::ITIL::Incident');
+pod_coverage_ok('ServiceNow::ITIL::Approval');
+pod_coverage_ok('ServiceNow::ITIL::Request');
+pod_coverage_ok('ServiceNow::ITIL::Task');
+pod_coverage_ok('ServiceNow::ITIL::SC_Task');
+pod_coverage_ok('ServiceNow::ITIL::Change');
+pod_coverage_ok('ServiceNow::ITIL::Dictionary');
+pod_coverage_ok('ServiceNow::ITIL::RequestedItem');
+pod_coverage_ok('ServiceNow::ITIL::Problem');
+pod_coverage_ok('ServiceNow::WS', { also_private => ['new'] });
+pod_coverage_ok('ServiceNow::Configuration', { also_private => [qw/getSoapEndPoint setSoapEndPoint/] });
+pod_coverage_ok('ServiceNow::WSResult');
+done_testing();
-- 
1.7.1

