diff -up kdebase-3.5.10/admin/acinclude.m4.in.me kdebase-3.5.10/admin/acinclude.m4.in
diff -up kdebase-3.5.10/configure.me kdebase-3.5.10/configure
--- kdebase-3.5.10/configure.me	2018-06-11 14:34:26.380732266 +0200
+++ kdebase-3.5.10/configure	2018-06-11 14:45:35.648135931 +0200
@@ -31828,7 +31828,7 @@ if { (eval echo "$as_me:$LINENO: \"$kde_
 	# if you're trying to debug this check and think it's incorrect,
 	# better check your installation. The check _is_ correct - your
 	# installation is not.
-	if test -f actest.cpp && grep klineedit actest.cpp > /dev/null; then
+	if test -f actest.cpp && grep KLineEdit actest.cpp > /dev/null; then
 		kde_cv_uic_plugins=yes
 	fi
 fi
diff -up kdebase-3.5.10/acinclude.m4.me kdebase-3.5.10/acinclude.m4
--- kdebase-3.5.10/acinclude.m4.me	2018-06-11 15:03:04.964896772 +0200
+++ kdebase-3.5.10/acinclude.m4	2018-06-11 15:03:40.078021996 +0200
@@ -1658,7 +1658,7 @@ if AC_TRY_EVAL(kde_line); then
 	# if you're trying to debug this check and think it's incorrect,
 	# better check your installation. The check _is_ correct - your
 	# installation is not.
-	if test -f actest.cpp && grep klineedit actest.cpp > /dev/null; then
+	if test -f actest.cpp && grep KLineEdit actest.cpp > /dev/null; then
 		kde_cv_uic_plugins=yes
 	fi
 fi
diff -up kdebase-3.5.10/admin/acinclude.m4.in.me kdebase-3.5.10/admin/acinclude.m4.in
--- kdebase-3.5.10/admin/acinclude.m4.in.me	2018-06-11 15:06:48.067692419 +0200
+++ kdebase-3.5.10/admin/acinclude.m4.in	2018-06-11 15:07:06.873759486 +0200
@@ -1658,7 +1658,7 @@ if AC_TRY_EVAL(kde_line); then
 	# if you're trying to debug this check and think it's incorrect,
 	# better check your installation. The check _is_ correct - your
 	# installation is not.
-	if test -f actest.cpp && grep klineedit actest.cpp > /dev/null; then
+	if test -f actest.cpp && grep KLineEdit actest.cpp > /dev/null; then
 		kde_cv_uic_plugins=yes
 	fi
 fi
