From 68bc00321d10861bf44cc127ea1abce468dc52a3 Mon Sep 17 00:00:00 2001
From: Gris Ge <fge@redhat.com>
Date: Sat, 28 Feb 2026 15:07:22 +0800
Subject: [PATCH] Change dependent toml version from 1.0.3 to 1.0

Signed-off-by: Gris Ge <fge@redhat.com>
---
 rust/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rust/Cargo.toml b/rust/Cargo.toml
index 97edfb97..0b9600f7 100644
--- a/rust/Cargo.toml
+++ b/rust/Cargo.toml
@@ -26,7 +26,7 @@ once_cell = "1.12.0"
 env_logger = "0.11.6"
 clap = { version = "3.1", features = ["cargo"] }
 time = { version = "0.3", default-features = false, features = ["std", "formatting", "parsing"] }
-toml = "1.0.3"
+toml = "1.0"
 tokio = { version = "1.30", features = ["rt", "net", "time"] }
 
 [workspace.metadata.vendor-filter]
-- 
2.53.0

