diff --git a/src/Makefile b/src/Makefile
index 7b02d19..0a499d7 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -2,10 +2,10 @@ DESTDIR ?=
 PREFIX ?= /usr/local
 BINDIR = $(PREFIX)/bin
 SHAREDIR = $(PREFIX)/share/imapfilter
-MANDIR = $(PREFIX)/man
+MANDIR = $(PREFIX)/share/man
 
 SSLCAPATH = /etc/ssl/certs
-SSLCAFILE = /etc/ssl/cert.pem
+SSLCAFILE = $(SSLCAPATH)/ca-bundle.crt
 
 MYCFLAGS =
 MYLDFLAGS =
@@ -19,7 +19,7 @@ LIBPCRE = -lpcre2-8
 LIBSSL = -lssl
 LIBCRYPTO = -lcrypto
 
-CFLAGS = -Wall -Wextra -O \
+CFLAGS += -Wall -Wextra -O \
 	 -DCONFIG_SHAREDIR='"$(SHAREDIR)"' \
 	 -DCONFIG_SSL_CAPATH='"$(SSLCAPATH)"' \
 	 -DCONFIG_SSL_CAFILE='"$(SSLCAFILE)"' \
