diff -up ./config/defaults.inc.php.rpm ./config/defaults.inc.php
--- ./config/defaults.inc.php.rpm	2025-12-11 17:54:02.000000000 +0100
+++ ./config/defaults.inc.php	2025-12-12 15:08:17.363515508 +0100
@@ -590,11 +590,11 @@ $config['user_aliases'] = false;
 // use this folder to store log files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
 // This is used by the 'file' log driver.
-$config['log_dir'] = RCUBE_INSTALL_PATH . 'logs/';
+$config['log_dir'] = '/var/log/roundcubemail/';
 
 // Location of temporary saved files such as attachments and cache files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
-$config['temp_dir'] = RCUBE_INSTALL_PATH . 'temp/';
+$config['temp_dir'] = '/var/lib/roundcubemail/temp/';
 
 // expire files in temp_dir after 48 hours
 // possible units: s, m, h, d, w
@@ -828,7 +828,7 @@ $config['mime_magic'] = null;
 // This is used to derive mime-types from the filename extension or vice versa.
 // Such a file is usually part of the apache webserver. If you don't find a file named mime.types on your system,
 // download it from https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types
-$config['mime_types'] = null;
+$config['mime_types'] = '/etc/mime.types';
 
 // path to imagemagick identify binary (if not set we'll use Imagick or GD extensions)
 $config['im_identify_path'] = null;
@@ -1568,3 +1568,7 @@ $config['message_show_email'] = false;
 // 0 - Reply-All always
 // 1 - Reply-List if mailing list is detected
 $config['reply_all_mode'] = 0;
+
+// Keys directory for all users.
+// Must be writeable by PHP process, and not in the web server document root
+$config['enigma_pgp_homedir'] = '/var/lib/roundcubemail/enigma';
diff -up ./plugins/enigma/config.inc.php.dist.rpm ./plugins/enigma/config.inc.php.dist
--- ./plugins/enigma/config.inc.php.dist.rpm	2025-12-11 17:54:02.000000000 +0100
+++ ./plugins/enigma/config.inc.php.dist	2025-12-12 15:08:17.364179921 +0100
@@ -14,10 +14,6 @@ $config['enigma_smime_driver'] = 'phpssl
 // Enables logging of enigma operations (including Crypt_GPG debug info)
 $config['enigma_debug'] = false;
 
-// REQUIRED! Keys directory for all users.
-// Must be writeable by PHP process, and not in the web server document root
-$config['enigma_pgp_homedir'] = null;
-
 // Location of gpg binary. By default it will be auto-detected.
 // This is also a way to force gpg2 use if there are both 1.x and 2.x on the system.
 $config['enigma_pgp_binary'] = '';
diff -up ./program/include/iniset.php.rpm ./program/include/iniset.php
--- ./program/include/iniset.php.rpm	2025-12-11 17:54:03.000000000 +0100
+++ ./program/include/iniset.php	2025-12-12 15:09:11.387611544 +0100
@@ -33,7 +33,7 @@ if (!defined('INSTALL_PATH')) {
 }
 
 if (!defined('RCMAIL_CONFIG_DIR')) {
-    define('RCMAIL_CONFIG_DIR', getenv('ROUNDCUBE_CONFIG_DIR') ?: (INSTALL_PATH . 'config'));
+    define('RCMAIL_CONFIG_DIR', '/etc/roundcubemail');
 }
 
 if (!defined('RCUBE_LOCALIZATION_DIR')) {
@@ -41,7 +41,7 @@ if (!defined('RCUBE_LOCALIZATION_DIR'))
 }
 
 define('RCUBE_INSTALL_PATH', INSTALL_PATH);
-define('RCUBE_CONFIG_DIR', RCMAIL_CONFIG_DIR . '/');
+define('RCUBE_CONFIG_DIR', '/etc/roundcubemail/');
 
 // Show basic error message on fatal PHP error
 register_shutdown_function('rcmail_error_handler');
diff -up ./program/lib/Roundcube/bootstrap.php.rpm ./program/lib/Roundcube/bootstrap.php
--- ./program/lib/Roundcube/bootstrap.php.rpm	2025-12-11 17:54:03.000000000 +0100
+++ ./program/lib/Roundcube/bootstrap.php	2025-12-12 15:08:17.364326762 +0100
@@ -67,7 +67,7 @@ if (!defined('RCUBE_INSTALL_PATH')) {
 }
 
 if (!defined('RCUBE_CONFIG_DIR')) {
-    define('RCUBE_CONFIG_DIR', RCUBE_INSTALL_PATH . 'config/');
+    define('RCUBE_CONFIG_DIR', '/etc/roundcubemail/');
 }
 
 if (!defined('RCUBE_PLUGINS_DIR')) {
