From eb274ca10d035c176e5eac79420ca734c02613ae Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Harabie=C5=84?= <rafalh92@outlook.com>
Date: Mon, 9 Oct 2023 23:30:57 +0200
Subject: [PATCH] Fix build with log v0.4.20

Fixes #87
---
 src/log_macros.rs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/rust/vendor/rust-fatfs/src/log_macros.rs b/rust/vendor/rust-fatfs/src/log_macros.rs
index 5d2e9d2..36a195d 100644
--- a/rust/vendor/rust-fatfs/src/log_macros.rs
+++ b/rust/vendor/rust-fatfs/src/log_macros.rs
@@ -51,7 +51,12 @@ macro_rules! log {
             log::log!(target: $target, lvl, $($arg)+);
         }
     });
-    ($lvl:expr, $($arg:tt)+) => (log!(target: log::__log_module_path!(), $lvl, $($arg)+))
+    ($lvl:expr, $($arg:tt)+) => ({
+        let lvl = $lvl;
+        if lvl <= $crate::log_macros::MAX_LOG_LEVEL {
+            log::log!(lvl, $($arg)+);
+        }
+    })
 }
 
 #[macro_export]
