--- addons/EditLine/source/IoEditLine.c~	2017-08-11 13:47:18.000000000 -0500
+++ addons/EditLine/source/IoEditLine.c	2024-02-05 15:36:02.950438252 -0600
@@ -56,7 +56,7 @@
 
 	el_source(DATA(self)->editline, NULL);
 
-	IoState_registerProtoWithFunc_((IoState *)state, self, IoEditLine_proto);
+	IoState_registerProtoWithFunc_((IoState *)state, self, (const char *)IoEditLine_proto);
 
 	IoObject_addMethodTable_(self, methodTable);
 
--- addons/Libxml2/source/IoXmlReader.c~	2017-08-11 13:47:18.000000000 -0500
+++ addons/Libxml2/source/IoXmlReader.c	2024-02-05 15:47:29.642730890 -0600
@@ -41,7 +41,7 @@
 
 	IoObject_setDataPointer_(self, calloc(1, sizeof(IoXmlReaderData)));
 
-	IoState_registerProtoWithFunc_(state, self, IoXmlReader_proto);
+	IoState_registerProtoWithFunc_(state, self, (const char *)IoXmlReader_proto);
 	IoMethodTable methodTable[] = {
 		{"parseFile", IoXmlReader_parseFile},
 		{"parseString", IoXmlReader_parseString},
--- addons/Libxml2/source/IoXmlWriter.h~	2017-08-11 13:47:18.000000000 -0500
+++ addons/Libxml2/source/IoXmlWriter.h	2024-02-05 16:06:23.258583187 -0600
@@ -8,6 +8,7 @@
 #include "IoObject.h"
 #include "IoSeq.h"
 #include <libxml/xmlwriter.h>
+#include <libxml/xmlerror.h>
 #include "IoFile.h"
 
 #define ISXMLWRITER(self) IoObject_hasCloneFunc_(self, (IoTagCloneFunc *)IoXmlWriter_rawClone)
--- addons/Loudmouth/source/IoLoudmouth.c~	2017-08-11 13:47:18.000000000 -0500
+++ addons/Loudmouth/source/IoLoudmouth.c	2024-02-05 16:13:47.393290790 -0600
@@ -118,7 +118,7 @@
   IoObject *self = IoObject_new(state);
   IoObject_tag_(self, IoLoudmouth_newTag(state));
 
-  IoState_registerProtoWithFunc_(state, self, IoLoudmouth_proto);
+  IoState_registerProtoWithFunc_(state, self, (const char *)IoLoudmouth_proto);
 
   {
     IoMethodTable methodTable[] = {
