diff --git a/src/qml/LandingComponent.qml b/src/qml/LandingComponent.qml
--- a/src/qml/LandingComponent.qml
+++ b/src/qml/LandingComponent.qml
@@ -43,12 +43,11 @@
         readonly property bool isLandscape: width >= height
 
         source: {
-            const landscapeFile = "5120x2880.png";
-            const portraitFile  = "1440x2960.png";
-
-            const lightFolder = "wallpapers/Next/contents/images/";
-            const darkFolder  = "wallpapers/Next/contents/images_dark/";
-            const themedFolder = Prepare.PrepareUtil.usingDarkTheme ? darkFolder : lightFolder;
+            const landscapeFile = "oreon-day.jpg";
+            const portraitFile  = "oreon-day.jpg";
+            const lightFolder = "backgrounds/";
+            const darkFolder  = "backgrounds/";
+            const themedFolder = lightFolder;
 
             function locate(file) {
                 return StandardPaths.locate(StandardPaths.GenericDataLocation, themedFolder + file)
