diff --git a/dbf/_index.py b/dbf/_index.py
index 6894106..e301f29 100755
--- a/dbf/_index.py
+++ b/dbf/_index.py
@@ -9,8 +9,8 @@ class IndexFile(object):
         filename += '.pdx'
         if not os.path.exists(filename):
             self.index_file = open(filename, 'r+b')
-            self.index_file.write('\xea\xaf\x37\xbf'    # signature
-                '\x00'*8                                # two non-existant lists
+            self.index_file.write('\xea\xaf\x37\xbf' +    # signature
+                '\x00'*8 +                               # two non-existant lists
                 '\x00'*500)                             # and no indices
             return
         index_file = self.index_file = open(filename, 'r+b')
