--- ceph-20.3.0/src/pybind/rbd/rbd.pyx.orig	2026-03-11 07:32:09.000000000 -0400
+++ ceph-20.3.0/src/pybind/rbd/rbd.pyx	2026-03-17 14:03:14.089738661 -0400
@@ -399,10 +399,10 @@
     return <rados_ioctx_t>ioctx.io
 {{endif}}
 
-cdef int progress_callback(uint64_t offset, uint64_t total, void* ptr) with gil:
+cdef int progress_callback(uint64_t offset, uint64_t total, void* ptr) noexcept with gil:
     return (<object>ptr)(offset, total)
 
-cdef int no_op_progress_callback(uint64_t offset, uint64_t total, void* ptr):
+cdef int no_op_progress_callback(uint64_t offset, uint64_t total, void* ptr) noexcept:
     return 0
 
 def cstr(val, name, encoding="utf-8", opt=False):
@@ -454,7 +454,7 @@
 
 cdef class Completion
 
-cdef void __aio_complete_cb(rbd_completion_t completion, void *args) with gil:
+cdef void __aio_complete_cb(rbd_completion_t completion, void *args) noexcept with gil:
     """
     Callback to oncomplete() for asynchronous operations
     """
