From c636ff223ec4981dd7083d6f35857e0fe5771cf8 Mon Sep 17 00:00:00 2001
From: Thomas Moschny <thomas.moschny@gmx.de>
Date: Sat, 30 Aug 2025 15:46:05 +0200
Subject: [PATCH] Fix compiler warning about variably modified array at file
 scope.

---
 src/stat/basic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stat/basic.c b/src/stat/basic.c
index 286d389..88066c7 100755
--- a/src/stat/basic.c
+++ b/src/stat/basic.c
@@ -56,7 +56,7 @@
  */
 #define MAX_LIFETIME	100.0	/* max. conn. lifetime in seconds */
 #define BIN_WIDTH	1e-3
-#define NUM_BINS	((u_int) (MAX_LIFETIME / BIN_WIDTH))
+enum { NUM_BINS = (u_int)(MAX_LIFETIME / BIN_WIDTH) };
 
 static struct {
 	u_long           num_conns_issued;	/* total # of connections * issued */
-- 
2.51.0

