diff --git a/src/main/java/com/lambdaworks/jni/LibraryLoaders.java b/src/main/java/com/lambdaworks/jni/LibraryLoaders.java
index 1d87e3a..e710c45 100644
--- a/src/main/java/com/lambdaworks/jni/LibraryLoaders.java
+++ b/src/main/java/com/lambdaworks/jni/LibraryLoaders.java
@@ -19,16 +19,6 @@ public class LibraryLoaders {
      * @return the loader.
      */
     public static LibraryLoader loader() {
-        String type = System.getProperty("com.lambdaworks.jni.loader");
-
-        if (type != null) {
-            if (type.equals("sys")) return new SysLibraryLoader();
-            if (type.equals("nil")) return new NilLibraryLoader();
-            if (type.equals("jar")) return new JarLibraryLoader();
-            throw new IllegalStateException("Illegal value for com.lambdaworks.jni.loader: " + type);
-        }
-
-        String vmSpec = System.getProperty("java.vm.specification.name");
-        return vmSpec.startsWith("Java") ? new JarLibraryLoader() : new SysLibraryLoader();
+        return new NilLibraryLoader();
     }
 }
