--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -546,6 +546,10 @@
 set_target_properties(${PROJECT_NAME} PROPERTIES
     VERSION 5.5.0
     SOVERSION 5)
+if(MINGW)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SUFFIX "-5${CMAKE_SHARED_LIBRARY_SUFFIX}")
+endif(MINGW)
+
 list(APPEND CMAKE_TARGETS ${PROJECT_NAME})
 # If json-c is used as subroject it set to target correct interface -I flags and allow
 # to build external target without extra include_directories(...)
