--- ebview-0.3.6.2/src/jcode.c.inline	2009-08-16 07:50:28.000000000 +0900
+++ ebview-0.3.6.2/src/jcode.c	2015-06-17 22:02:01.694526835 +0900
@@ -275,7 +275,7 @@ gchar *iconv_convert2(const gchar *icode
 	return(result);
 }
 
-inline gboolean isjisp(const gchar *buff){
+gboolean isjisp(const gchar *buff){
         g_assert(buff != NULL);
 
         if((buff[0] >= 0x21) && (buff[0] <= 0x74) &&
--- ebview-0.3.6.2/src/jcode.h.inline	2004-02-15 22:59:41.000000000 +0900
+++ ebview-0.3.6.2/src/jcode.h	2015-06-17 22:01:44.623394112 +0900
@@ -31,7 +31,7 @@ enum {
 
 gchar *iconv_convert(const gchar *icode, const gchar *ocode, const gchar *inbuf);
 gchar *iconv_convert2(const gchar *icode, const gchar *ocode, const gchar *orig);
-inline gboolean isjisp(const gchar *buff);
+gboolean isjisp(const gchar *buff);
 gboolean iseuckanji(const guchar *buff);
 gboolean iseuchiragana(const guchar *buff);
 gboolean iseuckatakana(const guchar *buff);
