diff -up email2trac-2.8.4/Makefile.in.installperms email2trac-2.8.4/Makefile.in
--- email2trac-2.8.4/Makefile.in.installperms	2010-03-24 11:09:47.000000000 +0100
+++ email2trac-2.8.4/Makefile.in	2014-05-10 13:59:42.802745304 +0200
@@ -37,15 +37,15 @@ install: all install-conf
 	for script in $(PYTHON_FILES) ; \
 	do \
           name=`basename $$script '.py'` ;\
-	  $(INSTALL) -m 755 -o $(INSTALL_USER) $$script  $(DESTDIR)$(EXEC_PREFIX)/$$name ;\
+	  $(INSTALL) -m 755 $$script  $(DESTDIR)$(EXEC_PREFIX)/$$name ;\
 	done
 
-	$(INSTALL) -m 4111 -o  $(INSTALL_USER) run_email2trac $(DESTDIR)$(EXEC_PREFIX)
+	$(INSTALL) -m 755 run_email2trac $(DESTDIR)$(EXEC_PREFIX)
 
 install-conf:
 	if [ ! -f  $(DESTDIR)/$(CONFIG)/email2trac.conf ] ; \
 	then \
-		$(INSTALL) -c -m 644 -o $(INSTALL_USER) email2trac.conf $(DESTDIR)/$(CONFIG)/email2trac.conf ;\
+		$(INSTALL) -c -m 644 email2trac.conf $(DESTDIR)/$(CONFIG)/email2trac.conf ;\
 	fi
  
 clean:
