diff --git a/Makefile b/Makefile
index e0284b8..ee2feec 100644
--- a/Makefile
+++ b/Makefile
@@ -43,13 +43,11 @@ dist: clean
 	@rm -rf dvtm-${VERSION}
 
 install: dvtm
-	@echo stripping executable
-	@${STRIP} dvtm
 	@echo installing executable file to ${DESTDIR}${PREFIX}/bin
 	@mkdir -p ${DESTDIR}${PREFIX}/bin
-	@cp -f dvtm ${DESTDIR}${PREFIX}/bin
+	@cp -pf dvtm ${DESTDIR}${PREFIX}/bin
 	@chmod 755 ${DESTDIR}${PREFIX}/bin/dvtm
-	@cp -f dvtm-status ${DESTDIR}${PREFIX}/bin
+	@cp -pf dvtm-status ${DESTDIR}${PREFIX}/bin
 	@chmod 755 ${DESTDIR}${PREFIX}/bin/dvtm-status
 	@echo installing manual page to ${DESTDIR}${MANPREFIX}/man1
 	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
diff --git a/config.mk b/config.mk
index 273aa45..e4a2e85 100644
--- a/config.mk
+++ b/config.mk
@@ -12,8 +12,8 @@ TERMINFO := ${DESTDIR}${PREFIX}/share/terminfo
 INCS = -I.
 LIBS = -lc -lutil -lncursesw
 CPPFLAGS = -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED
-CFLAGS += -std=c99 ${INCS} -DVERSION=\"${VERSION}\" -DNDEBUG ${CPPFLAGS}
-LDFLAGS += ${LIBS}
+CFLAGS += -std=c99 ${INCS} -DVERSION=\"${VERSION}\" -DNDEBUG ${CPPFLAGS} ${RPM_OPT_FLAGS}
+LDFLAGS += ${LIBS} -znow
 
 DEBUG_CFLAGS = ${CFLAGS} -UNDEBUG -O0 -g -ggdb -Wall -Wextra -Wno-unused-parameter
 
