From b073caac367e5c1dd0362240296187df1adc7b07 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADt=20Ondruch?= <vondruch@redhat.com>
Date: Mon, 7 Nov 2022 18:18:06 +0100
Subject: [PATCH] Fix kwargs handling compatibility with rspec-mocks 3.12+

---
 spec/memfs/file_spec.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/spec/memfs/file_spec.rb b/spec/memfs/file_spec.rb
index a9681f1..ab8a686 100644
--- a/spec/memfs/file_spec.rb
+++ b/spec/memfs/file_spec.rb
@@ -1073,7 +1073,7 @@ module MemFs
         it 'passes the contained options to +open+' do
           expect(described_class).to \
             receive(:open)
-            .with('/test-file', File::RDONLY, encoding: 'UTF-8')
+            .with('/test-file', File::RDONLY, { encoding: 'UTF-8' })
             .and_return(subject)
 
           described_class.read '/test-file', encoding: 'UTF-8'
