diff -ur iwidgets4.0.1/demos/catalog iwidgets4.0.1-wish85/demos/catalog
--- iwidgets4.0.1/demos/catalog	2001-09-06 22:10:18.000000000 +0700
+++ iwidgets4.0.1-wish85/demos/catalog	2012-08-28 15:59:13.803922818 +0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env tclsh
 # ----------------------------------------------------------------------
 #  PROGRAM: demo program for [incr Widgets]
 # ----------------------------------------------------------------------
@@ -9,8 +9,7 @@
 # ======================================================================
 #  Copyright (c) 1993-1998  Lucent Technologies, Inc.
 # ======================================================================
-#\
-exec wish8.4 "$0"
+#
 
 package require Iwidgets 4.0
 
