diff -up xpilot-ng-4.7.3/src/server/suibotdef.c~ xpilot-ng-4.7.3/src/server/suibotdef.c
--- xpilot-ng-4.7.3/src/server/suibotdef.c~	2026-01-24 18:58:21.000000000 +0100
+++ xpilot-ng-4.7.3/src/server/suibotdef.c	2026-01-24 18:59:54.806510851 +0100
@@ -378,7 +378,6 @@ typedef struct {
 } object_proximity_t;
 
 
-static bool Get_object_proximity();
 static bool Get_object_proximity(player_t *pl, object_t *shot, double sqmaxdist,int maxtime, object_proximity_t *object_proximity){
    /* get square of closest distance between player and object
     * compare with sqmaxdist and maxtime and return sqdistance and time
