From 5562bdc06749ae63a37986a55a7ea06b7f85aeff Mon Sep 17 00:00:00 2001
From: spike <spike@fedoraproject.org>
Date: Mon, 1 Nov 2021 10:02:04 +0100
Subject: [PATCH] Remove hard coded install_dir and use libdir instead

---
 meson.build | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 49c04f5..f35dbf6 100644
--- a/meson.build
+++ b/meson.build
@@ -267,9 +267,7 @@ buildtarget = shared_library('pam_mysql', [
   'src/password.c',
   'src/md5.c',
   'src/pam_calls.c',
-], dependencies: deps, install: true, include_directories: configuration_inc, install_dir: '/lib/security')
-
-meson.add_install_script('install.sh')
+], dependencies: deps, install: true, include_directories: configuration_inc, install_dir: join_paths(get_option('libdir'), 'security'))
 
 test_acct_mgmt = executable('test_acct_mgmt',
   sources: [
-- 
2.31.1

