--- anstream-0.6.21/Cargo.toml	1969-12-31 16:00:01.000000000 -0800
+++ anstream-0.6.21/Cargo.toml	2026-08-10 08:04:19.441103734 -0700
@@ -89,10 +89,8 @@
 auto = ["dep:anstyle-query"]
 default = [
     "auto",
-    "wincon",
 ]
 test = []
-wincon = ["dep:anstyle-wincon"]
 
 [lib]
 name = "anstream"
@@ -142,9 +140,6 @@
 [dev-dependencies.strip-ansi-escapes]
 version = "0.2.1"
 
-[target."cfg(windows)".dependencies.anstyle-wincon]
-version = "3.0.5"
-optional = true
 
 [lints.clippy]
 bool_assert_comparison = "allow"
