From a0250c672be4620bce7c05325b2525c24a5ed9b4 Mon Sep 17 00:00:00 2001
From: Andy Li <andy@onthewings.net>
Date: Tue, 2 Nov 2021 02:09:20 +0800
Subject: [PATCH] remove stdlib-shims dep

---
 dune         | 1 -
 dune-project | 1 -
 sha.opam     | 1 -
 3 files changed, 3 deletions(-)

diff --git a/dune b/dune
index 92b596c..d6b303d 100644
--- a/dune
+++ b/dune
@@ -9,7 +9,6 @@
  (synopsis "SHA-1 and SHA-2 family implementations")
  (wrapped false)
  (modules sha1 sha256 sha512)
- (libraries stdlib-shims)
  (foreign_stubs
   (language c)
   (flags
diff --git a/dune-project b/dune-project
index a21fcf4..6c9e9da 100644
--- a/dune-project
+++ b/dune-project
@@ -32,7 +32,6 @@ interface than the MD5 digest included in the OCaml standard library.
 It's currently providing SHA1, SHA256 and SHA512 hash functions.")
  (depends
   (ocaml (>= 4.02))
-  (stdlib-shims (>= 0.3.0))
   (ounit2 :with-test)))
 
 (generate_opam_files true)
diff --git a/sha.opam b/sha.opam
index 9ccea31..eebd8e8 100644
--- a/sha.opam
+++ b/sha.opam
@@ -28,7 +28,6 @@ bug-reports: "https://github.com/djs55/ocaml-sha/issues"
 depends: [
   "dune" {>= "2.9"}
   "ocaml" {>= "4.02"}
-  "stdlib-shims" {>= "0.3.0"}
   "ounit2" {with-test}
   "odoc" {with-doc}
 ]
-- 
2.33.0.windows.2

