--- data/css_decompressor
+++ data/css_decompressor
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 
--- data/cssmin.py
+++ data/cssmin.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 """`cssmin` - A Python port of the YUI CSS compressor."""
--- data/jsmin.py
+++ data/jsmin.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 # This code is original from jsmin by Douglas Crockford, it was translated to
 # Python by Baruch Even. It was rewritten by Dave St.Germain for speed.
--- data/json_prettyprint.py
+++ data/json_prettyprint.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import json
 import sys
--- data/lorem-ipsum-generator
+++ data/lorem-ipsum-generator
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import sys
 from os.path import abspath, exists
--- src/plugin_zencoding/zencoding/actions/basic.py
+++ src/plugin_zencoding/zencoding/actions/basic.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 """
--- src/plugin_zencoding/zencoding/actions/__init__.py
+++ src/plugin_zencoding/zencoding/actions/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 # import all actions
@@ -12,4 +12,4 @@ for file in os.listdir(os.path.dirname(_
 		
 __import__(__name__, globals(), locals(), __sub_modules)
 
-del __sub_modules
\ No newline at end of file
+del __sub_modules
--- src/plugin_zencoding/zencoding/actions/token.py
+++ src/plugin_zencoding/zencoding/actions/token.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
--- src/plugin_zencoding/zencoding/filters/comment.py
+++ src/plugin_zencoding/zencoding/filters/comment.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -44,4 +44,4 @@ def process(tree, profile):
 			add_comments(item, i)
 		process(item, profile)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/css.py
+++ src/plugin_zencoding/zencoding/filters/css.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 '''
 Process CSS properties: replaces snippets, augumented with ! char, with 
@@ -20,4 +20,4 @@ def process(tree, profile):
 		
 		process(item, profile)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/escape.py
+++ src/plugin_zencoding/zencoding/filters/escape.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -28,4 +28,4 @@ def process(tree, profile=None):
 		
 		process(item)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/format-css.py
+++ src/plugin_zencoding/zencoding/filters/format-css.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -21,4 +21,4 @@ def process(tree, profile):
 		
 		process(item, profile)
 		
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/format.py
+++ src/plugin_zencoding/zencoding/filters/format.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 """
@@ -179,4 +179,4 @@ def process(tree, profile, level=0):
 			
 		process(item, profile, level + 1)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/haml.py
+++ src/plugin_zencoding/zencoding/filters/haml.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -157,4 +157,4 @@ def process(tree, profile, level=0):
 		
 		process(item, profile, level + 1)
 		
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/html.py
+++ src/plugin_zencoding/zencoding/filters/html.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
--- src/plugin_zencoding/zencoding/filters/__init__.py
+++ src/plugin_zencoding/zencoding/filters/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 import os.path
@@ -14,4 +14,4 @@ for file in os.listdir(os.path.dirname(_
 		
 __import__(__name__, globals(), locals(), __sub_modules)
 
-del __sub_modules
\ No newline at end of file
+del __sub_modules
--- src/plugin_zencoding/zencoding/filters/single-line.py
+++ src/plugin_zencoding/zencoding/filters/single-line.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 '''
 Output abbreviation on a single line (i.e. no line breaks)
@@ -26,4 +26,4 @@ def process(tree, profile):
 		
 		process(item, profile)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/trim.py
+++ src/plugin_zencoding/zencoding/filters/trim.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 '''
 Trim filter: removes characters at the beginning of the text
@@ -19,4 +19,4 @@ def process(tree, profile):
 		
 		process(item, profile)
 	
-	return tree
\ No newline at end of file
+	return tree
--- src/plugin_zencoding/zencoding/filters/xsl.py
+++ src/plugin_zencoding/zencoding/filters/xsl.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -30,4 +30,4 @@ def process(tree, profile):
 		if item.type == 'tag' and item.name.lower() in tags and item.children:
 			trim_attribute(item)
 		
-		process(item, profile)
\ No newline at end of file
+		process(item, profile)
--- src/plugin_zencoding/zencoding/html_matcher.py
+++ src/plugin_zencoding/zencoding/html_matcher.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 '''
@@ -270,4 +270,4 @@ def _find_pair(html, start_ix, mode='xht
 				
 			ix += 1
 	
-	return action(opening_tag, closing_tag, start_ix)
\ No newline at end of file
+	return action(opening_tag, closing_tag, start_ix)
--- src/plugin_zencoding/zencoding/resources.py
+++ src/plugin_zencoding/zencoding/resources.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 '''
 @author Sergey Chikuyonok (serge.che@gmail.com)
--- src/plugin_zencoding/zencoding/utils.py
+++ src/plugin_zencoding/zencoding/utils.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 '''
 Core Zen Coding library. Contains various text manipulation functions:
