diff --git a/src/ocr/ppocr/ppocrv5.cpp b/src/ocr/ppocr/ppocrv5.cpp
index 2a7c9f3..0a11093 100644
--- a/src/ocr/ppocr/ppocrv5.cpp
+++ b/src/ocr/ppocr/ppocrv5.cpp
@@ -309,11 +309,11 @@ void PaddleOCRAppV5::rec(const std::vector<cv::Mat> &detectImg)
         input.substract_mean_normalize(mean_vals, norm_vals);
         auto outIndexes = recNet->output_indexes();
         ncnn::Extractor extractor = recNet->create_extractor();
-        if (recNet->opt.use_vulkan_compute) {
-            if (maxThreadsUsed > 1 && i % maxThreadsUsed != 1) {
-                extractor.set_vulkan_compute(false);
-            }
-        }
+        //if (recNet->opt.use_vulkan_compute) {
+        //    if (maxThreadsUsed > 1 && i % maxThreadsUsed != 1) {
+        //        extractor.set_vulkan_compute(false);
+        //    }
+        //}
 
         extractor.input(0, input);
         ncnn::Mat out;
