diff -r -U4 freedroid-1.2.3--orig/configure.ac freedroid-1.2.3--patched/configure.ac
--- freedroid-1.2.3--orig/configure.ac	2024-01-08 18:13:47.000000000 +0100
+++ freedroid-1.2.3--patched/configure.ac	2025-01-26 13:36:04.055353087 +0100
@@ -89,11 +89,11 @@
 (see see http://www.libsdl.org/)
 --> compiling without sound support
 --------------------------------------------------]))
 
-AC_CHECK_LIB([vorbis], [main],, AC_MSG_WARN([
+AC_CHECK_LIB([vorbisfile], [main],, AC_MSG_WARN([
 --------------------------------------------------
-libvorbis not found!
+libvorbisfile not found!
 You need the Vorbis libs installed if you want
 Freedroid to be able to play Ogg files (e.g. the Intro theme)
 --------------------------------------------------]))
 
