--- dolby_vision-3.4.0/Cargo.toml
+++ dolby_vision-3.4.0/Cargo.toml
@@ -59,11 +59,7 @@
 name = "dolby_vision"
 path = "src/lib.rs"
 doctest = false
-
-[[bench]]
-name = "bench_main"
-path = "benches/bench_main.rs"
-harness = false
+crate-type = ["lib", "cdylib"]
 
 [dependencies.anyhow]
 version = "1.0.102"
@@ -101,9 +97,6 @@
 version = "1.11.0"
 features = ["rustc_1_55"]
 
-[dev-dependencies.criterion]
-version = "0.8.2"
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
