diff -up simde-0.8.2/simde/arm/neon/mul_lane.h.orig simde-0.8.2/simde/arm/neon/mul_lane.h
--- simde-0.8.2/simde/arm/neon/mul_lane.h.orig	2024-04-30 18:35:50.000000000 +0200
+++ simde-0.8.2/simde/arm/neon/mul_lane.h	2025-06-23 13:15:24.431097445 +0200
@@ -386,7 +386,7 @@ simde_vmul_laneq_s32(simde_int32x2_t a,
   #else
     SIMDE_VECTORIZE
     for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
-      r_.values[i] = a_.values[i] * b_.values[lane];
+      r_.values[i] = (uint32_t)a_.values[i] * (uint32_t)b_.values[lane];
     }
   #endif
 
@@ -772,7 +772,7 @@ simde_vmulq_laneq_s32(simde_int32x4_t a,
   #else
     SIMDE_VECTORIZE
     for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
-      r_.values[i] = a_.values[i] * b_.values[lane];
+      r_.values[i] = (uint32_t)a_.values[i] * (uint32_t)b_.values[lane];
     }
   #endif
 
