From 0de7c0986fba5492c27fb8159f6be4d57cb2fdb6 Mon Sep 17 00:00:00 2001
From: topazus <topazus@outlook.com>
Date: Tue, 24 Oct 2023 22:16:12 +0800
Subject: [PATCH] Enable testing with CMake and ctest

---
 CMakeLists.txt       | 1 +
 tests/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c01de65..ba69d2d 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ if(SCIPLOT_BUILD_EXAMPLES)
 endif()
 
 if(SCIPLOT_BUILD_TESTS)
+    enable_testing()
     add_subdirectory(tests)
 endif()
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index d15c55a..84c8e00 100755
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -3,6 +3,7 @@ file(GLOB_RECURSE FILES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.test.cxx)
 add_executable(sciplot-cpptests ${FILES})
 target_link_libraries(sciplot-cpptests sciplot)
 target_include_directories(sciplot-cpptests PUBLIC ${PROJECT_SOURCE_DIR})
+add_test(NAME sciplot-cpptests COMMAND sciplot-cpptests)
 
 # Test for linking errors (due to missing inline specifiers)
 add_subdirectory(testing-project)
