--- gix-index-0.45.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ gix-index-0.45.1/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -118,16 +118,7 @@
 
 [dev-dependencies]
 
-[target."cfg(not(windows))".dependencies.libc]
-version = "0.2.178"
-
-[target."cfg(not(windows))".dependencies.rustix]
-version = "1.1.2"
-features = [
-    "std",
-    "fs",
-]
-default-features = false
+
 
 [lints.clippy]
 bool_to_int_with_if = "allow"
