diff -up bijiben-40.1 bijiben-40
diff -up bijiben-40.1/meson.build.1 bijiben-40.1/meson.build
--- bijiben-40.1/meson.build.1	2022-07-20 09:56:11.815124613 +0200
+++ bijiben-40.1/meson.build	2022-07-20 09:56:57.023115055 +0200
@@ -120,13 +120,13 @@ bijiben_deps = [
   dependency('goa-1.0'),
   dependency('gtk+-3.0', version: '>= 3.19.3'),
   dependency('json-glib-1.0'),
-  dependency('libecal-2.0', version: '>= 3.33.2'),
-  dependency('libedataserver-1.2', version: '>= 3.13.90'),
+  dependency('libecal-2.0', version: '>= 3.45.1'),
+  dependency('libedataserver-1.2', version: '>= 3.45.1'),
   dependency('libhandy-1', version: '>= 1.0.0'),
   dependency('libxml-2.0'),
   dependency('libcurl'),
   dependency('uuid'),
-  dependency('webkit2gtk-4.1', version: '>= 2.26'),
+  dependency('webkit2gtk-4.1', version: '>= 2.36'),
   cc.find_library('m')
 ]
 
