--- hashbrown-0.17.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hashbrown-0.17.0/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -148,10 +148,6 @@
 version = "3.13.0"
 features = ["allocator-api2"]
 
-[dev-dependencies.criterion]
-version = "0.7"
-features = ["html_reports"]
-
 [dev-dependencies.fnv]
 version = "1.0.7"
 
@@ -165,7 +161,7 @@
 [dev-dependencies.serde_test]
 version = "1.0"
 
-[target."cfg(any())".dependencies.serde]
+[dependencies.serde]
 version = "1.0.220"
 optional = true
 default-features = false
