--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -566,8 +566,6 @@

 # Install root-level documents
 if(NOT SKBUILD)
-    install(FILES LICENSE CHANGELOG.md README.md THIRD-PARTY.md DESTINATION "${CMAKE_INSTALL_DOCDIR}")
-
     set(MATERIALX_GEN_CONFIG_PATH "${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME}")

     include(CMakePackageConfigHelpers)
--- a/source/MaterialXTest/CMakeLists.txt
+++ b/source/MaterialXTest/CMakeLists.txt
@@ -104,11 +104,6 @@
     add_dependencies(MaterialXTestData MaterialXBuildData)
 endif()

-if(MATERIALX_BUILD_GEN_MDL)
-    install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../source/MaterialXGenMdl/mdl/"
-        DESTINATION ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${MATERIALX_INSTALL_MDL_MODULE_PATH}/mdl)
-endif()
-
 add_custom_command(TARGET MaterialXTest POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E copy_directory
     ${CMAKE_CURRENT_SOURCE_DIR}/../../resources ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/resources)
