--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@ set(MATERIALX_LIBRARY_VERSION ${MATERIALX_MAJOR_VERSION}.${MATERIALX_MINOR_VERSI

 # CMake setup
 cmake_minimum_required(VERSION 3.26)
+include(GNUInstallDirs)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 set(CMAKE_MACOSX_RPATH ON)
@@ -120,10 +121,10 @@ set(MATERIALX_PYTHON_PYBIND11_DIR "" CACHE PATH
     "Path to a folder containing the PyBind11 source to be used in building MaterialX Python.")

 # Settings to define installation layout
-set(MATERIALX_INSTALL_INCLUDE_PATH "include" CACHE STRING "Install header include path (e.g. 'inc', 'include').")
-set(MATERIALX_INSTALL_BIN_PATH "bin" CACHE STRING "Install bin path (e.g. 'bin').")
-set(MATERIALX_INSTALL_LIB_PATH "lib" CACHE STRING "Install lib path (e.g. 'libs', 'lib').")
-set(MATERIALX_INSTALL_STDLIB_PATH "libraries" CACHE STRING "Install path for mtlx std libs (e.g. 'libraries').")
+set(MATERIALX_INSTALL_INCLUDE_PATH "${CMAKE_INSTALL_INCLUDEDIR}" CACHE STRING "Install header include path (e.g. 'inc', 'include').")
+set(MATERIALX_INSTALL_BIN_PATH "${CMAKE_INSTALL_BINDIR}" CACHE STRING "Install bin path (e.g. 'bin').")
+set(MATERIALX_INSTALL_LIB_PATH "${CMAKE_INSTALL_LIBDIR}" CACHE STRING "Install lib path (e.g. 'libs', 'lib').")
+set(MATERIALX_INSTALL_STDLIB_PATH "${CMAKE_INSTALL_DATADIR}/materialx/libraries" CACHE STRING "Install path for mtlx std libs (e.g. 'libraries').")

 # Helpers for OSL validation
 set(MATERIALX_OSL_BINARY_OSLC "" CACHE FILEPATH "Full path to the OSL compiler binary.")
@@ -565,9 +566,9 @@ endif()

 # Install root-level documents
 if(NOT SKBUILD)
-    install(FILES LICENSE CHANGELOG.md README.md THIRD-PARTY.md DESTINATION .)
+    install(FILES LICENSE CHANGELOG.md README.md THIRD-PARTY.md DESTINATION "${CMAKE_INSTALL_DOCDIR}")

-    set(MATERIALX_GEN_CONFIG_PATH "${MATERIALX_INSTALL_LIB_PATH}/cmake/${CMAKE_PROJECT_NAME}")
+    set(MATERIALX_GEN_CONFIG_PATH "${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME}")

     include(CMakePackageConfigHelpers)

--- a/resources/CMakeLists.txt
+++ b/resources/CMakeLists.txt
@@ -1,4 +1,4 @@
 if(NOT SKBUILD)
     install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
-            DESTINATION "resources" MESSAGE_NEVER)
+            DESTINATION "${CMAKE_INSTALL_DATADIR}/materialx/resources" MESSAGE_NEVER)
 endif()
