--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -14,5 +14,6 @@ endif()
 if(MATERIALX_INSTALL_PYTHON AND PYTHON_EXECUTABLE AND NOT SKBUILD)
-  set(SETUP_PY "${CMAKE_INSTALL_PREFIX}/python/setup.py")
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in" "${SETUP_PY}")
+  set(SETUP_PY "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in" "${SETUP_PY}")
+  install(FILES "${SETUP_PY}" DESTINATION "python")
   install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} -m pip install . WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/python)")
 endif()
