diff --git a/CMakeLists.txt b/CMakeLists.txt
index f46144d..3918095 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ project(target-isns "C")
 set(TARGET_ISNS_VERSION "0.6.8")
 
 cmake_minimum_required(VERSION 3.1)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wno-stringop-overflow -Wno-stringop-truncation")
 
 option(SUPPORT_SYSTEMD "Support service control via systemd" OFF)
 
