diff -up pyscf-2.6.0/pyscf/lib/CMakeLists.txt.rpath pyscf-2.6.0/pyscf/lib/CMakeLists.txt
--- pyscf-2.6.0/pyscf/lib/CMakeLists.txt.rpath	2024-06-05 07:49:31.865204187 +0000
+++ pyscf-2.6.0/pyscf/lib/CMakeLists.txt	2024-06-05 07:50:07.691565634 +0000
@@ -123,19 +123,6 @@ configure_file(
 # to find config.h
 include_directories("${PROJECT_BINARY_DIR}")
 
-# See also https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
-if (WIN32)
-  #?
-elseif (APPLE)
-  set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-  set(CMAKE_INSTALL_RPATH "@loader_path;@loader_path/deps/lib;@loader_path/deps/lib64")
-  set(CMAKE_BUILD_RPATH "@loader_path;@loader_path/deps/lib;@loader_path/deps/lib64")
-else ()
-  set(CMAKE_SKIP_BUILD_RPATH  True)
-  set(CMAKE_BUILD_WITH_INSTALL_RPATH True)
-  set(CMAKE_INSTALL_RPATH "\$ORIGIN:\$ORIGIN/deps/lib:\$ORIGIN/deps/lib64")
-endif ()
-
 option(ENABLE_FFTW "Using fftw3" OFF)
 option(BUILD_FFTW "Building fftw3" OFF)
 
