diff --git a/generic/mysqltcl.c b/generic/mysqltcl.c
index 53195b9..f915abc 100755
--- a/generic/mysqltcl.c
+++ b/generic/mysqltcl.c
@@ -47,7 +47,7 @@
 #include <ctype.h>
 #include <stdlib.h>
 
-#define MYSQL_SMALL_SIZE  TCL_RESULT_SIZE /* Smaller buffer size. */
+#define MYSQL_SMALL_SIZE  200 /* Smaller buffer size. */
 #define MYSQL_NAME_LEN     80    /* Max. database name length. */
 /* #define PREPARED_STATEMENT */
 
@@ -117,11 +117,11 @@ static int Mysqltcl_State(ClientData clientData, Tcl_Interp *interp, int objc, T
 static int Mysqltcl_InsertId(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[]);
 static int Mysqltcl_Query(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[]);
 static int Mysqltcl_Receive(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[]);
-static int MysqlHandleSet _ANSI_ARGS_((Tcl_Interp *interp,Tcl_Obj *objPtr));
-static void MysqlHandleFree _ANSI_ARGS_((Tcl_Obj *objPtr));
-static int MysqlNullSet _ANSI_ARGS_((Tcl_Interp *interp,Tcl_Obj *objPtr));
+static int MysqlHandleSet(Tcl_Interp *interp,Tcl_Obj *objPtr);
+static void MysqlHandleFree(Tcl_Obj *objPtr);
+static int MysqlNullSet(Tcl_Interp *interp,Tcl_Obj *objPtr);
 static Tcl_Obj *Mysqltcl_NewNullObj(MysqltclState *mysqltclState);
-static void UpdateStringOfNull _ANSI_ARGS_((Tcl_Obj *objPtr));
+static void UpdateStringOfNull(Tcl_Obj *objPtr);
 
 /* handle object type 
  * This section defince funtions for Handling new Tcl_Obj type */
@@ -301,7 +301,7 @@ clear_msg(Tcl_Interp *interp)
  * mysqlstatus(command).
  */
 
-static void mysql_reassemble(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[])
+static void mysql_reassemble(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[])
 {
    set_statusArr(interp,MYSQL_STATUS_CMD,Tcl_NewListObj(objc, objv));
 }
@@ -333,7 +333,7 @@ static void freeResult(MysqlTclHandle *handle)
  *
  */
 
-static int mysql_prim_confl(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],char *msg)
+static int mysql_prim_confl(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[],char *msg)
 {
   set_statusArr(interp,MYSQL_STATUS_CODE,Tcl_NewIntObj(-1));
 
@@ -356,7 +356,7 @@ static int mysql_prim_confl(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],ch
  * If no error occurs it returns TCL_OK
  */
 
-static int mysql_server_confl(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],MYSQL * connection)
+static int mysql_server_confl(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[],MYSQL * connection)
 {
   const char* mysql_errorMsg;
   if (mysql_errno(connection)) {
@@ -380,7 +380,7 @@ static int mysql_server_confl(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],
   }
 }
 
-static  MysqlTclHandle *get_handle(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],int check_level) 
+static  MysqlTclHandle *get_handle(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[],int check_level) 
 {
   MysqlTclHandle *handle;
   if (GetHandleFromObj(interp, objv[1], &handle) != TCL_OK) {
@@ -417,8 +417,12 @@ static  MysqlTclHandle *get_handle(Tcl_Interp *interp,int objc,Tcl_Obj *CONST ob
 static int mysql_QueryTclObj(MysqlTclHandle *handle,Tcl_Obj *obj)
 {
   char *query;
-  int result,queryLen;
-
+  int result;
+#ifdef TCL_SIZE_MAX
+  Tcl_Size queryLen;
+#else
+  int queryLen;
+#endif
   Tcl_DString queryDS;
 
   query=Tcl_GetStringFromObj(obj, &queryLen);
@@ -460,7 +464,7 @@ static MysqlTclHandle *createMysqlHandle(MysqltclState *statePtr)
   handle=(MysqlTclHandle *)Tcl_Alloc(sizeof(MysqlTclHandle));
   memset(handle,0,sizeof(MysqlTclHandle));
   if (handle == 0) {
-    panic("no memory for handle");
+    Tcl_Panic("no memory for handle");
     return handle;
   }
   handle->type = HT_CONNECTION;
@@ -528,7 +532,7 @@ static void closeHandle(MysqlTclHandle *handle)
  * SIDE EFFECT: Sets the Tcl result on failure.
  */
 
-static MysqlTclHandle *mysql_prologue(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],int req_min_args,int req_max_args,int check_level,char *usage_msg)
+static MysqlTclHandle *mysql_prologue(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[],int req_min_args,int req_max_args,int check_level,char *usage_msg)
 {
   /* Check number of args. */
   if (objc < req_min_args || objc > req_max_args) {
@@ -556,11 +560,11 @@ static MysqlTclHandle *mysql_prologue(Tcl_Interp *interp,int objc,Tcl_Obj *CONST
  * SIDE EFFECT: Sets the result and status on failure.
  */
 
-static Tcl_Obj *mysql_colinfo(Tcl_Interp *interp,int objc,Tcl_Obj *CONST objv[],MYSQL_FIELD* fld,Tcl_Obj * keyw)
+static Tcl_Obj *mysql_colinfo(Tcl_Interp *interp,int objc,Tcl_Obj *const objv[],MYSQL_FIELD* fld,Tcl_Obj * keyw)
 {
   int idx ;
 
-  static CONST char* MysqlColkey[] =
+  static const char* MysqlColkey[] =
     {
       "table", "name", "type", "length", "prim_key", "non_null", "numeric", "decimals", NULL
     };
@@ -720,7 +724,7 @@ static void Mysqltcl_Kill(ClientData clientData)
  *      TCL_ERROR - connect not successful - error message returned
  */
 
-static CONST char* MysqlConnectOpt[] =
+static const char* MysqlConnectOpt[] =
     {
       "-host", "-user", "-password", "-db", "-port", "-socket","-encoding",
       "-ssl", "-compress", "-noschema","-odbc",
@@ -900,7 +904,7 @@ static int Mysqltcl_Connect(ClientData clientData, Tcl_Interp *interp, int objc,
   handle = createMysqlHandle(statePtr);
 
   if (handle == 0) {
-    panic("no memory for handle");
+    Tcl_Panic("no memory for handle");
     return TCL_ERROR;
 
   }
@@ -963,7 +967,11 @@ static int Mysqltcl_Connect(ClientData clientData, Tcl_Interp *interp, int objc,
 
 static int Mysqltcl_Use(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[])
 {
+#ifdef TCL_SIZE_MAX
+  Tcl_Size len;
+#else
   int len;
+#endif
   char *db;
   MysqlTclHandle *handle;  
 
@@ -999,7 +1007,11 @@ static int Mysqltcl_Use(ClientData clientData, Tcl_Interp *interp, int objc, Tcl
 
 static int Mysqltcl_Escape(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[])
 {
+#ifdef TCL_SIZE_MAX
+  Tcl_Size len;
+#else
   int len;
+#endif
   char *inString, *outString;
   MysqlTclHandle *handle;
   
@@ -1050,7 +1062,7 @@ static int Mysqltcl_Sel(ClientData clientData, Tcl_Interp *interp, int objc, Tcl
   unsigned long *lengths;
 
 
-  static CONST char* selOptions[] = {"-list", "-flatlist", NULL};
+  static const char* selOptions[] = {"-list", "-flatlist", NULL};
   /* Warning !! no option number */
   int i,selOption=2,colCount;
   
@@ -1194,14 +1206,14 @@ static int Mysqltcl_Exec(ClientData clientData, Tcl_Interp *interp, int objc, Tc
 	MysqlTclHandle *handle;
 	int affected;
 	Tcl_Obj *resList;
-    if ((handle = mysql_prologue(interp, objc, objv, 3, 3, CL_CONN,"handle sql-statement")) == 0)
-    	return TCL_ERROR;
+	if ((handle = mysql_prologue(interp, objc, objv, 3, 3, CL_CONN,"handle sql-statement")) == 0)
+		return TCL_ERROR;
 
   	/* Flush any previous result. */
 	freeResult(handle);
 
 	if (mysql_QueryTclObj(handle,objv[2]))
-    	return mysql_server_confl(interp,objc,objv,handle->connection);
+		return mysql_server_confl(interp,objc,objv,handle->connection);
 
 	if ((affected=mysql_affected_rows(handle->connection)) < 0) affected=0;
 
@@ -1335,7 +1347,11 @@ static int Mysqltcl_Map(ClientData clientData, Tcl_Interp *interp, int objc, Tcl
 
   MysqlTclHandle *handle;
   int idx;
+#ifdef TCL_SIZE_MAX
+  Tcl_Size listObjc;
+#else
   int listObjc;
+#endif
   Tcl_Obj *tempObj,*varNameObj;
   MYSQL_ROW row;
   int *val;
@@ -1437,7 +1453,11 @@ static int Mysqltcl_Receive(ClientData clientData, Tcl_Interp *interp, int objc,
 
   MysqlTclHandle *handle;
   int idx;
+#ifdef TCL_SIZE_MAX
+  Tcl_Size listObjc;
+#else
   int listObjc;
+#endif
   Tcl_Obj *tempObj,*varNameObj;
   MYSQL_ROW row;
   int *val = NULL;
@@ -1545,7 +1565,7 @@ static int Mysqltcl_Info(ClientData clientData, Tcl_Interp *interp, int objc, Tc
   MYSQL_ROW row ;
   const char* val ;
   Tcl_Obj *resList;
-  static CONST char* MysqlDbOpt[] =
+  static const char* MysqlDbOpt[] =
     {
       "dbname", "dbname?", "tables", "host", "host?", "databases",
       "info","serverversion",
@@ -1678,7 +1698,7 @@ static int Mysqltcl_BaseInfo(ClientData clientData, Tcl_Interp *interp, int objc
   int idx ;
   Tcl_Obj *resList;
   char **option;
-  static CONST char* MysqlInfoOpt[] =
+  static const char* MysqlInfoOpt[] =
     {
       "connectparameters", "clientversion",
 #if (MYSQL_VERSION_ID >= 40107)
@@ -1738,7 +1758,7 @@ static int Mysqltcl_Result(ClientData clientData, Tcl_Interp *interp, int objc,
 {
   int idx ;
   MysqlTclHandle *handle;
-  static CONST char* MysqlResultOpt[] =
+  static const char* MysqlResultOpt[] =
     {
      "rows", "rows?", "cols", "cols?", "current", "current?", NULL
     };
@@ -1823,7 +1843,11 @@ static int Mysqltcl_Col(ClientData clientData, Tcl_Interp *interp, int objc, Tcl
   int current_db ;
   MysqlTclHandle *handle;
   int idx ;
-  int listObjc ;
+#ifdef TCL_SIZE_MAX
+  Tcl_Size listObjc;
+#else
+  int listObjc;
+#endif
   Tcl_Obj **listObjv, *colinfo, *resList, *resSubList;
   MYSQL_FIELD* fld ;
   MYSQL_RES* result ;
@@ -2001,7 +2025,11 @@ static int Mysqltcl_Ping(ClientData clientData, Tcl_Interp *interp, int objc, Tc
 static int Mysqltcl_ChangeUser(ClientData clientData, Tcl_Interp *interp, int objc, Tcl_Obj *const objv[])
 {
   MysqlTclHandle *handle;
+#ifdef TCL_SIZE_MAX
+  Tcl_Size len;
+#else
   int len;
+#endif
   char *user,*password,*database=NULL;
   
   if ((handle = mysql_prologue(interp, objc, objv, 4, 5, CL_CONN,
@@ -2256,7 +2284,7 @@ static int Mysqltcl_NewNull(ClientData clientData, Tcl_Interp *interp, int objc,
  *
  */
 #if (MYSQL_VERSION_ID >= 40107)
-static CONST char* MysqlServerOpt[] =
+static const char* MysqlServerOpt[] =
     {
       "-multi_statment_on", "-multi_statment_off", "-auto_reconnect_on", "-auto_reconnect_off", NULL
     };
@@ -2755,7 +2783,7 @@ int Mysqltcl_Init(interp)
    if (strlen(MysqlHandlePrefix) == MYSQL_HPREFIX_LEN)
      return TCL_OK;
    else {
-     panic("*** mysqltcl (mysqltcl.c): handle prefix inconsistency!\n");
+     Tcl_Panic("*** mysqltcl (mysqltcl.c): handle prefix inconsistency!\n");
      return TCL_ERROR ;
    }
 }
