diff -r -u dnssec-tools-2.2.3.orig/validator/libval/val_context.c dnssec-tools-2.2.3/validator/libval/val_context.c
--- dnssec-tools-2.2.3.orig/validator/libval/val_context.c	2024-11-05 11:39:33.696730353 +0100
+++ dnssec-tools-2.2.3/validator/libval/val_context.c	2024-11-05 11:44:01.428711438 +0100
@@ -17,9 +17,10 @@
 
 #include <openssl/conf.h>
 #include <openssl/evp.h>
+#ifndef OPENSSL_NO_ENGINE
 #include <openssl/engine.h>
+#endif /* !defined(OPENSSL_NO_ENGINE) */
 #include <openssl/err.h>
-
 #include "val_support.h"
 #include "val_policy.h"
 #include "val_cache.h"
@@ -853,7 +854,9 @@
     /* Whole bunch of openssl cleanup routines */
     CONF_modules_unload(1); 
     EVP_cleanup(); 
+#ifndef OPENSSL_NO_ENGINE
     ENGINE_cleanup(); 
+#endif
     CRYPTO_cleanup_all_ex_data(); 
     ERR_remove_state(0); 
     ERR_free_strings(); 
diff -r -u dnssec-tools-2.2.3.orig/validator/libval/val_crypto.c dnssec-tools-2.2.3/validator/libval/val_crypto.c
--- dnssec-tools-2.2.3.orig/validator/libval/val_crypto.c	2024-11-05 11:39:33.695730345 +0100
+++ dnssec-tools-2.2.3/validator/libval/val_crypto.c	2024-11-05 11:40:17.569919704 +0100
@@ -22,7 +22,9 @@
 #include <crypto/sha2.h>
 #endif
 #include <openssl/dsa.h>
+#ifndef OPENSSL_NO_ENGINE
 #include <openssl/engine.h>
+#endif /* !defined(OPENSSL_NO_ENGINE) */
 #include <openssl/md5.h>
 #include <openssl/rsa.h>
 #include <openssl/err.h>
