--- stgit-2.5.4/Cargo.toml	2025-08-12T20:41:59+00:00
+++ stgit-2.5.4/Cargo.toml	2025-08-12T01:39:57.231186+00:00
@@ -31,7 +31,7 @@ anyhow = "1.0"
 anstyle = { version = "1.0", features = ["std"] }
 bstr = { version = "1.11", default-features = false, features = ["std"] }
 bzip2-rs = "0.1"
-clap = { version = "~4.5", default-features = false, features = [
+clap = { version = "4.5", default-features = false, features = [
   "color",
   "help",
   "std",
@@ -56,7 +56,7 @@ strsim = "0.11"
 tar = "0.4"
 tempfile = "3"
 termcolor = "1.4"
-thiserror = "~2.0"
+thiserror = "2.0"
 winnow = "0.7.10"

 curl = { version = "0.4", optional = true }
