diff -rupN --no-dereference svg2svgt-0.9.6/gui/exportdialog.cpp svg2svgt-0.9.6-new/gui/exportdialog.cpp
--- svg2svgt-0.9.6/gui/exportdialog.cpp	2017-08-31 12:01:05.000000000 +0200
+++ svg2svgt-0.9.6-new/gui/exportdialog.cpp	2025-07-18 09:24:05.415834995 +0200
@@ -21,6 +21,7 @@
 
 #include <svg2svgt/tracer.h>
 
+#include <QtGui/QValidator>
 #include <QtWidgets/QColorDialog>
 
 using svg2svgt::Tracer;
diff -rupN --no-dereference svg2svgt-0.9.6/gui/svglistitemdelegate.cpp svg2svgt-0.9.6-new/gui/svglistitemdelegate.cpp
--- svg2svgt-0.9.6/gui/svglistitemdelegate.cpp	2017-08-31 12:01:05.000000000 +0200
+++ svg2svgt-0.9.6-new/gui/svglistitemdelegate.cpp	2025-07-18 09:24:05.416088562 +0200
@@ -19,6 +19,7 @@
 #include "svglistitemdelegate.h"
 
 #include <QtGui/QPainter>
+#include <QtGui/QPainterPath>
 #include <QtGui/QPixmapCache>
 #include <QtSvg/QSvgRenderer>
 
