diff -ur pyregion-2.3.0.a/pyproject.toml pyregion-2.3.0.b/pyproject.toml
--- pyregion-2.3.0.a/pyproject.toml	2025-03-22 19:32:39.312998843 +0100
+++ pyregion-2.3.0.b/pyproject.toml	2025-03-22 19:34:57.077909177 +0100
@@ -48,9 +48,7 @@
 
 [tool.setuptools]
 include-package-data = true
-
-[tool.setuptools.packages]
-find = { namespaces = false }
+packages = ["pyregion", "pyregion.tests", "pyregion.tests.data"]
 
 [tool.setuptools.package-data]
 "pyregion.tests.data" = [ "*.header", "*.reg" ]
