--- a/Config.mk.in	2021-04-12 10:54:36.120860743 +0200
+++ b/Config.mk.in	2021-04-12 14:30:56.637020085 +0200
@@ -30,8 +30,8 @@
     cflags	:= -O0 -ggdb3
     ldflags	:= -g -rdynamic
 else
-    cflags	:= -Os -g0 -DNDEBUG=1
-    ldflags	:= -s
+    cflags	:= -Os -g0 -DNDEBUG=1 -ggdb3 -fPIE
+    ldflags	:= -g
 endif
 CFLAGS		:= -Wall -Wextra -Wredundant-decls -Wshadow
 cflags		+= -std=c11 @pkg_cflags@ ${CFLAGS}
