diff -Naur c4fs-1abba005074022f07d2d9d11ec6a786f2a1f1380-original/CMakeLists.txt c4fs-1abba005074022f07d2d9d11ec6a786f2a1f1380/CMakeLists.txt
--- c4fs-1abba005074022f07d2d9d11ec6a786f2a1f1380-original/CMakeLists.txt	2021-12-12 19:03:57.000000000 -0500
+++ c4fs-1abba005074022f07d2d9d11ec6a786f2a1f1380/CMakeLists.txt	2021-12-27 17:31:09.231599661 -0500
@@ -7,8 +7,6 @@
 c4_project(VERSION 0.0.1 STANDALONE
     AUTHOR "Joao Paulo Magalhaes <dev@jpmag.me>")
 
-c4_require_subproject(c4core SUBDIRECTORY ${C4FS_EXT_DIR}/c4core)
-
 c4_add_library(c4fs
     SOURCES c4/fs/export.hpp c4/fs/fs.hpp c4/fs/fs.cpp
     SOURCE_ROOT ${C4FS_SRC_DIR}
