--- notify-debouncer-mini-0.7.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ notify-debouncer-mini-0.7.0/Cargo.toml	2026-02-01T20:49:25.564146+00:00
@@ -53,12 +53,10 @@
     "dep:crossbeam-channel",
     "notify/crossbeam-channel",
 ]
-default = ["macos_fsevent"]
+default = []
 flume = [
     "dep:flume",
     "notify/flume",
 ]
-macos_fsevent = ["notify/macos_fsevent"]
-macos_kqueue = ["notify/macos_kqueue"]
 serde = ["notify-types/serde"]
 serialization-compat-6 = ["notify/serialization-compat-6"]
