From 271220e290048329dac61f11026afc418592ef98 Mon Sep 17 00:00:00 2001
From: "Kai A. Hiller" <V02460@gmail.com>
Date: Wed, 15 Nov 2023 23:14:05 +0100
Subject: [PATCH] Fix TestAsciiHex

---
 packet-v1_test.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packet-v1_test.go b/packet-v1_test.go
index 81f8ca1..026bb81 100644
--- a/packet-v1_test.go
+++ b/packet-v1_test.go
@@ -86,7 +86,7 @@ func truncate(d string) string {
 func TestAsciiHex(t *testing.T) {
 	c := qt.New(t)
 	for b := 0; b < 256; b++ {
-		n, err := strconv.ParseInt(string(b), 16, 8)
+		n, err := strconv.ParseInt(string(byte(b)), 16, 8)
 		value, ok := asciiHex(byte(b))
 		if err != nil || unicode.IsUpper(rune(b)) {
 			c.Assert(ok, qt.Equals, false)
-- 
2.41.0

