--- hap-1.62/lib/Kelvin/cw-functions.gi.orig	2024-02-01 05:23:39.000000000 -0700
+++ hap-1.62/lib/Kelvin/cw-functions.gi	2024-07-16 20:33:49.512607476 -0600
@@ -1039,7 +1039,7 @@ filepng:=Filename( tmpdir , "HAPtmpImage
 
     # convert the binary array to an upscaled 500x500 png
     Exec(
-        Concatenation("convert ",filetxt," ","-scale ",res,"x",res," ",file,".png")
+        Concatenation("magick",filetxt," ","-scale ",res,"x",res," ",file,".png")
     );
     # delete the old txt file
     Exec(
--- hap-1.62/lib/PolyComplexes/pureCubicalComplexes.gi.orig	2024-02-01 05:23:39.000000000 -0700
+++ hap-1.62/lib/PolyComplexes/pureCubicalComplexes.gi	2024-07-16 20:35:16.343085944 -0600
@@ -307,7 +307,7 @@ tmpdir := DirectoryTemporary();;
 file1:=Filename( tmpdir , "im.txt" );
 file2:=Filename( tmpdir , "im.g" );
 
-i:=Concatenation("convert -colorspace RGB -depth 8  ",file," ",file1);
+i:=Concatenation("magick -colorspace RGB -depth 8  ",file," ",file1);
 Exec(i);
 i:=Concatenation("perl ",prog," ",file1," > ",file2);
 Exec(i);
@@ -338,8 +338,8 @@ for f in file do
 tmpdir := DirectoryTemporary();;
 file1:=Filename( tmpdir , "im.txt" );
 file2:=Filename( tmpdir , "im.g" );
-#i:=Concatenation("convert ",f," /tmp/im.txt");
-i:=Concatenation("convert ",f," ",file1);
+#i:=Concatenation("magick ",f," /tmp/im.txt");
+i:=Concatenation("magick ",f," ",file1);
 Exec(i);
 #i:=Concatenation("perl ",prog," /tmp/im.txt >/tmp/im.g");
 i:=Concatenation("perl ",prog," ",file1," > ",file2 );
@@ -989,7 +989,7 @@ AppendTo(filetxt,i-1,",",j-1,": ",colour
 od;
 od;
 
-i:=Concatenation("convert ",filetxt," ",file,".",ext);
+i:=Concatenation("magick ",filetxt," ",file,".",ext);
 Exec(i);
 i:=Concatenation("rm ",filetxt);
 Exec(i);
