--- electronics.menu	2007-12-07 19:25:39.000000000 +0100
+++ electronics.menu.submenus	2009-07-08 22:30:00.510572856 +0200
@@ -8,5 +8,103 @@
     <Include>
       <Category>Electronics</Category>
     </Include>
+    <Exclude>
+      <Filename>dreal.desktop</Filename>
+      <Filename>graal.desktop</Filename>
+      <Filename>xfsm.desktop</Filename>
+      <Filename>xpat.desktop</Filename>
+      <Filename>xsch.desktop</Filename>
+      <Filename>xvpn.desktop</Filename>
+      <Filename>xgra.desktop</Filename>
+      <Filename>electric.desktop</Filename>
+      <Filename>magic.desktop</Filename>
+      <Filename>netgen.desktop</Filename>
+      <Filename>xcircuit.desktop</Filename>
+      <Filename>toped.desktop</Filename>
+      <Filename>tkgate.desktop</Filename>
+      <Filename>ktechlab.desktop</Filename>
+      <Filename>piklab.desktop</Filename>
+      <Filename>pikloops.desktop</Filename>
+      <Filename>geda-gattrib.desktop</Filename>
+      <Filename>gerbv.desktop</Filename>
+      <Filename>geda-gschem.desktop</Filename>
+      <Filename>pcb.desktop</Filename>
+      <Filename>kicad.desktop</Filename>
+      <Filename>gspiceui.desktop</Filename>
+      <Filename>gwave.desktop</Filename>
+      <Filename>dinotrace.desktop</Filename>
+      <Filename>GNUSim8085.desktop</Filename>
+      <Filename>gsim85.desktop</Filename>
+      <Filename>mcu8051ide.desktop</Filename>
+    </Exclude>
+    <Menu>
+      <Name>Automation</Name>
+      <Directory>electronics-Automation.directory</Directory>
+      <Include>
+        <Category>Automation</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Analog Mixed-Signal Design</Name>
+      <Directory>electronics-Analog.directory</Directory>
+      <Include>
+        <Category>Analog</Category>
+        <Filename>electric.desktop</Filename>
+        <Filename>magic.desktop</Filename>
+        <Filename>netgen.desktop</Filename>
+        <Filename>xcircuit.desktop</Filename>
+        <Filename>toped.desktop</Filename>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Digital IC Design</Name>
+      <Directory>electronics-Digital.directory</Directory>
+      <Include>
+        <Category>Digital</Category>
+        <Filename>dreal.desktop</Filename>
+        <Filename>graal.desktop</Filename>
+        <Filename>xfsm.desktop</Filename>
+        <Filename>xpat.desktop</Filename>
+        <Filename>xsch.desktop</Filename>
+        <Filename>xgra.desktop</Filename>
+        <Filename>xvpn.desktop</Filename>
+        <Filename>tkgate.desktop</Filename>
+        <Filename>dinotrace.desktop</Filename>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Embedded Design</Name>
+      <Directory>electronics-Embedded.directory</Directory>
+      <Include>
+        <Category>Embedded</Category>
+        <Filename>ktechlab.desktop</Filename>
+        <Filename>piklab.desktop</Filename>
+        <Filename>pikloops.desktop</Filename>
+        <Filename>GNUSim8085.desktop</Filename>
+        <Filename>gsim85.desktop</Filename>
+        <Filename>mcu8051ide.desktop</Filename>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Schematic to PCB Layout</Name>
+      <Directory>electronics-SchematicPCB.directory</Directory>
+      <Include>
+        <Category>Schematic,PCB</Category>
+        <Filename>geda-gattrib.desktop</Filename>
+        <Filename>gerbv.desktop</Filename>
+        <Filename>geda-gschem.desktop</Filename>
+        <Filename>pcb.desktop</Filename>
+        <Filename>kicad.desktop</Filename>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Spice Simulation</Name>
+      <Directory>electronics-Spice.directory</Directory>
+      <Include>
+        <Category>Spice</Category>
+        <Filename>gspiceui.desktop</Filename>
+        <Filename>gwave.desktop</Filename>
+      </Include>
+    </Menu>
   </Menu>
 </Menu>
