From 68f0c1dd3a40dd42182ce4fc8530302b93d085b1 Mon Sep 17 00:00:00 2001
From: Andrew Bauer <zonexpertconsulting@outlook.com>
Date: Sat, 24 Jan 2026 09:12:19 -0600
Subject: [PATCH] Fix implicit declaration of function AfpErr2name

---
 etc/afpd/directory.c | 1 +
 etc/afpd/extattrs.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/etc/afpd/directory.c b/etc/afpd/directory.c
index da37145eb23..8e1a335d764 100644
--- a/etc/afpd/directory.c
+++ b/etc/afpd/directory.c
@@ -23,6 +23,7 @@
 
 #include <atalk/adouble.h>
 #include <atalk/afp.h>
+#include <atalk/afp_util.h>
 #include <atalk/cnid.h>
 #include <atalk/errchk.h>
 #include <atalk/fce_api.h>
diff --git a/etc/afpd/extattrs.c b/etc/afpd/extattrs.c
index d9271659c99..0da9478da94 100644
--- a/etc/afpd/extattrs.c
+++ b/etc/afpd/extattrs.c
@@ -23,6 +23,7 @@
 
 #include <atalk/adouble.h>
 #include <atalk/afp.h>
+#include <atalk/afp_util.h>
 #include <atalk/ea.h>
 #include <atalk/globals.h>
 #include <atalk/logger.h>
