diff -up PEAR_Command_Packaging-0.3.0/PEAR/Command/Packaging.php.old PEAR_Command_Packaging-0.3.0/PEAR/Command/Packaging.php
--- PEAR_Command_Packaging-0.3.0/PEAR/Command/Packaging.php.old	2012-10-24 09:06:12.000000000 +0200
+++ PEAR_Command_Packaging-0.3.0/PEAR/Command/Packaging.php	2012-10-24 09:07:23.000000000 +0200
@@ -531,8 +531,9 @@ Wrote: /path/to/rpm-build-tree/RPMS/noar
         
         // Construct a fake registry inside the ultimate destination
         // temporary directory, and load the necessary channel into it
-        $regdir = $instroot . $this->config->get('php_dir');
-        $fakereg = new PEAR_Registry($regdir);
+        $phpdir = $instroot . $this->config->get('php_dir');
+        $regdir = $instroot . $this->config->get('metadata_dir');
+        $fakereg = new PEAR_Registry($phpdir, false, false, $regdir);
         $fakereg->addChannel($chan);
 
         $tmp = $this->config->get('verbose');
