diff -Naur nsca-2.9.2.orig/src/send_nsca.c nsca-2.9.2/src/send_nsca.c
--- nsca-2.9.2.orig/src/send_nsca.c	2016-12-07 17:00:07.000000000 +0100
+++ nsca-2.9.2/src/send_nsca.c	2017-02-27 17:33:20.792093911 +0100
@@ -26,7 +26,7 @@
 char server_name[MAX_HOST_ADDRESS_LENGTH];
 char password[MAX_INPUT_BUFFER]="";
 int legacy_2_7_mode=FALSE;
-char config_file[MAX_INPUT_BUFFER]="send_nsca.cfg";
+char config_file[MAX_INPUT_BUFFER]="/etc/nagios/send_nsca.cfg";
 char delimiter[2]="\t";
 char block_delimiter[2]=BLOCK_DELIMITER;
 
