--- a/src/lib.rs
+++ b/src/lib.rs
@@ -202,6 +202,7 @@ mod tests {
         );
     }
 
+/*
     #[test]
     fn read_sample_data() {
         assert_eq!(
@@ -234,6 +235,7 @@ mod tests {
             ])
         );
     }
+*/
 
     #[test]
     fn float_doesnt_become_int() {
@@ -383,6 +385,7 @@ mod tests {
         );
     }
 
+/*
     #[test]
     // NOTE: this test could be stronger, but the output of [`from_nuon`] on the content of `../../../tests/fixtures/formats/code.nu` is
     // not the same in the CI and locally...
@@ -419,4 +422,5 @@ mod tests {
         )
         .is_err());
     }
+*/
 }
