diff -up modsecurity-v2.9.12/apache2/msc_lua.c.lua55 modsecurity-v2.9.12/apache2/msc_lua.c
--- modsecurity-v2.9.12/apache2/msc_lua.c.lua55	2026-03-14 17:26:08.306090891 -0400
+++ modsecurity-v2.9.12/apache2/msc_lua.c	2026-03-14 17:27:24.031487783 -0400
@@ -433,12 +433,12 @@ int lua_execute(msc_script *script, char
 #else
 
     /* Create new state. */
-#if LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 504 || LUA_VERSION_NUM == 501
+#if LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 504 || LUA_VERSION_NUM == 505 || LUA_VERSION_NUM == 501
     L = luaL_newstate();
 #elif LUA_VERSION_NUM == 500
     L = lua_open();
 #else
-#error We are only tested under Lua 5.0, 5.1, 5.2, 5.3, or 5.4.
+#error We are only tested under Lua 5.0, 5.1, 5.2, 5.3, 5.4, or 5.5.
 #endif
     luaL_openlibs(L);
 
@@ -463,10 +463,10 @@ int lua_execute(msc_script *script, char
     /* Register functions. */
 #if LUA_VERSION_NUM == 500 || LUA_VERSION_NUM == 501
     luaL_register(L, "m", mylib);
-#elif LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 504
+#elif LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 504 || LUA_VERSION_NUM == 505
     luaL_setfuncs(L, mylib, 0);
 #else
-#error We are only tested under Lua 5.0, 5.1, 5.2, 5.3, or 5.4.
+#error We are only tested under Lua 5.0, 5.1, 5.2, 5.3, 5.4, or 5.5.
 #endif
 
     lua_setglobal(L, "m");
