diff -up credentials-fetcher-v.1.3.8/api/tests/CMakeLists.txt.fedora-check credentials-fetcher-v.1.3.8/api/tests/CMakeLists.txt
--- credentials-fetcher-v.1.3.8/api/tests/CMakeLists.txt.fedora-check	2025-07-18 09:16:15.632452901 -0400
+++ credentials-fetcher-v.1.3.8/api/tests/CMakeLists.txt	2025-07-18 09:17:04.215138092 -0400
@@ -1,5 +1,13 @@
 cmake_minimum_required(VERSION 3.10)
 
+# Check if the system is Fedora
+if(EXISTS "/etc/fedora-release")
+    set(IS_FEDORA TRUE)
+else()
+    set(IS_FEDORA FALSE)
+endif()
+
+if (NOT IS_FEDORA)
 # set the project name
 project(api-tests)
 
@@ -41,3 +49,5 @@ if (CMAKE_C_LINK_PIE_SUPPORTED)
     set_property(TARGET gmsa_test_client
                 PROPERTY POSITION_INDEPENDENT_CODE TRUE)
 endif ()
+
+endif()
