--- zipios++-0.1.5/src/appendzip.cpp~	2000-08-25 14:12:53.000000000 -0500
+++ zipios++-0.1.5/src/appendzip.cpp	2011-07-12 09:53:13.824178438 -0500
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include "zipios++/zipios-config.h"
 
 #include "zipios++/meta-iostreams.h"
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' zipios++~/src/directory.h zipios++/src/directory.h
--- zipios++~/src/directory.h	2011-08-24 16:01:00.000000000 +0200
+++ zipios++/src/directory.h	2011-08-24 16:04:56.503892154 +0200
@@ -31,6 +31,7 @@
 #include <string>
 #include <ctime>
 #include <stdexcept>
+#include <cstddef>
 
 // #include <boost.h>  Contents of boost.h
 
