diff -up rudesocket-1.3.0/src/socket_connect_normal.cpp.timeout rudesocket-1.3.0/src/socket_connect_normal.cpp
--- rudesocket-1.3.0/src/socket_connect_normal.cpp.timeout	2008-02-06 16:52:17.000000000 +0100
+++ rudesocket-1.3.0/src/socket_connect_normal.cpp	2012-04-04 16:38:58.607570481 +0200
@@ -128,10 +128,8 @@ int Socket_Connect_Normal::connecttimeou
 		FD_ZERO(&wset); 
 		FD_SET(socket, &wset); 
  
-		tv.tv_sec = 5; 
-		// Why the hell is this times 1000?????
-		//tv.tv_usec = 1000 * (msec % 1000); 
-		tv.tv_usec = 0;// 1000 * msec % 1000;
+		tv.tv_sec = msec / 1000;
+		tv.tv_usec = 1000 * msec % 1000;
  
 		ret = select(socket+1, NULL, &wset, NULL, &tv); 
  
