From 95d037bc50e2c4bef4ce6d669b83bdd084374cf2 Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Tue, 9 Dec 2025 12:23:28 +0100
Subject: [PATCH 1/2] meson: adapt for RPM package build environment

---
 meson.build | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 5404757..4b4f904 100644
--- a/meson.build
+++ b/meson.build
@@ -8,19 +8,14 @@ project(
 dependency('gstreamer-1.0', version: '>=1.26.0')
 dependency('glycin-2', version: '>=2.0.0')
 
-if get_option('profile') == 'release'
-  rust_target = 'release'
-  cargo_profile = 'release'
-else
-  rust_target = 'debug'
-  cargo_profile = 'dev'
-endif
+rust_target = 'rpm'
+cargo_profile = 'rpm'
 
 bindir = get_option('prefix') / get_option('bindir')
 datadir = get_option('prefix') / get_option('datadir')
 
 cargo_target_dir = meson.project_build_root() / 'cargo-target'
-cargo_home = meson.project_build_root() / 'cargo-home'
+cargo_home = meson.project_build_root()
 cargo_manifest = meson.project_source_root() / 'Cargo.toml'
 
 cargo_bin = find_program('cargo')
-- 
2.53.0

