diff -up Net-RawIP-0.23/RawIP.xs~ Net-RawIP-0.23/RawIP.xs
--- Net-RawIP-0.23/RawIP.xs~	2007-07-19 07:58:24.000000000 +0200
+++ Net-RawIP-0.23/RawIP.xs	2008-07-30 05:28:09.000000000 +0200
@@ -611,7 +611,7 @@ CODE:
         croak("gettimeofday()");
     }
     else {
-        RETVAL = newSVpvf("%u.%06u",tv.tv_sec,tv.tv_usec);
+        RETVAL = newSVpvf("%lld.%06u",(long long)tv.tv_sec,(unsigned)tv.tv_usec);
     }
 OUTPUT:
     RETVAL
