diff -ur beanstalkd-1.10.org/ct/ct.c beanstalkd-1.10/ct/ct.c
--- beanstalkd-1.10.org/ct/ct.c	2014-08-04 21:39:54.000000000 -0600
+++ beanstalkd-1.10/ct/ct.c	2016-01-08 11:44:55.012786874 -0700
@@ -182,7 +183,7 @@
 {
     t->fd = tmpfd();
     strcpy(t->dir, TmpDirPat);
-    mktemp(t->dir);
+    mkdtemp(t->dir);
     t->pid = fork();
     if (t->pid < 0) {
         die(1, errno, "fork");
@@ -285,7 +286,7 @@
     int outfd = tmpfd();
     int durfd = tmpfd();
     strcpy(b->dir, TmpDirPat);
-    mktemp(b->dir);
+    mkdtemp(b->dir);
     int pid = fork();
     if (pid < 0) {
         die(1, errno, "fork");
