--- a/hyph/croatian.lh
+++ b/hyph/croatian.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % hrhyph.tex
 % Hyphenation patterns for Croatian language words for use with 
--- a/hyph/czech.lh
+++ b/hyph/czech.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is a Czech hyphenation file created by Jeff Kingston
 %  on 16 July 1995 from a TeX file supplied by Pavel Vana.
--- a/hyph/danish.lh
+++ b/hyph/danish.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Danish hyphenation file, created by
 %  Jeffrey H. Kingston 22 April 1994
--- a/hyph/dutch.lh
+++ b/hyph/dutch.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % Created by Jeff Kingston 7 October 1994 from a TeX hyphenation
 % patterns file supplied by Ronald Rietman, who says it is nehyph1.tex
--- a/hyph/english.lh
+++ b/hyph/english.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Created by Jeff Kingston, 20 April 1994, from a TeX
 %  hyphenation file.
--- a/hyph/engluk.lh
+++ b/hyph/engluk.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Created by Jeff Kingston, 16 November 1996, from a TeX
 %  hyphenation file with the following header comment:
--- a/hyph/esperanto.lh
+++ b/hyph/esperanto.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % Created by Ludovic Courtès, 2005-05-20, based on the file `eohyph.tex'
 % available from the `textex-texmf-3.0.tar.gz' package.
--- a/hyph/finnish.lh
+++ b/hyph/finnish.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % Created by Jeff Kingston 19 April 1994 from a TeX hyphenation
 % patterns file with the following header comment:
--- a/hyph/french.lh
+++ b/hyph/french.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Created by Jeff Kingston, 19 May 1996, from a TeX
 %  hyphenation file.  Apart from commenting out things that
--- a/hyph/german.lh
+++ b/hyph/german.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % This is a German hyphenation file constructed by Jeff Kingston
 % on 21 April 1994 from a TeX file supplied by Sverre Slotte.
--- a/hyph/italian.lh
+++ b/hyph/italian.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % This is an Italian hyphenation file constructed by Daniele Giacomini
 % (daniele@evo.it) on 03 February 1999 from a TeX file supplied by
--- a/hyph/magyar.lh
+++ b/hyph/magyar.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is a Hun hyphenation file created by Héri Tamás.
 %  The only comment received with the file was this one:
--- a/hyph/norweg.lh
+++ b/hyph/norweg.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % Norwegian hyphenation file, created by
 % Jeffrey H. Kingston 24 September 1994 from
--- a/hyph/polish.lh
+++ b/hyph/polish.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 % This is a Polish hyphenation file - converted from a TeX hyphenation file
 % plhyph.tex by Jacek Naglak <nag@promail.pl> on 3 February 2001
--- a/hyph/portugal.lh
+++ b/hyph/portugal.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Portuguese hyphenation file, created by José Roberto B. de A. Monteiro
 %   at 29 october 1999 from scratch
--- a/hyph/programming.lh
+++ b/hyph/programming.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  Created by Jeff Kingston, 30 August 2008, from scratch
 %  Used for hyphenating programming language identifiers
--- a/hyph/slovak.lh
+++ b/hyph/slovak.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is a Slovak hyphenation file created
 %  by Peter Valach <pvalach@gmx.net> on 26 June 2001
--- a/hyph/slovenia.lh
+++ b/hyph/slovenia.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is a Slovenian hyphenation file created by Jeff Kingston
 %  on 8 June 1995 from a TeX file supplied by Ljupco Taseski.
--- a/hyph/swedish.lh
+++ b/hyph/swedish.lh
@@ -14,8 +14,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is a Swedish hyphenation file created by Jeff Kingston
 %  on 20 March 1994 from a TeX file supplied by Sverre Slotte.
--- a/hyph/uppersorbian.lh
+++ b/hyph/uppersorbian.lh
@@ -13,8 +13,7 @@ Lout hyphenation information
 %  GNU General Public License for more details.
 %
 %  You should have received a copy of the GNU General Public License
-%  along with this program; if not, write to the Free Software
-%  Foundation, Inc., 59 Temple Place, Suite 330, Boston MA 02111-1307 USA
+%  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 %  This is an Upper Sorbian hyphenation file created by E.Werner, 2000
 %
--- a/README
+++ b/README
@@ -19,8 +19,7 @@ MERCHANTABILITY or FITNESS FOR A PARTICU
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
-(file COPYING in this directory); if not, write to the Free Software
-Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 This directory contains everything needed to install the Lout document
 formatting system.  It contains the following directories:
