From fd7af3f51e9a9e60d677d4d811d52cb4c703cfe8 Mon Sep 17 00:00:00 2001
From: Pierre Pronchery <khorben@defora.org>
Date: Wed, 6 Aug 2014 01:23:05 +0200
Subject: [PATCH] Avoid a crash on invalid input values

---
 src/makepasswd.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/makepasswd.c b/src/makepasswd.c
index c000bef..ed35fef 100644
--- a/src/makepasswd.c
+++ b/src/makepasswd.c
@@ -741,28 +741,32 @@ int main(int argc, char * argv[])
 					return _usage();
 				break;
 			case 'i':
-				if((prefs.iterations = _parse_unsigned(optarg))
-						<= 0)
+				if((o = _parse_unsigned(optarg)) <= 0)
 					return _usage();
+				prefs.iterations = o;
 				break;
 			case 'l':
-				if((prefs.max = _parse_unsigned(optarg)) <= 0)
+				if((o = _parse_unsigned(optarg)) <= 0)
 					return _usage();
+				prefs.max = o;
 				prefs.min = prefs.max;
 				break;
 			case 'M':
-				if((prefs.max = _parse_unsigned(optarg)) <= 0)
+				if((o = _parse_unsigned(optarg)) <= 0)
 					return _usage();
+				prefs.max = o;
 				break;
 			case 'm':
-				if((prefs.min = _parse_unsigned(optarg)) <= 0)
+				if((o = _parse_unsigned(optarg)) <= 0)
 					return _usage();
+				prefs.min = o;
 				if(prefs.max < prefs.min)
 					prefs.max = prefs.min;
 				break;
 			case 'n':
-				if((prefs.count = _parse_unsigned(optarg)) <= 0)
+				if((o = _parse_unsigned(optarg)) <= 0)
 					return _usage();
+				prefs.count = o;
 				break;
 			case 'p':
 				prefs.password = optarg;
-- 
1.9.3

