diff --git a/lib/Slic3r.pm b/lib/Slic3r.pm
index 154af90..ef8274c 100644
--- a/lib/Slic3r.pm
+++ b/lib/Slic3r.pm
@@ -42,7 +42,7 @@ warn "Running Slic3r under Perl 5.16 is neither supported nor recommended\n"
 
 use FindBin;
 # Path to the images.
-my $varpath = decode_path($FindBin::Bin) . "/var";
+my $varpath = "/usr/share/slic3r";
 if ($^O eq 'darwin' && !-d $varpath) {
     $varpath = decode_path($FindBin::Bin) . "/../Resources/var";
 }
