--- a/CMakeLists.orig.txt	2025-07-20 19:14:34.037260183 +0200
+++ b/CMakeLists.txt	2025-07-20 19:16:10.397953590 +0200
@@ -1,13 +1,7 @@
 # CMake 3.30 or later is required for FindBoost.
 # It can be easily obtained and used without superuser privileges
 # even if the system has an older version of CMake installed.
-cmake_minimum_required(VERSION 3.5...3.10)
-
-if (CMAKE_MAJOR_VERSION LESS 3)
-  # CMake 3.0 is used for generator expressions in install(FILES).
-  message(WARNING
-    "MP: CMake version 3.0 or newer is required to install all targets")
-endif ()
+cmake_minimum_required(VERSION 3.3...4.0)
 
 # Minimum MacOS target to enable C++17 std::filesystem
 set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15)
