From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 21 Sep 2017 13:35:27 +0300
Subject: [PATCH] Disable support for Prometheus in Fedora

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/README.md b/README.md
index f9ec465..96be29e 100644
--- a/README.md
+++ b/README.md
@@ -238,7 +238,6 @@ Independent of which stats plugin you use, the `ok` and `blown` metrics are incr
 * `fuse_stats_ets`: Maintains simple counts in an [ETS](http://www.erlang.org/doc/man/ets.html) table, which you can retrieve with a call to e.g. `fuse_stats_ets:counters(foo).`
 * `fuse_stats_folsom`: Maintains [folsom](https://github.com/boundary/folsom) spirals `foo.ok`, `foo.blown` and `foo.melt`.
 * `fuse_stats_exometer`: Maintains [exometer](https://github.com/Feuerlabs/exometer) spirals `[fuse, foo, ok]`, `[fuse, foo, blown]` and `[fuse, foo, melt]`.
-* `fuse_stats_prometheus`: Maintains [prometheus](https://github.com/deadtrickster/prometheus.erl) counters `foo_responses_total[type="ok|blown"]` and `foo_melts_total`.
 
 ### Using a plugin
 
diff --git a/src/fuse_stats_prometheus.erl b/src/fuse_stats_prometheus.erl
deleted file mode 100644
index fca6241..0000000
--- a/src/fuse_stats_prometheus.erl
+++ /dev/null
@@ -1,45 +0,0 @@
-%%% @doc fuse_stats_prometheus - use <a href="https://github.com/deadtrickster/prometheus.erl">prometheus</a> counters for fuse stats.
-%%% Assumes that you have already arranged to start prometheus.
--module(fuse_stats_prometheus).
--behaviour(fuse_stats_plugin).
--export([init/1, increment/2]).
-
--define(RESPONSES_COUNTER_NAME(NameBin), <<NameBin/binary, "_responses_total">>).
-
--define(MELTS_COUNTER_NAME(NameBin), <<NameBin/binary, "_melts_total">>).
-
-%% @doc Initializes prometheus counters for  `Name'.
-%% Exports the following metrics:
-%% <ul>
-%% <li>`name_responses_total[type]'</li>
-%% <li>`name_melts_total'.</li>
-%% </ul>
-%% Uses `default' registry.
-%% @end
-%% init/1
--spec init(Name :: atom()) -> ok.
-init(Name) ->
-    NameBin = atom_to_binary(Name, utf8),
-
-    prometheus_counter:new([{name, ?RESPONSES_COUNTER_NAME(NameBin)},
-                            {help, <<NameBin/binary, " fuse responses counter">>},
-                            {labels, [type]}]),
-    prometheus_counter:new([{name, ?MELTS_COUNTER_NAME(NameBin)},
-                            {help, <<NameBin/binary, " fuse melts counter">>}]),
-    ok.
-
-%% @doc Increments `Name''s `Counter'.
-%% @end
-%% increment/2
--spec increment(Name :: atom(), Counter :: ok | blown | melt) -> ok.
-increment(Name, Counter) ->
-    NameBin = atom_to_binary(Name, utf8),
-    case Counter of
-        ok ->
-            prometheus_counter:inc(?RESPONSES_COUNTER_NAME(NameBin), [ok]);
-        blown ->
-            prometheus_counter:inc(?RESPONSES_COUNTER_NAME(NameBin), [blown]);
-        melt ->
-            prometheus_counter:inc(?MELTS_COUNTER_NAME(NameBin))
-    end,
-    ok.
