--- fribidi-1.0.16/bin/meson.build
+++ fribidi-1.0.16-new/bin/meson.build
@@ -2,7 +2,7 @@
 # need to build it for internal usage, we just won't install it.
 gen_man_pages = 'fribidi.1'
 fribidi = executable('fribidi',
-  'fribidi-main.c', 'getopt.c', 'getopt1.c', fribidi_unicode_version_h,
+  'fribidi-main.c', fribidi_unicode_version_h,
   c_args: ['-DHAVE_CONFIG_H'] + fribidi_static_cargs,
   include_directories: incs,
   link_with: libfribidi,
@@ -14,7 +14,7 @@
 endif
 
 executable('fribidi-benchmark',
-  'fribidi-benchmark.c', 'getopt.c', 'getopt1.c', fribidi_unicode_version_h,
+  'fribidi-benchmark.c', fribidi_unicode_version_h,
   c_args: ['-DHAVE_CONFIG_H'] + fribidi_static_cargs,
   include_directories: incs,
   link_with: libfribidi,
