diff -up evolution-rspam-0.6.0/src/rspam.c.evo313 evolution-rspam-0.6.0/src/rspam.c
--- evolution-rspam-0.6.0/src/rspam.c.evo313	2014-09-22 11:04:28.857208274 +0200
+++ evolution-rspam-0.6.0/src/rspam.c	2014-09-22 11:33:05.977140234 +0200
@@ -505,8 +505,12 @@ spamcop_report(CamelMimeMessage *msg, gc
 	text = em_utils_message_to_html (msg, NULL, 0, &len, NULL, NULL, NULL);
 #else
 #if EVOLUTION_VERSION > 30400
+#if EVOLUTION_VERSION < 31303
 	text = em_utils_message_to_html (CAMEL_SESSION (session), msg, NULL, 0, NULL, NULL, NULL);
 #else
+	text = em_utils_message_to_html (CAMEL_SESSION (session), msg, NULL, 0, NULL, NULL, NULL, NULL);
+#endif
+#else
 	text = em_utils_message_to_html (msg, NULL, 0, NULL, NULL, NULL);
 #endif
 #endif
