diff -up gxemul-0.7.0/configure.no-rpath gxemul-0.7.0/configure
--- gxemul-0.7.0/configure.no-rpath	2021-07-15 09:43:37.539248898 -0400
+++ gxemul-0.7.0/configure	2021-07-15 09:44:01.107395072 -0400
@@ -333,7 +333,7 @@ if [ z$NOX11 = z ]; then
 	XINCLUDE=-I/usr/X11R6/include
 	$CC $CFLAGS _test_x11.c -c -o _test_x11.o $XINCLUDE 2> /dev/null
 
-	XLIB="-L/usr/X11R6/lib -lX11 -Wl,-rpath,/usr/X11R6/lib"
+	XLIB="-L/usr/X11R6/lib -lX11"
 	$CC $CFLAGS _test_x11.o -o _test_x11 $XLIB 2> /dev/null
 
 	if [ -x _test_x11 ]; then
@@ -348,7 +348,7 @@ if [ z$NOX11 = z ]; then
 		XINCLUDE=-I/usr/X11R7/include
 		$CC $CFLAGS _test_x11.c -c -o _test_x11.o $XINCLUDE 2> /dev/null
 
-		XLIB="-L/usr/X11R7/lib -lX11 -Wl,-rpath,/usr/X11R7/lib"
+		XLIB="-L/usr/X11R7/lib -lX11"
 		$CC $CFLAGS _test_x11.o -o _test_x11 $XLIB 2> /dev/null
 
 		if [ -x _test_x11 ]; then
@@ -363,7 +363,7 @@ if [ z$NOX11 = z ]; then
 		XINCLUDE=-I/usr/local/include
 		$CC $CFLAGS _test_x11.c -c -o _test_x11.o $XINCLUDE 2> /dev/null
 
-		XLIB="-L/usr/local/lib -lX11 -Wl,-rpath,/usr/local/lib"
+		XLIB="-L/usr/local/lib -lX11"
 		$CC $CFLAGS _test_x11.o -o _test_x11 $XLIB 2> /dev/null
 
 		if [ -x _test_x11 ]; then
