diff -up gxemul-0.6.0.1/configure.cleanup gxemul-0.6.0.1/configure
--- gxemul-0.6.0.1/configure.cleanup	2014-08-17 04:45:08.000000000 -0400
+++ gxemul-0.6.0.1/configure	2014-08-20 18:39:09.811542814 -0400
@@ -1145,6 +1145,17 @@ echo "checking for 'make install' man di
 
 ###############################################################################
 
+# On some platforms, the man pages are in /usr/man, but on Linux, /usr/share/man is the 
+# correct location. This can be overridden by passing MANDIR=/foo/bar at configure.
+
+DEFAULTMANDIR=/usr/share/man
+
+if [ "z$MANDIR" = z ]; then
+	MANDIR="$DEFAULTMANDIR"
+fi
+
+###############################################################################
+
 INCLUDE=-Iinclude/
 DINCLUDE=-I../include/
 INCLUDE2=-I../../include/
diff -up gxemul-0.6.0.1/Makefile.skel.cleanup gxemul-0.6.0.1/Makefile.skel
--- gxemul-0.6.0.1/Makefile.skel.cleanup	2014-08-17 04:45:08.000000000 -0400
+++ gxemul-0.6.0.1/Makefile.skel	2014-08-20 18:39:09.811542814 -0400
@@ -4,6 +4,7 @@
 
 BIN=gxemul
 LIBS=$(XLIB) $(OTHERLIBS)
+DESTDIR=
 
 all: build
 
