diff --git a/rigcommander.cpp b/rigcommander.cpp
index a842187..5a81a4d 100644
--- a/rigcommander.cpp
+++ b/rigcommander.cpp
@@ -4157,6 +4157,7 @@ void rigCommander::determineRigCaps()
             rigCaps.useRTSforPTT = true;
             rigCaps.hasDataModes = false;
             rigCaps.attenuators.push_back('\x20');
+            rigCaps.preamps.push_back('\x01');
             rigCaps.bands = standardHF;
             rigCaps.bands.insert(rigCaps.bands.end(), standardVU.begin(), standardVU.end());
             rigCaps.bands.push_back(bandDefGen);
