From 655c0908155758e7c94858af2fb99dc992709075 Mon Sep 17 00:00:00 2001
From: "Fletcher T. Penney" <fletcher@fletcherpenney.net>
Date: Sat, 20 Jul 2024 16:42:23 -0400
Subject: [PATCH] UPDATED: Increase required CMake version to silence warning

---
 CMakeLists.txt       | 2 +-
 lemon/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d847cb65..9809f00d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@
 #	See the LICENSE file for copyright and licensing information.
 #
 
-cmake_minimum_required(VERSION 2.9)
+cmake_minimum_required(VERSION 3.5)
 
 
 # ===================
diff --git a/lemon/CMakeLists.txt b/lemon/CMakeLists.txt
index 87b42fcb..27d8b0fa 100644
--- a/lemon/CMakeLists.txt
+++ b/lemon/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.5)
 
 # Quick file to build lemon parser generator
 
