diff -ur gtk+-1.2.10.orig/configure gtk+-1.2.10/configure
--- gtk+-1.2.10.orig/configure	2019-09-09 21:49:43.602211637 +0200
+++ gtk+-1.2.10/configure	2019-09-09 22:40:10.672294726 +0200
@@ -1151,7 +1151,7 @@
 #line 1152 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:1157: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -1867,7 +1867,7 @@
 #line 1868 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:1873: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -2457,6 +2457,7 @@
 #line 2458 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
@@ -2575,7 +2576,7 @@
 #include "confdefs.h"
 
 int main() {
-} $ac_kw foo() {
+} $ac_kw int foo() {
 ; return 0; }
 EOF
 if { (eval echo configure:2582: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
@@ -4689,6 +4690,8 @@
 #line 4690 "configure"
 #include "confdefs.h"
 
+extern char ${x_direct_test_function}(void);
+
 int main() {
 ${x_direct_test_function}()
 ; return 0; }
@@ -6008,6 +6011,7 @@
 #line 6009 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
diff -ur gtk+-1.2.10.orig/gdk/gdkinput.c gtk+-1.2.10/gdk/gdkinput.c
--- gtk+-1.2.10.orig/gdk/gdkinput.c	2000-01-27 22:10:44.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkinput.c	2019-09-09 22:37:05.956183052 +0200
@@ -27,6 +27,7 @@
 #include <stdlib.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <string.h>
 #include "config.h"
 #include "gdk.h"
 #include "gdkx.h"
diff -ur gtk+-1.2.10.orig/ltconfig gtk+-1.2.10/ltconfig
--- gtk+-1.2.10.orig/ltconfig	2023-04-14 09:50:57.358511727 +0200
+++ gtk+-1.2.10/ltconfig	2023-04-14 10:03:17.965002981 +0200
@@ -625,7 +625,7 @@
 else
   ac_cv_exeext="no"
   $rm conftest*
-  echo 'main () { return 0; }' > conftest.c
+  echo 'int main (void) { return 0; }' > conftest.c
   echo "$progname:629: checking for executable suffix" >& 5
   if { (eval echo $progname:630: \"$ac_link\") 1>&5; (eval $ac_link) 2>conftest.err; }; then
     # Append any warnings to the config.log.
@@ -952,7 +952,7 @@
 
 echo $ac_n "checking if $compiler static flag $link_static_flag works... $ac_c" 1>&6
 $rm conftest*
-echo 'main(){return(0);}' > conftest.c
+echo 'int main(void){return(0);}' > conftest.c
 save_LDFLAGS="$LDFLAGS"
 LDFLAGS="$LDFLAGS $link_static_flag"
 echo "$progname:958: checking if $compiler static flag $link_static_flag works" >&5
@@ -1629,7 +1629,7 @@
 #ifdef __cplusplus
 }
 #endif
-main(){nm_test_var='a';nm_test_func();return(0);}
+int main(void){nm_test_var='a';nm_test_func();return(0);}
 EOF
 
   echo "$progname:1635: checking if global_symbol_pipe works" >&5
