diff -up frysk-0.4/frysk-imports/include/frysk-asm.h.2 frysk-0.4/frysk-imports/include/frysk-asm.h
--- frysk-0.4/frysk-imports/include/frysk-asm.h.2	2009-02-09 16:17:15.000000000 -0500
+++ frysk-0.4/frysk-imports/include/frysk-asm.h	2009-02-09 16:18:06.000000000 -0500
@@ -564,7 +564,6 @@
 	.cfi_offset 30, 8          ; \
 	stw    31,  12(1)          ; \
 	.cfi_offset 31, 12         ;
-#elif
 #else
 #  warning "No function-prologue compound instruction defined"
 #endif
