diff --git a/frysk-sys/lib/dwfl/jni/Elf.cxx b/frysk-sys/lib/dwfl/jni/Elf.cxx
index c22fa58ee..92f19417b 100644
--- a/frysk-sys/lib/dwfl/jni/Elf.cxx
+++ b/frysk-sys/lib/dwfl/jni/Elf.cxx
@@ -174,7 +174,7 @@ lib::dwfl::Elf::elf_newehdr(jnixx::env env, jint wordSize) {
     runtimeException(env, "Bad parameter to elf_newehdr (word size %d)",
 		     wordSize);
   }
-  if (::gelf_newehdr(elf, elfClass) < 0)
+  if (::gelf_newehdr(elf, elfClass) == NULL)
     throw_last_elf_error(env);
 }
 
