diff -rupN --no-dereference netgen-6.2.2601/CMakeLists.txt netgen-6.2.2601-new/CMakeLists.txt
--- netgen-6.2.2601/CMakeLists.txt	2026-01-18 10:30:32.331891329 +0100
+++ netgen-6.2.2601-new/CMakeLists.txt	2026-01-18 10:30:32.443344726 +0100
@@ -643,7 +643,7 @@ if(USE_PYTHON AND NOT SKBUILD)
   file( WRITE "netgen_mesher-py3.egg-info"
 "Metadata-Version: 2.1
 Name: netgen-mesher
-Version: ${NETGEN_VERSION_MAJOR}.${NETGEN_VERSION_MINOR}.${NETGEN_VERSION_PATCH}.post${NETGEN_VERSION_TWEAK}
+Version: ${NETGEN_VERSION_MAJOR}.${NETGEN_VERSION_MINOR}.${NETGEN_VERSION_PATCH}
 ")
   install(FILES netgen_mesher-py3.egg-info DESTINATION ${NG_INSTALL_DIR_PYTHON} COMPONENT netgen)
 endif()
