From d9f52fd8c1bb453d239d918af28ce64ff438d581 Mon Sep 17 00:00:00 2001
From: Mikio Braun <mikiobraun@gmail.com>
Date: Sat, 28 Nov 2020 11:59:47 +0100
Subject: [PATCH] options check for dynamic-libs had a typo.

---
 config/configure.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/configure.rb b/config/configure.rb
index 64b3919f36..f067895a99 100644
--- a/config/configure.rb
+++ b/config/configure.rb
@@ -92,7 +92,7 @@ options summary:
                            You might still have to set libpath correctly
 EOS
 
-unless $opts.check?([:lapack, :libpath, :download_lapack, :static_libs, :dyanmic_libs,
+unless $opts.check?([:lapack, :libpath, :download_lapack, :static_libs, :dynamic_libs,
                      :ptatlas, :arch_flavor, :libs, :lapack_build, :build_type])
   puts 'Configuration aborted.'
   exit
