--- a/modules/meson.build
+++ b/modules/meson.build
@@ -43,3 +43,5 @@
 subdir('pam_warn')
 subdir('pam_wheel')
 subdir('pam_xauth')
+subdir('pam_chroot')
+subdir('pam_postgresok')
--- a/modules/module-meson.build
+++ b/modules/module-meson.build
@@ -489,6 +489,13 @@
     env: ['MESON_BUILD_SUBDIR=' + meson.current_build_dir()],
   )
 endif
+if module == 'pam_chroot'
+  install_data(
+    'chroot.conf',
+    install_dir: vendor_sconfigdir,
+    install_tag: 'config',
+  )
+endif
 
 
 if module == 'pam_timestamp' and cdata.get('WITH_OPENSSL', 0) != 1
