From c03c01f36f65b7242d13da7d6cf4d6a025caa1a3 Mon Sep 17 00:00:00 2001
From: Joost Yervante Damad <andete@debian.org>
Date: Tue, 26 Jun 2018 22:31:16 +0200
Subject: improve error message

Bug-Debian: https://bugs.debian.org/335295
---
 timidity/timidity.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/timidity/timidity.c b/timidity/timidity.c
index 4a4d601..9989635 100644
--- a/timidity/timidity.c
+++ b/timidity/timidity.c
@@ -5886,11 +5886,11 @@ int main(int argc, char **argv)
 				}
 			}
 			ctl->cmsg(CMSG_FATAL, VERB_NORMAL,
-					"%s: Can't read any configuration file.\n"
+					"%s: Error reading configuration file.\n"
 					"Please check %s or %s", program_name, config1, config2);
 #else
 			ctl->cmsg(CMSG_FATAL, VERB_NORMAL,
-					"%s: Can't read any configuration file.\n"
+					"%s: Error reading configuration file.\n"
 					"Please check " CONFIG_FILE, program_name);
 #endif /* __W32__ */
 		} else
