--- plotters-0.3.7/Cargo.toml	1970-01-01T00:00:01+00:00
+++ plotters-0.3.7/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -197,10 +197,6 @@
 version = "1.0.82"
 
 [features]
-ab_glyph = [
-    "dep:ab_glyph",
-    "once_cell",
-]
 all_elements = [
     "errorbar",
     "candlestick",
@@ -222,16 +218,10 @@
 colormaps = []
 datetime = ["chrono"]
 default = [
-    "bitmap_backend",
-    "bitmap_encoder",
-    "bitmap_gif",
     "svg_backend",
-    "chrono",
-    "ttf",
-    "image",
-    "deprecated_items",
-    "all_series",
-    "all_elements",
+    "area_series",
+    "line_series",
+    "point_series",
     "full_palette",
     "colormaps",
 ]
@@ -243,81 +233,12 @@
     "bitmap_backend",
     "plotters-svg/bitmap_encoder",
 ]
-fontconfig-dlopen = ["font-kit/source-fontconfig-dlopen"]
 full_palette = []
 histogram = []
 line_series = []
 point_series = []
 surface_series = []
 svg_backend = ["plotters-svg"]
-ttf = [
-    "font-kit",
-    "ttf-parser",
-    "lazy_static",
-    "pathfinder_geometry",
-]
-
-[target.'cfg(all(target_arch = "wasm32", not(target_os = "wasi")))'.dependencies.wasm-bindgen]
-version = "0.2.89"
-
-[target.'cfg(all(target_arch = "wasm32", not(target_os = "wasi")))'.dependencies.web-sys]
-version = "0.3.66"
-features = [
-    "Document",
-    "DomRect",
-    "Element",
-    "HtmlElement",
-    "Node",
-    "Window",
-    "HtmlCanvasElement",
-    "CanvasRenderingContext2d",
-]
-
-[target.'cfg(all(target_arch = "wasm32", not(target_os = "wasi")))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3.39"
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.ab_glyph]
-version = "0.2.12"
-optional = true
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.font-kit]
-version = "0.14.2"
-optional = true
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.image]
-version = "0.24.3"
-features = [
-    "jpeg",
-    "png",
-    "bmp",
-]
-optional = true
-default-features = false
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.lazy_static]
-version = "1.4.0"
-optional = true
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.once_cell]
-version = "1.8.0"
-optional = true
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.pathfinder_geometry]
-version = "0.5.1"
-optional = true
-
-[target.'cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"))))'.dependencies.ttf-parser]
-version = "0.20.0"
-optional = true
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.rand]
-version = "0.8.3"
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.rand_distr]
-version = "0.4.0"
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.rand_xorshift]
-version = "0.3.0"
 
 [lints.rust.deprecated]
 level = "allow"
