--- a/src/Makefile.orig.am	2024-05-13 08:25:27.000000000 +0200
+++ b/src/Makefile.am	2024-07-16 21:37:38.681076855 +0200
@@ -71,7 +71,7 @@
 	./mkjs
 
 mkjs: mkjs.cc
-	g++ mkjs.cc -o mkjs
+	g++ mkjs.cc -o mkjs.fake
 
 AM_CPPFLAGS = -DIN_GIAC -I$(srcdir) -I$(top_srcdir) -I$(builddir) \
 	      -I$(top_builddir) $(GMP_CFLAGS) $(NTL_CFLAGS) \
--- a/src/Makefile.orig.in	2024-05-23 17:09:52.000000000 +0200
+++ b/src/Makefile.in	2024-07-16 21:36:49.122491632 +0200
@@ -1470,7 +1470,7 @@
 	./mkjs
 
 mkjs: mkjs.cc
-	g++ mkjs.cc -o mkjs
+	g++ mkjs.cc -o mkjs.fake
 
 install-exec-hook:
 	rm -f $(DESTDIR)$(bindir)/cas_help
