--- pygrib-2.1.6.unchanged/test/test_gaussian.py	2021-09-17 23:19:37.000000000 +0200
+++ pygrib-2.1.6/test/test_gaussian.py	2024-07-19 14:37:26.210892388 +0200
@@ -15,7 +15,7 @@ data = grb.values
 data,lons1 = add_cyclic_point(data, coord=lons1)
 lons,lats = np.meshgrid(lons1,lats1)
 
-@pytest.mark.mpl_image_compare(tolerance=20,remove_text=True)
+@pytest.mark.mpl_image_compare(tolerance=32,remove_text=True)
 def test_gaussian():
     fig = plt.figure()
     ax = plt.axes(projection=ccrs.PlateCarree(central_longitude=0))
