--- sequoia-openpgp-2.4.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sequoia-openpgp-2.4.1/Cargo.toml	2026-01-01T00:00:00+00:00
@@ -562,23 +562,8 @@
 version = "0.2"
 features = ["js"]
 
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = "0.5.1"
-features = [
-    "rand",
-    "block-cipher",
-]
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["bcrypt"]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.zeroize]
-version = "1"
-optional = true
+
+
 
 [lints.clippy.complexity]
 level = "allow"
