diff -up openvpn-auth-ldap-auth-ldap-2.0.4/tools/Makefile.in.tools-CFLAGS openvpn-auth-ldap-auth-ldap-2.0.4/tools/Makefile.in
--- openvpn-auth-ldap-auth-ldap-2.0.4/tools/Makefile.in.tools-CFLAGS	2019-10-25 10:13:21.568169882 -0600
+++ openvpn-auth-ldap-auth-ldap-2.0.4/tools/Makefile.in	2019-10-25 10:13:50.804045318 -0600
@@ -11,7 +11,6 @@ include ${top_builddir}/Mk/subdir.mk
 LEMON_OBJS=	lemon.o
 
 MAKEHEADERS_OBJS=	makeheaders.o
-CFLAGS=
 
 all:: lemon makeheaders
 
