--- interim-0.2.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ interim-0.2.1/Cargo.toml	2025-05-22T10:06:06.875790+00:00
@@ -36,7 +36,6 @@
 [features]
 chrono_0_4 = ["dep:chrono"]
 default = ["std"]
-jiff_0_1 = ["dep:jiff_0_1"]
 jiff_0_2 = ["dep:jiff_0_2"]
 std = []
 time_0_3 = ["dep:time"]
@@ -54,12 +53,6 @@
 optional = true
 default-features = false
 
-[dependencies.jiff_0_1]
-version = "0.1"
-optional = true
-default-features = false
-package = "jiff"
-
 [dependencies.jiff_0_2]
 version = "0.2"
 optional = true
@@ -67,7 +60,7 @@
 package = "jiff"
 
 [dependencies.logos]
-version = "0.15.0"
+version = ">=0.14,<0.16"
 
 [dependencies.time]
 version = "0.3.9"
@@ -82,11 +75,6 @@
 [dev-dependencies.chrono-tz]
 version = "0.10.0"
 
-[dev-dependencies.jiff_0_1]
-version = "0.1"
-features = ["std"]
-package = "jiff"
-
 [dev-dependencies.jiff_0_2]
 version = "0.2"
 features = ["std"]
