--- a/CMakeLists.txt	2024-09-17 14:34:00.473020176 +0300
+++ b/CMakeLists.txt	2024-09-17 15:22:17.528161824 +0300
@@ -131,11 +131,11 @@
     add_library(ittnotify STATIC ${ITT_SRCS} ${ITT_PUBLIC_HDRS} ${ITT_PT} ${ITT_FORTRAN})
     add_library(advisor STATIC ${ADVISOR_ANNOTATION})
 else()
-    add_library(ittnotify STATIC ${ITT_SRCS} ${ITT_PUBLIC_HDRS} ${ITT_PT})
+	add_library(ittnotify SHARED ${ITT_SRCS} ${ITT_PUBLIC_HDRS} ${ITT_PT})
 endif()
 
 set(JITPROFILING_SRC "src/ittnotify/jitprofiling.c")
-add_library(jitprofiling STATIC ${JITPROFILING_SRC})
+add_library(jitprofiling SHARED ${JITPROFILING_SRC})
 
 if(WIN32)
     set_target_properties(ittnotify PROPERTIES OUTPUT_NAME libittnotify)
@@ -168,6 +168,13 @@
     PRIVATE src/ittnotify
 )
 
+# version
+set_target_properties(ittnotify PROPERTIES
+	VERSION 3.25.4
+	SOVERSION 3
+	OUTPUT_NAME "ittnotify"
+)
+
 # install
 
 include(CMakePackageConfigHelpers)
