diff --git a/.github/workflows/main.yml b/.github/workflows/main.yml
index 5f51fdad..f91eaf40 100644
--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["2.7", "3.7", "3.8", "3.9", "3.10"]
+        python-version: ["2.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
 
     steps:
     - uses: actions/checkout@v2
diff --git a/docs/apps/docapps.py b/docs/apps/docapps.py
index 07401800..1a00ee85 100755
--- a/docs/apps/docapps.py
+++ b/docs/apps/docapps.py
@@ -1,15 +1,18 @@
 #!/usr/bin/python
 
 import os
-import imp
+import sys
 import importlib
 from subprocess import Popen, PIPE
 
-path = [importlib.util.find_spec("prody").submodule_search_locations[0]]
-apps = imp.load_module('prody.apps',
-                           *imp.find_module('apps', path))
+from prody.utilities.misctools import impLoadModule
 
-for cmd, subcmds in [('prody', apps.PRODY_APPS), ('evol', apps.EVOL_APPS)]:
+path_apps = importlib.util.find_spec("prody.apps").submodule_search_locations[0]
+
+prody_apps = impLoadModule('prody.apps.prody_apps', path_apps + '/prody_apps/', '__init__')
+evol_apps = impLoadModule('prody.apps.evol_apps', path_apps + '/evol_apps/', '__init__')
+
+for cmd, subcmds in [('prody', prody_apps.PRODY_APPS), ('evol', evol_apps.EVOL_APPS)]:
 
     pipe = Popen([cmd, '-h'], stdout=PIPE, stderr=PIPE)
     with open(os.path.join(cmd, cmd + '.txt'), 'w') as rst:
diff --git a/prody/apps/evol_apps/__init__.py b/prody/apps/evol_apps/__init__.py
index 9bc350aa..e18a688d 100644
--- a/prody/apps/evol_apps/__init__.py
+++ b/prody/apps/evol_apps/__init__.py
@@ -1,6 +1,5 @@
 """This module defines some sequence evolution applications."""
 
-import imp
 import importlib
 import sys
 
@@ -9,14 +8,22 @@ try:
 except ImportError:
     from .. import argparse
 
-from ..apptools import *
+from prody.apps.apptools import *
+from prody.utilities.misctools import impLoadModule
 
 if sys.version_info[0] == 2:
+    import imp
     path_prody = imp.find_module('prody')[1]
 else:
     path_prody = importlib.util.find_spec("prody").submodule_search_locations[0]
-path_apps = imp.find_module('apps', [path_prody])[1]
-path_apps = imp.find_module('evol_apps', [path_apps])[1]
+
+try:
+    import imp
+    path_apps = imp.find_module('apps', [path_prody])[1]
+    path_apps = imp.find_module('evol_apps', [path_apps])[1]
+except ModuleNotFoundError:
+    path_apps = importlib.util.find_spec("prody.apps").submodule_search_locations[0]
+    path_apps += '/evol_apps/'
 
 EVOL_APPS = ['search', 'fetch', 'filter', 'refine', 'merge', 'occupancy',
              'conserv', 'coevol', 'rankorder']
@@ -81,8 +88,7 @@ evol_commands = evol_parser.add_subparsers(
 
 for cmd in EVOL_APPS:
     cmd = 'evol_' + cmd
-    mod = imp.load_module('prody.apps.evol_apps.' + cmd,
-                          *imp.find_module(cmd, [path_apps]))
+    mod = impLoadModule('prody.apps.evol_apps.', cmd, path_apps)
     mod.APP.addApplication(evol_commands)
 
 
diff --git a/prody/apps/prody_apps/__init__.py b/prody/apps/prody_apps/__init__.py
index a31be6cf..c1eeee03 100644
--- a/prody/apps/prody_apps/__init__.py
+++ b/prody/apps/prody_apps/__init__.py
@@ -1,6 +1,5 @@
 """This module defines structure and dynamics analysis applications."""
 
-import imp
 import importlib
 import sys
 
@@ -9,14 +8,22 @@ try:
 except ImportError:
     from .. import argparse
 
-from ..apptools import *
+from prody.apps.apptools import *
+from prody.utilities.misctools import impLoadModule
 
 if sys.version_info[0] == 2:
+    import imp
     path_prody = imp.find_module('prody')[1]
 else:
     path_prody = importlib.util.find_spec("prody").submodule_search_locations[0]
-path_apps = imp.find_module('apps', [path_prody])[1]
-path_apps = imp.find_module('prody_apps', [path_apps])[1]
+
+try:
+    import imp
+    path_apps = imp.find_module('apps', [path_prody])[1]
+    path_apps = imp.find_module('prody_apps', [path_apps])[1]
+except ModuleNotFoundError:
+    path_apps = importlib.util.find_spec("prody.apps").submodule_search_locations[0]
+    path_apps += '/prody_apps/'
 
 PRODY_APPS = ['anm', 'gnm', 'pca', 'eda', 'align', 'blast', 'biomol',
                   'catdcd', 'contacts', 'fetch', 'select', 'energy', 
@@ -44,11 +51,9 @@ prody_commands = prody_parser.add_subparsers(
 
 for cmd in PRODY_APPS:
     cmd = 'prody_' + cmd
-    mod = imp.load_module('prody.apps.prody_apps.' + cmd,
-                          *imp.find_module(cmd, [path_apps]))
+    mod = impLoadModule('prody.apps.prody_apps.', cmd, path_apps)
     mod.addCommand(prody_commands)
 
-
 def prody_main():
 
     if len(sys.argv) == 1:
diff --git a/prody/dynamics/rtbtools.c b/prody/dynamics/rtbtools.c
index ff4d2c5c..1bbf3a8b 100644
--- a/prody/dynamics/rtbtools.c
+++ b/prody/dynamics/rtbtools.c
@@ -419,7 +419,7 @@ void cross(double x[], double y[], double z[])
 int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB, 
 			int nres, int nblx, int bmx)
 {
-  double **X, **I, **IC, *CM, *W, **A, **ISQT;
+  double **X, **Ii, **IC, *CM, *W, **A, **ISQT;
   double x, tr, dd, df;
   int *IDX, nbp, b, i, j, k, ii, jj, aa, bb, elm;
 
@@ -429,7 +429,7 @@ int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB,
   X = dmatrix(1, bmx, 1, 3);
   IDX = ivector(1, bmx);
   CM = dvector(1, 3);
-  I = dmatrix(1, 3, 1, 3);
+  Ii = dmatrix(1, 3, 1, 3);
   IC = dmatrix(1, 3, 1, 3);
   W = dvector(1, 3);
   A = dmatrix(1, 3, 1, 3);
@@ -442,7 +442,7 @@ int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB,
     for(j=1; j<=3; j++)
     {
       CM[j] = 0.0;
-      for(i=1; i<=3; i++) I[i][j] = 0.0;
+      for(i=1; i<=3; i++) Ii[i][j] = 0.0;
       for(i=1; i<=bmx; i++) X[i][j] = 0.0;
     }
 
@@ -478,11 +478,11 @@ int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB,
       }
       for(i=1; i<=3; i++)
       {
-        I[i][i] += (dd - X[k][i] * X[k][i]);
+        Ii[i][i] += (dd - X[k][i] * X[k][i]);
         for(j=i+1; j<=3; j++)
         {
-          I[i][j] -= X[k][i]*X[k][j];
-          I[j][i] = I[i][j];
+          Ii[i][j] -= X[k][i]*X[k][j];
+          Ii[j][i] = Ii[i][j];
 	      }
       }
     }
@@ -490,7 +490,7 @@ int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB,
     /* DIAGONALIZE INERTIA TENSOR */
     for (i=1; i<=3; i++)
       for (j=1; j<=3; j++)
-	      IC[i][j]=I[i][j];
+	      IC[i][j]=Ii[i][j];
     dsvdcmp(IC, 3, 3, W, A);
     deigsrt(W, A, 3);
     righthand2(W, A, 3);
@@ -545,7 +545,7 @@ int dblock_projections2(dSparse_Matrix *PP, PDB_File *PDB,
   free_dmatrix(X, 1, bmx, 1, 3);
   free_ivector(IDX, 1, bmx);
   free_dvector(CM, 1, 3);
-  free_dmatrix(I, 1, 3, 1, 3);
+  free_dmatrix(Ii, 1, 3, 1, 3);
   free_dmatrix(IC, 1, 3, 1, 3);
   free_dvector(W, 1, 3);
   free_dmatrix(A, 1, 3, 1, 3);
diff --git a/prody/tests/dynamics/test_lda.py b/prody/tests/dynamics/test_lda.py
new file mode 100644
index 00000000..839fba37
--- /dev/null
+++ b/prody/tests/dynamics/test_lda.py
@@ -0,0 +1,83 @@
+"""This module contains unit tests for :mod:`~prody.dynamics`."""
+
+import numpy as np
+from numpy import arange
+from numpy.testing import *
+from prody.utilities import importDec
+dec = importDec()
+
+from prody import *
+from prody import LOGGER
+from prody.tests import unittest
+from prody.tests.datafiles import *
+
+LOGGER.verbosity = 'none'
+
+ATOL = 1e-5
+RTOL = 0
+
+ATOMS = parseDatafile('2k39_ca')
+COORDSETS = ATOMS.getCoordsets()
+ENSEMBLE = refineEnsemble(PDBEnsemble(ATOMS), lower=0., upper=5.)
+
+RMSD_MATRIX = ENSEMBLE.getRMSDs(pairwise=True)
+RMSD_TREE = calcTree(ENSEMBLE.getLabels(), RMSD_MATRIX)
+SUBGROUPS = findSubgroups(RMSD_TREE, 2.9)
+CLASSES = [np.nonzero(np.array([i in sg for sg in SUBGROUPS]))[0][0] 
+           for i in ENSEMBLE.getLabels()]
+
+LDA_EVALUES = parseDatafile('lda2k39_evalues')
+LDA_EVECTORS = parseDatafile('lda2k39_vectors')
+
+lda = LDA()
+lda.calcModes(ENSEMBLE, CLASSES)
+
+n_modes = 1
+lda_n_modes = LDA()
+lda_n_modes.calcModes(ENSEMBLE, CLASSES, n_modes=n_modes)
+
+class TestLDA(unittest.TestCase):
+
+    def setUp(self):
+
+        self.model = LDA()
+        self.calcModes = self.model.calcModes
+
+    def testCalcCoordsWrongType(self):
+        """Test response to wrong type *coords* argument."""
+
+        self.assertRaises(TypeError, self.calcModes, 'nogood', CLASSES)
+
+    def testCalcWrongCoords(self):
+        """Test response to wrong coords.dtype."""
+
+        array = np.array([['a','a','a'] for _ in range(10)])
+        self.assertRaises(ValueError, self.calcModes, array, CLASSES)
+
+    def testCalcLabelsWrongType(self):
+        """Test response to wrong type *coords* argument."""
+
+        self.assertRaises(TypeError, self.calcModes, ENSEMBLE, 'nogood')
+
+
+class TestLDAResults(TestLDA):
+
+    def testEigenvalues(self):
+        assert_allclose(lda.getEigvals(), LDA_EVALUES,
+                        rtol=RTOL, atol=ATOL*100,
+                        err_msg='failed to get correct eigenvalues')
+
+    def testEigenvectors(self):
+        _temp = np.abs(np.dot(lda.getEigvecs().T, LDA_EVECTORS))
+        assert_allclose(_temp, np.eye(_temp.shape[0]), 
+                        rtol=1e-10, atol=0.1, # may not be so close to orthonormal
+                       err_msg='failed to get correct eigenvectors')
+
+    def testNumModesAll(self):
+        assert_equal(lda._n_modes, len(LDA_EVALUES))
+
+    def testNumModesSet(self):
+        assert_equal(lda_n_modes._n_modes, n_modes)
+
+if __name__ == '__main__':
+    unittest.main()
diff --git a/prody/tests/proteins/test_fixer.py b/prody/tests/proteins/test_fixer.py
new file mode 100644
index 00000000..a01567dc
--- /dev/null
+++ b/prody/tests/proteins/test_fixer.py
@@ -0,0 +1,38 @@
+"""This module contains unit tests for :mod:`~prody.proteins`."""
+
+from numpy.testing import *
+from prody.utilities import importDec
+dec = importDec()
+
+from prody import *
+from prody import LOGGER
+from prody.tests import unittest
+from prody.tests.datafiles import *
+
+import os
+
+LOGGER.verbosity = 'none'
+
+PATH_1UBI = pathDatafile('1ubi')
+SPLIT_PATH = os.path.split(PATH_1UBI)
+EXPECT_PATH = os.path.join(SPLIT_PATH[0], 'addH_' + SPLIT_PATH[1])
+
+class TestFixer(unittest.TestCase):
+
+    def setUp(self):
+        """Set PDB file data and parse the PDB file."""
+
+        self.ubi_orig = DATA_FILES['1ubi']
+        self.ubi_addH = DATA_FILES['1ubi_addH']
+
+    @dec.slow
+    def testPDBFixer(self):
+
+        if prody.PY3K:
+            self.filename = addMissingAtoms(PATH_1UBI, method='pdbfixer')
+            self.assertEqual(self.filename, EXPECT_PATH,
+                            'addMissing atoms wrote the file in the wrong place')
+            
+            atoms = parsePDB(self.filename)
+            self.assertEqual(len(atoms.hydrogen), DATA_FILES['1ubi_addH']['n_h'],
+                            'addMissing atoms did not add the right number of hydrogens')
diff --git a/prody/tests/proteins/test_waterbridges.py b/prody/tests/proteins/test_waterbridges.py
new file mode 100644
index 00000000..a6b7c1fc
--- /dev/null
+++ b/prody/tests/proteins/test_waterbridges.py
@@ -0,0 +1,142 @@
+"""This module contains unit tests for :mod:`~prody.proteins`."""
+
+import numpy as np
+from numpy.testing import *
+from prody.utilities import importDec
+dec = importDec()
+
+from prody import *
+from prody import LOGGER
+from prody.tests import unittest
+from prody.tests.datafiles import *
+
+import os
+
+LOGGER.verbosity = 'none'
+
+PATH_PDB = pathDatafile('3enl_addH')
+SPLIT_PATH = os.path.split(PATH_PDB)
+TARGET_PATH = os.path.join(SPLIT_PATH[0], 'wb_cluster_' + SPLIT_PATH[1])
+
+class TestWaterBridges(unittest.TestCase):
+
+    def setUp(self):
+        """Parse the PDB file and run the core calculations."""
+        self.atoms = parsePDB(PATH_PDB)
+
+        if prody.PY3K:
+            self.waterBridgesChain = calcWaterBridges(self.atoms)
+            self.chainMultiWater = self.waterBridgesChain[37]
+
+            self.waterBridgesClust = calcWaterBridges(self.atoms, method='cluster')
+            self.clustMultiWater = self.waterBridgesClust[36]
+
+            self.waterBridgesClustInfo = calcWaterBridges(self.atoms, method='cluster', 
+                                                        output='info')
+            self.clustInfoMultiWater = self.waterBridgesClustInfo[36]
+
+            self.pro0 = self.clustMultiWater.proteins[0]
+            self.pro1 = self.clustMultiWater.proteins[1]
+            self.w0 = self.clustMultiWater.waters[0]
+            self.w1 = self.clustMultiWater.waters[1]
+        
+    @dec.slow
+    def testSingleWaterBridgesChain(self):
+        if prody.PY3K:
+            self.assertIsInstance(self.waterBridgesChain, list,
+                                  'calcWaterBridges chain output should be a list')
+            
+            self.assertEqual(len(self.waterBridgesChain), 77,
+                             'calcWaterBridges chain list should have 77 items')
+            
+            
+            self.assertIsInstance(self.chainMultiWater, prody.proteins.waterbridges.AtomicOutput,
+                                  'calcWaterBridges output should contain AtomicOutput')
+            
+            self.assertEqual(len(self.chainMultiWater.proteins), 2,
+                             'water bridges chains should have 2 protein items')
+            self.assertIsInstance(self.chainMultiWater.proteins[0], Atom,
+                                  'protein items in water bridges chains should be Atom')
+            
+            self.assertEqual(len(self.chainMultiWater.waters), 2,
+                             'water bridges chain should have 2 water items')
+            self.assertIsInstance(self.chainMultiWater.waters[0], Atom,
+                                  'waters items in water bridges chains should be Atom')
+            
+    @dec.slow
+    def testSingleWaterBridgesCluster(self):
+        if prody.PY3K:
+            self.assertIsInstance(self.waterBridgesClust, list,
+                                  'calcWaterBridges clust output should be a list')
+            
+            self.assertEqual(len(self.waterBridgesClust), 74,
+                             'calcWaterBridges clust list should have 74 items')
+            
+            self.assertIsInstance(self.clustMultiWater, prody.proteins.waterbridges.AtomicOutput,
+                                  'calcWaterBridges output should contain AtomicOutput')
+            
+            self.assertEqual(len(self.clustMultiWater.proteins), 2,
+                             'tested water bridges multi-water cluster should have 3 protein items')
+            self.assertIsInstance(self.clustMultiWater.proteins[0], Atom,
+                                  'protein items in water bridges clusters should be Atom')
+            
+            self.assertEqual(len(self.clustMultiWater.waters), 2,
+                             'tested water bridges multi-water cluster should have 2 waters items')
+            self.assertIsInstance(self.clustMultiWater.waters[0], Atom,
+                                  'waters items in water bridges clusters should be Atom')
+
+    @dec.slow
+    def testSaveWaterBridges(self):
+        if prody.PY3K:
+            savePDBWaterBridges(self.waterBridgesClust, self.atoms, TARGET_PATH)
+            self.saved = parsePDB(TARGET_PATH)
+
+            self.assertFalse(np.allclose(self.atoms.protein.getBetas(), self.saved.protein.getBetas()),
+                             'savePDBWaterBridges did not change protein b-factors')
+            
+            self.assertFalse(np.allclose(self.atoms.protein.getOccupancies(), self.saved.protein.getOccupancies()),
+                             'savePDBWaterBridges did not change protein occupancies')
+            
+            self.assertFalse(np.equal(self.atoms.water.numAtoms(), self.saved.water.numAtoms()),
+                             'savePDBWaterBridges did not change the number of water atoms')
+            
+    @dec.slow
+    def testSingleWaterBridgesOutputInfo(self):
+        if prody.PY3K:
+            self.assertIsInstance(self.waterBridgesClustInfo, list,
+                                  'calcWaterBridges clust info output should be a list')
+            
+            self.assertEqual(len(self.waterBridgesClustInfo), 74,
+                             'calcWaterBridges clust info list should have 74 items')
+            
+            self.assertIsInstance(self.clustInfoMultiWater, list,
+                                  'tested water bridges multi-water cluster info output should contain lists')
+            
+            self.assertEqual(len(self.clustInfoMultiWater), 9,
+                             'tested water bridges multi-water cluster info should have 9 items')
+            self.assertIsInstance(self.clustInfoMultiWater[0], str,
+                                  'info items in water bridges clusters should be str')
+
+
+            self.assertEqual(self.clustInfoMultiWater[0], self.pro0.getResname() + str(self.pro0.getResnum()),
+                             'item 0 from selected info should be resname + resnum of atom 0')
+            self.assertEqual(self.clustInfoMultiWater[1], self.pro0.getName() + '_' + str(self.pro0.getIndex()),
+                             'item 1 from selected info should be name + _ + index of atom 0')
+            self.assertEqual(self.clustInfoMultiWater[2], self.pro0.getChid(),
+                             'item 2 from selected info should be chid of atom 0')
+
+            self.assertEqual(self.clustInfoMultiWater[3], self.pro1.getResname() + str(self.pro1.getResnum()),
+                             'item 3 from selected info should be resname + resnum of atom 1')
+            self.assertEqual(self.clustInfoMultiWater[4], self.pro1.getName() + '_' + str(self.pro1.getIndex()),
+                             'item 4 from selected info should be name + _ + index of atom 1')
+            self.assertEqual(self.clustInfoMultiWater[5], self.pro1.getChid(),
+                             'item 5 from selected info should be chid of atom 1')
+
+            self.assertEqual(self.clustInfoMultiWater[6], calcDistance(self.pro0, self.pro1),
+                             'item 6 from selected info should be distance from protein atom 0 to protein atom 1')
+            self.assertEqual(self.clustInfoMultiWater[7], len(self.clustMultiWater.waters),
+                             'item 7 from selected info should be number of waters in the cluster')
+            self.assertEqual(self.clustInfoMultiWater[8], 
+                             list(map(lambda w: w.getChid() + "_" + str(w.getIndex()), self.clustMultiWater.waters)),
+                             'item 8 from selected info should be chid and index for 2 water atoms')
+           
diff --git a/prody/utilities/doctools.py b/prody/utilities/doctools.py
index dc9e843b..98d1dcae 100644
--- a/prody/utilities/doctools.py
+++ b/prody/utilities/doctools.py
@@ -45,7 +45,7 @@ def joinTerms(terms, sep=', ', last=None, sort=False):
 
 
 def wrapText(text, width=70, join='\n', **kwargs):
-    """Returns wrapped lines from :func:`textwrap.wrap` after *join*\ing them.
+    """Returns wrapped lines from :func:`textwrap.wrap` after *join*ing them.
     """
 
     try:
diff --git a/prody/utilities/misctools.py b/prody/utilities/misctools.py
index 20b7ef6f..5dd0c92f 100644
--- a/prody/utilities/misctools.py
+++ b/prody/utilities/misctools.py
@@ -21,7 +21,8 @@ __all__ = ['Everything', 'Cursor', 'ImageCursor', 'rangeString', 'alnum', 'impor
            'getDataPath', 'openData', 'chr2', 'toChararray', 'interpY', 'cmp', 'pystr',
            'getValue', 'indentElement', 'isPDB', 'isURL', 'isListLike', 'isSymmetric', 'makeSymmetric',
            'getDistance', 'fastin', 'createStringIO', 'div0', 'wmean', 'bin2dec', 'wrapModes', 
-           'fixArraySize', 'decToHybrid36', 'hybrid36ToDec', 'DTYPE', 'checkIdentifiers', 'split', 'mad']
+           'fixArraySize', 'decToHybrid36', 'hybrid36ToDec', 'DTYPE', 'checkIdentifiers', 'split', 'mad',
+           'importDec', 'impLoadModule']
 
 DTYPE = array(['a']).dtype.char  # 'S' for PY2K and 'U' for PY3K
 CURSORS = []
@@ -784,3 +785,40 @@ def mad(x):
                 return median(abs(x - med))
     
     return _mad(x)
+
+
+def importDec():
+    """Returns an imported module equivalent to numpy testing decorators."""
+
+    try:
+        import numpy.testing.decorators as dec
+    except ModuleNotFoundError:
+        try:
+            from numpy.testing import dec
+        except ImportError:
+            try:
+                import numpy.testing._private.decorators as dec
+            except ModuleNotFoundError:
+                from pytest import mark as dec    
+
+    return dec
+
+
+def impLoadModule(name, cmd, path):
+    """Returns an  an imported module equivalent to imp."""
+
+    if not name.endswith('.'):
+        name += '.'
+
+    try:
+        import imp
+        mod = imp.load_module(name + cmd,
+                              *imp.find_module(cmd, [path]))
+    except ImportError:
+        import importlib
+        loader = importlib.machinery.SourceFileLoader(name + cmd,
+                                                      path + cmd + '.py')
+        mod = importlib.util.types.ModuleType(loader.name)
+        loader.exec_module(mod)
+
+    return mod
diff --git a/pyproject.toml b/pyproject.toml
index 23456bfd..8f38e849 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,2 +1,2 @@
 [build-system]
-requires = ["setuptools", "wheel", "numpy>=1.10,<1.24", "pyparsing", "scipy"]
+requires = ["setuptools", "wheel", "numpy>=1.10", "pyparsing<=3.1.1", "scipy<=1.13.1"]
diff --git a/scripts/evol b/scripts/evol
index 326faafe..bb3581da 100755
--- a/scripts/evol
+++ b/scripts/evol
@@ -1,13 +1,10 @@
 #!/usr/bin/env python
 
-import imp
+from prody.utilities.misctools import impLoadModule
 import importlib
-if sys.version_info[0] == 2:
-    path_prody = imp.find_module('prody')[1]
-else:
-    path_prody = importlib.util.find_spec("prody").submodule_search_locations[0]
-apps = imp.find_module('apps', [path])
-apps = imp.load_module('prody.apps', *apps)
+
+path_apps = importlib.util.find_spec("prody.apps").submodule_search_locations[0]
+apps = impLoadModule('prody.apps.evol_apps', path_apps + '/evol_apps/', '__init__')
 
 if __name__ == '__main__':
     apps.evol_main()
\ No newline at end of file
diff --git a/scripts/prody b/scripts/prody
index ddd8cc0c..d0582985 100755
--- a/scripts/prody
+++ b/scripts/prody
@@ -1,13 +1,10 @@
 #!/usr/bin/env python
 
-import imp
+from prody.utilities.misctools import impLoadModule
 import importlib
-if sys.version_info[0] == 2:
-    path_prody = imp.find_module('prody')[1]
-else:
-    path_prody = importlib.util.find_spec("prody").submodule_search_locations[0]
-apps = imp.find_module('apps', [path])
-apps = imp.load_module('prody.apps', *apps)
+
+path_apps = importlib.util.find_spec("prody.apps").submodule_search_locations[0]
+apps = impLoadModule('prody.apps.prody_apps', path_apps + '/prody_apps/', '__init__')
 
 if __name__ == '__main__':
     apps.prody_main()
\ No newline at end of file
diff --git a/setup.py b/setup.py
index f8a3b454..df43491a 100644
--- a/setup.py
+++ b/setup.py
@@ -12,9 +12,9 @@ if sys.version_info[:2] < (2, 7):
     sys.exit()
 
 if sys.version_info[:2] == (2, 7) or sys.version_info[:2] <= (3, 5):
-    INSTALL_REQUIRES=['numpy>=1.10,<1.24', 'biopython<=1.76', 'pyparsing', 'scipy']
+    INSTALL_REQUIRES=['numpy>=1.10', 'biopython<=1.76', 'pyparsing', 'scipy']
 else:
-    INSTALL_REQUIRES=['numpy>=1.10,<1.24', 'biopython<=1.79', 'pyparsing', 'scipy', 'setuptools']
+    INSTALL_REQUIRES=['numpy>=1.10', 'biopython', 'pyparsing', 'scipy', 'setuptools']
 
 if sys.version_info[0] == 3:
     if sys.version_info[1] < 6:
