From: Adrian Reber <adrian@lisas.de>
Date: Fri, 27 Jul 2012 17:12:26 +0200
Subject: [PATCH] another try to fix the build with -O2


diff --git a/envos.fs.in b/envos.fs.in
index f9d0fd0e..0a073d1c 100644
--- a/envos.fs.in
+++ b/envos.fs.in
@@ -32,7 +32,7 @@ set-current
 
 : libtool-cc ( -- c-addr u ) \ gforth
 \G invocation string of the C compiler used for libtool
-    s" @CC@" ;
+    s" --tag=CXX @CC@ " ;
 
 : libtool-flags ( -- c-addr u ) \ gforth
 \G force no undefined for cygwin
