From bb3963d887fe7983bb17047b79de7b7f8128e585 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sun, 10 Feb 2019 00:58:28 -0500
Subject: [PATCH 1/2] Disable self-update.

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 cli/cli.go | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/cli/cli.go b/cli/cli.go
index fe113ec..0f82811 100644
--- a/cli/cli.go
+++ b/cli/cli.go
@@ -16,7 +16,6 @@ import (
 
 	"github.com/blang/semver"
 	"github.com/exercism/cli/debug"
-	update "github.com/inconshreveable/go-update"
 )
 
 var (
@@ -134,7 +133,7 @@ func (c *CLI) Upgrade() error {
 	}
 	defer bin.Close()
 
-	return update.Apply(bin, update.Options{})
+	return fmt.Errorf("Use dnf to update Fedora-packaged exercism")
 }
 
 func (c *CLI) fetchLatestRelease() error {
-- 
2.50.0

