--- ghc-9.0.2/docs/users_guide/flags.py~	2023-07-27 11:29:29.164886520 +0800
+++ ghc-9.0.2/docs/users_guide/flags.py	2023-07-27 11:31:05.475659936 +0800
@@ -52,7 +52,7 @@
 from sphinx.domains import ObjType
 from sphinx.roles import XRefRole
 from sphinx.errors import SphinxError
-from distutils.version import LooseVersion
+from packaging.version import parse
 from utils import build_table_from_list
 
 import os.path
@@ -629,8 +629,8 @@
 
 def setup(app):
     # The override argument to add_directive_to_domain is only supported by >= 1.8
-    sphinx_version = LooseVersion(sphinx.__version__)
-    override_arg = {'override': True} if sphinx_version >= LooseVersion('1.8') else {}
+    sphinx_version = parse(sphinx.__version__)
+    override_arg = {'override': True} if sphinx_version >= parse('1.8') else {}
 
     # Yuck: We can't use app.add_object_type since we need to provide the
     # Directive instance ourselves.
