diff -up ./dvblinkremote/src/command_interpreter.cpp.orig ./dvblinkremote/src/command_interpreter.cpp
--- ./dvblinkremote/src/command_interpreter.cpp.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./dvblinkremote/src/command_interpreter.cpp	2020-08-13 09:41:34.566724605 +0200
@@ -26,6 +26,8 @@
 #include <sstream>
 #include <cctype>
 #include <ctime>
+#include <cstdio>
+#include <cstring>
 #include "util.h"
 
 using namespace dvblinkremote;
@@ -145,7 +147,7 @@ void CommandInterpreter::SearchEpg()
             ChannelEpgData* channelEpgData = (ChannelEpgData*)*it;
             EpgData& epgData = channelEpgData->GetEpgData();
 
-            printf("EPG data found for channel with id (%s)\n", channelEpgData->GetChannelID());
+            printf("EPG data found for channel with id (%s)\n", channelEpgData->GetChannelID().c_str());
 
             for (std::vector<Program*>::iterator pIt = epgData.begin(); pIt < epgData.end(); pIt++) 
             {
@@ -231,7 +233,7 @@ void CommandInterpreter::PlayChannel()
         Stream* stream = new Stream();
 
         if ((status = m_dvblinkRemoteCommunication.PlayChannel(*streamRequest, *stream)) == DVBLINK_REMOTE_STATUS_OK) {
-          printf("Use the following url to play the stream:\n%s\n\n", stream->GetUrl());
+          printf("Use the following url to play the stream:\n%s\n\n", stream->GetUrl().c_str());
           std::cout << "Press any key to stop stream..." << std::endl;
           std::string s;
           getline(std::cin, s);
diff -up ./dvblinkremote/src/util.cpp.orig ./dvblinkremote/src/util.cpp
--- ./dvblinkremote/src/util.cpp.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./dvblinkremote/src/util.cpp	2020-08-13 09:41:34.566724605 +0200
@@ -28,7 +28,7 @@
 
 void ToLowerCase(std::string &str)
 {
-  std::transform(str.begin(), str.end(), str.begin(), std::tolower);
+  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
 }
 
 bool StringToLong(long& output, const std::string& s, std::ios_base& (*f)(std::ios_base&))
diff -up ./libdvblinkremote/src/curlhttpclient.cpp.orig ./libdvblinkremote/src/curlhttpclient.cpp
--- ./libdvblinkremote/src/curlhttpclient.cpp.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/curlhttpclient.cpp	2020-08-13 09:41:34.566724605 +0200
@@ -22,6 +22,7 @@
  ***************************************************************************/
 
 #include "curlhttpclient.h"
+#include <cstring>
 
 using namespace dvblinkremotehttp;
 
diff -up ./libdvblinkremote/src/dvblinkremotecommunication.cpp.orig ./libdvblinkremote/src/dvblinkremotecommunication.cpp
--- ./libdvblinkremote/src/dvblinkremotecommunication.cpp.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/dvblinkremotecommunication.cpp	2020-08-13 09:41:51.621306746 +0200
@@ -21,6 +21,8 @@
  *
  ***************************************************************************/
 
+#include <cstdarg>
+
 #include "dvblinkremoteconnection.h"
 #include "xml_object_serializer.h"
 #include "generic_response.h"
@@ -258,7 +260,7 @@ DVBLinkRemoteStatusCode DVBLinkRemoteCom
   ClearErrorBuffer();
 
   if ((status = SerializeRequestObject(command, request, xmlData)) != DVBLINK_REMOTE_STATUS_OK) {
-    WriteError("Serialization of request object failed with error code %d (%s).\n", status, GetStatusCodeDescription(status));
+    WriteError("Serialization of request object failed with error code %d (%s).\n", status, GetStatusCodeDescription(status).c_str());
     return status;
   }
 
@@ -274,14 +276,14 @@ DVBLinkRemoteStatusCode DVBLinkRemoteCom
 
   if (!m_httpClient.SendRequest(*httpRequest)) {
     status = DVBLINK_REMOTE_STATUS_CONNECTION_ERROR;
-    WriteError("HTTP request failed with error code %d (%s).\n", status, GetStatusCodeDescription(status));
+    WriteError("HTTP request failed with error code %d (%s).\n", status, GetStatusCodeDescription(status).c_str());
   }
   else {
     dvblinkremotehttp::HttpWebResponse* httpResponse = m_httpClient.GetResponse();
 
     if (httpResponse->GetStatusCode() == 401) {
       status = DVBLINK_REMOTE_STATUS_UNAUTHORISED;
-      WriteError("HTTP response returned status code %d (%s).\n", httpResponse->GetStatusCode(), GetStatusCodeDescription(status));
+      WriteError("HTTP response returned status code %d (%s).\n", httpResponse->GetStatusCode(), GetStatusCodeDescription(status).c_str());
     }
     else if (httpResponse->GetStatusCode() != 200) {
       status = DVBLINK_REMOTE_STATUS_ERROR;
@@ -291,7 +293,7 @@ DVBLinkRemoteStatusCode DVBLinkRemoteCom
       std::string responseData = httpResponse->GetResponseData();
       
       if ((status = DeserializeResponseData(command, responseData, responseObject)) != DVBLINK_REMOTE_STATUS_OK) {
-        WriteError("Deserialization of response data failed with error code %d (%s).\n", status, GetStatusCodeDescription(status));
+        WriteError("Deserialization of response data failed with error code %d (%s).\n", status, GetStatusCodeDescription(status).c_str());
       }
     }
 
diff -up ./libdvblinkremote/src/dvblinkremoteconnection.h.orig ./libdvblinkremote/src/dvblinkremoteconnection.h
--- ./libdvblinkremote/src/dvblinkremoteconnection.h.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/dvblinkremoteconnection.h	2020-08-13 09:41:34.567724580 +0200
@@ -31,6 +31,12 @@
 
 namespace dvblinkremote 
 {
+
+  #ifndef _MSC_VER
+  #define vsprintf_s vsprintf
+  #define _snprintf_s(a,b,c,...) snprintf(a,b,__VA_ARGS__)
+  #endif
+
   class DVBLinkRemoteCommunication : public IDVBLinkRemoteConnection
   {
   public:
diff -up ./libdvblinkremote/src/playback_container.cpp.orig ./libdvblinkremote/src/playback_container.cpp
--- ./libdvblinkremote/src/playback_container.cpp.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/playback_container.cpp	2020-08-13 09:41:34.568724556 +0200
@@ -27,11 +27,11 @@
 using namespace dvblinkremote;
 using namespace dvblinkremoteserialization;
 
-PlaybackContainer::PlaybackContainer(const std::string& objectId, const std::string& parentId, const std::string& name, const DVBLinkPlaybackContainerType containerType, const DVBLinkPlaybackContainerContentType containerContentType)
+PlaybackContainer::PlaybackContainer(const std::string& objectId, const std::string& parentId, const std::string& name, DVBLinkPlaybackContainerType& containerType, DVBLinkPlaybackContainerContentType& containerContentType)
   : PlaybackObject(PlaybackObject::PLAYBACK_OBJECT_TYPE_CONTAINER, objectId, parentId),
     m_name(name),
-    m_containerType((DVBLinkPlaybackContainerType)containerType),
-    m_containerContentType((DVBLinkPlaybackContainerContentType)containerContentType),
+    m_containerType((DVBLinkPlaybackContainerType&)containerType),
+    m_containerContentType((DVBLinkPlaybackContainerContentType&)containerContentType),
     Description(""), 
     Logo(""), 
     TotalCount(0),
diff -up ./libdvblinkremote/src/recording_settings.cpp.orig ./libdvblinkremote/src/recording_settings.cpp
--- ./libdvblinkremote/src/recording_settings.cpp.orig	2020-08-13 09:41:09.615336242 +0200
+++ ./libdvblinkremote/src/recording_settings.cpp	2020-08-13 09:41:34.568724556 +0200
@@ -28,6 +28,12 @@
 using namespace dvblinkremote;
 using namespace dvblinkremoteserialization;
 
+RecordingSettings::RecordingSettings()
+{ }
+
+RecordingSettings::~RecordingSettings()
+{ }
+
 GetRecordingSettingsRequest::GetRecordingSettingsRequest()
 { }
 
diff -up ./libdvblinkremote/src/request.h.orig ./libdvblinkremote/src/request.h
--- ./libdvblinkremote/src/request.h.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/request.h	2020-08-13 09:41:34.568724556 +0200
@@ -616,9 +616,11 @@ namespace dvblinkremote {
     std::string m_clientId;
   };
   
-  class AddScheduleRequest : public Request, public Schedule
+  class AddScheduleRequest : public Request, public virtual Schedule
   {
   public:
+	//AddScheduleRequest(const DVBLinkScheduleType scheduleType, const std::string& channelId);
+	AddScheduleRequest();
     virtual ~AddScheduleRequest() = 0;
   };
 
@@ -627,7 +629,7 @@ namespace dvblinkremote {
     * This is used as input parameter for the IDVBLinkRemoteConnection::AddSchedule method.
     * @see IDVBLinkRemoteConnection::AddSchedule()
     */
-  class AddManualScheduleRequest : public ManualSchedule
+  class AddManualScheduleRequest : public ManualSchedule, public AddScheduleRequest
   {
   public:
     /**
@@ -653,7 +655,7 @@ namespace dvblinkremote {
     * This is used as input parameter for the IDVBLinkRemoteConnection::AddSchedule method.
     * @see IDVBLinkRemoteConnection::AddSchedule m()
     */
-  class AddScheduleByEpgRequest : public EpgSchedule
+  class AddScheduleByEpgRequest : public EpgSchedule, public AddScheduleRequest
   {
   public:
     /**
diff -up ./libdvblinkremote/src/response.h.orig ./libdvblinkremote/src/response.h
--- ./libdvblinkremote/src/response.h.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/response.h	2020-08-13 09:41:34.568724556 +0200
@@ -806,7 +806,7 @@ namespace dvblinkremote {
       * @param containerContentType a constant DVBLinkPlaybackContainerContentType instance representing the content 
       * type of the playback items in this playback container.
       */
-    PlaybackContainer(const std::string& objectId, const std::string& parentId, const std::string& name, const DVBLinkPlaybackContainerType containerType, const DVBLinkPlaybackContainerContentType containerContentType);
+    PlaybackContainer(const std::string& objectId, const std::string& parentId, const std::string& name, DVBLinkPlaybackContainerType& containerType, DVBLinkPlaybackContainerContentType& containerContentType);
 
     /**
       * Destructor for cleaning up allocated memory.
diff -up ./libdvblinkremote/src/scheduling.cpp.orig ./libdvblinkremote/src/scheduling.cpp
--- ./libdvblinkremote/src/scheduling.cpp.orig	2020-08-13 09:41:09.616336217 +0200
+++ ./libdvblinkremote/src/scheduling.cpp	2020-08-13 09:41:34.569724531 +0200
@@ -27,6 +27,11 @@
 using namespace dvblinkremote;
 using namespace dvblinkremoteserialization;
 
+Schedule::Schedule()
+{
+
+}
+
 Schedule::Schedule(const DVBLinkScheduleType scheduleType, const std::string& channelId, const int recordingsToKeep) 
   : m_scheduleType(scheduleType), 
     m_channelId(channelId), 
@@ -137,8 +142,19 @@ std::string& EpgSchedule::GetProgramID()
   return m_programId; 
 }
 
+AddScheduleRequest::AddScheduleRequest()
+{
+
+}
+
+AddScheduleRequest::~AddScheduleRequest()
+{
+
+}
+
+
 AddManualScheduleRequest::AddManualScheduleRequest(const std::string& channelId, const long startTime, const long duration, const long dayMask, const std::string& title)
-  : ManualSchedule(channelId, startTime, duration, dayMask, title)
+  : ManualSchedule(channelId, startTime, duration, dayMask, title), AddScheduleRequest(), Schedule(Schedule::SCHEDULE_TYPE_MANUAL, channelId)
 {
 
 }
@@ -149,7 +165,7 @@ AddManualScheduleRequest::~AddManualSche
 }
 
 AddScheduleByEpgRequest::AddScheduleByEpgRequest(const std::string& channelId, const std::string& programId, const bool repeat, const bool newOnly, const bool recordSeriesAnytime)
-  : EpgSchedule(channelId, programId, repeat, newOnly, recordSeriesAnytime)
+  : EpgSchedule(channelId, programId, repeat, newOnly, recordSeriesAnytime), AddScheduleRequest(), Schedule(Schedule::SCHEDULE_TYPE_BY_EPG, channelId)
 {
 
 }
@@ -240,7 +256,7 @@ bool GetSchedulesRequestSerializer::Writ
 }
 
 StoredManualSchedule::StoredManualSchedule(const std::string& id, const std::string& channelId, const long startTime, const long duration, const long dayMask, const std::string& title)
-  : ManualSchedule(id, channelId, startTime, duration, dayMask, title)
+  : ManualSchedule(id, channelId, startTime, duration, dayMask, title), Schedule(Schedule::SCHEDULE_TYPE_MANUAL,id, channelId)
 {
 
 }
@@ -257,7 +273,7 @@ StoredManualScheduleList::~StoredManualS
 }
 
 StoredEpgSchedule::StoredEpgSchedule(const std::string& id, const std::string& channelId, const std::string& programId, const bool repeat, const bool newOnly, const bool recordSeriesAnytime)
-  : EpgSchedule(id, channelId, programId, repeat, newOnly, recordSeriesAnytime)
+  : EpgSchedule(id, channelId, programId, repeat, newOnly, recordSeriesAnytime), Schedule(Schedule::SCHEDULE_TYPE_BY_EPG,id, channelId)
 {
 
 }
diff -up ./libdvblinkremote/src/scheduling.h.orig ./libdvblinkremote/src/scheduling.h
--- ./libdvblinkremote/src/scheduling.h.orig	2012-10-22 01:07:36.000000000 +0200
+++ ./libdvblinkremote/src/scheduling.h	2020-08-13 09:41:34.569724531 +0200
@@ -103,7 +103,15 @@ namespace dvblinkremote {
       */
     DVBLinkScheduleType& GetScheduleType();
 
+  protected:
+
+    /**
+      * Protected constructor used to solve diamond problem in adding schedules
+      */	 
+	Schedule();
+
   private:
+
     /**
       * The identifier for the schedule.
       */
@@ -119,11 +127,11 @@ namespace dvblinkremote {
       */
     DVBLinkScheduleType m_scheduleType;
   };
-
+  
   /**
     * Abstract base class for manual schedules. 
     */
-  class ManualSchedule : public Schedule
+  class ManualSchedule :  public virtual Schedule
   {
   public:
     /**
@@ -214,7 +222,7 @@ namespace dvblinkremote {
   /**
     * Abstract base class for electronic program guide (EPG) schedules. 
     */
-  class EpgSchedule : public Schedule
+  class EpgSchedule : public virtual Schedule
   {
   public:
     /**
