From 259f79cb24ded7a09eefab7b801b50df41e7f84b Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Sun, 21 Sep 2025 15:34:49 +0200
Subject: [PATCH 3/3] Relax ipnetwork dependency to allow both v0.20 and v0.21

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index e6fcd4c..85226c3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ getopts = "0.2"
 diesel = { version = "2.2.11", features = ["postgres"] }
 dotenv = "0.15.0"
 dns-lookup = "2.0.4"
-ipnetwork = "0.21.1"
+ipnetwork = ">=0.20,<0.22"
 indicatif = "0.18.0"
 console = "0.16.0"
 tracing = "0.1.41"
-- 
2.51.0

