--- ftnchek-3.3.1/dcl2inc.in.orig	1999-11-02 17:23:36.000000000 -0700
+++ ftnchek-3.3.1/dcl2inc.in	2006-03-07 14:45:33.000000000 -0700
@@ -24,5 +24,5 @@
 # The following lines are edited by configure when it builds dcl2inc.
 prefix=@prefix@
 exec_prefix=@exec_prefix@  
-libdir=@libdir@/ftnchek 
-exec @AWK@ -f ${libdir}/dcl2inc.awk "$@"
+datadir=@datadir@/ftnchek 
+exec @AWK@ -f ${datadir}/dcl2inc.awk "$@"
--- ftnchek-3.3.1/Makefile.in.orig	2006-03-07 14:45:46.000000000 -0700
+++ ftnchek-3.3.1/Makefile.in	2006-03-07 14:47:00.000000000 -0700
@@ -99,7 +99,7 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-datadir		= @datadir@
+datadir		= @datadir@/ftnchek
 mandir		= @mandir@
 manext		= .1
 sgimansubdir	= @sgimansubdir@
@@ -496,9 +496,9 @@
 	$(CP) ftnchek$(EXE) $(bindir)
 	-$(STRIP) $(bindir)/ftnchek$(EXE)
 	$(CHMOD) 755 $(bindir)/ftnchek$(EXE)
-	-$(MKDIR) -p $(libdir)
-	$(CP) dcl2inc.awk $(libdir)/dcl2inc.awk
-	$(CHMOD) 644 $(libdir)/dcl2inc.awk
+	-$(MKDIR) -p $(datadir)
+	$(CP) dcl2inc.awk $(datadir)/dcl2inc.awk
+	$(CHMOD) 644 $(datadir)/dcl2inc.awk
 	$(CP) dcl2inc$(CMD) $(bindir)/dcl2inc$(CMD)
 	$(CHMOD) 755 $(bindir)/dcl2inc$(CMD)
 	-$(RM) $(bindir)/fcl2vcg
@@ -580,14 +580,14 @@
 uninstall:
 	-$(RM) $(bindir)/dcl2inc$(CMD)
 	-$(RM) $(bindir)/ftnchek$(CMD)
-	-$(RM) $(libdir)/dcl2inc.awk
+	-$(RM) $(datadir)/dcl2inc.awk
 	-$(RM) $(mandir)/cat1/dcl2inc$(manext)
 	-$(RM) $(mandir)/cat1/ftnchek$(manext)
 	-$(RM) $(mandir)/dcl2inc.z $(mandir)/ftnchek.z # SGI
 	-$(RM) $(mandir)/man1/dcl2inc$(manext)
 	-$(RM) $(mandir)/man1/ftnchek$(manext)
 	-$(RM) $(lispdir)/ftnchek.el
-	-$(RMDIR) $(libdir)
+	-$(RMDIR) $(datadir)
 
 # WARNING: do NOT execute this target, unless you have nroff/troff or groff
 # to recreate the formatted documentation files.
