diff --git a/conda/activate.py b/conda/activate.py
index f79a03f..951bed4 100644
--- a/conda/activate.py
+++ b/conda/activate.py
@@ -914,13 +914,7 @@ class PosixActivator(_Activator):
     set_var_tmpl = "%s='%s'"
     run_script_tmpl = ". \"`cygpath '%s'`\"" if on_win else '. "%s"'
 
-    hook_source_path = Path(
-        CONDA_PACKAGE_ROOT,
-        "shell",
-        "etc",
-        "profile.d",
-        "conda.sh",
-    )
+    hook_source_path = Path("/etc/profile.d/conda.sh")
     inline_hook_source = True
 
     def _update_prompt(self, set_vars, conda_prompt_modifier):
@@ -956,13 +950,7 @@ class CshActivator(_Activator):
     set_var_tmpl = "set %s='%s'"
     run_script_tmpl = "source \"`cygpath '%s'`\"" if on_win else 'source "%s"'
 
-    hook_source_path = Path(
-        CONDA_PACKAGE_ROOT,
-        "shell",
-        "etc",
-        "profile.d",
-        "conda.csh",
-    )
+    hook_source_path = Path("/etc/profile.d/conda.csh")
     # TCSH/CSH removes newlines when doing command substitution (see `man tcsh`),
     # source conda.csh directly and use line terminators to separate commands
     inline_hook_source = False
@@ -1001,7 +989,7 @@ class XonshActivator(_Activator):
         else 'source-bash --suppress-skip-message -n "%s"'
     )
 
-    hook_source_path = Path(CONDA_PACKAGE_ROOT, "shell", "conda.xsh")
+    hook_source_path = Path("/etc/profile.d/conda.xsh")
     inline_hook_source = True
 
     def template_path_var(self, key: str, value: str) -> str:
@@ -1055,14 +1043,7 @@ class FishActivator(_Activator):
     set_var_tmpl = 'set -g %s "%s"'
     run_script_tmpl = 'source "%s"'
 
-    hook_source_path = Path(
-        CONDA_PACKAGE_ROOT,
-        "shell",
-        "etc",
-        "fish",
-        "conf.d",
-        "conda.fish",
-    )
+    hook_source_path = Path("/etc/fish/conf.d/conda.fish")
     inline_hook_source = True
 
 
diff --git a/conda/base/context.py b/conda/base/context.py
index adf6d6e..612ea4a 100644
--- a/conda/base/context.py
+++ b/conda/base/context.py
@@ -848,7 +848,7 @@ class Context(Configuration):
     def av_data_dir(self) -> PathType:
         """Where critical artifact verification data (e.g., various public keys) can be found."""
         # TODO (AV): Find ways to make this user configurable?
-        return join(self.conda_prefix, "etc", "conda")
+        return '/etc/conda'
 
     @property
     def signing_metadata_url_base(self) -> str | None:
