diff -up OpenSubdiv-3_5_0/documentation/OpenSubdiv.doxy.reproducible OpenSubdiv-3_5_0/documentation/OpenSubdiv.doxy
--- OpenSubdiv-3_5_0/documentation/OpenSubdiv.doxy.reproducible	2022-09-26 14:17:44.000000000 -0400
+++ OpenSubdiv-3_5_0/documentation/OpenSubdiv.doxy	2023-01-02 19:24:18.827692805 -0500
@@ -154,7 +154,7 @@ INLINE_INHERITED_MEMB  = NO
 # path before files name in the file list and in the header files. If set
 # to NO the shortest path that makes the file name unique will be used.
 
-FULL_PATH_NAMES        = YES
+FULL_PATH_NAMES        = NO
 
 # If the FULL_PATH_NAMES tag is set to YES then the STRIP_FROM_PATH tag
 # can be used to strip a user-defined part of the path. Stripping is
@@ -180,7 +180,7 @@ STRIP_FROM_INC_PATH    =
 # (but less readable) file names. This can be useful if your file system
 # doesn't support long names like on DOS, Mac, or CD-ROM.
 
-SHORT_NAMES            = YES
+SHORT_NAMES            = NO
 
 # If the JAVADOC_AUTOBRIEF tag is set to YES then Doxygen
 # will interpret the first line (until the first dot) of a JavaDoc-style
@@ -1031,7 +1031,7 @@ HTML_COLORSTYLE_GAMMA  = 80
 # page will contain the date and time when the page was generated. Setting
 # this to NO can help when comparing the output of multiple runs.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO
 
 # If the HTML_DYNAMIC_SECTIONS tag is set to YES then the generated HTML
 # documentation will contain sections that can be hidden and shown after the
