--- configure~	2011-08-12 21:35:20.000000000 +0200
+++ configure	2011-08-12 21:35:20.000000000 +0200
@@ -3036,7 +3036,7 @@
   else
     GTK_PREFIX=`$PKG_CONFIG $GTK --variable=prefix`
     GTK_CFLAGS=`$PKG_CONFIG $GTK --cflags`
-    GTK_LIBS=`$PKG_CONFIG $GTK --libs`
+    GTK_LIBS=-Wl,--as-needed\ `$PKG_CONFIG $GTK --libs`
     gtk_config_major_version=`$PKG_CONFIG $GTK --modversion | \
            sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1/'`
     gtk_config_minor_version=`$PKG_CONFIG $GTK --modversion | \
