diff --git a/awscli/customizations/wizard/app.py b/awscli/customizations/wizard/app.py
--- a/awscli/customizations/wizard/app.py
+++ b/awscli/customizations/wizard/app.py
@@ -10,9 +10,9 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
+import asyncio
 import json
 import os
-from asyncio import new_event_loop, set_event_loop
 from collections.abc import MutableMapping
 
 from prompt_toolkit.application import Application
@@ -77,14 +77,12 @@
         )
 
     def run(self, pre_run=None, **kwargs):
-        loop = new_event_loop()
-        try:
-            set_event_loop(loop)
+        def create_event_loop():
+            loop = asyncio.new_event_loop()
             loop.set_exception_handler(self._handle_exception)
-            f = self.run_async(pre_run=pre_run, set_exception_handler=False)
-            return loop.run_until_complete(f)
-        finally:
-            loop.close()
+            return loop
+        f = self.run_async(pre_run=pre_run, set_exception_handler=False)
+        return asyncio.run(f, loop_factory=create_event_loop)
 
     def _handle_exception(self, loop, context):
         self.exit(exception=UnexpectedWizardException(context['exception']))
diff --git a/tests/__init__.py b/tests/__init__.py
--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -475,16 +475,9 @@
         # This is the function that will be passed to our thread to
         # actually run the application
         try:
-            # When we run the app in a separate thread, there is no
-            # default event loop. This ensures we create one as it is
-            # likely the application will try to grab the default loop
-            loop = asyncio.new_event_loop()
-            asyncio.set_event_loop(loop)
             app_run_context.return_value = target(*target_args)
         except BaseException as e:
             app_run_context.raised_exception = e
-        finally:
-            loop.close()
 
     def _wait_until_app_is_done_updating(self):
         self._wait_until_app_is_done_rendering()
