--- tests/standalone_minimal_maximization.ml.orig      2017-08-23 06:49:52.000000000 -0600
+++ tests/standalone_minimal_maximization.ml   2019-05-27 21:09:08.353940039 -0600
@@ -97,10 +97,11 @@ let () =
   *)
 
   (* x <= 5 *)
-  let sim = Sim.Assert.var sim "x" three (Ex.singleton "x>=3") None Ex.empty in
+  let sim, _ =
+    Sim.Assert.var sim "x" three (Ex.singleton "x>=3") None Ex.empty in
 
   (* s == x + y <= 10 *)
-  let sim =
+  let sim, _ =
     Sim.Assert.poly sim x_y "s" None Ex.empty ten (Ex.singleton "x+y<=10") in
 
   aux sim x_y;
--- tests/standalone_minimal.ml.orig   2017-08-23 06:49:52.000000000 -0600
+++ tests/standalone_minimal.ml        2019-05-27 21:08:36.913019394 -0600
@@ -63,14 +63,16 @@ let () =
   let m_one = Some (Rat.m_one, Rat.zero) in
 
   (* x >= 0 *)
-  let sim = Sim.Assert.var sim "x" zero (Ex.singleton "x>=0") None Ex.empty in
+  let sim, _ =
+    Sim.Assert.var sim "x" zero (Ex.singleton "x>=0") None Ex.empty in
 
   (* y >= 0 *)
-  let sim = Sim.Assert.var sim "y" zero (Ex.singleton "y>=0") None Ex.empty in
+  let sim, _ =
+    Sim.Assert.var sim "y" zero (Ex.singleton "y>=0") None Ex.empty in
   let x_y = Sim.Core.P.from_list ["x", Rat.one; "y", Rat.one] in
 
   (* z == x + y <= -1 *)
-  let sim =
+  let sim, _ =
     Sim.Assert.poly sim x_y "z" None Ex.empty m_one (Ex.singleton "x+y<=-1") in
   let sim = Sim.Solve.solve sim in
   match Sim.Result.get None sim with
