diff --git a/src/follow.c b/src/follow.c
index 09456c9..bf80da7 100644
--- a/src/follow.c
+++ b/src/follow.c
@@ -21,12 +21,7 @@
 # ifdef APOLLO
 #  include "/sys5.3/usr/include/sys/termio.h"
 # else
-/*
- * including both termio.h and termios.h might be an overkill, and gives
- * many warnings, but seems to be necessary at times. works anyway.
- */
 #  include <termios.h>
-#  include <termio.h>
 # endif
 /* #else USE_SGTTY */
 #endif
diff --git a/src/tty.c b/src/tty.c
index a1e8590..123a5d2 100644
--- a/src/tty.c
+++ b/src/tty.c
@@ -40,12 +40,7 @@
 #  ifdef APOLLO
 #    include "/sys5.3/usr/include/sys/termio.h"
 #  else
-/*
- * including both termio.h and termios.h might be an overkill, and gives
- * many warnings, but seems to be necessary at times. works anyway.
- */
 #    include <termios.h>
-#    include <termio.h>
 #  endif
 /* #else USE_SGTTY */
 #endif
