diff -up huami-token/pyproject.toml.orig huami-token/pyproject.toml
--- huami-token/pyproject.toml.orig	2024-09-24 12:26:00.000000000 +0200
+++ huami-token/pyproject.toml	2024-09-25 09:06:13.210627706 +0200
@@ -26,7 +26,7 @@ dependencies = [
 "Bug Tracker" = "https://codeberg.org/argrento/huami-token/issues"
 
 [project.scripts]
-huami_token = "huami_token:main"
+huami_token = "huami_token.huami_token:main"
 
 
 [project.optional-dependencies]
