diff -up mctc-lib-0.3.2/config/install-mod.py.python3 mctc-lib-0.3.2/config/install-mod.py
--- mctc-lib-0.3.2/config/install-mod.py.python3	2024-09-06 08:15:30.540204664 +0000
+++ mctc-lib-0.3.2/config/install-mod.py	2024-09-06 08:16:14.494427266 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # This file is part of mctc-lib.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
