From 092f16cafa0a1ef5f001a35b54dd896f601f1428 Mon Sep 17 00:00:00 2001
From: Vladimir Vuksan <44271-vvuksan-fastly@users.noreply.drupalcode.org>
Date: Mon, 5 May 2025 10:16:00 -0400
Subject: [PATCH 1/2] Sanitize input for timezone

---
 graph.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/graph.php b/graph.php
index e8a7a9c..fdf9c86 100644
--- a/graph.php
+++ b/graph.php
@@ -946,7 +946,7 @@ function rrdgraph_cmd_build($rrdtool_graph,
 
   $command = '';
   if (isset($_SESSION['tz']) && ($_SESSION['tz'] != ''))
-    $command .= "TZ='" . $_SESSION['tz'] . "' ";
+    $command .= "TZ='" . escapeshellcmd($_SESSION['tz']) . "' ";
 
   $command .=
     $conf['rrdtool'] .
@@ -1123,7 +1123,7 @@ function output_data_to_external_format($rrdtool_graph_series,
 
   $command = '';
   if (isset($_SESSION['tz']) && ($_SESSION['tz'] != ''))
-    $command .= "TZ='" . $_SESSION['tz'] . "' ";
+    $command .= "TZ='" . escapeshellcmd($_SESSION['tz']) . "' ";
 
   $command .= $rrdtool .
     " xport --start '" . $rrdtool_graph_start .
-- 
2.51.0

