diff -rupN jack-keyboard-2.7.1.old/CMakeLists.txt jack-keyboard-2.7.1/CMakeLists.txt
--- jack-keyboard-2.7.1.old/CMakeLists.txt	2011-12-27 04:50:32.000000000 -0500
+++ jack-keyboard-2.7.1/CMakeLists.txt	2012-06-13 00:09:44.009221984 -0400
@@ -26,6 +26,7 @@
 # Makefile for Jack Keyboard
 #
 cmake_minimum_required(VERSION 2.8)
+include(FindPkgConfig)
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/")
 set(VERSION "2.7.2")
@@ -39,8 +40,9 @@ project(jack-keyboard)
 add_executable(jack-keyboard src/jack-keyboard src/pianokeyboard)
 
 find_package(GTK2 2.2 REQUIRED gtk)
-include_directories(${GTK2_INCLUDE_DIRS})
-target_link_libraries(jack-keyboard ${GTK2_LIBRARIES})
+pkg_check_modules(GTHREAD2 REQUIRED gthread-2.0)
+include_directories(${GTK2_INCLUDE_DIRS} ${GTHREAD2_INCLUDE_DIRS})
+target_link_libraries(jack-keyboard ${GTK2_LIBRARIES} ${GTHREAD2_LIBRARIES})
 
 if(JackEnable)
 find_package(JACK)
