diff -rupN --no-dereference qtshadertools-everywhere-src-6.10.0/src/3rdparty/glslang/glslang/MachineIndependent/SymbolTable.h qtshadertools-everywhere-src-6.10.0-new/src/3rdparty/glslang/glslang/MachineIndependent/SymbolTable.h
--- qtshadertools-everywhere-src-6.10.0/src/3rdparty/glslang/glslang/MachineIndependent/SymbolTable.h	2025-08-17 21:32:16.000000000 +0200
+++ qtshadertools-everywhere-src-6.10.0-new/src/3rdparty/glslang/glslang/MachineIndependent/SymbolTable.h	2025-09-02 18:30:05.080527096 +0200
@@ -65,6 +65,8 @@
 //   are tracked in the intermediate representation, not the symbol table.
 //
 
+#include <cstdint>
+
 #include "../Include/Common.h"
 #include "../Include/intermediate.h"
 #include "../Include/InfoSink.h"
