diff -up texlive-base-20230311-build/texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp.me texlive-base-20230311-build/texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp
--- texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp.me	2026-01-22 16:05:28.236115941 +0100
+++ texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp	2026-01-22 16:10:11.204612649 +0100
@@ -1970,7 +1970,7 @@ void Clipper::DeleteFromSEL(TEdge *e)
 #ifdef use_xyz
 void Clipper::SetZ(IntPoint& pt, TEdge& e1, TEdge& e2)
 {
-  if (pt.Z != 0 || !m_ZFill) return;
+  if (pt.Z != ClipperLib::ZType(0) || !m_ZFill) return;
   else if (pt == e1.Bot) pt.Z = e1.Bot.Z;
   else if (pt == e1.Top) pt.Z = e1.Top.Z;
   else if (pt == e2.Bot) pt.Z = e2.Bot.Z;
diff -up texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp.me texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp
--- texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp.me	2026-01-22 19:44:39.628015473 +0100
+++ texlive-base-20230311/source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp	2026-01-22 19:45:03.071537295 +0100
@@ -24,7 +24,8 @@
 #include <string>
 
 struct MemoryFontData {
-	MemoryFontData () =delete;
+   constexpr MemoryFontData(const char *d, size_t s) noexcept
+      : data(d), size(s) {}
 	const char *data;
 	size_t size;
 };
