From 39d3dcc8f8ae4e7e751cb0ba633301630e32f54e Mon Sep 17 00:00:00 2001
From: Aki Tuomi <aki.tuomi@open-xchange.com>
Date: Tue, 20 May 2025 11:47:24 +0300
Subject: [PATCH] configure: Use libssl_iostream_openssl.so with installed
 dovecot

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index d8c3405..df881b5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -25,8 +25,8 @@ AS_IF([test "$DOVECOT_INSTALLED" = 'no'], [
 	LIBDOVECOT=$abs_dovecotdir/src/lib-dovecot/libdovecot.la
 	LIBDOVECOT_SSL=$abs_dovecotdir/src/lib-ssl-iostream/libssl_iostream_openssl.la
 ], [
-	LIBDOVECOT=$dovecot_pkglibdir/libdovecot.la
-	LIBDOVECOT_SSL=$dovecot_moduledir/libssl_iostream_openssl.la
+	LIBDOVECOT=$dovecot_pkglibdir/libdovecot.so
+	LIBDOVECOT_SSL=$dovecot_moduledir/libssl_iostream_openssl.so
 ])
 AC_SUBST([LIBDOVECOT_SSL])
 
