--- gix-config-value-0.19.1/Cargo.toml
+++ gix-config-value-0.19.1/Cargo.toml
@@ -36,10 +36,7 @@
 features = ["document-features"]
 
 [features]
-serde = [
-    "dep:serde",
-    "bstr/serde",
-]
+serde = ["dep:serde", "bstr/serde"]
 
 [lib]
 name = "gix_config_value"
@@ -70,9 +67,6 @@
 [dependencies.thiserror]
 version = "2.0.18"
 
-[target."cfg(not(windows))".dependencies.libc]
-version = "0.2"
-
 [lints.clippy]
 bool_to_int_with_if = "allow"
 borrow_as_ptr = "allow"
