diff -u tests/socket_tests-ip.adb tests/socket_tests-ip.adb
--- tests/socket_tests-ip.adb	2016-06-29 12:26:01.000000000 +0200
+++ tests/socket_tests-ip.adb	2016-07-11 17:27:27.499316612 +0200
@@ -331,6 +331,7 @@
       Grp  : constant IPv6_Addr_Type
         := To_IPv6_Addr (Str => "ff04:0000:0000:0000:0000:0000:0001:0002");
    begin
+      Skip(Message => "Doesn't work in Koji.");
       if Test_Utils.OS = Test_Utils.BSD then
          Skip (Message => "Not supported");
       end if;
