--- gambas-3.14.2/gb.pdf/configure.ac
+++ gambas-3.14.2/gb.pdf/configure.ac
@@ -39,6 +39,8 @@ if test "$have_poppler" = "yes"; then
   AC_DEFINE_UNQUOTED(POPPLER_VERSION_0_72, $((1-$?)), Poppler version >= 0.72)
   pkg-config --atleast-version=0.76.0 poppler
   AC_DEFINE_UNQUOTED(POPPLER_VERSION_0_76, $((1-$?)), Poppler version >= 0.76)
+  pkg-config --atleast-version=0.83.0 poppler
+  AC_DEFINE_UNQUOTED(POPPLER_VERSION_0_83, $((1-$?)), Poppler version >= 0.83)
 fi
 
 AC_OUTPUT( \
--- gambas-3.14.2/gb.pdf/src/main.cpp
+++ gambas-3.14.2/gb.pdf/src/main.cpp
@@ -55,11 +55,15 @@ int EXPORT GB_INIT(void)
 {
 	if (!globalParams)
 	{
+#if POPPLER_VERSION_0_83
+		globalParams = std::unique_ptr<GlobalParams>(new GlobalParams());
+#else
 #if POPPLER_VERSION_0_6
 		globalParams = new GlobalParams();
 #else
 		globalParams = new GlobalParams("/etc/xpdfrc");
 #endif
+#endif
 
 #if POPPLER_VERSION_0_5
 #else
