--- CMakeLists.txt~	2019-09-19 11:56:56.000000000 -0500
+++ CMakeLists.txt	2019-09-19 12:00:30.716178022 -0500
@@ -326,15 +326,8 @@
         set(GLUT_FOUND TRUE)
     endif ()
 
-    if(GLUT_FOUND)
-        message(STATUS "Glut found: '${GLUT_INCLUDE_DIR}', '${GLUT_LIBRARIES}'")
-    else()
-        message(STATUS "Glut include and library variables: '${GLUT_INCLUDE_DIR}', '${GLUT_LIBRARIES}'")
-        message(FATAL_ERROR "Glut not found.  Building without OpenGL is possible.")
-    endif()
-
-    include_directories(${GLUT_INCLUDE_DIR})
-    list(APPEND DEP_LIBS ${GLUT_LIBRARIES})
+        include_directories(/usr/include/GL)
+        list(APPEND DEP_LIBS -lglut)
 endif()
 
 
