--- cryptominisat-release-5.13.0/CMakeLists.txt.orig	2025-04-19 13:14:50.000000000 -0600
+++ cryptominisat-release-5.13.0/CMakeLists.txt	2025-04-24 16:16:59.342920705 -0600
@@ -177,8 +177,6 @@ if (ENABLE_ASSERTIONS)
 else()
     # Note this definition doesn't appear in the cache variables.
     add_definitions(-DNDEBUG)
-    add_cxx_flag_if_supported("-fno-stack-protector")
-    add_definitions(-D_FORTIFY_SOURCE=0)
 endif()
 
 # Note: O3 gives slight speed increase, 1 more solved from SAT Comp'14 @ 3600s
@@ -278,9 +276,6 @@ if (NOT WIN32)
     add_link_options("-fPIE")
     #add_cxx_flag_if_supported("-mtune=native")
 
-    add_cxx_flag_if_supported("-msse4.2")
-    add_cxx_flag_if_supported("-mpopcnt")
-    add_cxx_flag_if_supported("-march=sandybridge")
     if(CMAKE_BUILD_TYPE STREQUAL "Release")
         #NOTE: out-satrace19-8373595 has confirmed that O3+flto only hurts compared to O2
         #      on gcc version 7.3.0
@@ -290,9 +285,6 @@ if (NOT WIN32)
         add_cxx_flag_if_supported("-Wextra")
         add_cxx_flag_if_supported("-Wunused")
         add_cxx_flag_if_supported("-Wsign-compare")
-        if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
-            add_cxx_flag_if_supported("-fno-omit-frame-pointer")
-        endif()
         add_cxx_flag_if_supported("-Wtype-limits")
         add_cxx_flag_if_supported("-Wuninitialized")
         add_cxx_flag_if_supported("-Wno-deprecated")
@@ -305,10 +297,8 @@ if (NOT WIN32)
         add_cxx_flag_if_supported("-Wparentheses")
         add_cxx_flag_if_supported("-Wunreachable-code")
         add_cxx_flag_if_supported("-g")
-        add_cxx_flag_if_supported("-Wno-class-memaccess")
         add_cxx_flag_if_supported("-Wextra-semi-stmt")
         add_cxx_flag_if_supported("-Wnoweak-vtables")
-        add_cxx_flag_if_supported("-ggdb3")
 
         # Apparently needed before OS X Maverics (2013)
         #add_c_flag_if_supported("-stdlib=libc++")
@@ -324,10 +314,6 @@ endif()
 # -----------------------------------------------------------------------------
 # Uncomment these for static compilation under Linux (messes up Valgrind)
 # -----------------------------------------------------------------------------
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" AND NOT SANITIZE)
-    set(CMAKE_EXE_LINKER_FLAGS " ${CMAKE_EXE_LINKER_FLAGS} -Wl,--discard-all -Wl,--build-id=sha1")
-endif()
-
 if ((${CMAKE_SYSTEM_NAME} MATCHES "Linux") OR (${CMAKE_SYSTEM_NAME} MATCHES "Darwin"))
     if(NOT BUILD_SHARED_LIBS)
         MESSAGE(STATUS "Compiling statically")
