--- cryptominisat-release-5.13.0/CMakeLists.txt.orig	2025-04-24 16:23:28.090781385 -0600
+++ cryptominisat-release-5.13.0/CMakeLists.txt	2025-04-25 08:31:05.603168353 -0600
@@ -89,10 +89,11 @@ foreach (build_config ${build_types})
     endforeach()
 endforeach()
 
-PROJECT(cryptominisat5)
-
+project(cryptominisat5)
+set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 enable_language( CXX )
 enable_language( C )
 include (GNUInstallDirs)
@@ -272,8 +273,6 @@ if (NOT WIN32)
     if(NOT ENABLE_TESTING AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
         add_cxx_flag_if_supported("-fvisibility=hidden")
     endif()
-    add_compile_options("-fPIC")
-    add_link_options("-fPIE")
     #add_cxx_flag_if_supported("-mtune=native")
 
     if(CMAKE_BUILD_TYPE STREQUAL "Release")
