--- a/gzip.h
+++ b/gzip.h
@@ -117,11 +117,11 @@
 #define tab_suffix window
 #ifndef MAXSEG_64K
 #  define tab_prefix prev    /* hash link (see deflate.c) */
-#  define head (prev+WSIZE)  /* hash head (see deflate.c) */
+#  define gzip_hash_head (prev+WSIZE)  /* hash head (see deflate.c) */
    EXTERN(ush, tab_prefix);  /* prefix code (see unlzw.c) */
 #else
 #  define tab_prefix0 prev
-#  define head tab_prefix1
+#  define gzip_hash_head tab_prefix1
    EXTERN(ush, tab_prefix0); /* prefix for even codes */
    EXTERN(ush, tab_prefix1); /* prefix for odd  codes */
 #endif
--- a/deflate.c
+++ b/deflate.c
@@ -110,7 +110,7 @@
    error: cannot overlay window with tab_suffix and prev with tab_prefix0
 #endif
 #if HASH_BITS > BITS-1
-   error: cannot overlay head with tab_prefix1
+   error: cannot overlay gzip_hash_head with tab_prefix1
 #endif
 
 #define HASH_SIZE (unsigned)(1<<HASH_BITS)
@@ -290,8 +290,8 @@
  */
 #define INSERT_STRING(s, match_head) \
    (UPDATE_HASH(ins_h, window[(s) + MIN_MATCH-1]), \
-    prev[(s) & WMASK] = match_head = head[ins_h], \
-    head[ins_h] = (s))
+    prev[(s) & WMASK] = match_head = gzip_hash_head[ins_h], \
+    gzip_hash_head[ins_h] = (s))
 
 /* ===========================================================================
  * Initialize the "longest match" routines for a new file
@@ -306,9 +306,9 @@
 
     /* Initialize the hash table. */
 #if defined MAXSEG_64K && HASH_BITS == 15
-    for (j = 0;  j < HASH_SIZE; j++) head[j] = NIL;
+    for (j = 0;  j < HASH_SIZE; j++) gzip_hash_head[j] = NIL;
 #else
-    memzero((char*)head, HASH_SIZE*sizeof(*head));
+    memzero((char*)gzip_hash_head, HASH_SIZE*sizeof(*gzip_hash_head));
 #endif
     /* prev will be initialized on the fly */
 
@@ -555,8 +555,8 @@
         block_start -= (long) WSIZE;
 
         for (n = 0; n < HASH_SIZE; n++) {
-            m = head[n];
-            head[n] = (Pos)(m >= WSIZE ? m-WSIZE : NIL);
+            m = gzip_hash_head[n];
+            gzip_hash_head[n] = (Pos)(m >= WSIZE ? m-WSIZE : NIL);
         }
         for (n = 0; n < WSIZE; n++) {
             m = prev[n];
