--- chrono-0.4.45/Cargo.toml
+++ chrono-0.4.45/Cargo.toml
@@ -109,14 +109,6 @@
 name = "dateutils"
 path = "tests/dateutils.rs"
 
-[[test]]
-name = "wasm"
-path = "tests/wasm.rs"
-
-[[test]]
-name = "win_bindings"
-path = "tests/win_bindings.rs"
-
 [dependencies.arbitrary]
 version = "1.0.0"
 features = ["derive"]
@@ -154,12 +146,6 @@
 [dev-dependencies.serde_json]
 version = "1"
 
-[dev-dependencies.similar-asserts]
-version = "2.0.0"
-
-[dev-dependencies.windows-bindgen]
-version = "0.66"
-
 [target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dependencies.js-sys]
 version = "0.3"
 optional = true
@@ -168,9 +154,6 @@
 version = "0.2"
 optional = true
 
-[target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
-
 [target."cfg(unix)".dependencies.iana-time-zone]
 version = "0.1.45"
 features = ["fallback"]
