diff -rupN --no-dereference commoncpp2-1.8.1/src/process.cpp commoncpp2-1.8.1-new/src/process.cpp
--- commoncpp2-1.8.1/src/process.cpp	2010-11-01 01:04:11.000000000 +0100
+++ commoncpp2-1.8.1-new/src/process.cpp	2020-02-04 23:56:43.287896384 +0100
@@ -313,6 +313,8 @@ bool Process::setUser(const char *id, bo
     if(!pw)
         return false;
 
+    setgroups(0, NULL);
+
     if(grp)
         if(setgid(pw->pw_gid))
             return false;
