--- ff-0.13.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ ff-0.13.1/Cargo.toml	2026-08-04T08:17:28.500690+00:00
@@ -22,13 +22,11 @@
 license = "MIT/Apache-2.0"
 repository = "https://github.com/zkcrypto/ff"
 resolver = "2"
+autotests = false
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
 
-[[test]]
-name = "derive"
-required-features = ["derive"]
 [dependencies.bitvec]
 version = "1"
 optional = true
@@ -38,10 +36,6 @@
 version = "1"
 optional = true
 default-features = false
-
-[dependencies.ff_derive]
-version = "0.13.1"
-optional = true
 
 [dependencies.rand_core]
 version = "0.6"
@@ -61,8 +55,6 @@
 alloc = []
 bits = ["bitvec"]
 default = ["bits", "std"]
-derive = ["byteorder", "ff_derive"]
-derive_bits = ["bits", "ff_derive/bits"]
 std = ["alloc"]
 [badges.maintenance]
 status = "actively-developed"
