--- carbon-6a016c2533716d7bf6f44f4459c43976421bf650.orig/conf/carbon.conf.example	2016-09-13 22:44:40.000000000 +0200
+++ carbon-6a016c2533716d7bf6f44f4459c43976421bf650/conf/carbon.conf.example	2016-09-18 22:11:02.358319492 +0200
@@ -30,6 +30,13 @@
 #
 #LOCAL_DATA_DIR = /opt/graphite/storage/whisper/
 
+STORAGE_DIR    = /var/lib/carbon/
+LOCAL_DATA_DIR = /var/lib/carbon/whisper/
+WHITELISTS_DIR = /var/lib/carbon/lists/
+CONF_DIR       = /etc/carbon/
+LOG_DIR        = /var/log/carbon/
+PID_DIR        = /var/run/
+
 # Specify the database library used to store metric data on disk. Each database
 # may have configurable options to change the behaviour of how it writes to
 # persistent storage.
@@ -41,7 +48,7 @@
 # Specify the user to drop privileges to
 # If this is blank carbon-cache runs as the user that invokes it
 # This user must have write access to the local data directory
-USER =
+USER = carbon
 
 # Limit the size of the cache to avoid swapping or becoming CPU bound.
 # Sorts and serving cache queries gets more expensive as the cache grows.
