diff -up psi4-1.9.1/psi4/src/CMakeLists.txt.nostrip psi4-1.9.1/psi4/src/CMakeLists.txt
--- psi4-1.9.1/psi4/src/CMakeLists.txt.nostrip	2024-02-08 21:08:35.000000000 +0000
+++ psi4-1.9.1/psi4/src/CMakeLists.txt	2024-09-03 19:05:21.153525742 +0000
@@ -16,10 +16,6 @@ target_link_libraries(core
     pybind11::windows_extras
 )
 pybind11_extension(core)
-if(NOT MSVC AND NOT ${CMAKE_BUILD_TYPE} MATCHES Debug|RelWithDebInfo)
-    # Strip unnecessary sections of the binary on Linux/macOS
-    pybind11_strip(core)
-endif()
 set_target_properties(core PROPERTIES CXX_VISIBILITY_PRESET "hidden" VISIBILITY_INLINES_HIDDEN 1)
 
 ### >> Go into psi4 subdirectory to compile libraries and modules <<
diff -up psi4-1.9.1/psi4/src/psi4/CMakeLists.txt.nostrip psi4-1.9.1/psi4/src/psi4/CMakeLists.txt
