diff -rupN --no-dereference tcl9.0.3/win/configure.ac tcl9.0.3-new/win/configure.ac
--- tcl9.0.3/win/configure.ac	2025-11-11 17:22:58.000000000 +0100
+++ tcl9.0.3-new/win/configure.ac	2026-05-15 20:06:54.829745008 +0200
@@ -61,9 +61,9 @@ fi
 AC_PROG_CC
 AC_C_INLINE
 
-AC_CHECK_TOOL(AR, ar)
-AC_CHECK_TOOL(RANLIB, ranlib)
-AC_CHECK_TOOL(RC, windres)
+AC_CHECK_TOOL(AR, ${target_alias}-ar)
+AC_CHECK_TOOL(RANLIB, ${target_alias}-ranlib)
+AC_CHECK_TOOL(RC, ${target_alias}-windres)
 
 #--------------------------------------------------------------------
 # Checks to see if the make program sets the $MAKE variable.
