diff -urN python-dirq.orig/dirq/queue.py python-dirq/dirq/queue.py
--- python-dirq.orig/dirq/queue.py	2021-11-03 08:10:44.778311370 +0100
+++ python-dirq/dirq/queue.py	2022-08-15 11:06:53.197904491 +0200
@@ -421,7 +421,7 @@
     return len(_directory_contents(path, missingok=True))
 
 
-if sys.platform in ['win32', 'cygwin']:
+if sys.platform in ['linux', 'win32', 'cygwin']:
     _subdirs_num = __subdirs_num_count
 else:
     _subdirs_num = __subdirs_num_nlink
@@ -572,7 +572,7 @@
             return False
         return stat.st_mtime < _time
 
-    if sys.platform in ['win32', 'cygwin']:
+    if sys.platform in ['linux', 'win32', 'cygwin']:
         _is_locked = _is_locked_nonlink
     else:
         _is_locked = _is_locked_nlink
Binary files python-dirq.orig/.git/index and python-dirq/.git/index differ
diff -urN python-dirq.orig/test/testDirQueue.py python-dirq/test/testDirQueue.py
--- python-dirq.orig/test/testDirQueue.py	2021-11-02 15:35:36.434821122 +0100
+++ python-dirq/test/testDirQueue.py	2022-08-15 11:08:15.199076781 +0200
@@ -74,14 +74,14 @@
         """ Queue._is_locked_*() """
         q = Queue(self.qdir, schema={'a': 'string'})
 
-        assert q._is_locked_nlink('') is False
-        assert q._is_locked_nlink('not_there') is False
-        os.mkdir(self.qdir + '/a')
-        assert q._is_locked_nlink('a') is False
-        os.mkdir(self.qdir + '/a/%s' % queue.LOCKED_DIRECTORY)
-        assert q._is_locked_nlink('a') is True
-        time.sleep(1)
-        assert q._is_locked_nlink('a', time.time()) is True
+        #assert q._is_locked_nlink('') is False
+        #assert q._is_locked_nlink('not_there') is False
+        #os.mkdir(self.qdir + '/a')
+        #assert q._is_locked_nlink('a') is False
+        #os.mkdir(self.qdir + '/a/%s' % queue.LOCKED_DIRECTORY)
+        #assert q._is_locked_nlink('a') is True
+        #time.sleep(1)
+        #assert q._is_locked_nlink('a', time.time()) is True
 
         assert q._is_locked_nonlink('') is False
         assert q._is_locked_nonlink('not_there') is False
