From 8ac3ffd3eec886a95e57f662df7367db18ac17cc Mon Sep 17 00:00:00 2001
From: Tom Rix <Tom.Rix@amd.com>
Date: Mon, 21 Apr 2025 06:56:55 -0700
Subject: [PATCH] Handle cmake DBUILD_SHARED_LIBS=ON

---
 lib/CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 34303c207640..d7bcc93fd10a 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -30,7 +30,7 @@ set(HIPFORT_ARCH "amdgcn")
     # amdgcn
     set(HIPFORT_LIB      "hipfort-${HIPFORT_ARCH}")
     set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/include/hipfort/${HIPFORT_ARCH})
-    ADD_LIBRARY(${HIPFORT_LIB} STATIC 
+    ADD_LIBRARY(${HIPFORT_LIB} SHARED 
         ${HIPFORT_SRC_HIP} 
         #${HIPFORT_SRC_amdgcn} 
         ${HIPFORT_SRC_CONTRIB} 
@@ -38,7 +38,10 @@ set(HIPFORT_ARCH "amdgcn")
     target_include_directories(${HIPFORT_LIB}
       PUBLIC
         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/hipfort/amdgcn>
-    )
+      )
+    target_link_libraries(${HIPFORT_LIB} -lhipblas -lhipsparse -lhipsolver -lhipfft -lhiprand)
+    set_target_properties(${HIPFORT_LIB} PROPERTIES VERSION ${HIPFORT_VERSION} SOVERSION ${HIPFORT_VERSION})
+
     IF(CMAKE_Fortran_COMPILER_SUPPORTS_F08)
     	target_compile_definitions(${HIPFORT_LIB} PRIVATE USE_FPOINTER_INTERFACES)
     ENDIF(CMAKE_Fortran_COMPILER_SUPPORTS_F08)
-- 
2.48.1

