diff -Nrbu ipxripd/ipxd.c ipxripd.OK/ipxd.c
--- ipxripd/ipxd.c	2004-08-03 15:36:01.000000000 +0400
+++ ipxripd.OK/ipxd.c	2004-08-03 15:35:33.000000000 +0400
@@ -19,6 +19,7 @@
 
 */
 
+#include <stdlib.h>
 #include <signal.h>
 #include <unistd.h>
 #include <string.h>
diff -Nrbu ipxripd/ipxd.h ipxripd.OK/ipxd.h
--- ipxripd/ipxd.h	1996-02-08 19:07:56.000000000 +0300
+++ ipxripd.OK/ipxd.h	2004-08-03 15:34:42.000000000 +0400
@@ -23,6 +23,7 @@
 #define _IPXD_H_
 
 #include <stdio.h>
+#include <time.h>
 #include <sys/time.h>
 #include "ipxutil.h"
 #include "ipxkern.h"
@@ -55,7 +56,7 @@
 extern FILE *log_file;
 extern int passive;
 
-extern int check_request;
+extern volatile int check_request;
 
 struct ipx_interface *first_interface(void);
 struct ipx_interface *next_interface(struct ipx_interface *ifc);
